<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>class Bio::SQL::Sequence - BioRuby API documentation</title>

<link type="text/css" media="screen" href="../../rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "../../";
</script>

<script type="text/javascript" charset="utf-8" src="../../js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/darkfish.js"></script>


<body id="top" class="class">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="../../index.html">Home</a>
    <a href="../../table_of_contents.html#classes">Classes</a>
    <a href="../../table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio/db/biosql/sequence.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    <nav id="parent-class-section" class="section">
  <h3 class="section-header">Parent</h3>
  
  <p class="link"><a href="../../Object.html">Object</a>
  
</nav>

    
    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-new">::new</a>
    
    <li><a href="#method-i-cdsfeatures">#cdsfeatures</a>
    
    <li><a href="#method-i-comment-3D">#comment=</a>
    
    <li><a href="#method-i-comments">#comments</a>
    
    <li><a href="#method-i-database">#database</a>
    
    <li><a href="#method-i-database_desc">#database_desc</a>
    
    <li><a href="#method-i-definition">#definition</a>
    
    <li><a href="#method-i-definition-3D">#definition=</a>
    
    <li><a href="#method-i-delete">#delete</a>
    
    <li><a href="#method-i-description">#description</a>
    
    <li><a href="#method-i-description-3D">#description=</a>
    
    <li><a href="#method-i-division">#division</a>
    
    <li><a href="#method-i-division-3D">#division=</a>
    
    <li><a href="#method-i-entry_id">#entry_id</a>
    
    <li><a href="#method-i-entry_id-3D">#entry_id=</a>
    
    <li><a href="#method-i-feature-3D">#feature=</a>
    
    <li><a href="#method-i-features">#features</a>
    
    <li><a href="#method-i-get_seqfeature">#get_seqfeature</a>
    
    <li><a href="#method-i-identifier">#identifier</a>
    
    <li><a href="#method-i-identifier-3D">#identifier=</a>
    
    <li><a href="#method-i-length">#length</a>
    
    <li><a href="#method-i-length-3D">#length=</a>
    
    <li><a href="#method-i-name">#name</a>
    
    <li><a href="#method-i-name-3D">#name=</a>
    
    <li><a href="#method-i-organism">#organism</a>
    
    <li><a href="#method-i-organism-3D">#organism=</a>
    
    <li><a href="#method-i-other_seqids">#other_seqids</a>
    
    <li><a href="#method-i-primary_accession">#primary_accession</a>
    
    <li><a href="#method-i-primary_accession-3D">#primary_accession=</a>
    
    <li><a href="#method-i-reference-3D">#reference=</a>
    
    <li><a href="#method-i-references">#references</a>
    
    <li><a href="#method-i-save">#save</a>
    
    <li><a href="#method-i-seq">#seq</a>
    
    <li><a href="#method-i-seq-3D">#seq=</a>
    
    <li><a href="#method-i-sequence_version">#sequence_version</a>
    
    <li><a href="#method-i-sequence_version-3D">#sequence_version=</a>
    
    <li><a href="#method-i-species">#species</a>
    
    <li><a href="#method-i-species-3D">#species=</a>
    
    <li><a href="#method-i-taxonomy">#taxonomy</a>
    
    <li><a href="#method-i-to_biosequence">#to_biosequence</a>
    
    <li><a href="#method-i-to_biosql">#to_biosql</a>
    
    <li><a href="#method-i-to_fasta">#to_fasta</a>
    
    <li><a href="#method-i-to_fasta_reverse_complememt">#to_fasta_reverse_complememt</a>
    
    <li><a href="#method-i-version">#version</a>
    
    <li><a href="#method-i-version-3D">#version=</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="../../ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="../../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="../../LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="../../README_rdoc.html">README</a>
  
    <li class="file"><a href="../../README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="../../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="../../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="../../Bio.html">Bio</a>
  
    <li><a href="../../Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="../../Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="../../Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="../../Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="../../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="../../Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="../../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="../../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="../../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="../../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="../../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="../../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="../../Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="../../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="../../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="../../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="../../Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="../../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="../../Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="../../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="../../Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="../../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="../../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="../../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="../../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="../../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="../../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="../../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="../../Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="../../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="../../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="../../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="../../Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="../../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="../../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="../../Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="../../Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="../../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="../../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="../../Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="../../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="../../Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="../../Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="../../Bio/Command.html">Bio::Command</a>
  
    <li><a href="../../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="../../Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="../../Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="../../Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="../../Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="../../Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="../../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="../../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="../../Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="../../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="../../Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="../../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="../../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="../../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="../../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="../../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="../../Bio/DB.html">Bio::DB</a>
  
    <li><a href="../../Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="../../Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="../../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="../../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="../../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="../../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="../../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="../../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="../../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="../../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="../../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="../../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="../../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="../../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="../../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="../../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="../../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="../../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="../../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="../../Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="../../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="../../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="../../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="../../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="../../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="../../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="../../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="../../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="../../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="../../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="../../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="../../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="../../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="../../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="../../Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="../../Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="../../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="../../Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="../../Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="../../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="../../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="../../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="../../Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="../../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="../../Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="../../Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="../../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="../../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="../../Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="../../Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="../../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="../../Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="../../Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="../../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="../../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="../../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="../../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="../../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="../../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="../../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="../../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="../../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="../../Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="../../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="../../Bio/Features.html">Bio::Features</a>
  
    <li><a href="../../Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="../../Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="../../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="../../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="../../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="../../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="../../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="../../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="../../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="../../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="../../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="../../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="../../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="../../Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="../../Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="../../Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="../../Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="../../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="../../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="../../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="../../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="../../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="../../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="../../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="../../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="../../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="../../Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="../../Bio/GO.html">Bio::GO</a>
  
    <li><a href="../../Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="../../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="../../Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="../../Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="../../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="../../Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="../../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="../../Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="../../Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="../../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="../../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="../../Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="../../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="../../Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="../../Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="../../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="../../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="../../Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="../../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="../../Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="../../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="../../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="../../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="../../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="../../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="../../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="../../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="../../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="../../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="../../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="../../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="../../Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="../../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="../../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="../../Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="../../Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="../../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="../../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="../../Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="../../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="../../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="../../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="../../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="../../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="../../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="../../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="../../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="../../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="../../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="../../Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="../../Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="../../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="../../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="../../Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="../../Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="../../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="../../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="../../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="../../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="../../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="../../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="../../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="../../Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="../../Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="../../Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="../../Bio/Location.html">Bio::Location</a>
  
    <li><a href="../../Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="../../Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="../../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="../../Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="../../Bio/Map.html">Bio::Map</a>
  
    <li><a href="../../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="../../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="../../Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="../../Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="../../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="../../Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="../../Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="../../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="../../Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="../../Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="../../Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="../../Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="../../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="../../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="../../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="../../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="../../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="../../Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="../../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="../../Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="../../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="../../Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="../../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="../../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="../../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="../../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="../../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="../../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="../../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="../../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="../../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="../../Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="../../Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="../../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="../../Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="../../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="../../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="../../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="../../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="../../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="../../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="../../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="../../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="../../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="../../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="../../Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="../../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="../../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="../../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="../../Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="../../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="../../Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="../../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="../../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="../../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="../../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="../../Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="../../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="../../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="../../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="../../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="../../Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="../../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="../../Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="../../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="../../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="../../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="../../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="../../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="../../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="../../Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="../../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="../../Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="../../Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="../../Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="../../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="../../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="../../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="../../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="../../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="../../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="../../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="../../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="../../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="../../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="../../Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="../../Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="../../Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="../../Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="../../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="../../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="../../Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="../../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="../../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="../../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="../../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="../../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="../../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="../../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="../../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="../../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="../../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="../../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="../../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="../../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="../../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="../../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="../../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="../../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="../../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="../../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="../../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="../../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="../../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="../../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="../../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="../../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="../../Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="../../Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="../../Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="../../Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="../../Bio/References.html">Bio::References</a>
  
    <li><a href="../../Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="../../Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="../../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="../../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="../../Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="../../Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="../../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="../../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="../../Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="../../Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="../../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="../../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="../../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="../../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="../../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="../../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="../../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="../../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="../../Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="../../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="../../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="../../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="../../Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="../../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="../../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="../../Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="../../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="../../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="../../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="../../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="../../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="../../Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="../../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="../../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="../../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="../../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="../../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="../../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="../../Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="../../Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="../../Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="../../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="../../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="../../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="../../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="../../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="../../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="../../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="../../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="../../Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="../../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="../../Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="../../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="../../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="../../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="../../Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="../../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="../../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="../../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="../../Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="../../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="../../Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="../../Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="../../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="../../Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="../../Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="../../Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="../../Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="../../Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="../../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="../../Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="../../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="../../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="../../Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="../../Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="../../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="../../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="../../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="../../Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="../../Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="../../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="../../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="../../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="../../Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="../../Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="../../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="../../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="../../Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="../../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="../../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="../../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="../../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="../../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="../../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="../../Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="../../Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="../../Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="../../Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="../../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="../../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="../../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="../../Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="../../Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="../../Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="../../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="../../Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="../../Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="../../BiorubyController.html">BiorubyController</a>
  
    <li><a href="../../BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="../../BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="../../Hash.html">Hash</a>
  
    <li><a href="../../Object.html">Object</a>
  
    <li><a href="../../String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="class">class Bio::SQL::Sequence</h1>

  <div id="description" class="description">
    
  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    

    
    <!-- Attributes -->
    <section id="attribute-method-details" class="method-section section">
      <h3 class="section-header">Attributes</h3>

      
      <div id="attribute-i-entry" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">entry</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        
        
        </div>
      </div>
      
    </section><!-- attribute-method-details -->
    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(options={})</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 106</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">options</span>={})
  <span class="ruby-comment">#options.assert_valid_keys(:entry, :biodatabase,:biosequence)</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@entry</span> = <span class="ruby-identifier">options</span>[<span class="ruby-value">:entry</span>] <span class="ruby-keyword">unless</span> <span class="ruby-identifier">options</span>[<span class="ruby-value">:entry</span>].<span class="ruby-identifier">nil?</span>

  <span class="ruby-keyword">return</span> <span class="ruby-identifier">to_biosql</span>(<span class="ruby-identifier">options</span>[<span class="ruby-value">:biosequence</span>], <span class="ruby-identifier">options</span>[<span class="ruby-value">:biodatabase</span>]) <span class="ruby-keyword">unless</span> <span class="ruby-identifier">options</span>[<span class="ruby-value">:biosequence</span>].<span class="ruby-identifier">nil?</span> <span class="ruby-keyword">or</span> <span class="ruby-identifier">options</span>[<span class="ruby-value">:biodatabase</span>].<span class="ruby-identifier">nil?</span>

<span class="ruby-keyword">end</span></pre>
          </div><!-- new-source -->
          
        </div>

        

        
      </div><!-- new-method -->

    
    </section><!-- public-class-method-details -->
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Instance Methods</h3>

    
      <div id="method-i-cdsfeatures" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cdsfeatures</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>return the seqfeature mapped from BioSQL with a type_term like ‘CDS’</p>
          

          
          <div class="method-source-code" id="cdsfeatures-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 323</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cdsfeatures</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">cdsfeatures</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- cdsfeatures-source -->
          
        </div>

        

        
      </div><!-- cdsfeatures-method -->

    
      <div id="method-i-comment-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">comment=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="comment-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 420</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">comment=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-comment">#DELETE        comment=Comment.new({:bioentry=&gt;@entry, :comment_text=&gt;value, :rank=&gt;@entry.comments.count.succ})</span>
  <span class="ruby-identifier">comment</span> = <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">comments</span>.<span class="ruby-identifier">build</span>({<span class="ruby-value">:comment_text=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">value</span>, <span class="ruby-value">:rank=</span><span class="ruby-operator">&gt;</span><span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">comments</span>.<span class="ruby-identifier">count</span>.<span class="ruby-identifier">succ</span>})
  <span class="ruby-identifier">comment</span>.<span class="ruby-identifier">save</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- comment-3D-source -->
          
        </div>

        

        
      </div><!-- comment-3D-method -->

    
      <div id="method-i-comments" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">comments</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="comments-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 393</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">comments</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">comments</span>.<span class="ruby-identifier">map</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">comment</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">comment</span>.<span class="ruby-identifier">comment_text</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- comments-source -->
          
        </div>

        

        
      </div><!-- comments-method -->

    
      <div id="method-i-database" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">database</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="database-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 234</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">database</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biodatabase</span>.<span class="ruby-identifier">name</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- database-source -->
          
        </div>

        

        
      </div><!-- database-method -->

    
      <div id="method-i-database_desc" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">database_desc</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="database_desc-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 238</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">database_desc</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biodatabase</span>.<span class="ruby-identifier">description</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- database_desc-source -->
          
        </div>

        

        
      </div><!-- database_desc-method -->

    
      <div id="method-i-definition" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">definition</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-description">description</a>
        </div>
        
      </div><!-- definition-method -->

    
      <div id="method-i-definition-3D" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">definition=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-description-3D">description=</a>
        </div>
        
      </div><!-- definition-3D-method -->

    
      <div id="method-i-delete" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">delete</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="delete-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 85</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">delete</span>
  <span class="ruby-comment">#TODO: check is references connected to this bioentry are leaf or not.</span>
  <span class="ruby-comment">#actually I think it should be more sofisticated, check if there are</span>
  <span class="ruby-comment">#other bioentries connected to references; if not delete 'em</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">references</span>.<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">ref</span><span class="ruby-operator">|</span> <span class="ruby-identifier">ref</span>.<span class="ruby-identifier">delete</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">ref</span>.<span class="ruby-identifier">bioentries</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">==</span><span class="ruby-value">1</span>}
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">destroy</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- delete-source -->
          
        </div>

        

        
      </div><!-- delete-method -->

    
      <div id="method-i-description" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">description</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="description-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 260</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">description</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">description</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- description-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-definition">definition</a>
        </div>
        

        
      </div><!-- description-method -->

    
      <div id="method-i-description-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">description=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="description-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 265</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">description=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">description</span>=<span class="ruby-identifier">value</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- description-3D-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-definition-3D">definition=</a>
        </div>
        

        
      </div><!-- description-3D-method -->

    
      <div id="method-i-division" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">division</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="division-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 252</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">division</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">division</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- division-source -->
          
        </div>

        

        
      </div><!-- division-method -->

    
      <div id="method-i-division-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">division=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="division-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 256</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">division=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">division</span>=<span class="ruby-identifier">value</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- division-3D-source -->
          
        </div>

        

        
      </div><!-- division-3D-method -->

    
      <div id="method-i-entry_id" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">entry_id</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-name">name</a>
        </div>
        
      </div><!-- entry_id-method -->

    
      <div id="method-i-entry_id-3D" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">entry_id=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-name-3D">name=</a>
        </div>
        
      </div><!-- entry_id-3D-method -->

    
      <div id="method-i-feature-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">feature=</span><span
            class="method-args">(feat)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="feature-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 293</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">feature=</span>(<span class="ruby-identifier">feat</span>)
  <span class="ruby-comment">#ToDo: avoid Ontology find here, probably more efficient create class variables</span>
  <span class="ruby-comment">#DELETE        type_term_ontology = Ontology.find_or_create({:name=&gt;'SeqFeature Keys'})</span>
  <span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;feature:type_term = #{feat.feature}&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
  <span class="ruby-identifier">type_term</span> = <span class="ruby-constant">Term</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;name = ?&quot;</span>, <span class="ruby-identifier">feat</span>.<span class="ruby-identifier">feature</span>]) <span class="ruby-operator">||</span> <span class="ruby-constant">Term</span>.<span class="ruby-identifier">create</span>({<span class="ruby-value">:name=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">feat</span>.<span class="ruby-identifier">feature</span>, <span class="ruby-value">:ontology=</span><span class="ruby-operator">&gt;</span><span class="ruby-constant">Ontology</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;name = ?&quot;</span>,<span class="ruby-string">'SeqFeature Keys'</span>])})
  <span class="ruby-comment">#DELETE        source_term_ontology = Ontology.find_or_create({:name=&gt;'SeqFeature Sources'})</span>
  <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;feature:source_term&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
  <span class="ruby-identifier">source_term</span> = <span class="ruby-constant">Term</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;name = ?&quot;</span>,<span class="ruby-string">'EMBLGenBankSwit'</span>])
  <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;feature:seqfeature&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
  <span class="ruby-identifier">seqfeature</span> = <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">seqfeatures</span>.<span class="ruby-identifier">build</span>({<span class="ruby-value">:source_term=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">source_term</span>, <span class="ruby-value">:type_term=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">type_term</span>, <span class="ruby-value">:rank=</span><span class="ruby-operator">&gt;</span><span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">seqfeatures</span>.<span class="ruby-identifier">count</span>.<span class="ruby-identifier">succ</span>, <span class="ruby-value">:display_name=</span><span class="ruby-operator">&gt;</span><span class="ruby-string">''</span>})
  <span class="ruby-identifier">seqfeature</span>.<span class="ruby-identifier">save</span>
  <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;feature:location&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
  <span class="ruby-identifier">feat</span>.<span class="ruby-identifier">locations</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">loc</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">location</span> = <span class="ruby-identifier">seqfeature</span>.<span class="ruby-identifier">locations</span>.<span class="ruby-identifier">build</span>({<span class="ruby-value">:seqfeature=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">seqfeature</span>, <span class="ruby-value">:start_pos=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">loc</span>.<span class="ruby-identifier">from</span>, <span class="ruby-value">:end_pos=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">loc</span>.<span class="ruby-identifier">to</span>, <span class="ruby-value">:strand=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">loc</span>.<span class="ruby-identifier">strand</span>, <span class="ruby-value">:rank=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">seqfeature</span>.<span class="ruby-identifier">locations</span>.<span class="ruby-identifier">count</span>.<span class="ruby-identifier">succ</span>})
    <span class="ruby-identifier">location</span>.<span class="ruby-identifier">save</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-comment">#DELETE        qual_term_ontology = Ontology.find_or_create({:name=&gt;'Annotation Tags'})</span>

  <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;feature:qualifier&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
  <span class="ruby-identifier">feat</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">qualifier</span><span class="ruby-operator">|</span>
    <span class="ruby-comment">#DELETE          qual_term = Term.find_or_create({:name=&gt;qualifier.qualifier}, {:ontology=&gt;qual_term_ontology})</span>
    <span class="ruby-identifier">qual_term</span> = <span class="ruby-constant">Term</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;name = ?&quot;</span>, <span class="ruby-identifier">qualifier</span>.<span class="ruby-identifier">qualifier</span>]) <span class="ruby-operator">||</span> <span class="ruby-constant">Term</span>.<span class="ruby-identifier">create</span>({<span class="ruby-value">:name=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">qualifier</span>.<span class="ruby-identifier">qualifier</span>, <span class="ruby-value">:ontology=</span><span class="ruby-operator">&gt;</span><span class="ruby-constant">Ontology</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;name = ?&quot;</span>, <span class="ruby-string">'Annotation Tags'</span>])})
    <span class="ruby-identifier">qual</span> = <span class="ruby-identifier">seqfeature</span>.<span class="ruby-identifier">seqfeature_qualifier_values</span>.<span class="ruby-identifier">build</span>({<span class="ruby-value">:seqfeature=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">seqfeature</span>, <span class="ruby-value">:term=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">qual_term</span>, <span class="ruby-value">:value=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">qualifier</span>.<span class="ruby-identifier">value</span>.<span class="ruby-identifier">to_s</span>, <span class="ruby-value">:rank=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">seqfeature</span>.<span class="ruby-identifier">seqfeature_qualifier_values</span>.<span class="ruby-identifier">count</span>.<span class="ruby-identifier">succ</span>})
    <span class="ruby-identifier">qual</span>.<span class="ruby-identifier">save</span>

  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- feature-3D-source -->
          
        </div>

        

        
      </div><!-- feature-3D-method -->

    
      <div id="method-i-features" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">features</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="features-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 287</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">features</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">seqfeatures</span>.<span class="ruby-identifier">collect</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">sf</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">get_seqfeature</span>(<span class="ruby-identifier">sf</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- features-source -->
          
        </div>

        

        
      </div><!-- features-method -->

    
      <div id="method-i-get_seqfeature" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">get_seqfeature</span><span
            class="method-args">(sf)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="get_seqfeature-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 93</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">get_seqfeature</span>(<span class="ruby-identifier">sf</span>)

  <span class="ruby-comment">#in seqfeature BioSQL class</span>
  <span class="ruby-identifier">locations_str</span> = <span class="ruby-identifier">sf</span>.<span class="ruby-identifier">locations</span>.<span class="ruby-identifier">map</span>{<span class="ruby-operator">|</span><span class="ruby-identifier">loc</span><span class="ruby-operator">|</span> <span class="ruby-identifier">loc</span>.<span class="ruby-identifier">to_s</span>}.<span class="ruby-identifier">join</span>(<span class="ruby-string">','</span>)
  <span class="ruby-comment">#pp sf.locations.inspect</span>
  <span class="ruby-identifier">locations_str</span> = <span class="ruby-node">&quot;join(#{locations_str})&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">sf</span>.<span class="ruby-identifier">locations</span>.<span class="ruby-identifier">count</span><span class="ruby-operator">&gt;</span><span class="ruby-value">1</span>
  <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Feature</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">sf</span>.<span class="ruby-identifier">type_term</span>.<span class="ruby-identifier">name</span>, <span class="ruby-identifier">locations_str</span>,<span class="ruby-identifier">sf</span>.<span class="ruby-identifier">seqfeature_qualifier_values</span>.<span class="ruby-identifier">collect</span>{<span class="ruby-operator">|</span><span class="ruby-identifier">sfqv</span><span class="ruby-operator">|</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Feature</span><span class="ruby-operator">::</span><span class="ruby-constant">Qualifier</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">sfqv</span>.<span class="ruby-identifier">term</span>.<span class="ruby-identifier">name</span>,<span class="ruby-identifier">sfqv</span>.<span class="ruby-identifier">value</span>)})
<span class="ruby-keyword">end</span></pre>
          </div><!-- get_seqfeature-source -->
          
        </div>

        

        
      </div><!-- get_seqfeature-method -->

    
      <div id="method-i-identifier" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">identifier</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="identifier-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 270</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">identifier</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">identifier</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- identifier-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-other_seqids">other_seqids</a>
        </div>
        

        
      </div><!-- identifier-method -->

    
      <div id="method-i-identifier-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">identifier=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="identifier-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 275</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">identifier=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">identifier</span>=<span class="ruby-identifier">value</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- identifier-3D-source -->
          
        </div>

        

        
      </div><!-- identifier-3D-method -->

    
      <div id="method-i-length" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">length</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="length-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 365</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">length</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">length</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- length-source -->
          
        </div>

        

        
      </div><!-- length-method -->

    
      <div id="method-i-length-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">length=</span><span
            class="method-args">(len)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="length-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 102</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">length=</span>(<span class="ruby-identifier">len</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">length</span>=<span class="ruby-identifier">len</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- length-3D-source -->
          
        </div>

        

        
      </div><!-- length-3D-method -->

    
      <div id="method-i-name" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">name</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="name-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 195</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">name</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">name</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- name-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-entry_id">entry_id</a>
        </div>
        

        
      </div><!-- name-method -->

    
      <div id="method-i-name-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">name=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="name-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 200</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">name=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">name</span>=<span class="ruby-identifier">value</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- name-3D-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-entry_id-3D">entry_id=</a>
        </div>
        

        
      </div><!-- name-3D-method -->

    
      <div id="method-i-organism" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">organism</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>TODO      def secondary_accession</p>

<pre>  @entry.bioentry_qualifier_values
end</pre>
          

          
          <div class="method-source-code" id="organism-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 217</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">organism</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">nil?</span> <span class="ruby-operator">?</span> <span class="ruby-string">&quot;&quot;</span> <span class="ruby-operator">:</span> <span class="ruby-node">&quot;#{@entry.taxon.taxon_scientific_name.name}&quot;</span><span class="ruby-operator">+</span> (<span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">taxon_genbank_common_name</span> <span class="ruby-operator">?</span> <span class="ruby-node">&quot;(#{@entry.taxon.taxon_genbank_common_name.name})&quot;</span> <span class="ruby-operator">:</span> <span class="ruby-string">''</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- organism-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-species">species</a>
        </div>
        

        
      </div><!-- organism-method -->

    
      <div id="method-i-organism-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">organism=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="organism-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 222</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">organism=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-comment">#FIX there is a shortcut</span>
  <span class="ruby-identifier">taxon_name</span>=<span class="ruby-constant">TaxonName</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;name = ? and name_class = ?&quot;</span>,<span class="ruby-identifier">value</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">%r\s+\(.+\)/</span>,<span class="ruby-string">''</span>),<span class="ruby-string">'scientific name'</span>])
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">taxon_name</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;Error value doesn't exists in taxon_name table with scientific name constraint.&quot;</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">taxon_id</span>=<span class="ruby-identifier">taxon_name</span>.<span class="ruby-identifier">taxon_id</span>
    <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">save</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- organism-3D-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-species-3D">species=</a>
        </div>
        

        
      </div><!-- organism-3D-method -->

    
      <div id="method-i-other_seqids" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">other_seqids</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-identifier">identifier</a>
        </div>
        
      </div><!-- other_seqids-method -->

    
      <div id="method-i-primary_accession" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">primary_accession</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="primary_accession-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 205</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">primary_accession</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">accession</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- primary_accession-source -->
          
        </div>

        

        
      </div><!-- primary_accession-method -->

    
      <div id="method-i-primary_accession-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">primary_accession=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="primary_accession-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 209</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">primary_accession=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">accession</span>=<span class="ruby-identifier">value</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- primary_accession-3D-source -->
          
        </div>

        

        
      </div><!-- primary_accession-3D-method -->

    
      <div id="method-i-reference-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">reference=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="reference-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 399</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">reference=</span>(<span class="ruby-identifier">value</span>)       
  <span class="ruby-identifier">locations</span>=<span class="ruby-constant">Array</span>.<span class="ruby-identifier">new</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;journal=#{value.journal}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">journal</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;volume=#{value.volume}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">volume</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;issue=#{value.issue}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">issue</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;pages=#{value.pages}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">pages</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;year=#{value.year}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">year</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;pubmed=#{value.pubmed}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">pubmed</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;medline=#{value.medline}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">medline</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;doi=#{value.doi}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">doi</span>.<span class="ruby-identifier">nil?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;abstract=#{value.abstract}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">abstract</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;url=#{value.url}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">url</span>.<span class="ruby-identifier">nil?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;mesh=#{value.mesh}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">mesh</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;affiliations=#{value.affiliations}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">affiliations</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;comments=#{value.comments.join('~')}&quot;</span><span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">comments</span>.<span class="ruby-identifier">nil?</span>
  <span class="ruby-identifier">start_pos</span>, <span class="ruby-identifier">end_pos</span> = <span class="ruby-identifier">value</span>.<span class="ruby-identifier">sequence_position</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">sequence_position</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">%r\s*/</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">split</span>(<span class="ruby-string">'-'</span>) <span class="ruby-operator">:</span> [<span class="ruby-keyword">nil</span>,<span class="ruby-keyword">nil</span>]
  <span class="ruby-identifier">reference</span>= <span class="ruby-constant">Reference</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;title = ?&quot;</span>,<span class="ruby-identifier">value</span>.<span class="ruby-identifier">title</span>]) <span class="ruby-operator">||</span> <span class="ruby-constant">Reference</span>.<span class="ruby-identifier">create</span>({<span class="ruby-value">:title=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">value</span>.<span class="ruby-identifier">title</span>,<span class="ruby-value">:authors=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">value</span>.<span class="ruby-identifier">authors</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">' '</span>), <span class="ruby-value">:location=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">locations</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">'|'</span>)})
  <span class="ruby-identifier">bio_reference</span>=<span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">bioentry_references</span>.<span class="ruby-identifier">build</span>({<span class="ruby-value">:reference=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">reference</span>,<span class="ruby-value">:rank=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">value</span>.<span class="ruby-identifier">embl_gb_record_number</span>, <span class="ruby-value">:start_pos=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">start_pos</span>, <span class="ruby-value">:end_pos=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">end_pos</span>})
  <span class="ruby-identifier">bio_reference</span>.<span class="ruby-identifier">save</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- reference-3D-source -->
          
        </div>

        

        
      </div><!-- reference-3D-method -->

    
      <div id="method-i-references" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">references</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="references-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 369</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">references</span>
  <span class="ruby-comment">#return and array of hash, hash has these keys [&quot;title&quot;, &quot;dbxref_id&quot;, &quot;reference_id&quot;, &quot;authors&quot;, &quot;crc&quot;, &quot;location&quot;]</span>
  <span class="ruby-comment">#probably would be better to d a class refrence to collect these informations</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">bioentry_references</span>.<span class="ruby-identifier">collect</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">bio_ref</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">hash</span> = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>
    <span class="ruby-identifier">hash</span>[<span class="ruby-string">'authors'</span>] = <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">reference</span>.<span class="ruby-identifier">authors</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">%r\.\s/</span>, <span class="ruby-string">&quot;\.\s\|&quot;</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">%r\|/</span>)

    <span class="ruby-identifier">hash</span>[<span class="ruby-string">'sequence_position'</span>] = <span class="ruby-node">&quot;#{bio_ref.start_pos}-#{bio_ref.end_pos}&quot;</span> <span class="ruby-keyword">if</span> (<span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">start_pos</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">end_pos</span>)
    <span class="ruby-identifier">hash</span>[<span class="ruby-string">'title'</span>] = <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">reference</span>.<span class="ruby-identifier">title</span>
    <span class="ruby-identifier">hash</span>[<span class="ruby-string">'embl_gb_record_number'</span>] = <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">rank</span>
    <span class="ruby-comment">#TODO: solve the problem with specific comment per reference.</span>
    <span class="ruby-comment">#TODO: get dbxref</span>
    <span class="ruby-comment">#take a look when location is build up in def reference=(value)</span>

    <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">reference</span>.<span class="ruby-identifier">location</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">'|'</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">element</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">key</span>,<span class="ruby-identifier">value</span>=<span class="ruby-identifier">element</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">'='</span>)
      <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">key</span>]=<span class="ruby-identifier">value</span>
    <span class="ruby-keyword">end</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">reference</span>.<span class="ruby-identifier">location</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-identifier">hash</span>[<span class="ruby-string">'xrefs'</span>] = <span class="ruby-identifier">bio_ref</span>.<span class="ruby-identifier">reference</span>.<span class="ruby-identifier">dbxref</span> <span class="ruby-operator">?</span> <span class="ruby-node">&quot;#{bio_ref.reference.dbxref.dbname}; #{bio_ref.reference.dbxref.accession}.&quot;</span> <span class="ruby-operator">:</span> <span class="ruby-string">''</span>
    <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Reference</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">hash</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- references-source -->
          
        </div>

        

        
      </div><!-- references-method -->

    
      <div id="method-i-save" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">save</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="save-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 426</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">save</span>
  <span class="ruby-comment">#I should add chks for SQL errors</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">save</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">save</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- save-source -->
          
        </div>

        

        
      </div><!-- save-method -->

    
      <div id="method-i-seq" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">seq</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns the sequence. Returns a Bio::Sequence::Generic object.</p>
          

          
          <div class="method-source-code" id="seq-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 330</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">seq</span>
  <span class="ruby-identifier">s</span> = <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>
  <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">Generic</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">s</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span> <span class="ruby-operator">:</span> <span class="ruby-string">''</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- seq-source -->
          
        </div>

        

        
      </div><!-- seq-method -->

    
      <div id="method-i-seq-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">seq=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="seq-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 335</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">seq=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-comment">#TODO: revise this piece of code.</span>
  <span class="ruby-comment">#chk which type of alphabet is, NU/NA/nil</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-comment">#DELETE          puts &quot;intoseq1&quot;</span>
    <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span> = <span class="ruby-constant">Biosequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-value">:seq=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">value</span>)
    <span class="ruby-comment">#          biosequence = @entry.biosequence.build({:seq=&gt;value})</span>
    <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">save</span>
    <span class="ruby-comment">#          biosequence.save</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">seq</span>=<span class="ruby-identifier">value</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">length</span>=<span class="ruby-identifier">value</span>.<span class="ruby-identifier">length</span>
  <span class="ruby-comment">#DELETE        #@entry.biosequence.length=value.length</span>
  <span class="ruby-comment">#DELETE        #break</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">save</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- seq-3D-source -->
          
        </div>

        

        
      </div><!-- seq-3D-method -->

    
      <div id="method-i-sequence_version" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">sequence_version</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-version">version</a>
        </div>
        
      </div><!-- sequence_version-method -->

    
      <div id="method-i-sequence_version-3D" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">sequence_version=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-version-3D">version=</a>
        </div>
        
      </div><!-- sequence_version-3D-method -->

    
      <div id="method-i-species" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">species</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-organism">organism</a>
        </div>
        
      </div><!-- species-method -->

    
      <div id="method-i-species-3D" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">species=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-organism-3D">organism=</a>
        </div>
        
      </div><!-- species-3D-method -->

    
      <div id="method-i-taxonomy" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">taxonomy</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>report parents and exclude info with “no rank”. Now I report rank == class
but … Question ? Have to be reported taxonomy with rank==“class”?</p>
          

          
          <div class="method-source-code" id="taxonomy-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 354</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">taxonomy</span>
  <span class="ruby-identifier">tax</span> = []
  <span class="ruby-identifier">taxon</span> = <span class="ruby-constant">Taxon</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;taxon_id = ?&quot;</span>,<span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">parent_taxon_id</span>])
  <span class="ruby-keyword">while</span> <span class="ruby-identifier">taxon</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">taxon_id</span> <span class="ruby-operator">!=</span> <span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">parent_taxon_id</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">node_rank!</span>=<span class="ruby-string">'no rank'</span>
    <span class="ruby-identifier">tax</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">taxon_scientific_name</span>.<span class="ruby-identifier">name</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">node_rank!</span>=<span class="ruby-string">'class'</span>
    <span class="ruby-comment">#Note: I don't like this call very much, correct with a relationship in the ref class.</span>
    <span class="ruby-identifier">taxon</span> = <span class="ruby-constant">Taxon</span>.<span class="ruby-identifier">first</span>(<span class="ruby-value">:conditions=</span><span class="ruby-operator">&gt;</span>[<span class="ruby-string">&quot;taxon_id = ?&quot;</span>,<span class="ruby-identifier">taxon</span>.<span class="ruby-identifier">parent_taxon_id</span>])
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">tax</span>.<span class="ruby-identifier">reverse</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- taxonomy-source -->
          
        </div>

        

        
      </div><!-- taxonomy-method -->

    
      <div id="method-i-to_biosequence" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_biosequence</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="to_biosequence-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 439</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_biosequence</span>
  <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">adapter</span>(<span class="ruby-keyword">self</span>,<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">Adapter</span><span class="ruby-operator">::</span><span class="ruby-constant">BioSQL</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_biosequence-source -->
          
        </div>

        

        
      </div><!-- to_biosequence-method -->

    
      <div id="method-i-to_biosql" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_biosql</span><span
            class="method-args">(bs,biodatabase)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="to_biosql-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 114</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_biosql</span>(<span class="ruby-identifier">bs</span>,<span class="ruby-identifier">biodatabase</span>)
  <span class="ruby-comment">#DELETE        #Transcaction works greatly!!!</span>
  <span class="ruby-keyword">begin</span>
    <span class="ruby-comment">#DELETE          Bioentry.transaction do</span>
    <span class="ruby-ivar">@entry</span> = <span class="ruby-identifier">biodatabase</span>.<span class="ruby-identifier">bioentries</span>.<span class="ruby-identifier">build</span>({<span class="ruby-value">:name=</span><span class="ruby-operator">&gt;</span><span class="ruby-identifier">bs</span>.<span class="ruby-identifier">entry_id</span>})

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;primary&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">primary_accession</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">primary_accession</span>

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;def&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">definition</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">definition</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">definition</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;seqver&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">sequence_version</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">sequence_version</span> <span class="ruby-operator">||</span> <span class="ruby-value">0</span>

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;divi&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">division</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">division</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">division</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;identifier&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">identifier</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">other_seqids</span>.<span class="ruby-identifier">collect</span>{<span class="ruby-operator">|</span><span class="ruby-identifier">dblink</span><span class="ruby-operator">|</span> <span class="ruby-node">&quot;#{dblink.database}:#{dblink.id}&quot;</span>}.<span class="ruby-identifier">join</span>(<span class="ruby-string">';'</span>) <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">other_seqids</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">save</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;secacc&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>

    <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">secondary_accessions</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">sa</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;#{sa}&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
      <span class="ruby-comment">#write as qualifier every secondary accession into the array</span>
      <span class="ruby-keyword">self</span>.<span class="ruby-identifier">secondary_accessions</span> = <span class="ruby-identifier">sa</span>
    <span class="ruby-keyword">end</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">secondary_accessions</span>.<span class="ruby-identifier">nil?</span>


    <span class="ruby-comment">#to create the sequence entry needs to exists</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;seq&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">seq</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">seq</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">seq</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;mol&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>

    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">molecule_type</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">molecule_type</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">molecule_type</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;dc&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>

    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">data_class</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">data_class</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">data_class</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;top&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">topology</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">topology</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">topology</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;datec&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">date_created</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">date_created</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">date_created</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;datemod&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">date_modified</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">date_modified</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">date_modified</span>.<span class="ruby-identifier">nil?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;key&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>

    <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">keywords</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">kw</span><span class="ruby-operator">|</span>
      <span class="ruby-comment">#write as qualifier every secondary accessions into the array</span>
      <span class="ruby-keyword">self</span>.<span class="ruby-identifier">keywords</span> = <span class="ruby-identifier">kw</span>
    <span class="ruby-keyword">end</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">keywords</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;spec&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-comment">#self.species = bs.species unless bs.species.nil?</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">species</span> = <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">species</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">species</span>.<span class="ruby-identifier">empty?</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;Debug: #{bs.species}&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;Debug: feat..start&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>

    <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">features</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">feat</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">self</span>.<span class="ruby-identifier">feature</span>=<span class="ruby-identifier">feat</span>
    <span class="ruby-keyword">end</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">features</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;Debug: feat...end&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
    <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">references</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">reference</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">self</span>.<span class="ruby-identifier">reference</span>=<span class="ruby-identifier">reference</span>
    <span class="ruby-keyword">end</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">references</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">comments</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">comment</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">self</span>.<span class="ruby-identifier">comment</span>=<span class="ruby-identifier">comment</span>
    <span class="ruby-keyword">end</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">bs</span>.<span class="ruby-identifier">comments</span>.<span class="ruby-identifier">nil?</span>

    <span class="ruby-comment">#DELETE          end #transaction</span>
    <span class="ruby-keyword">return</span> <span class="ruby-keyword">self</span>
  <span class="ruby-keyword">rescue</span> <span class="ruby-constant">Exception</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">e</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;to_biosql exception: #{e}&quot;</span>
    <span class="ruby-identifier">puts</span> <span class="ruby-identifier">$!</span>
  <span class="ruby-keyword">end</span> <span class="ruby-comment">#rescue</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_biosql-source -->
          
        </div>

        

        
      </div><!-- to_biosql-method -->

    
      <div id="method-i-to_fasta" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_fasta</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="to_fasta-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 431</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_fasta</span>
  <span class="ruby-string">&quot;&gt;&quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">accession</span> <span class="ruby-operator">+</span> <span class="ruby-string">&quot;\n&quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-constant">Regexp</span>.<span class="ruby-identifier">new</span>(<span class="ruby-node">&quot;.{1,#{60}}&quot;</span>), <span class="ruby-string">&quot;\\0\n&quot;</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_fasta-source -->
          
        </div>

        

        
      </div><!-- to_fasta-method -->

    
      <div id="method-i-to_fasta_reverse_complememt" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_fasta_reverse_complememt</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="to_fasta_reverse_complememt-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 435</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_fasta_reverse_complememt</span>
  <span class="ruby-string">&quot;&gt;&quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">accession</span> <span class="ruby-operator">+</span> <span class="ruby-string">&quot;\n&quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">reverse_complement</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-constant">Regexp</span>.<span class="ruby-identifier">new</span>(<span class="ruby-node">&quot;.{1,#{60}}&quot;</span>), <span class="ruby-string">&quot;\\0\n&quot;</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_fasta_reverse_complememt-source -->
          
        </div>

        

        
      </div><!-- to_fasta_reverse_complememt-method -->

    
      <div id="method-i-version" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">version</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="version-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 242</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">version</span>
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">version</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- version-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-sequence_version">sequence_version</a>
        </div>
        

        
      </div><!-- version-method -->

    
      <div id="method-i-version-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">version=</span><span
            class="method-args">(value)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
          <div class="method-source-code" id="version-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/biosql/sequence.rb, line 247</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">version=</span>(<span class="ruby-identifier">value</span>)
  <span class="ruby-ivar">@entry</span>.<span class="ruby-identifier">version</span>=<span class="ruby-identifier">value</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- version-3D-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-sequence_version-3D">sequence_version=</a>
        </div>
        

        
      </div><!-- version-3D-method -->

    
    </section><!-- public-instance-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

