<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />

  <title>File: genes.rb [BioRuby API documentation]</title>

  <link type="text/css" media="screen" href="../../../../rdoc.css" rel="stylesheet" />

  <script src="../../../../js/jquery.js" type="text/javascript"
    charset="utf-8"></script>
  <script src="../../../../js/thickbox-compressed.js" type="text/javascript"
    charset="utf-8"></script>
  <script src="../../../../js/quicksearch.js" type="text/javascript"
    charset="utf-8"></script>
  <script src="../../../../js/darkfish.js" type="text/javascript"
    charset="utf-8"></script>
</head>

<body class="file file-popup">
  <div id="metadata">
    <dl>
      <dt class="modified-date">Last Modified</dt>
      <dd class="modified-date">2010-10-22 11:48:50 +0900</dd>

      
      <dt class="requires">Requires</dt>
      <dd class="requires">
        <ul>
        
          <li>bio/db/kegg/common</li>
        
        </ul>
      </dd>
      

      
    </dl>
  </div>

  <div id="documentation">
    
    <div class="description">
      <h2>Description</h2>
      
<h1>bio/db/kegg/genes.rb - KEGG/GENES database class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2002, 2006, 2010 Toshiaki Katayama
&lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>

<h2>KEGG GENES parser</h2>

<p>See <a
href="http://www.genome.jp/kegg/genes.html">www.genome.jp/kegg/genes.html</a></p>

<h3>Examples</h3>

<pre>require 'bio/io/fetch'
entry_string = Bio::Fetch.query('genes', 'b0002')

entry = Bio::KEGG::GENES.new(entry_string)

# ENTRY
p entry.entry       # =&gt; Hash

p entry.entry_id    # =&gt; String
p entry.division    # =&gt; String
p entry.organism    # =&gt; String

# NAME
p entry.name        # =&gt; String
p entry.names       # =&gt; Array

# DEFINITION
p entry.definition  # =&gt; String
p entry.eclinks     # =&gt; Array

# PATHWAY
p entry.pathway     # =&gt; String
p entry.pathways    # =&gt; Hash

# POSITION
p entry.position    # =&gt; String
p entry.chromosome  # =&gt; String
p entry.gbposition  # =&gt; String
p entry.locations   # =&gt; Bio::Locations

# MOTIF
p entry.motifs      # =&gt; Hash of Array

# DBLINKS
p entry.dblinks     # =&gt; Hash of Array

# STRUCTURE
p entry.structure   # =&gt; Array

# CODON_USAGE
p entry.codon_usage # =&gt; Hash
p entry.cu_list     # =&gt; Array

# AASEQ
p entry.aaseq       # =&gt; Bio::Sequence::AA
p entry.aalen       # =&gt; Fixnum

# NTSEQ
p entry.ntseq       # =&gt; Bio::Sequence::NA
p entry.naseq       # =&gt; Bio::Sequence::NA
p entry.ntlen       # =&gt; Fixnum
p entry.nalen       # =&gt; Fixnum</pre>

    </div>
    
  </div>
</body>
</html>

