<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>class Bio::Sequence::NA - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../../";
  var index_rel_prefix = "../../";
</script>

<script src="../../js/jquery.js"></script>
<script src="../../js/darkfish.js"></script>

<link href="../../css/fonts.css" rel="stylesheet">
<link href="../../css/rdoc.css" rel="stylesheet">



<body id="top" role="document" class="class">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../../table_of_contents.html#pages">Pages</a>
    <a href="../../table_of_contents.html#classes">Classes</a>
    <a href="../../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#class-Bio::Sequence::NA-label-DESCRIPTION">DESCRIPTION</a>
    <li><a href="#class-Bio::Sequence::NA-label-USAGE">USAGE</a>
  </ul>
</div>


  <div id="class-metadata">
    
    <div id="parent-class-section" class="nav-section">
  <h3>Parent</h3>

  
  <p class="link"><a href="../../String.html">String</a>
  
</div>

    <div id="includes-section" class="nav-section">
  <h3>Included Modules</h3>

  <ul class="link-list">
  
  
    <li><a class="include" href="Common.html">Bio::Sequence::Common</a>
  
  
  </ul>
</div>

    
    <!-- Method Quickref -->
<div id="method-list-section" class="nav-section">
  <h3>Methods</h3>

  <ul class="link-list" role="directory">
    
    <li class="calls-super" ><a href="#method-c-new">::new</a>
    
    <li ><a href="#method-c-randomize">::randomize</a>
    
    <li ><a href="#method-i-at_content">#at_content</a>
    
    <li ><a href="#method-i-at_skew">#at_skew</a>
    
    <li ><a href="#method-i-codon_usage">#codon_usage</a>
    
    <li ><a href="#method-i-complement">#complement</a>
    
    <li ><a href="#method-i-complement-21">#complement!</a>
    
    <li ><a href="#method-i-cut_with_enzyme">#cut_with_enzyme</a>
    
    <li ><a href="#method-i-cut_with_enzymes">#cut_with_enzymes</a>
    
    <li ><a href="#method-i-dna">#dna</a>
    
    <li ><a href="#method-i-dna-21">#dna!</a>
    
    <li ><a href="#method-i-forward_complement">#forward_complement</a>
    
    <li ><a href="#method-i-forward_complement-21">#forward_complement!</a>
    
    <li ><a href="#method-i-gc_content">#gc_content</a>
    
    <li ><a href="#method-i-gc_percent">#gc_percent</a>
    
    <li ><a href="#method-i-gc_skew">#gc_skew</a>
    
    <li ><a href="#method-i-illegal_bases">#illegal_bases</a>
    
    <li ><a href="#method-i-molecular_weight">#molecular_weight</a>
    
    <li ><a href="#method-i-names">#names</a>
    
    <li ><a href="#method-i-reverse_complement">#reverse_complement</a>
    
    <li ><a href="#method-i-reverse_complement-21">#reverse_complement!</a>
    
    <li ><a href="#method-i-rna">#rna</a>
    
    <li ><a href="#method-i-rna-21">#rna!</a>
    
    <li ><a href="#method-i-rna-3F">#rna?</a>
    
    <li ><a href="#method-i-to_midi">#to_midi</a>
    
    <li ><a href="#method-i-to_re">#to_re</a>
    
    <li ><a href="#method-i-translate">#translate</a>
    
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-labelledby="class-Bio::Sequence::NA">
  <h1 id="class-Bio::Sequence::NA" class="class">
    class Bio::Sequence::NA
  </h1>

  <section class="description">
    
<h1 id="class-Bio::Sequence::NA-label-DESCRIPTION">DESCRIPTION<span><a href="#class-Bio::Sequence::NA-label-DESCRIPTION">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p><a href="NA.html">Bio::Sequence::NA</a> represents a bare Nucleic Acid
sequence in bioruby.</p>

<h1 id="class-Bio::Sequence::NA-label-USAGE">USAGE<span><a href="#class-Bio::Sequence::NA-label-USAGE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<pre class="ruby"><span class="ruby-comment"># Create a Nucleic Acid sequence.</span>
<span class="ruby-identifier">dna</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;atgcatgcATGCATGCAAAA&#39;</span>)
<span class="ruby-identifier">rna</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;augcaugcaugcaugcaaaa&#39;</span>)

<span class="ruby-comment"># What are the names of all the bases?</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">names</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">rna</span>.<span class="ruby-identifier">names</span>

<span class="ruby-comment"># What is the GC percentage?</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">gc_percent</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">rna</span>.<span class="ruby-identifier">gc_percent</span>

<span class="ruby-comment"># What is the molecular weight?</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">molecular_weight</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">rna</span>.<span class="ruby-identifier">molecular_weight</span>

<span class="ruby-comment"># What is the reverse complement?</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">reverse_complement</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">complement</span>

<span class="ruby-comment"># Is this sequence DNA or RNA?</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">rna?</span>

<span class="ruby-comment"># Translate my sequence (see method docs for many options)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">translate</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">rna</span>.<span class="ruby-identifier">translate</span>
</pre>

  </section>

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    

    

    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Public Class Methods</h3>
       </header>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(str)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Generate an nucleic acid sequence object from a string.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&quot;aagcttggaccgttgaagt&quot;</span>)
</pre>

<p>or maybe (if you have an nucleic acid sequence in a file)</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>:<span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-constant">File</span>.<span class="ruby-identifier">open</span>(<span class="ruby-string">&#39;dna.txt&#39;</span>).<span class="ruby-identifier">read</span>)
</pre>

<p>Nucleic Acid sequences are <strong>always</strong> all lowercase in bioruby</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&quot;AAGcTtGG&quot;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &quot;aagcttgg&quot;</span>
</pre>

<p>Whitespace is stripped from the sequence</p>

<pre class="ruby"><span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&quot;atg\nggg\ttt\r  gc&quot;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &quot;atggggttgc&quot;</span>
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>str</em>: <a href="../../String.html">String</a></p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="NA.html">Bio::Sequence::NA</a> object</p>
</dd></dl>
          
          
            <div class="method-calls-super">
              Calls superclass method
              
            </div>
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 75</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-keyword">super</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">downcase!</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">&quot; \t\n\r&quot;</span>,<span class="ruby-string">&#39;&#39;</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-c-randomize" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">randomize</span><span
            class="method-args">(*arg, &block)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Generate a new random sequence with the given frequency of bases. The
sequence length is determined by their cumulative sum. (See also <a
href="Common.html#method-i-randomize">Bio::Sequence::Common#randomize</a>
which creates a new randomized sequence object using the base composition
of an existing  sequence instance).</p>

<pre class="ruby"><span class="ruby-identifier">counts</span> = {<span class="ruby-string">&#39;a&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">1</span>,<span class="ruby-string">&#39;c&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">2</span>,<span class="ruby-string">&#39;g&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">3</span>,<span class="ruby-string">&#39;t&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">4</span>}
<span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">randomize</span>(<span class="ruby-identifier">counts</span>)  <span class="ruby-comment">#=&gt; &quot;ggcttgttac&quot; (for example)</span>
</pre>

<p>You may also feed the output of randomize into a block</p>

<pre class="ruby"><span class="ruby-identifier">actual_counts</span> = {<span class="ruby-string">&#39;a&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">0</span>, <span class="ruby-string">&#39;c&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">0</span>, <span class="ruby-string">&#39;g&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">0</span>, <span class="ruby-string">&#39;t&#39;</span>=<span class="ruby-operator">&gt;</span><span class="ruby-value">0</span>}
<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">randomize</span>(<span class="ruby-identifier">counts</span>) {<span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span> <span class="ruby-identifier">actual_counts</span>[<span class="ruby-identifier">x</span>] <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>}
<span class="ruby-identifier">actual_counts</span>                     <span class="ruby-comment">#=&gt; {&quot;a&quot;=&gt;1, &quot;c&quot;=&gt;2, &quot;g&quot;=&gt;3, &quot;t&quot;=&gt;4}</span>
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>hash</em>: <a href="../../Hash.html">Hash</a> object</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="NA.html">Bio::Sequence::NA</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="randomize-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/compat.rb, line 82</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">randomize</span>(<span class="ruby-operator">*</span><span class="ruby-identifier">arg</span>, <span class="ruby-operator">&amp;</span><span class="ruby-identifier">block</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;&#39;</span>).<span class="ruby-identifier">randomize</span>(<span class="ruby-operator">*</span><span class="ruby-identifier">arg</span>, <span class="ruby-operator">&amp;</span><span class="ruby-identifier">block</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
    </section>
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Public Instance Methods</h3>
       </header>

    
      <div id="method-i-at_content" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">at_content</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Calculate the ratio of AT / ATGC bases. U is regarded as T.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">at_content</span>                       <span class="ruby-comment">#=&gt; 4/9</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">at_content</span>.<span class="ruby-identifier">to_f</span>                  <span class="ruby-comment">#=&gt; 0.444444444444444</span>
</pre>

<p>In older Ruby versions, Float is always returned.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">at_content</span>                       <span class="ruby-comment">#=&gt; 0.444444444444444</span>
</pre>

<p>Note that “u” is regarded as “t”. If there are no ATGC bases in the
sequence, 0.0 is returned.</p>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Rational or Float</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="at_content-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 346</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">at_content</span>
  <span class="ruby-identifier">count</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">composition</span>
  <span class="ruby-identifier">at</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;a&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;t&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;u&#39;</span>]
  <span class="ruby-identifier">gc</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;g&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;c&#39;</span>]
  <span class="ruby-identifier">total</span> = <span class="ruby-identifier">at</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">gc</span>
  <span class="ruby-keyword">return</span> <span class="ruby-value">0.0</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">total</span> <span class="ruby-operator">==</span> <span class="ruby-value">0</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">at</span>.<span class="ruby-identifier">quo</span>(<span class="ruby-identifier">total</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-at_skew" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">at_skew</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Calculate the ratio of (A - T) / (A + T) bases. U is regarded as T.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgttgttgttc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">at_skew</span>                          <span class="ruby-comment">#=&gt; (-3/4)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">at_skew</span>.<span class="ruby-identifier">to_f</span>                     <span class="ruby-comment">#=&gt; -0.75</span>
</pre>

<p>In older Ruby versions, Float is always returned.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgttgttgttc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">at_skew</span>                          <span class="ruby-comment">#=&gt; -0.75</span>
</pre>

<p>Note that “u” is regarded as “t”. If there are no AT bases in the sequence,
0.0 is returned.</p>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Rational or Float</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="at_skew-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 395</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">at_skew</span>
  <span class="ruby-identifier">count</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">composition</span>
  <span class="ruby-identifier">a</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;a&#39;</span>]
  <span class="ruby-identifier">t</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;t&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;u&#39;</span>]
  <span class="ruby-identifier">at</span> = <span class="ruby-identifier">a</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">t</span>
  <span class="ruby-keyword">return</span> <span class="ruby-value">0.0</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">at</span> <span class="ruby-operator">==</span> <span class="ruby-value">0</span>
  <span class="ruby-keyword">return</span> (<span class="ruby-identifier">a</span> <span class="ruby-operator">-</span> <span class="ruby-identifier">t</span>).<span class="ruby-identifier">quo</span>(<span class="ruby-identifier">at</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-codon_usage" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">codon_usage</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Returns counts of each codon in the sequence in a hash.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">codon_usage</span>                <span class="ruby-comment">#=&gt; {&quot;gcg&quot;=&gt;1, &quot;tga&quot;=&gt;1, &quot;atg&quot;=&gt;1}</span>
</pre>

<p>This method does not validate codons!  Any three letter group is a
&#39;codon&#39;. So,</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggNNtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">codon_usage</span>                <span class="ruby-comment">#=&gt; {&quot;tga&quot;=&gt;1, &quot;gnn&quot;=&gt;1, &quot;atg&quot;=&gt;1}</span>

<span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgg--tga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">codon_usage</span>                <span class="ruby-comment">#=&gt; {&quot;tga&quot;=&gt;1, &quot;g--&quot;=&gt;1, &quot;atg&quot;=&gt;1}</span>
</pre>

<p>Also, there is no option to work in any frame other than the first.</p>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../../Hash.html">Hash</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="codon_usage-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 273</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">codon_usage</span>
  <span class="ruby-identifier">hash</span> = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>(<span class="ruby-value">0</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">window_search</span>(<span class="ruby-value">3</span>, <span class="ruby-value">3</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">codon</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">codon</span>] <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">hash</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-complement" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">complement</span><span
            class="method-args">()</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Alias for <a
href="NA.html#method-i-reverse_complement">#reverse_complement</a></p>
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="NA.html#method-i-reverse_complement">reverse_complement</a>
        </div>
        
      </div>

    
      <div id="method-i-complement-21" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">complement!</span><span
            class="method-args">()</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Alias for <a
href="NA.html#method-i-reverse_complement-21">#reverse_complement!</a></p>
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="NA.html#method-i-reverse_complement-21">reverse_complement!</a>
        </div>
        
      </div>

    
      <div id="method-i-cut_with_enzyme" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cut_with_enzyme</span><span
            class="method-args">(*args)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Example:</p>

<pre class="ruby"><span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;gaattc&#39;</span>)
<span class="ruby-identifier">cuts</span> = <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">cut_with_enzyme</span>(<span class="ruby-string">&#39;EcoRI&#39;</span>)
</pre>

<p><em>or</em></p>

<pre class="ruby"><span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;gaattc&#39;</span>)
<span class="ruby-identifier">cuts</span> = <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">cut_with_enzyme</span>(<span class="ruby-string">&#39;g^aattc&#39;</span>)
</pre>
<hr>

<p>See <a
href="../RestrictionEnzyme/Analysis.html#method-c-cut">Bio::RestrictionEnzyme::Analysis.cut</a></p>
          
          

          
          <div class="method-source-code" id="cut_with_enzyme-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 530</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cut_with_enzyme</span>(<span class="ruby-operator">*</span><span class="ruby-identifier">args</span>)
  <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Analysis</span>.<span class="ruby-identifier">cut</span>(<span class="ruby-keyword">self</span>, <span class="ruby-operator">*</span><span class="ruby-identifier">args</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="NA.html#method-i-cut_with_enzymes">cut_with_enzymes</a>
        </div>
        

        
      </div>

    
      <div id="method-i-cut_with_enzymes" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">cut_with_enzymes</span><span
            class="method-args">(*args)</span>
          
        </div>
        

        <div class="method-description">
          
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="NA.html#method-i-cut_with_enzyme">cut_with_enzyme</a>
        </div>
        
      </div>

    
      <div id="method-i-dna" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dna</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Returns a new sequence object with any &#39;u&#39; bases changed to
&#39;t&#39;. The original sequence is not modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;augc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">dna</span>                              <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;augc&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>new <a href="NA.html">Bio::Sequence::NA</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="dna-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 474</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dna</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr</span>(<span class="ruby-string">&#39;u&#39;</span>, <span class="ruby-string">&#39;t&#39;</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-dna-21" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dna!</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Changes any &#39;u&#39; bases in the original sequence to &#39;t&#39;. The
original sequence is modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;augc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">dna!</span>                             <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>current <a href="NA.html">Bio::Sequence::NA</a> object (modified)</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="dna-21-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 486</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dna!</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">&#39;u&#39;</span>, <span class="ruby-string">&#39;t&#39;</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-forward_complement" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">forward_complement</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Returns a new complementary sequence object (without reversing). The
original sequence object is not modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">forward_complement</span>               <span class="ruby-comment">#=&gt; &#39;tacg&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>new <a href="NA.html">Bio::Sequence::NA</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="forward_complement-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 100</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">forward_complement</span>
  <span class="ruby-identifier">s</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">class</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">self</span>)
  <span class="ruby-identifier">s</span>.<span class="ruby-identifier">forward_complement!</span>
  <span class="ruby-identifier">s</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-forward_complement-21" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">forward_complement!</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Converts the current sequence into its complement (without reversing). The
original sequence object is modified.</p>

<pre class="ruby"><span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">forward_complement!</span>              <span class="ruby-comment">#=&gt; &#39;tacg&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;tacg&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>current <a href="NA.html">Bio::Sequence::NA</a> object (modified)</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="forward_complement-21-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 114</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">forward_complement!</span>
  <span class="ruby-keyword">if</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">rna?</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">&#39;augcrymkdhvbswn&#39;</span>, <span class="ruby-string">&#39;uacgyrkmhdbvswn&#39;</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">&#39;atgcrymkdhvbswn&#39;</span>, <span class="ruby-string">&#39;tacgyrkmhdbvswn&#39;</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">self</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-gc_content" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">gc_content</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Calculate the ratio of GC / ATGC bases. U is regarded as T.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_content</span>                       <span class="ruby-comment">#=&gt; (5/9)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_content</span>.<span class="ruby-identifier">to_f</span>                  <span class="ruby-comment">#=&gt; 0.5555555555555556</span>
</pre>

<p>In older Ruby versions, Float is always returned.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_content</span>                       <span class="ruby-comment">#=&gt; 0.555555555555556</span>
</pre>

<p>Note that “u” is regarded as “t”. If there are no ATGC bases in the
sequence, 0.0 is returned.</p>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Rational or Float</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="gc_content-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 321</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gc_content</span>
  <span class="ruby-identifier">count</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">composition</span>
  <span class="ruby-identifier">at</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;a&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;t&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;u&#39;</span>]
  <span class="ruby-identifier">gc</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;g&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;c&#39;</span>]
  <span class="ruby-identifier">total</span> = <span class="ruby-identifier">at</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">gc</span>
  <span class="ruby-keyword">return</span> <span class="ruby-value">0.0</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">total</span> <span class="ruby-operator">==</span> <span class="ruby-value">0</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">gc</span>.<span class="ruby-identifier">quo</span>(<span class="ruby-identifier">total</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-gc_percent" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">gc_percent</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Calculate the ratio of GC / ATGC bases as a percentage rounded to  the
nearest whole number. U is regarded as T.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_percent</span>                       <span class="ruby-comment">#=&gt; 55</span>
</pre>

<p>Note that this method only returns an integer value. When more digits after
decimal points are needed, use <a
href="NA.html#method-i-gc_content">#gc_content</a> and sprintf like below:</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">sprintf</span>(<span class="ruby-string">&quot;%3.2f&quot;</span>, <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_content</span> <span class="ruby-operator">*</span> <span class="ruby-value">100</span>)  <span class="ruby-comment">#=&gt; &quot;55.56&quot;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Fixnum</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="gc_percent-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 296</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gc_percent</span>
  <span class="ruby-identifier">count</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">composition</span>
  <span class="ruby-identifier">at</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;a&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;t&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;u&#39;</span>]
  <span class="ruby-identifier">gc</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;g&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;c&#39;</span>]
  <span class="ruby-keyword">return</span> <span class="ruby-value">0</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">at</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">gc</span> <span class="ruby-operator">==</span> <span class="ruby-value">0</span>
  <span class="ruby-identifier">gc</span> = <span class="ruby-value">100</span> <span class="ruby-operator">*</span> <span class="ruby-identifier">gc</span> <span class="ruby-operator">/</span> (<span class="ruby-identifier">at</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">gc</span>)
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">gc</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-gc_skew" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">gc_skew</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Calculate the ratio of (G - C) / (G + C) bases.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_skew</span>                          <span class="ruby-comment">#=&gt; 3/5</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_skew</span>.<span class="ruby-identifier">to_f</span>                     <span class="ruby-comment">#=&gt; 0.6</span>
</pre>

<p>In older Ruby versions, Float is always returned.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">gc_skew</span>                          <span class="ruby-comment">#=&gt; 0.6</span>
</pre>

<p>If there are no GC bases in the sequence, 0.0 is returned.</p>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Rational or Float</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="gc_skew-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 370</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gc_skew</span>
  <span class="ruby-identifier">count</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">composition</span>
  <span class="ruby-identifier">g</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;g&#39;</span>]
  <span class="ruby-identifier">c</span> = <span class="ruby-identifier">count</span>[<span class="ruby-string">&#39;c&#39;</span>]
  <span class="ruby-identifier">gc</span> = <span class="ruby-identifier">g</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">c</span>
  <span class="ruby-keyword">return</span> <span class="ruby-value">0.0</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">gc</span> <span class="ruby-operator">==</span> <span class="ruby-value">0</span>
  <span class="ruby-keyword">return</span> (<span class="ruby-identifier">g</span> <span class="ruby-operator">-</span> <span class="ruby-identifier">c</span>).<span class="ruby-identifier">quo</span>(<span class="ruby-identifier">gc</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-illegal_bases" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">illegal_bases</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Returns an alphabetically sorted array of any non-standard bases  (other
than &#39;atgcu&#39;).</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgStgQccR&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">illegal_bases</span>                    <span class="ruby-comment">#=&gt; [&quot;q&quot;, &quot;r&quot;, &quot;s&quot;]</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Array object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="illegal_bases-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 411</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">illegal_bases</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">/[^atgcu]/</span>).<span class="ruby-identifier">sort</span>.<span class="ruby-identifier">uniq</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-molecular_weight" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">molecular_weight</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Estimate molecular weight (using the values from BioPerl&#39;s  <a
href="http://doc.bioperl.org/releases/bioperl-1.0.1/Bio/Tools/SeqStats.html">SeqStats.pm</a>
module).</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">molecular_weight</span>                 <span class="ruby-comment">#=&gt; 2841.00708</span>
</pre>

<p>RNA and DNA do not have the same molecular weights,</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;auggcguga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">molecular_weight</span>                 <span class="ruby-comment">#=&gt; 2956.94708</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Float object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="molecular_weight-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 427</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">molecular_weight</span>
  <span class="ruby-keyword">if</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">rna?</span>
    <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-keyword">self</span>, <span class="ruby-keyword">true</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-keyword">self</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-names" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">names</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Generate the list of the names of each nucleotide along with the sequence
(full name).  Names used in bioruby are found in the Bio::AminoAcid::NAMES
hash.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atg&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">names</span>                    <span class="ruby-comment">#=&gt; [&quot;Adenine&quot;, &quot;Thymine&quot;, &quot;Guanine&quot;]</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Array object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="names-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 458</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">names</span>
  <span class="ruby-identifier">array</span> = []
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">each_byte</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">array</span>.<span class="ruby-identifier">push</span>(<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">names</span>[<span class="ruby-identifier">x</span>.<span class="ruby-identifier">chr</span>.<span class="ruby-identifier">upcase</span>])
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">array</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-reverse_complement" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">reverse_complement</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Returns a new sequence object with the reverse complement  sequence to the
original.  The original sequence is not modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">reverse_complement</span>               <span class="ruby-comment">#=&gt; &#39;gcat&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>new <a href="NA.html">Bio::Sequence::NA</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="reverse_complement-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 131</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">reverse_complement</span>
  <span class="ruby-identifier">s</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">class</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">self</span>)
  <span class="ruby-identifier">s</span>.<span class="ruby-identifier">reverse_complement!</span>
  <span class="ruby-identifier">s</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="NA.html#method-i-complement">complement</a>
        </div>
        

        
      </div>

    
      <div id="method-i-reverse_complement-21" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">reverse_complement!</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Converts the original sequence into its reverse complement.</p>

<p>The original sequence is modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">reverse_complement</span>               <span class="ruby-comment">#=&gt; &#39;gcat&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;gcat&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>current <a href="NA.html">Bio::Sequence::NA</a> object (modified)</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="reverse_complement-21-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 145</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">reverse_complement!</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">reverse!</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">forward_complement!</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="NA.html#method-i-complement-21">complement!</a>
        </div>
        

        
      </div>

    
      <div id="method-i-rna" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">rna</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Returns a new sequence object with any &#39;t&#39; bases changed to
&#39;u&#39;. The original sequence is not modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">dna</span>                              <span class="ruby-comment">#=&gt; &#39;augc&#39;  </span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>new <a href="NA.html">Bio::Sequence::NA</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="rna-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 498</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">rna</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr</span>(<span class="ruby-string">&#39;t&#39;</span>, <span class="ruby-string">&#39;u&#39;</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-rna-21" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">rna!</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Changes any &#39;t&#39; bases in the original sequence to &#39;u&#39;. The
original sequence is modified.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">dna!</span>                             <span class="ruby-comment">#=&gt; &#39;augc&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;augc&#39;</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>current <a href="NA.html">Bio::Sequence::NA</a> object (modified)</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="rna-21-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 510</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">rna!</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">&#39;t&#39;</span>, <span class="ruby-string">&#39;u&#39;</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-to_midi" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_midi</span><span
            class="method-args">(style = {}, drum = true)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>style:</p>

<pre class="ruby"><span class="ruby-constant">Hash</span> <span class="ruby-identifier">of</span> :<span class="ruby-identifier">tempo</span>, :<span class="ruby-identifier">scale</span>, :<span class="ruby-identifier">tones</span>
</pre>

<p>scale:</p>

<pre>C  C# D  D# E  F  F# G  G# A  A#  B
0  1  2  3  4  5  6  7  8  9  10  11</pre>

<p>tones:</p>

<pre>Hash of :prog, :base, :range -- tone, vol? or len?, octaves</pre>

<p>drum:</p>

<pre>true (with rhythm part), false (without rhythm part)</pre>
          
          

          
          <div class="method-source-code" id="to_midi-source">
            <pre><span class="ruby-comment"># File lib/bio/shell/plugin/midi.rb, line 351</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_midi</span>(<span class="ruby-identifier">style</span> = {}, <span class="ruby-identifier">drum</span> = <span class="ruby-keyword">true</span>)
  <span class="ruby-identifier">default</span> = <span class="ruby-constant">MidiTrack</span><span class="ruby-operator">::</span><span class="ruby-constant">Styles</span>[<span class="ruby-string">&quot;Ichinose&quot;</span>]
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">style</span>.<span class="ruby-identifier">is_a?</span>(<span class="ruby-constant">String</span>)
    <span class="ruby-identifier">style</span> = <span class="ruby-constant">MidiTrack</span><span class="ruby-operator">::</span><span class="ruby-constant">Styles</span>[<span class="ruby-identifier">style</span>] <span class="ruby-operator">||</span> <span class="ruby-identifier">default</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">tempo</span> = <span class="ruby-identifier">style</span>[<span class="ruby-value">:tempo</span>] <span class="ruby-operator">||</span> <span class="ruby-identifier">default</span>[<span class="ruby-value">:tempo</span>]
  <span class="ruby-identifier">scale</span> = <span class="ruby-identifier">style</span>[<span class="ruby-value">:scale</span>] <span class="ruby-operator">||</span> <span class="ruby-identifier">default</span>[<span class="ruby-value">:scale</span>]
  <span class="ruby-identifier">tones</span> = <span class="ruby-identifier">style</span>[<span class="ruby-value">:tones</span>] <span class="ruby-operator">||</span> <span class="ruby-identifier">default</span>[<span class="ruby-value">:tones</span>]

  <span class="ruby-identifier">track</span> = []

  <span class="ruby-identifier">tones</span>.<span class="ruby-identifier">each_with_index</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">tone</span>, <span class="ruby-identifier">i</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">ch</span> = <span class="ruby-identifier">i</span>
    <span class="ruby-identifier">ch</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">i</span> <span class="ruby-operator">&gt;=</span> <span class="ruby-value">9</span>         <span class="ruby-comment"># skip rythm track</span>
    <span class="ruby-identifier">track</span>.<span class="ruby-identifier">push</span> <span class="ruby-constant">MidiTrack</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">ch</span>, <span class="ruby-identifier">tone</span>[<span class="ruby-value">:prog</span>], <span class="ruby-identifier">tone</span>[<span class="ruby-value">:base</span>], <span class="ruby-identifier">tone</span>[<span class="ruby-value">:range</span>], <span class="ruby-identifier">scale</span>)
  <span class="ruby-keyword">end</span>

  <span class="ruby-keyword">if</span> <span class="ruby-identifier">drum</span>
    <span class="ruby-identifier">rhythm</span> = [<span class="ruby-value">0</span>, <span class="ruby-value">1</span>, <span class="ruby-value">2</span>, <span class="ruby-value">3</span>, <span class="ruby-value">4</span>, <span class="ruby-value">5</span>, <span class="ruby-value">6</span>, <span class="ruby-value">7</span>, <span class="ruby-value">8</span>, <span class="ruby-value">9</span>, <span class="ruby-value">10</span>, <span class="ruby-value">11</span>]
    <span class="ruby-identifier">track</span>.<span class="ruby-identifier">push</span>(<span class="ruby-constant">MidiTrack</span>.<span class="ruby-identifier">new</span>(<span class="ruby-value">9</span>, <span class="ruby-value">0</span>, <span class="ruby-value">35</span>, <span class="ruby-value">2</span>, <span class="ruby-identifier">rhythm</span>))
  <span class="ruby-keyword">end</span>

  <span class="ruby-identifier">cur</span> = <span class="ruby-value">0</span>
  <span class="ruby-identifier">window_search</span>(<span class="ruby-value">4</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">s</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">track</span>[<span class="ruby-identifier">cur</span> <span class="ruby-operator">%</span> <span class="ruby-identifier">track</span>.<span class="ruby-identifier">length</span>].<span class="ruby-identifier">push</span>(<span class="ruby-identifier">s</span>)
    <span class="ruby-identifier">cur</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-identifier">track</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">t</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">t</span>.<span class="ruby-identifier">push_silent</span>(<span class="ruby-value">12</span>)
  <span class="ruby-keyword">end</span>

  <span class="ruby-identifier">ans</span> = <span class="ruby-identifier">track</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">header</span>(<span class="ruby-identifier">track</span>.<span class="ruby-identifier">length</span>, <span class="ruby-identifier">tempo</span>)
  <span class="ruby-identifier">track</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">t</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">ans</span> <span class="ruby-operator">+=</span> <span class="ruby-identifier">t</span>.<span class="ruby-identifier">encode</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">ans</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-to_re" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_re</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Create a ruby regular expression instance  (<a
href="http://corelib.rubyonrails.org/classes/Regexp.html">Regexp)</a></p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">to_re</span>                            <span class="ruby-comment">#=&gt; /atggcgtga/</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Regexp object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="to_re-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 442</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_re</span>
  <span class="ruby-keyword">if</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">rna?</span>
    <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">to_re</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">dna</span>, <span class="ruby-keyword">true</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">to_re</span>(<span class="ruby-keyword">self</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-translate" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">translate</span><span
            class="method-args">(frame = 1, table = 1, unknown = 'X')</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Translate into an amino acid sequence.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atggcgtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>                        <span class="ruby-comment">#=&gt; &quot;MA*&quot;</span>
</pre>

<p>By default, translate starts in reading frame position 1, but you can start
in either 2 or 3 as well,</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">2</span>)                     <span class="ruby-comment">#=&gt; &quot;WR&quot;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">3</span>)                     <span class="ruby-comment">#=&gt; &quot;GV&quot;</span>
</pre>

<p>You may also translate the reverse complement in one step by using frame
values of -1, -2, and -3 (or 4, 5, and 6)</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">-1</span>)                    <span class="ruby-comment">#=&gt; &quot;SRH&quot;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">4</span>)                     <span class="ruby-comment">#=&gt; &quot;SRH&quot;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">reverse_complement</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">1</span>)  <span class="ruby-comment">#=&gt; &quot;SRH&quot;</span>
</pre>

<p>The default codon table in the translate function is the Standard
Eukaryotic codon table.  The translate function takes either a  number or a
<a href="../CodonTable.html">Bio::CodonTable</a> object for its table
argument.  The available tables are  (<a
href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=t">NCBI</a>):</p>

<pre>1. &quot;Standard (Eukaryote)&quot;
2. &quot;Vertebrate Mitochondrial&quot;
3. &quot;Yeast Mitochondorial&quot;
4. &quot;Mold, Protozoan, Coelenterate Mitochondrial and Mycoplasma/Spiroplasma&quot;
5. &quot;Invertebrate Mitochondrial&quot;
6. &quot;Ciliate Macronuclear and Dasycladacean&quot;
9. &quot;Echinoderm Mitochondrial&quot;
10. &quot;Euplotid Nuclear&quot;
11. &quot;Bacteria&quot;
12. &quot;Alternative Yeast Nuclear&quot;
13. &quot;Ascidian Mitochondrial&quot;
14. &quot;Flatworm Mitochondrial&quot;
15. &quot;Blepharisma Macronuclear&quot;
16. &quot;Chlorophycean Mitochondrial&quot;
21. &quot;Trematode Mitochondrial&quot;
22. &quot;Scenedesmus obliquus mitochondrial&quot;
23. &quot;Thraustochytrium Mitochondrial&quot;</pre>

<p>If you are using anything other than the default table, you must specify 
frame in the translate method call,</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>                <span class="ruby-comment">#=&gt; &quot;MA*&quot;  (using defaults)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">1</span>,<span class="ruby-value">1</span>)           <span class="ruby-comment">#=&gt; &quot;MA*&quot;  (same as above, but explicit)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">1</span>,<span class="ruby-value">2</span>)           <span class="ruby-comment">#=&gt; &quot;MAW&quot;  (different codon table)</span>
</pre>

<p>and using a <a href="../CodonTable.html">Bio::CodonTable</a> instance in
the translate method call,</p>

<pre class="ruby"><span class="ruby-identifier">mt_table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-value">2</span>]
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">1</span>, <span class="ruby-identifier">mt_table</span>)           <span class="ruby-comment">#=&gt; &quot;MAW&quot;</span>
</pre>

<p>By default, any invalid or unknown codons (as could happen if the  sequence
contains ambiguities) will be represented by &#39;X&#39; in the  translated
sequence.  You may change this to any character of your choice.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgcNNtga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>                        <span class="ruby-comment">#=&gt; &quot;MX*&quot;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>(<span class="ruby-value">1</span>,<span class="ruby-value">1</span>,<span class="ruby-string">&#39;9&#39;</span>)               <span class="ruby-comment">#=&gt; &quot;M9*&quot;</span>
</pre>

<p>The translate method considers gaps to be unknown characters and treats 
them as such (i.e. does not collapse sequences prior to translation), so</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc--tga&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">translate</span>                        <span class="ruby-comment">#=&gt; &quot;MX*&quot;</span>
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>frame</em>:  one of 1,2,3,4,5,6,-1,-2,-3 (default 1)</p>
</li><li>
<p>(optional) <em>table</em>: Fixnum in range 1,23 or <a
href="../CodonTable.html">Bio::CodonTable</a> object (default 1)</p>
</li><li>
<p>(optional) <em>unknown</em>: Character (default &#39;X&#39;)</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="AA.html">Bio::Sequence::AA</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="translate-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 232</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">translate</span>(<span class="ruby-identifier">frame</span> = <span class="ruby-value">1</span>, <span class="ruby-identifier">table</span> = <span class="ruby-value">1</span>, <span class="ruby-identifier">unknown</span> = <span class="ruby-string">&#39;X&#39;</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">table</span>.<span class="ruby-identifier">is_a?</span>(<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>)
    <span class="ruby-identifier">ct</span> = <span class="ruby-identifier">table</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">ct</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-identifier">table</span>]
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">naseq</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">dna</span>
  <span class="ruby-keyword">case</span> <span class="ruby-identifier">frame</span>
  <span class="ruby-keyword">when</span> <span class="ruby-value">1</span>, <span class="ruby-value">2</span>, <span class="ruby-value">3</span>
    <span class="ruby-identifier">from</span> = <span class="ruby-identifier">frame</span> <span class="ruby-operator">-</span> <span class="ruby-value">1</span>
  <span class="ruby-keyword">when</span> <span class="ruby-value">4</span>, <span class="ruby-value">5</span>, <span class="ruby-value">6</span>
    <span class="ruby-identifier">from</span> = <span class="ruby-identifier">frame</span> <span class="ruby-operator">-</span> <span class="ruby-value">4</span>
    <span class="ruby-identifier">naseq</span>.<span class="ruby-identifier">complement!</span>
  <span class="ruby-keyword">when</span> <span class="ruby-value">-1</span>, <span class="ruby-value">-2</span>, <span class="ruby-value">-3</span>
    <span class="ruby-identifier">from</span> = <span class="ruby-value">-1</span> <span class="ruby-operator">-</span> <span class="ruby-identifier">frame</span>
    <span class="ruby-identifier">naseq</span>.<span class="ruby-identifier">complement!</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">from</span> = <span class="ruby-value">0</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">nalen</span> = <span class="ruby-identifier">naseq</span>.<span class="ruby-identifier">length</span> <span class="ruby-operator">-</span> <span class="ruby-identifier">from</span>
  <span class="ruby-identifier">nalen</span> <span class="ruby-operator">-=</span> <span class="ruby-identifier">nalen</span> <span class="ruby-operator">%</span> <span class="ruby-value">3</span>
  <span class="ruby-identifier">aaseq</span> = <span class="ruby-identifier">naseq</span>[<span class="ruby-identifier">from</span>, <span class="ruby-identifier">nalen</span>].<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">/.{3}/</span>) {<span class="ruby-operator">|</span><span class="ruby-identifier">codon</span><span class="ruby-operator">|</span> <span class="ruby-identifier">ct</span>[<span class="ruby-identifier">codon</span>] <span class="ruby-keyword">or</span> <span class="ruby-identifier">unknown</span>}
  <span class="ruby-keyword">return</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">AA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">aaseq</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
    </section>
  
     <section id="protected-instance-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Protected Instance Methods</h3>
       </header>

    
      <div id="method-i-rna-3F" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">rna?</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          
          
          

          
          <div class="method-source-code" id="rna-3F-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/na.rb, line 514</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">rna?</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">index</span>(<span class="ruby-string">&#39;u&#39;</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
    </section>
  
  </section>
</main>


<footer id="validator-badges" role="contentinfo">
  <p><a href="http://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://rdoc.github.io/rdoc">RDoc</a> 5.0.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

