<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>class Bio::PAML::Baseml - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../../";
  var index_rel_prefix = "../../";
</script>

<script src="../../js/jquery.js"></script>
<script src="../../js/darkfish.js"></script>

<link href="../../css/fonts.css" rel="stylesheet">
<link href="../../css/rdoc.css" rel="stylesheet">



<body id="top" role="document" class="class">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../../table_of_contents.html#pages">Pages</a>
    <a href="../../table_of_contents.html#classes">Classes</a>
    <a href="../../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#class-Bio::PAML::Baseml-label-Description">Description</a>
    <li><a href="#class-Bio::PAML::Baseml-label-Examples">Examples</a>
  </ul>
</div>


  <div id="class-metadata">
    
    <div id="parent-class-section" class="nav-section">
  <h3>Parent</h3>

  
  <p class="link">Common
  
</div>

    
    
    
  </div>
</nav>

<main role="main" aria-labelledby="class-Bio::PAML::Baseml">
  <h1 id="class-Bio::PAML::Baseml" class="class">
    class Bio::PAML::Baseml
  </h1>

  <section class="description">
    
<h2 id="class-Bio::PAML::Baseml-label-Description">Description<span><a href="#class-Bio::PAML::Baseml-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p><a href="Baseml.html">Bio::PAML::Baseml</a> is a wrapper for running <a
href="../PAML.html">PAML</a> baseml program.</p>

<p>Because most of the methods in this class are inherited from <a
href="Common.html">Bio::PAML::Common</a>, see documents of <a
href="Common.html">Bio::PAML::Common</a> for details.</p>

<h2 id="class-Bio::PAML::Baseml-label-Examples">Examples<span><a href="#class-Bio::PAML::Baseml-label-Examples">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Example 1:</p>

<pre class="ruby"><span class="ruby-identifier">require</span> <span class="ruby-string">&#39;bio&#39;</span>
<span class="ruby-comment"># Reads multi-fasta formatted file and gets a Bio::Alignment object.</span>
<span class="ruby-identifier">alignment</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">FlatFile</span>.<span class="ruby-identifier">open</span>(<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Alignment</span><span class="ruby-operator">::</span><span class="ruby-constant">MultiFastaFormat</span>,
                               <span class="ruby-string">&#39;example.fst&#39;</span>).<span class="ruby-identifier">alignment</span>
<span class="ruby-comment"># Reads newick tree from a file</span>
<span class="ruby-identifier">tree</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">FlatFile</span>.<span class="ruby-identifier">open</span>(<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Newick</span>, <span class="ruby-string">&#39;example.tree&#39;</span>).<span class="ruby-identifier">tree</span>
<span class="ruby-comment"># Creates a Baseml object</span>
<span class="ruby-identifier">baseml</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">PAML</span><span class="ruby-operator">::</span><span class="ruby-constant">Baseml</span>.<span class="ruby-identifier">new</span>
<span class="ruby-comment"># Sets parameters</span>
<span class="ruby-identifier">baseml</span>.<span class="ruby-identifier">parameters</span>[:<span class="ruby-identifier">runmode</span>] = <span class="ruby-value">0</span>
<span class="ruby-identifier">baseml</span>.<span class="ruby-identifier">parameters</span>[:<span class="ruby-constant">RateAncestor</span>] = <span class="ruby-value">1</span>
<span class="ruby-comment"># You can also set many parameters at a time.</span>
<span class="ruby-identifier">baseml</span>.<span class="ruby-identifier">parameters</span>.<span class="ruby-identifier">update</span>({ :<span class="ruby-identifier">alpha</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">0.5</span>, :<span class="ruby-identifier">fix_alpha</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">0</span> })
<span class="ruby-comment"># Executes baseml with the alignment and the tree</span>
<span class="ruby-identifier">report</span> = <span class="ruby-identifier">baseml</span>.<span class="ruby-identifier">query</span>(<span class="ruby-identifier">alignment</span>, <span class="ruby-identifier">tree</span>)
</pre>

  </section>

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    
    <section class="constants-list">
      <header>
        <h3>Constants</h3>
      </header>
      <dl>
      
        <dt id="DEFAULT_PARAMETERS">DEFAULT_PARAMETERS
        
        <dd><p>Default parameters when running baseml.</p>

<p>The parameters whose values are different from the baseml defalut value
(described in pamlDOC.pdf) in <a href="../PAML.html">PAML</a> 4.1 are:</p>

<pre>seqfile, outfile, treefile, ndata, noisy, verbose</pre>
        
      
        <dt id="DEFAULT_PROGRAM">DEFAULT_PROGRAM
        
        <dd><p>Default program name</p>
        
      
      </dl>
    </section>
    

    

    
  </section>
</main>


<footer id="validator-badges" role="contentinfo">
  <p><a href="http://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://rdoc.github.io/rdoc">RDoc</a> 5.0.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

