<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>Changes-1.3 - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
  var index_rel_prefix = "../";
</script>

<script src="../js/navigation.js" defer></script>
<script src="../js/search.js" defer></script>
<script src="../js/search_index.js" defer></script>
<script src="../js/searcher.js" defer></script>
<script src="../js/darkfish.js" defer></script>

<link href="../css/fonts.css" rel="stylesheet">
<link href="../css/rdoc.css" rel="stylesheet">


<body id="top" role="document" class="file">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../table_of_contents.html#pages">Pages</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#label-Incompatible+and+important+changes+since+the+BioRuby+1.2.1+release">Incompatible and important changes since the BioRuby 1.2.1 release</a>
    <li><a href="#label-New+features">New features</a>
    <li><a href="#label-Support+for+sequence+output+with+improvements+of+Bio-3A-3ASequence">Support for sequence output with improvements of Bio::Sequence</a>
    <li><a href="#label-BioSQL+support">BioSQL support</a>
    <li><a href="#label-Bio-3A-3ABlast">Bio::Blast</a>
    <li><a href="#label-Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3">Bio::GFF::GFF2 and Bio::GFF::GFF3</a>
    <li><a href="#label-Bio-3A-3AHinv">Bio::Hinv</a>
    <li><a href="#label-Bio-3A-3ANCBI-3A-3AREST">Bio::NCBI::REST</a>
    <li><a href="#label-Bio-3A-3APAML-3A-3ACodeml+and+Bio-3A-3APAML-3A-3ACodeml-3A-3AReport">Bio::PAML::Codeml and Bio::PAML::Codeml::Report</a>
    <li><a href="#label-Bio-3A-3ALocations">Bio::Locations</a>
    <li><a href="#label-Bio-3A-3ATogoWS-3A-3AREST">Bio::TogoWS::REST</a>
    <li><a href="#label-Deprecated+classes">Deprecated classes</a>
    <li><a href="#label-Bio-3A-3AFeatures">Bio::Features</a>
    <li><a href="#label-Bio-3A-3AReferences">Bio::References</a>
    <li><a href="#label-Incompatible+changes">Incompatible changes</a>
    <li><a href="#label-Bio-3A-3ABIORUBY_VERSION">Bio::BIORUBY_VERSION</a>
    <li><a href="#label-Bio-3A-3ASequence">Bio::Sequence</a>
    <li><a href="#label-Bio-3A-3ALocations+and+Bio-3A-3ALocation">Bio::Locations and Bio::Location</a>
    <li><a href="#label-Bio-3A-3ABlast">Bio::Blast</a>
    <li><a href="#label-Bio-3A-3ABlast-3A-3ADefault-3A-3AReport+and+Bio-3A-3ABlast-3A-3AWU-3A-3AReport">Bio::Blast::Default::Report and Bio::Blast::WU::Report</a>
    <li><a href="#label-Bio-3A-3ABlat">Bio::Blat</a>
    <li><a href="#label-Bio-3A-3AGFF-2C+Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3">Bio::GFF, Bio::GFF::GFF2 and Bio::GFF::GFF3</a>
    <li><a href="#label-Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3">Bio::GFF::GFF2 and Bio::GFF::GFF3</a>
    <li><a href="#label-Bio-3A-3AGFF-3A-3AGFF2">Bio::GFF::GFF2</a>
    <li><a href="#label-Bio-3A-3AGFF-3A-3AGFF3">Bio::GFF::GFF3</a>
    <li><a href="#label-Bio-3A-3APathway">Bio::Pathway</a>
    <li><a href="#label-Bio-3A-3ASQL+and+BioSQL+related+classes">Bio::SQL and BioSQL related classes</a>
  </ul>
</div>


  <div id="project-metadata">
    
<div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
    <li><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
    <li><a href="../README_rdoc.html">README</a>
    <li><a href="../README_DEV_rdoc.html">README_DEV</a>
    <li><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
    <li><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
    <li><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
    <li><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
    <li><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
    <li><a href="../doc/RELEASE_NOTES-1_4_3_rdoc.html">RELEASE_NOTES-1.4.3</a>
    <li><a href="../doc/RELEASE_NOTES-1_5_0_rdoc.html">RELEASE_NOTES-1.5.0</a>
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-label="Page doc/Changes-1.3.rdoc">

<h1 id="label-Incompatible+and+important+changes+since+the+BioRuby+1.2.1+release">Incompatible and important changes since the BioRuby 1.2.1 release<span><a href="#label-Incompatible+and+important+changes+since+the+BioRuby+1.2.1+release">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>A lot of changes have been made to the BioRuby after the version 1.2.1 is released.</p>

<h2 id="label-New+features">New features<span><a href="#label-New+features">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Support+for+sequence+output+with+improvements+of+Bio-3A-3ASequence">Support for sequence output with improvements of <a href="../Bio/Sequence.html"><code>Bio::Sequence</code></a><span><a href="#label-Support+for+sequence+output+with+improvements+of+Bio-3A-3ASequence">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>The outputting of EMBL and GenBank formatted text are now supported in the <a href="../Bio/Sequence.html"><code>Bio::Sequence</code></a> class. See the document of <a href="../Bio/Sequence/Format.html#method-i-output"><code>Bio::Sequence#output</code></a> for details. You can also create <a href="../Bio/Sequence.html"><code>Bio::Sequence</code></a> objects from many kinds of data such as <a href="../Bio/GenBank.html"><code>Bio::GenBank</code></a>, <a href="../Bio/EMBL.html"><code>Bio::EMBL</code></a>, and <a href="../Bio/FastaFormat.html"><code>Bio::FastaFormat</code></a> by using the to_biosequence method.</p>

<h3 id="label-BioSQL+support">BioSQL support<span><a href="#label-BioSQL+support">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>BioSQL support is completely rewritten by using ActiveRecord.</p>

<h3 id="label-Bio-3A-3ABlast"><a href="../Bio/Blast.html"><code>Bio::Blast</code></a><span><a href="#label-Bio-3A-3ABlast">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Bio::Blast#reports can parse NCBI default (-m 0) format and tabular (-m 8) format, in addition to XML (-m 7) format.</p>

<p><a href="../Bio/Blast/Report.html"><code>Bio::Blast::Report</code></a> now supports XML format with multiple query sequences generated by blastall 2.2.14 or later.</p>

<p><a href="../Bio/Blast.html#method-c-remote"><code>Bio::Blast.remote</code></a> supports DDBJ, in addition to GenomeNet. In addition, a list of available blast databases on remote sites can be obtained by using Bio::Blast::Remote::DDBJ.databases and Bio::Blast::Remote::GenomeNet.databases methods. Note that the above remote blast methods may be changed in the future to support NCBI.</p>

<p><a href="../Bio/Blast/RPSBlast/Report.html"><code>Bio::Blast::RPSBlast::Report</code></a> is newly added, a parser for NCBI RPS Blast (Reversed Position Specific Blast) default (-m 0 option) results.</p>

<h3 id="label-Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3"><a href="../Bio/GFF/GFF2.html"><code>Bio::GFF::GFF2</code></a> and <a href="../Bio/GFF/GFF3.html"><code>Bio::GFF::GFF3</code></a><span><a href="#label-Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>The outputting of GFF2/GFF3-formatted text is now supported. However, many incompatible changes have been made (See below for details).</p>

<h3 id="label-Bio-3A-3AHinv"><a href="../Bio/Hinv.html"><code>Bio::Hinv</code></a><span><a href="#label-Bio-3A-3AHinv">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>H-Invitational Database web service (REST) client class is newly added.</p>

<h3 id="label-Bio-3A-3ANCBI-3A-3AREST"><a href="../Bio/NCBI/REST.html"><code>Bio::NCBI::REST</code></a><span><a href="#label-Bio-3A-3ANCBI-3A-3AREST">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>NCBI E-Utilities client class is newly added.</p>

<h3 id="label-Bio-3A-3APAML-3A-3ACodeml+and+Bio-3A-3APAML-3A-3ACodeml-3A-3AReport"><a href="../Bio/PAML/Codeml.html"><code>Bio::PAML::Codeml</code></a> and <a href="../Bio/PAML/Codeml/Report.html"><code>Bio::PAML::Codeml::Report</code></a><span><a href="#label-Bio-3A-3APAML-3A-3ACodeml+and+Bio-3A-3APAML-3A-3ACodeml-3A-3AReport">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/PAML/Codeml.html"><code>Bio::PAML::Codeml</code></a>, wrapper for PAML codeml program, and <a href="../Bio/PAML/Codeml/Report.html"><code>Bio::PAML::Codeml::Report</code></a>, parser for codeml result are newly added, though some of them are still under construction and too specific to particular use cases.</p>

<h3 id="label-Bio-3A-3ALocations"><a href="../Bio/Locations.html"><code>Bio::Locations</code></a><span><a href="#label-Bio-3A-3ALocations">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>New method <a href="../Bio/Locations.html#method-i-to_s"><code>Bio::Locations#to_s</code></a> is added to support output of features.</p>

<h3 id="label-Bio-3A-3ATogoWS-3A-3AREST"><a href="../Bio/TogoWS/REST.html"><code>Bio::TogoWS::REST</code></a><span><a href="#label-Bio-3A-3ATogoWS-3A-3AREST">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>TogoWS REST client class is newly added. Information about TogoWS REST service can be found on <a href="http://togows.dbcls.jp/site/en/rest.html">togows.dbcls.jp/site/en/rest.html</a>.</p>

<h2 id="label-Deprecated+classes">Deprecated classes<span><a href="#label-Deprecated+classes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3AFeatures"><a href="../Bio/Features.html"><code>Bio::Features</code></a><span><a href="#label-Bio-3A-3AFeatures">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/Features.html"><code>Bio::Features</code></a> is obsoleted and changed to an array of <a href="../Bio/Feature.html"><code>Bio::Feature</code></a> object with some backward compatibility methods.  The backward compatibility methods will soon be removed in the future.</p>

<h3 id="label-Bio-3A-3AReferences"><a href="../Bio/References.html"><code>Bio::References</code></a><span><a href="#label-Bio-3A-3AReferences">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/References.html"><code>Bio::References</code></a> is obsoleted and changed to an array of <a href="../Bio/Reference.html"><code>Bio::Reference</code></a> object with some backward compatibility methods.  The backward compatibility methods will soon be removed in the future.</p>

<h2 id="label-Incompatible+changes">Incompatible changes<span><a href="#label-Incompatible+changes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ABIORUBY_VERSION">Bio::BIORUBY_VERSION<span><a href="#label-Bio-3A-3ABIORUBY_VERSION">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Definition of the constant Bio::BIORUBY_VERSION is moved from lib/bio.rb to lib/bio/version.rb. Normally, the autoload mechanism of Ruby correctly loads the version.rb, but special scripts directly using bio.rb may be needed to be changed.</p>

<p>Bio::BIORUBY_VERSION is changed to be frozen.</p>

<p>New constants Bio::BIORUBY_EXTRA_VERSION and Bio::BIORUBY_VERSION_ID are added. See their RDoc for details.</p>

<h3 id="label-Bio-3A-3ASequence"><a href="../Bio/Sequence.html"><code>Bio::Sequence</code></a><span><a href="#label-Bio-3A-3ASequence">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Bio::Sequence#date is removed.  Alternatively, date_created or date_modified can be used.</p>

<p><a href="../Bio/Sequence.html#attribute-i-taxonomy"><code>Bio::Sequence#taxonomy</code></a> is changed to be an alias of classification, and the data type is changed to an array of string.</p>

<h3 id="label-Bio-3A-3ALocations+and+Bio-3A-3ALocation"><a href="../Bio/Locations.html"><code>Bio::Locations</code></a> and <a href="../Bio/Location.html"><code>Bio::Location</code></a><span><a href="#label-Bio-3A-3ALocations+and+Bio-3A-3ALocation">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>A carat in a location (e.g. “123^124”) is now parsed, instead of being replaced by “..”.  To distinguish from normal “..”, a new attribute <a href="../Bio/Location.html#attribute-i-carat"><code>Bio::Location#carat</code></a> is used.</p>

<p>“order(…)” or “group(…)” are also parsed, instead of being regarded as “join(…)”.  To distinguish from “join(…)”, a new attribute <a href="../Bio/Locations.html#attribute-i-operator"><code>Bio::Locations#operator</code></a> is used.  When “order(…)” or “group(…)”, the attribute is set to :order or :group, respectively.  Note that “group(…)” is already deprecated in EMBL/GenBank/DDBJ.</p>

<h3 id="label-Bio-3A-3ABlast"><a href="../Bio/Blast.html"><code>Bio::Blast</code></a><span><a href="#label-Bio-3A-3ABlast">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Return value of Bio::Blast#exec_* is changed to String instead of Report object. Parsing the string is now processed in <a href="../Bio/Blast.html#method-i-query"><code>Bio::Blast#query</code></a> method.</p>

<p>Bio::Blast#exec_genomenet_tab and <a href="../Bio/Blast.html#method-i-server-3D"><code>Bio::Blast#server=</code></a>“genomenet_tab” is deprecated.</p>

<p><a href="../Bio/Blast.html#method-i-options-3D"><code>Bio::Blast#options=()</code></a> can now change the following attributes: program, db, format, matrix, and filter.</p>

<p><a href="../Bio/Blast.html#method-c-reports"><code>Bio::Blast.reports</code></a> now supports default (-m 0) and tabular (-m 8) formats. Old implementation (only supports XML) is renamed to <a href="../Bio/Blast.html#method-c-reports_xml"><code>Bio::Blast.reports_xml</code></a>, to keep compatibility for older BLAST XML documents which might not be parsed by the new <a href="../Bio/Blast.html#method-c-reports"><code>Bio::Blast.reports</code></a> nor <a href="../Bio/FlatFile.html"><code>Bio::FlatFile</code></a>, although we are not sure whether such documents really exist or not.</p>

<h3 id="label-Bio-3A-3ABlast-3A-3ADefault-3A-3AReport+and+Bio-3A-3ABlast-3A-3AWU-3A-3AReport"><a href="../Bio/Blast/Default/Report.html"><code>Bio::Blast::Default::Report</code></a> and <a href="../Bio/Blast/WU/Report.html"><code>Bio::Blast::WU::Report</code></a><span><a href="#label-Bio-3A-3ABlast-3A-3ADefault-3A-3AReport+and+Bio-3A-3ABlast-3A-3AWU-3A-3AReport">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Iteration#lambda, kappa, entropy, gapped_lambda, gapped_kappa, and gapped_entropy, and the same methods in the Report class are changed to return float or nil instead of string or nil.</p>

<h3 id="label-Bio-3A-3ABlat"><a href="../Bio/Blat.html"><code>Bio::Blat</code></a><span><a href="#label-Bio-3A-3ABlat">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>When reading BLAT psl (or pslx) data by using <a href="../Bio/FlatFile.html"><code>Bio::FlatFile</code></a>, it checks each query name and returns a new entry object when the query name is changed from previous queries. This is, data is stored to two or more <a href="../Bio/Blat/Report.html"><code>Bio::Blat::Report</code></a> objects, instead of previous version’s behavior (always reads all data at once and stores to a <a href="../Bio/Blat/Report.html"><code>Bio::Blat::Report</code></a> object).</p>

<h3 id="label-Bio-3A-3AGFF-2C+Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3"><a href="../Bio/GFF.html"><code>Bio::GFF</code></a>, <a href="../Bio/GFF/GFF2.html"><code>Bio::GFF::GFF2</code></a> and <a href="../Bio/GFF/GFF3.html"><code>Bio::GFF::GFF3</code></a><span><a href="#label-Bio-3A-3AGFF-2C+Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/GFF/Record.html#method-i-comments"><code>Bio::GFF::Record#comments</code></a> is renamed to comment, and comments= is renamed to comment=, because they only allow a single String (or nil) and the plural form “comments” may be confusable.  The “comments” and “comments=” methods can still be used, but warning messages will be shown when using in GFF2::Record and GFF3::Record objects.</p>

<p>See below about GFF2 and/or GFF3 specific changes.</p>

<h3 id="label-Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3"><a href="../Bio/GFF/GFF2.html"><code>Bio::GFF::GFF2</code></a> and <a href="../Bio/GFF/GFF3.html"><code>Bio::GFF::GFF3</code></a><span><a href="#label-Bio-3A-3AGFF-3A-3AGFF2+and+Bio-3A-3AGFF-3A-3AGFF3">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/GFF/GFF2/Record.html#method-c-new"><code>Bio::GFF::GFF2::Record.new</code></a> and <a href="../Bio/GFF/GFF3/Record.html#method-c-new"><code>Bio::GFF::GFF3::Record.new</code></a> can also get 9 arguments corresponding to GFF columns, which helps to create Record object directly without formatted text.</p>

<p><a href="../Bio/GFF/Record.html#attribute-i-start"><code>Bio::GFF::GFF2::Record#start</code></a>, end, and frame return Integer or nil, and score returns Float or nil, instead of String or nil. The same changes are also made to <a href="../Bio/GFF/GFF3/Record.html"><code>Bio::GFF::GFF3::Record</code></a>.</p>

<p><a href="../Bio/GFF/Record.html#attribute-i-attributes"><code>Bio::GFF::GFF2::Record#attributes</code></a> and <a href="../Bio/GFF/Record.html#attribute-i-attributes"><code>Bio::GFF::GFF3::Record#attributes</code></a> are changed to return a nested Array, containing [ tag, value ] pairs, because of supporting multiple tags in the same tag names.  If you want to get a Hash, use Record#attributes_to_hash method, though some tag-value pairs in the same tag names may be lost.  Note that Bio::GFF::Record#attribute still returns a Hash for compatibility.</p>

<p>New methods for getting, setting and manipulating attributes are added to <a href="../Bio/GFF/GFF2/Record.html"><code>Bio::GFF::GFF2::Record</code></a> and <a href="../Bio/GFF/GFF3/Record.html"><code>Bio::GFF::GFF3::Record</code></a> classes: attribute, get_attribute, get_attributes, set_attribute, replace_attributes, add_attribute, delete_attribute, delete_attributes, sort_attributes_by_tag!. It is recommended to use these methods instead of directly manipulating the array returned by Record#attributes.</p>

<p><a href="../Bio/GFF/GFF2.html#method-i-to_s"><code>Bio::GFF::GFF2#to_s</code></a>, <a href="../Bio/GFF/GFF3.html#method-i-to_s"><code>Bio::GFF::GFF3#to_s</code></a>, <a href="../Bio/GFF/GFF2/Record.html#method-i-to_s"><code>Bio::GFF::GFF2::Record#to_s</code></a>, and <a href="../Bio/GFF/GFF3/Record.html#method-i-to_s"><code>Bio::GFF::GFF3::Record#to_s</code></a> are added to support output of GFF2/GFF3 data.</p>

<h3 id="label-Bio-3A-3AGFF-3A-3AGFF2"><a href="../Bio/GFF/GFF2.html"><code>Bio::GFF::GFF2</code></a><span><a href="#label-Bio-3A-3AGFF-3A-3AGFF2">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>GFF2 attribute values are now automatically unescaped.  In addition, if a value of an attribute is consisted of two or more tokens delimited by spaces, an object of the new class <a href="../Bio/GFF/GFF2/Record/Value.html"><code>Bio::GFF::GFF2::Record::Value</code></a> is returned instead of String.  The new class <a href="../Bio/GFF/GFF2/Record/Value.html"><code>Bio::GFF::GFF2::Record::Value</code></a> aims to store a parsed value of an attribute.  If you really want to get unparsed string, <a href="../Bio/GFF/GFF2/Record/Value.html#method-i-to_s"><code>Bio::GFF::GFF2::Record::Value#to_s</code></a> can be used.</p>

<p>The metadata (lines beginning with “##”) are parsed to <a href="../Bio/GFF/GFF2/MetaData.html"><code>Bio::GFF::GFF2::MetaData</code></a> objects and are stored to <a href="../Bio/GFF/GFF2.html#attribute-i-metadata"><code>Bio::GFF::GFF2#metadata</code></a> as an array, except the “##gff-version” line.  The “##gff-version” version string is stored to the <a href="../Bio/GFF/GFF2.html#attribute-i-gff_version"><code>Bio::GFF::GFF2#gff_version</code></a> as a string.</p>

<h3 id="label-Bio-3A-3AGFF-3A-3AGFF3"><a href="../Bio/GFF/GFF3.html"><code>Bio::GFF::GFF3</code></a><span><a href="#label-Bio-3A-3AGFF-3A-3AGFF3">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Aliases of columns which are renamed in the GFF3 specification are added to the <a href="../Bio/GFF/GFF3/Record.html"><code>Bio::GFF::GFF3::Record</code></a> class: seqid (column 1; alias of “seqname”), feature_type (column 3; alias of “feature”; in the GFF3 spec, it is called “type”, but because “type” is already used by Ruby, we use “feature_type”), phase (column 8; formerly “frame”). Original names can still be used because they are only aliases.</p>

<p>Sequences bundled within GFF3 after “##FASTA” are now supported (<a href="../Bio/GFF/GFF3.html#attribute-i-sequences"><code>Bio::GFF::GFF3#sequences</code></a>).</p>

<p>GFF3 attribute keys and values are automatically unescaped. Each attribute value is stored as a string, except for special attributes listed below:</p>
<ul><li>
<p><a href="../Bio/GFF/GFF3/Record/Target.html"><code>Bio::GFF::GFF3::Record::Target</code></a> to store a “Target” attribute.</p>
</li><li>
<p><a href="../Bio/GFF/GFF3/Record/Gap.html"><code>Bio::GFF::GFF3::Record::Gap</code></a> to store a “Gap” attribute.</p>
</li></ul>

<p>The metadata (lines beginning with “##”) are parsed to Bio::GFF::GFF3::MetaData objects and stored to <a href="../Bio/GFF/GFF3.html#attribute-i-metadata"><code>Bio::GFF::GFF3#metadata</code></a> as an array, except “##gff-version”, “##sequence-region”, “###”, and “##FASTA” lines.</p>
<ul><li>
<p>“##gff-version” version string is stored to <a href="../Bio/GFF/GFF3.html#attribute-i-gff_version"><code>Bio::GFF::GFF3#gff_version</code></a>.</p>
</li><li>
<p>“##sequence-region” lines are parsed to <a href="../Bio/GFF/GFF3/SequenceRegion.html"><code>Bio::GFF::GFF3::SequenceRegion</code></a> objects and stored to <a href="../Bio/GFF/GFF3.html#attribute-i-sequence_regions"><code>Bio::GFF::GFF3#sequence_regions</code></a> as an array.</p>
</li><li>
<p>“###” lines are parsed to <a href="../Bio/GFF/GFF3/RecordBoundary.html"><code>Bio::GFF::GFF3::RecordBoundary</code></a> objects.</p>
</li><li>
<p>“##FASTA” is regarded as the beginning of bundled sequences.</p>
</li></ul>

<h3 id="label-Bio-3A-3APathway"><a href="../Bio/Pathway.html"><code>Bio::Pathway</code></a><span><a href="#label-Bio-3A-3APathway">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/Pathway.html#method-i-cliquishness"><code>Bio::Pathway#cliquishness</code></a> is changed to calculate cliquishness (clustering coefficient) for not only undirected graphs but also directed graphs.</p>

<p>In <a href="../Bio/Pathway.html#method-i-to_matrix"><code>Bio::Pathway#to_matrix</code></a>, dump_matrix, dump_list, and depth_first_search methods, to avoid dependency to the order of objects in Hash#each (and each_keys etc.), <a href="../Bio/Pathway.html#attribute-i-index"><code>Bio::Pathway#index</code></a> is used to specify preferences of nodes in a graph.</p>

<h3 id="label-Bio-3A-3ASQL+and+BioSQL+related+classes">Bio::SQL and BioSQL related classes<span><a href="#label-Bio-3A-3ASQL+and+BioSQL+related+classes">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>BioSQL support is completely rewritten by using ActiveRecord. See documents in lib/bio/io/sql.rb, lib/bio/io/biosql, and lib/bio/db/biosql for details of changes and usage of the classes/modules.</p>

</main>



<footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.4.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

