<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>RELEASE_NOTES-1.4.2 - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
</script>

<script src="../js/jquery.js"></script>
<script src="../js/darkfish.js"></script>

<link href="../css/fonts.css" rel="stylesheet">
<link href="../css/rdoc.css" rel="stylesheet">



<body id="top" role="document" class="file">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../table_of_contents.html#pages">Pages</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#label-BioRuby+1.4.2+RELEASE+NOTES">BioRuby 1.4.2 RELEASE NOTES</a>
    <li><a href="#label-New+features">New features</a>
    <li><a href="#label-Speed-up+of+Bio-3A-3ARestrictionEnzyme-3A-3AAnalysis.cut">Speed-up of Bio::RestrictionEnzyme::Analysis.cut</a>
    <li><a href="#label-New+classes+Bio-3A-3ADDBJ-3A-3AREST-2C+REST+interface+for+DDBJ+web+service">New classes Bio::DDBJ::REST, REST interface for DDBJ web service</a>
    <li><a href="#label-Bio-3A-3ABlast+with+remote+DDBJ+server+uses+REST+instead+of+SOAP">Bio::Blast with remote DDBJ server uses REST instead of SOAP</a>
    <li><a href="#label-Tutorial+is+updated">Tutorial is updated</a>
    <li><a href="#label-Many+unit+tests+are+added">Many unit tests are added</a>
    <li><a href="#label-Other+new+features">Other new features</a>
    <li><a href="#label-Bug+fixes">Bug fixes</a>
    <li><a href="#label-Bio-3A-3ABlast">Bio::Blast</a>
    <li><a href="#label-Bio-3A-3ASPTR+-28Bio-3A-3AUniProt-29">Bio::SPTR (Bio::UniProt)</a>
    <li><a href="#label-Other+bug+fixes">Other bug fixes</a>
    <li><a href="#label-Incompatible+changes">Incompatible changes</a>
    <li><a href="#label-Bio-3A-3ASequence-23output-28-3Afastq-29">Bio::Sequence#output(:fastq)</a>
    <li><a href="#label-Bio-3A-3ASPTR+CC+line+topic+-22WEB+RESOURCE-22">Bio::SPTR CC line topic “WEB RESOURCE”</a>
    <li><a href="#label-Bio-3A-3ABlast+with+the+remote+GenomeNet+server">Bio::Blast with the remote GenomeNet server</a>
    <li><a href="#label-Bio-3A-3ABlast+with+the+remote+DDBJ+server">Bio::Blast with the remote DDBJ server</a>
    <li><a href="#label-Bio-3A-3ARestrictionEnzyme+internal+data+structure+change">Bio::RestrictionEnzyme internal data structure change</a>
    <li><a href="#label-Known+issues">Known issues</a>
    <li><a href="#label-Other+important+news">Other important news</a>
  </ul>
</div>


  <div id="project-metadata">
    <div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
  
    <li><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li><a href="../README_rdoc.html">README</a>
  
    <li><a href="../README_DEV_rdoc.html">README_DEV</a>
  
    <li><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_3_rdoc.html">RELEASE_NOTES-1.4.3</a>
  
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-label="Page doc/RELEASE_NOTES-1.4.2.rdoc">

<h1 id="label-BioRuby+1.4.2+RELEASE+NOTES">BioRuby 1.4.2 RELEASE NOTES<span><a href="#label-BioRuby+1.4.2+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>A lot of changes have been made to the BioRuby 1.4.2 after the version
1.4.1 is released. This document describes important and/or incompatible
changes since the BioRuby 1.4.1 release.</p>

<p>For known problems, see <a
href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a>.</p>

<h2 id="label-New+features">New features<span><a href="#label-New+features">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Speed-up+of+Bio-3A-3ARestrictionEnzyme-3A-3AAnalysis.cut">Speed-up of <a href="../Bio/RestrictionEnzyme/Analysis.html#method-c-cut">Bio::RestrictionEnzyme::Analysis.cut</a><span><a href="#label-Speed-up+of+Bio-3A-3ARestrictionEnzyme-3A-3AAnalysis.cut">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>The running speed of <a
href="../Bio/RestrictionEnzyme/Analysis.html#method-c-cut">Bio::RestrictionEnzyme::Analysis.cut</a>
is significantly increased. The new code is 50 to 80 fold faster than the
previous code when cutting 1Mbp sequence running on Ruby 1.9.2p180. The
code is written by Tomoaki NISHIYAMA and Naohisa Goto.</p>

<h3 id="label-New+classes+Bio-3A-3ADDBJ-3A-3AREST-2C+REST+interface+for+DDBJ+web+service">New classes Bio::DDBJ::REST, REST interface for DDBJ web service<span><a href="#label-New+classes+Bio-3A-3ADDBJ-3A-3AREST-2C+REST+interface+for+DDBJ+web+service">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>For DDBJ Web API for Biology (WABI) web service, in additon to SOAP, REST
(REpresentational State Transfer) interface is added as Bio::DDBJ::REST.
Currently, only selected APIs are implemented.</p>

<h3 id="label-Bio-3A-3ABlast+with+remote+DDBJ+server+uses+REST+instead+of+SOAP"><a href="../Bio/Blast.html">Bio::Blast</a> with remote DDBJ server uses REST instead of SOAP<span><a href="#label-Bio-3A-3ABlast+with+remote+DDBJ+server+uses+REST+instead+of+SOAP">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/Blast.html">Bio::Blast</a> with remote DDBJ server uses
REST instead of SOAP, because Soap4r (SOAP library for Ruby) does not work
well with Ruby 1.9. We can now use remote DDBJ BLAST server with Ruby 1.9.</p>

<h3 id="label-Tutorial+is+updated">Tutorial is updated<span><a href="#label-Tutorial+is+updated">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>The Tutorial.rd is updated by Pjotr Prins and Michael O&#39;Keefe.</p>

<h3 id="label-Many+unit+tests+are+added">Many unit tests are added<span><a href="#label-Many+unit+tests+are+added">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Added many unit tests for <a href="../Bio/GenBank.html">Bio::GenBank</a>,
<a href="../Bio/GenPept.html">Bio::GenPept</a>, <a
href="../Bio/NBRF.html">Bio::NBRF</a>, <a
href="../Bio/PDB.html">Bio::PDB</a> and so on. Most of them are developed
by Kazuhiro Hayashi during the Google Summer of Code 2010.</p>

<h3 id="label-Other+new+features">Other new features<span><a href="#label-Other+new+features">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>New method <a href="../Bio/Fastq.html#method-i-to_s">Bio::Fastq#to_s</a>
for convenience. Note that the use of the method may cause loss of
performance. To get each input sequence entry as-is, consider using <a
href="../Bio/FlatFile.html#method-i-entry_raw">Bio::FlatFile#entry_raw</a>.
To output fastq format data, consider using <a
href="../Bio/Sequence/Format.html#method-i-output">Bio::Sequence::Format#output</a>(:fastq).</p>
</li><li>
<p>New methods <a
href="../Bio/NCBI/REST/EFetch/Methods.html#method-i-nucleotide">Bio::NCBI::REST::EFetch::Methods#nucleotide</a>
and protein, to get data from “nucleotide” and “protein” database
respectively. Because NCBI changed not to accept “gb” format for the
database “sequence”, the two new methods are added for convenience.</p>
</li><li>
<p>In BioRuby Shell, efetch method uses the above new methods.</p>
</li><li>
<p>In GenomeNet remote BLAST execution, database “mine-aa” and “mine-nt” with
KEGG organism codes are now supported.</p>
</li><li>
<p>Support for Ruby 1.9.2 / 1.9.3 is improved.</p>
</li></ul>

<h2 id="label-Bug+fixes">Bug fixes<span><a href="#label-Bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ABlast"><a href="../Bio/Blast.html">Bio::Blast</a><span><a href="#label-Bio-3A-3ABlast">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>Failure of remote BLAST execution is fixed, due to the changes in GenomeNet
and DDBJ.</p>
</li><li>
<p>When executing remote BLAST with “genomenet” server, options “-b” and “-v”
are now correctly used to limit the number of hits to be reported.</p>
</li></ul>

<h3 id="label-Bio-3A-3ASPTR+-28Bio-3A-3AUniProt-29">Bio::SPTR (Bio::UniProt)<span><a href="#label-Bio-3A-3ASPTR+-28Bio-3A-3AUniProt-29">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>Due to the UniProtKB format changes, ID, DE, and WEB RESOURCE of CC lines
were not correctly parsed. See also below about incompatible change of the
fix.</p>
</li></ul>

<h3 id="label-Other+bug+fixes">Other bug fixes<span><a href="#label-Other+bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p><a
href="../Bio/Reference.html#method-i-pubmed_url">Bio::Reference#pubmed_url</a>
is updated to follow recent NCBI changes.</p>
</li><li>
<p>Fixed: <a
href="../Bio/Newick.html#method-i-reparse">Bio::Newick#reparse</a> failure.</p>
</li><li>
<p>Fixed: In <a
href="../Bio/MEDLINE.html#method-i-reference">Bio::MEDLINE#reference</a>,
doi field should be filled.</p>
</li><li>
<p>Fixed: <a
href="../Bio/Reference.html#method-i-endnote">Bio::Reference#endnote</a>
fails when url is not set.</p>
</li><li>
<p>Fixed: <a
href="../Bio/FastaFormat.html#method-i-query">Bio::FastaFormat#query</a>
passes nil to the given factory object.</p>
</li><li>
<p>Fixed: In BioRuby Shell, efetch() with no additional arguments fails
because of the NCBI site changes.</p>
</li><li>
<p>Fixed: In BioRuby Shell, getent() fails when EMBOSS seqret is not found.</p>
</li><li>
<p>Fixed: In BioRuby Shell, demo() fails due to the above two issues.</p>
</li></ul>

<h2 id="label-Incompatible+changes">Incompatible changes<span><a href="#label-Incompatible+changes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ASequence-23output-28-3Afastq-29"><a href="../Bio/Sequence/Format.html#method-i-output">Bio::Sequence::Format#output</a>(:fastq)<span><a href="#label-Bio-3A-3ASequence-23output-28-3Afastq-29">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>In Fastq output formatter, default width value is changed from 70 to nil.
The nil means “without wrapping”. The new default behavior without wrapping
is generally good with many recent applications that read fastq.</p>

<h3 id="label-Bio-3A-3ASPTR+CC+line+topic+-22WEB+RESOURCE-22">Bio::SPTR CC line topic “WEB RESOURCE”<span><a href="#label-Bio-3A-3ASPTR+CC+line+topic+-22WEB+RESOURCE-22">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>In the return value of Bio::SPTR#cc(&#39;WEB RESOURCE&#39;), “NAME” and
“NOTE” are now renamed to “Name” and “Note”, respectively. The change is
due to the UniProt format change since UniProtKB release 12.2 of
11-Sep-2007. (See <a
href="http://www.uniprot.org/docs/sp_news.htm#rel12.2">www.uniprot.org/docs/sp_news.htm#rel12.2</a>
for details.) Note that “Name” and “Note” are used even when parsing older
format. The change would also affect Marshal.dump (and YAML.dump) data.</p>

<h3 id="label-Bio-3A-3ABlast+with+the+remote+GenomeNet+server"><a href="../Bio/Blast.html">Bio::Blast</a> with the remote GenomeNet server<span><a href="#label-Bio-3A-3ABlast+with+the+remote+GenomeNet+server">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>When executing remote BLAST with “genomenet” server, options “-b” and “-v”
are now correctly used to limit the number of hits to be reported. In 1.4.1
and before, “-B” and “-V” were mistakenly used for the purpose.</p>

<h3 id="label-Bio-3A-3ABlast+with+the+remote+DDBJ+server"><a href="../Bio/Blast.html">Bio::Blast</a> with the remote DDBJ server<span><a href="#label-Bio-3A-3ABlast+with+the+remote+DDBJ+server">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/Blast.html">Bio::Blast</a> with remote DDBJ server uses
REST instead of SOAP.</p>

<h3 id="label-Bio-3A-3ARestrictionEnzyme+internal+data+structure+change"><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a> internal data structure change<span><a href="#label-Bio-3A-3ARestrictionEnzyme+internal+data+structure+change">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Due to the speedup, internal data structure of the following classes are
changed: <a
href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>,
<a
href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>,
<a
href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>.
This indicates that Marshal.dump (and YAML.dump) data generated by older
versions cannot be loaded by the new version, and vice versa, although
public APIs of the classes keep compatibility.</p>

<h2 id="label-Known+issues">Known issues<span><a href="#label-Known+issues">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>The following issues are added or updated. See <a
href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a> for other already
known issues.</p>
<ul><li>
<p>Bio::SPTR should be updated to follow UniProtKB format changes.</p>
</li><li>
<p>Problems observed only with Ruby 1.8.5 or earlier will not be fixed.</p>
</li><li>
<p>Descriptions about very old RubyGems 0.8.11 or earlier and about CVS
repository are moved from <a href="../README_rdoc.html">README.rdoc</a>.</p>
</li></ul>

<h2 id="label-Other+important+news">Other important news<span><a href="#label-Other+important+news">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Required ruby version is now Ruby 1.8.6 or later (except 1.9.0).</p>
</li></ul>
</main>



<footer id="validator-badges" role="contentinfo">
  <p><a href="http://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="http://docs.seattlerb.org/rdoc/">RDoc</a> 4.2.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

