<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>RELEASE_NOTES-1.4.1 - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
</script>

<script src="../js/jquery.js"></script>
<script src="../js/darkfish.js"></script>

<link href="../css/fonts.css" rel="stylesheet">
<link href="../css/rdoc.css" rel="stylesheet">



<body id="top" role="document" class="file">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../table_of_contents.html#pages">Pages</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#label-BioRuby+1.4.1+RELEASE+NOTES">BioRuby 1.4.1 RELEASE NOTES</a>
    <li><a href="#label-New+features">New features</a>
    <li><a href="#label-PAML+Codeml+support+is+significantly+improved">PAML Codeml support is significantly improved</a>
    <li><a href="#label-KEGG+PATHWAY+and+KEGG+MODULE+parser">KEGG PATHWAY and KEGG MODULE parser</a>
    <li><a href="#label-Bio-3A-3AKEGG+improvements">Bio::KEGG improvements</a>
    <li><a href="#label-Test+codes+are+added+and+improved.">Test codes are added and improved.</a>
    <li><a href="#label-Other+new+methods">Other new methods</a>
    <li><a href="#label-Bug+fixes">Bug fixes</a>
    <li><a href="#label-Bio-3A-3ATree">Bio::Tree</a>
    <li><a href="#label-Bio-3A-3AKEGG-3A-3AGENES+and+Bio-3A-3AKEGG-3A-3AGENOME">Bio::KEGG::GENES and Bio::KEGG::GENOME</a>
    <li><a href="#label-Other+bug+fixes">Other bug fixes</a>
    <li><a href="#label-Incompatible+changes">Incompatible changes</a>
    <li><a href="#label-Bio-3A-3APAML-3A-3ACodeml-3A-3AReport">Bio::PAML::Codeml::Report</a>
    <li><a href="#label-Bio-3A-3AKEGG-3A-3AORTHOLOGY">Bio::KEGG::ORTHOLOGY</a>
    <li><a href="#label-Bio-3A-3AAAindex2">Bio::AAindex2</a>
    <li><a href="#label-Bio-3A-3AMEDLINE">Bio::MEDLINE</a>
    <li><a href="#label-Known+issues">Known issues</a>
    <li><a href="#label-String+escaping+of+command-line+arguments+in+Ruby+1.9.X+on+Windows">String escaping of command-line arguments in Ruby 1.9.X on Windows</a>
  </ul>
</div>


  <div id="project-metadata">
    <div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
  
    <li><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li><a href="../README_rdoc.html">README</a>
  
    <li><a href="../README_DEV_rdoc.html">README_DEV</a>
  
    <li><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
    <li><a href="../doc/RELEASE_NOTES-1_4_3_rdoc.html">RELEASE_NOTES-1.4.3</a>
  
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-label="Page doc/RELEASE_NOTES-1.4.1.rdoc">

<h1 id="label-BioRuby+1.4.1+RELEASE+NOTES">BioRuby 1.4.1 RELEASE NOTES<span><a href="#label-BioRuby+1.4.1+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>A lot of changes have been made to the BioRuby 1.4.1 after the version
1.4.0 is released. This document describes important and/or incompatible
changes since the BioRuby 1.4.0 release.</p>

<p>For known problems, see <a
href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a>.</p>

<h2 id="label-New+features">New features<span><a href="#label-New+features">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-PAML+Codeml+support+is+significantly+improved">PAML Codeml support is significantly improved<span><a href="#label-PAML+Codeml+support+is+significantly+improved">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>PAML Codeml result parser is completely rewritten and is significantly
improved. The code is developed by Pjotr Prins.</p>

<h3 id="label-KEGG+PATHWAY+and+KEGG+MODULE+parser">KEGG PATHWAY and KEGG MODULE parser<span><a href="#label-KEGG+PATHWAY+and+KEGG+MODULE+parser">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Parsers for KEGG PATHWAY and KEGG MODULE data are added. The code is
developed by Kozo Nishida and Toshiaki Katayama.</p>

<h3 id="label-Bio-3A-3AKEGG+improvements"><a href="../Bio/KEGG.html">Bio::KEGG</a> improvements<span><a href="#label-Bio-3A-3AKEGG+improvements">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Following new methods are added.</p>
<ul><li>
<p><a
href="../Bio/KEGG/GENES.html#method-i-keggclass">Bio::KEGG::GENES#keggclass</a>,
keggclasses, names_as_array, names, motifs_as_strings, motifs_as_hash,
motifs</p>
</li><li>
<p><a
href="../Bio/KEGG/GENOME.html#method-i-original_databases">Bio::KEGG::GENOME#original_databases</a></p>
</li></ul>

<h3 id="label-Test+codes+are+added+and+improved.">Test codes are added and improved.<span><a href="#label-Test+codes+are+added+and+improved.">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Test codes are added and improved. Tney are developed by Kazuhiro Hayashi,
Kozo Nishida, John Prince, and Naohisa Goto.</p>

<h3 id="label-Other+new+methods">Other new methods<span><a href="#label-Other+new+methods">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p><a href="../Bio/Fastq.html#method-i-mask">Bio::Fastq#mask</a></p>
</li><li>
<p><a
href="../Bio/Sequence/Format.html#method-i-output_fasta">Bio::Sequence::Format#output_fasta</a></p>
</li><li>
<p><a
href="../Bio/ClustalW/Report.html#method-i-get_sequence">Bio::ClustalW::Report#get_sequence</a></p>
</li><li>
<p>Bio::Reference#==</p>
</li><li>
<p>Bio::Location#==</p>
</li><li>
<p>Bio::Locations#==</p>
</li><li>
<p><a
href="../Bio/FastaNumericFormat.html#method-i-to_biosequence">Bio::FastaNumericFormat#to_biosequence</a></p>
</li></ul>

<h2 id="label-Bug+fixes">Bug fixes<span><a href="#label-Bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ATree"><a href="../Bio/Tree.html">Bio::Tree</a><span><a href="#label-Bio-3A-3ATree">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Following methods did not work correctly.</p>
<ul><li>
<p><a
href="../Bio/Tree.html#method-i-collect_edge-21">Bio::Tree#collect_edge!</a></p>
</li><li>
<p><a
href="../Bio/Tree.html#method-i-remove_edge_if">Bio::Tree#remove_edge_if</a></p>
</li></ul>

<h3 id="label-Bio-3A-3AKEGG-3A-3AGENES+and+Bio-3A-3AKEGG-3A-3AGENOME"><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a> and <a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a><span><a href="#label-Bio-3A-3AKEGG-3A-3AGENES+and+Bio-3A-3AKEGG-3A-3AGENOME">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>Fixed bugs in <a
href="../Bio/KEGG/GENES.html#method-i-pathway">Bio::KEGG::GENES#pathway</a>.</p>
</li><li>
<p>Fixed parser errors due to the format changes of KEGG GENES and KEGG
GENOME.</p>
</li></ul>

<h3 id="label-Other+bug+fixes">Other bug fixes<span><a href="#label-Other+bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>In <a href="../Bio/Command.html">Bio::Command</a>, changed not to call
fork(2) on platforms that do not support it.</p>
</li><li>
<p>Bio::MEDLINE#initialize should handle continuation of lines.</p>
</li><li>
<p>Typo and a missing field in <a
href="../Bio/GO/GeneAssociation.html#method-i-to_str">Bio::GO::GeneAssociation#to_str</a>.</p>
</li><li>
<p>Bug fix of <a
href="../Bio/FastaNumericFormat.html#method-i-to_biosequence">Bio::FastaNumericFormat#to_biosequence</a>.</p>
</li><li>
<p>Fixed UniProt GN parsing issue in Bio::SPTR.</p>
</li></ul>

<h2 id="label-Incompatible+changes">Incompatible changes<span><a href="#label-Incompatible+changes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3APAML-3A-3ACodeml-3A-3AReport"><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a><span><a href="#label-Bio-3A-3APAML-3A-3ACodeml-3A-3AReport">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>The code is completely rewritten. See the RDoc for details.</p>

<h3 id="label-Bio-3A-3AKEGG-3A-3AORTHOLOGY"><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a><span><a href="#label-Bio-3A-3AKEGG-3A-3AORTHOLOGY">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a
href="../Bio/KEGG/ORTHOLOGY.html#method-i-pathways">Bio::KEGG::ORTHOLOGY#pathways</a>
is changed to return a hash. The old pathway method is renamed to
pathways_in_keggclass for compatibility.</p>

<h3 id="label-Bio-3A-3AAAindex2"><a href="../Bio/AAindex2.html">Bio::AAindex2</a><span><a href="#label-Bio-3A-3AAAindex2">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/AAindex2.html">Bio::AAindex2</a> now copies each symmetric
element for lower triangular matrix to the upper right part, because the
Matrix class in Ruby 1.9.2 no longer accepts any dimension mismatches. We
think the previous behavior is a bug.</p>

<h3 id="label-Bio-3A-3AMEDLINE"><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a><span><a href="#label-Bio-3A-3AMEDLINE">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p><a href="../Bio/MEDLINE.html#method-i-reference">Bio::MEDLINE#reference</a>
no longer puts empty values in the returned <a
href="../Bio/Reference.html">Bio::Reference</a> object. We think the
previous behavior is a bug. We also think the effect is very small.</p>

<h2 id="label-Known+issues">Known issues<span><a href="#label-Known+issues">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>The following issues are added or updated. See <a
href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a> for other already
known issues.</p>

<h3 id="label-String+escaping+of+command-line+arguments+in+Ruby+1.9.X+on+Windows"><a href="../String.html">String</a> escaping of command-line arguments in Ruby 1.9.X on Windows<span><a href="#label-String+escaping+of+command-line+arguments+in+Ruby+1.9.X+on+Windows">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>After BioRuby 1.4.1, in Ruby 1.9.X running on Windows, escaping of
command-line arguments are processed by the Ruby interpreter. Before
BioRuby 1.4.0, the escaping is executed in <a
href="../Bio/Command.html#method-i-escape_shell_windows">Bio::Command#escape_shell_windows</a>,
and the behavior is different from the Ruby interpreter&#39;s one.</p>

<p>Curreltly, due to the change, test/functional/bio/test_command.rb may fail
on Windows with Ruby 1.9.X.</p>
</main>



<footer id="validator-badges" role="contentinfo">
  <p><a href="http://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="http://docs.seattlerb.org/rdoc/">RDoc</a> 4.2.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

