<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>RELEASE_NOTES-1.4.0 - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
  var index_rel_prefix = "../";
</script>

<script src="../js/navigation.js" defer></script>
<script src="../js/search.js" defer></script>
<script src="../js/search_index.js" defer></script>
<script src="../js/searcher.js" defer></script>
<script src="../js/darkfish.js" defer></script>

<link href="../css/fonts.css" rel="stylesheet">
<link href="../css/rdoc.css" rel="stylesheet">


<body id="top" role="document" class="file">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../table_of_contents.html#pages">Pages</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>



  <ul class="link-list" role="directory">
              <li>
            <details open>
              <summary>      <a href="#label-BioRuby+1.4.0+RELEASE+NOTES">BioRuby 1.4.0 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>
            <details open>
              <summary>      <a href="#label-New+features">New features</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-PhyloXML+support">PhyloXML support</a>
          <li>      <a href="#label-FASTQ+file+format+support">FASTQ file format support</a>
          <li>      <a href="#label-DNA+chromatogram+support">DNA chromatogram support</a>
          <li>      <a href="#label-MEME+-28motif-based+sequence+analysis+tools-29+support">MEME (motif-based sequence analysis tools) support</a>
          <li>      <a href="#label-Improvement+of+KEGG+parser+classes">Improvement of KEGG parser classes</a>
          <li>      <a href="#label-Many+sample+scripts+are+added">Many sample scripts are added</a>
          <li>      <a href="#label-Unit+tests+can+test+installed+BioRuby">Unit tests can test installed BioRuby</a>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-Deprecated+features">Deprecated features</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-ChangeLog+is+replaced+by+git+log">ChangeLog is replaced by git log</a>
          <li>      <a href="#label-22if+__FILE__+-3D-3D+-240-22+convention">“if __FILE__ == $0” convention</a>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-Incompatible+changes">Incompatible changes</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Bio-3A-3ANCBI-3A-3AREST">Bio::NCBI::REST</a>
          <li>
            <details open>
              <summary>      <a href="#label-Bio-3A-3AKEGG">Bio::KEGG</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-dblinks+method">dblinks method</a>
          <li>      <a href="#label-pathways+method">pathways method</a>
          <li>      <a href="#label-orthologs+method">orthologs method</a>
          <li>      <a href="#label-genes+method">genes method</a>
          <li>      <a href="#label-Bio-3A-3AKEGG-3AREACTION-23rpairs">Bio::KEGG:REACTION#rpairs</a>
          <li>      <a href="#label-Bio-3A-3AKEGG-3A-3AORTHOLOGY">Bio::KEGG::ORTHOLOGY</a>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-Bio-3A-3ARestrictionEnzyme">Bio::RestrictionEnzyme</a>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-Known+problems">Known problems</a>

              </ul>
            </details>
          </li>

  </ul>
</div>


  <div id="project-metadata">
    
<div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
    <li><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
    <li><a href="../README_rdoc.html">README</a>
    <li><a href="../README_DEV_rdoc.html">README_DEV</a>
    <li><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
    <li><details open><summary>doc</summary>
    <ul class="link-list">
      <li><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
      <li><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
      <li><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
      <li><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
      <li><a href="../doc/RELEASE_NOTES-1_4_3_rdoc.html">RELEASE_NOTES-1.4.3</a>
      <li><a href="../doc/RELEASE_NOTES-1_5_0_rdoc.html">RELEASE_NOTES-1.5.0</a>
    </ul></details>
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-label="Page doc/RELEASE_NOTES-1.4.0.rdoc">

<h1 id="label-BioRuby+1.4.0+RELEASE+NOTES">BioRuby 1.4.0 RELEASE NOTES<span><a href="#label-BioRuby+1.4.0+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>A lot of changes have been made to the BioRuby 1.4.0 after the version 1.3.1 is released. This document describes important and/or incompatible changes since the BioRuby 1.3.1 release.</p>

<h2 id="label-New+features">New features<span><a href="#label-New+features">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-PhyloXML+support">PhyloXML support<span><a href="#label-PhyloXML+support">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Support for reading and writing PhyloXML file format is added. New classes Bio::PhyloXML::Parser and Bio::PhyloXML::Writer are used to read and write a PhyloXML file, respectively.</p>

<p>The code is developed by Diana Jaunzeikare, mentored by Christian M Zmasek and co-mentors, supported by Google Summer of Code 2009 in collaboration with the National Evolutionary Synthesis Center (NESCent).</p>

<h3 id="label-FASTQ+file+format+support">FASTQ file format support<span><a href="#label-FASTQ+file+format+support">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Support for reading and writing FASTQ file format is added. All of the three FASTQ format variants are supported.</p>

<p>To read a FASTQ file, <a href="../Bio/FlatFile.html"><code>Bio::FlatFile</code></a> can be used. File format auto-detection of the FASTQ format is supported (although the three format variants should be specified later by users if quality scores are needed).</p>

<p>New class <a href="../Bio/Fastq.html"><code>Bio::Fastq</code></a> is the parser class for the FASTQ format. An object of the <a href="../Bio/Fastq.html"><code>Bio::Fastq</code></a> class can be converted to a <a href="../Bio/Sequence.html"><code>Bio::Sequence</code></a> object with the “to_biosequnece” method. <a href="../Bio/Sequence/Format.html#method-i-output"><code>Bio::Sequence#output</code></a> now supports output of the FASTQ format.</p>

<p>The code is written by Naohisa Goto, with the help of discussions in the open-bio-l mailing list. The prototype of <a href="../Bio/Fastq.html"><code>Bio::Fastq</code></a> class was first developed during the BioHackathon 2009 held in Okinawa.</p>

<h3 id="label-DNA+chromatogram+support">DNA chromatogram support<span><a href="#label-DNA+chromatogram+support">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Support for reading DNA chromatogram files are added. SCF and and ABIF file formats are supported. The code is developed by Anthony Underwood.</p>

<h3 id="label-MEME+-28motif-based+sequence+analysis+tools-29+support">MEME (motif-based sequence analysis tools) support<span><a href="#label-MEME+-28motif-based+sequence+analysis+tools-29+support">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Support for running MAST (Motif Aliginment &amp; Search Tool, part of the MEME Suite, motif-based sequence analysis tools) and parsing its results are added. The code is developed by Adam Kraut.</p>

<h3 id="label-Improvement+of+KEGG+parser+classes">Improvement of KEGG parser classes<span><a href="#label-Improvement+of+KEGG+parser+classes">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Some new methods are added to parse new fields added to some KEGG file formats. Unit tests for KEGG parsers are also added and improved. In addition, return value types of some methods are also changed for unifying APIs among KEGG parser classes. See incompatible changes below for details. Most of them are contributed by Kozo Nishida.</p>

<h3 id="label-Many+sample+scripts+are+added">Many sample scripts are added<span><a href="#label-Many+sample+scripts+are+added">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Many sample scripts showing demonstrations of usages of classes are added. They are moved from primitive test codes for the classes described in the “if __FILE__ == $0” convention in the library files.</p>

<h3 id="label-Unit+tests+can+test+installed+BioRuby">Unit tests can test installed BioRuby<span><a href="#label-Unit+tests+can+test+installed+BioRuby">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Mechanism to load library and to find test data in the unit tests are changed, and the library path and test data path can be specified with environment variables. BIORUBY_TEST_LIB is the path to be added to the Ruby’s $LOAD_PATH. For example, to test BioRuby installed in /usr/local/lib/site_ruby/1.8, run</p>

<pre>env BIORUBY_TEST_LIB=/usr/local/lib/site_ruby/1.8 ruby test/runner.rb</pre>

<p>BIORUBY_TEST_DATA is the path of the test data, and BIORUBY_TEST_DEBUG is a flag to turn on debug of the tests.</p>

<h2 id="label-Deprecated+features">Deprecated features<span><a href="#label-Deprecated+features">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-ChangeLog+is+replaced+by+git+log">ChangeLog is replaced by git log<span><a href="#label-ChangeLog+is+replaced+by+git+log">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>ChangeLog is replaced by the output of git-log command, and ChangeLog before the 1.3.1 release is moved to doc/ChangeLog-before-1.3.1.</p>

<h3 id="label-22if+__FILE__+-3D-3D+-240-22+convention">“if __FILE__ == $0” convention<span><a href="#label-22if+__FILE__+-3D-3D+-240-22+convention">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Primitive test codes in the “if __FILE__ == $0” convention are removed and the codes are moved to the sample scripts named sample/demo_*.rb (except some older or deprecated files).</p>

<h2 id="label-Incompatible+changes">Incompatible changes<span><a href="#label-Incompatible+changes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ANCBI-3A-3AREST"><a href="../Bio/NCBI/REST.html"><code>Bio::NCBI::REST</code></a><span><a href="#label-Bio-3A-3ANCBI-3A-3AREST">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>NCBI announces that all Entrez E-utility requests must contain email and tool parameters, and requests without them will return error after June 2010.</p>

<p>To set default email address and tool name, following methods are added.</p>
<ul><li>
<p><a href="../Bio/NCBI.html#method-c-default_email-3D"><code>Bio::NCBI.default_email=(email)</code></a></p>
</li><li>
<p><a href="../Bio/NCBI.html#method-c-default_tool-3D"><code>Bio::NCBI.default_tool=(tool_name)</code></a></p>
</li></ul>

<p>For every query, <a href="../Bio/NCBI/REST.html"><code>Bio::NCBI::REST</code></a> checks the email and tool parameters and raises error if they are empty.</p>

<p>IMPORTANT NOTE: No default email address is preset in BioRuby. Programmers using BioRuby must set their own email address or implement to get user’s email address in some way (from input form, configuration file, etc).</p>

<p>Default tool name is set as “#{$0} (bioruby/#{Bio::BIORUBY_VERSION_ID})”. For example, if you run “ruby my_script.rb” with BioRuby 1.4.0, the value is “my_script.rb (bioruby/1.4.0)”.</p>

<h3 id="label-Bio-3A-3AKEGG"><a href="../Bio/KEGG.html"><code>Bio::KEGG</code></a><span><a href="#label-Bio-3A-3AKEGG">&para;</a> <a href="#top">&uarr;</a></span></h3>

<h4 id="label-dblinks+method">dblinks method<span><a href="#label-dblinks+method">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>In <a href="../Bio/KEGG/COMPOUND.html"><code>Bio::KEGG::COMPOUND</code></a>, DRUG, ENZYME, GLYCAN and ORTHOLOGY, the method dblinks is changed to return a Hash. Each key of the hash is a database name and its value is an array of entry IDs in the database. If old behavior (returns raw entry lines as an array of strings) is needed, use dblinks_as_strings.</p>

<h4 id="label-pathways+method">pathways method<span><a href="#label-pathways+method">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>In <a href="../Bio/KEGG/COMPOUND.html"><code>Bio::KEGG::COMPOUND</code></a>, DRUG, ENZYME, GENES, GLYCAN and REACTION, the method pathways is changed to return a Hash. Each key of the hash is a pathway ID and its value is the description of the pathway.</p>

<p>In <a href="../Bio/KEGG/GENES.html"><code>Bio::KEGG::GENES</code></a>, if old behavior (returns pathway IDs as an Array) is needed, use pathways.keys.</p>

<p>In <a href="../Bio/KEGG/COMPOUND.html"><code>Bio::KEGG::COMPOUND</code></a>, DRUG, ENZYME, GLYCAN, and REACTION, if old behavior (returns raw entry lines as an array of strings) is needed, use pathways_as_strings.</p>

<p>Note that <a href="../Bio/KEGG/ORTHOLOGY.html#method-i-pathways"><code>Bio::KEGG::ORTHOLOGY#pathways</code></a> is not changed (returns an array containing pathway IDs).</p>

<h4 id="label-orthologs+method">orthologs method<span><a href="#label-orthologs+method">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>In <a href="../Bio/KEGG/ENZYME.html"><code>Bio::KEGG::ENZYME</code></a>, GENES, GLYCAN and REACTION, the method orthologs is changed to return a Hash. Each key of the hash is a ortholog ID and its value is the name of the ortholog. If old behavior (returns raw entry lines as an array of strings) is needed, use orthologs_as_strings.</p>

<h4 id="label-genes+method">genes method<span><a href="#label-genes+method">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>In <a href="../Bio/KEGG/ENZYME.html#method-i-genes"><code>Bio::KEGG::ENZYME#genes</code></a> and <a href="../Bio/KEGG/ORTHOLOGY.html#method-i-genes"><code>Bio::KEGG::ORTHOLOGY#genes</code></a> is changed to return a Hash that is the same as <a href="../Bio/KEGG/ORTHOLOGY.html#method-i-genes_as_hash"><code>Bio::KEGG::ORTHOLOGY#genes_as_hash</code></a>. If old behavior (returns raw entry lines as an array of strings) is needed, use genes_as_strings.</p>

<h4 id="label-Bio-3A-3AKEGG-3AREACTION-23rpairs"><a href="../Bio/KEGG.html"><code>Bio::KEGG</code></a>:REACTION#rpairs<span><a href="#label-Bio-3A-3AKEGG-3AREACTION-23rpairs">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p><a href="../Bio/KEGG/REACTION.html#method-i-rpairs"><code>Bio::KEGG::REACTION#rpairs</code></a> is changed to return a Hash. Each key of the hash is a KEGG Rpair ID and its value is an array containing name and type. If old behavior (returns as tokens) is needed, use rpairs_as_tokens.</p>

<h4 id="label-Bio-3A-3AKEGG-3A-3AORTHOLOGY"><a href="../Bio/KEGG/ORTHOLOGY.html"><code>Bio::KEGG::ORTHOLOGY</code></a><span><a href="#label-Bio-3A-3AKEGG-3A-3AORTHOLOGY">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p><a href="../Bio/KEGG.html"><code>Bio::KEGG</code></a>:ORTHOLOGY#dblinks_as_hash does not lower-case database names.</p>

<h3 id="label-Bio-3A-3ARestrictionEnzyme"><a href="../Bio/RestrictionEnzyme.html"><code>Bio::RestrictionEnzyme</code></a><span><a href="#label-Bio-3A-3ARestrictionEnzyme">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Format validation when creating an object is turned off because of efficiency.</p>

<h2 id="label-Known+problems">Known problems<span><a href="#label-Known+problems">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>See <a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a> for details.</p>

</main>



<footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.5.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

