<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>module Bio - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "./";
  var index_rel_prefix = "./";
</script>

<script src="./js/navigation.js" defer></script>
<script src="./js/search.js" defer></script>
<script src="./js/search_index.js" defer></script>
<script src="./js/searcher.js" defer></script>
<script src="./js/darkfish.js" defer></script>

<link href="./css/fonts.css" rel="stylesheet">
<link href="./css/rdoc.css" rel="stylesheet">


<body id="top" role="document" class="module">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="./index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="./table_of_contents.html#pages">Pages</a>
    <a href="./table_of_contents.html#classes">Classes</a>
    <a href="./table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#module-Bio-label-bio.rb+-+Loading+all+BioRuby+modules">bio.rb - Loading all BioRuby modules</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fbl2seq-2Freport.rb+-+bl2seq+-28BLAST+2+sequences-29+parser">bio/appl/bl2seq/report.rb - bl2seq (BLAST 2 sequences) parser</a>
    <li><a href="#module-Bio-label-Acknowledgements">Acknowledgements</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fblast-2Fformat8.rb+-+BLAST+tab-delimited+output+-28-m+8-29+parser">bio/appl/blast/format8.rb - BLAST tab-delimited output (-m 8) parser</a>
    <li><a href="#module-Bio-label-Note">Note</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fblast-2Fremote.rb+-+remote+BLAST+wrapper+basic+module">bio/appl/blast/remote.rb - remote BLAST wrapper basic module</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fblast-2Fwublast.rb+-+WU-BLAST+default+output+parser">bio/appl/blast/wublast.rb - WU-BLAST default output parser</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Femboss.rb+-+EMBOSS+wrapper">bio/appl/emboss.rb - EMBOSS wrapper</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fgcg-2Fseq.rb+-+GCG+sequence+file+format+class+-28.seq-2F.pep+file-29">bio/appl/gcg/seq.rb - GCG sequence file format class (.seq/.pep file)</a>
    <li><a href="#module-Bio-label-About+Bio-3A-3AGCG-3A-3ASeq">About Bio::GCG::Seq</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fiprscan-2Freport.rb+-+a+class+for+iprscan+output.">bio/appl/iprscan/report.rb - a class for iprscan output.</a>
    <li><a href="#module-Bio-label-Report+classes+for+the+iprscan+program.">Report classes for the iprscan program.</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fmeme-2Fmotif.rb+-+Class+to+represent+a+sequence+motif">bio/appl/meme/motif.rb - Class to represent a sequence motif</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fmuscle.rb+-+MUSCLE+application+wrapper+class">bio/appl/muscle.rb - MUSCLE application wrapper class</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fphylip-2Falignment.rb+-+phylip+multiple+alignment+format+parser">bio/appl/phylip/alignment.rb - phylip multiple alignment format parser</a>
    <li><a href="#module-Bio-label-About+Bio-3A-3APhylip-3A-3APhylipFormat">About Bio::Phylip::PhylipFormat</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fprobcons.rb+-+ProbCons+application+wrapper+class">bio/appl/probcons.rb - ProbCons application wrapper class</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fsim4-2Freport.rb+-+sim4+result+parser">bio/appl/sim4/report.rb - sim4 result parser</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Fsosui-2Freport.rb+-+SOSUI+report+class">bio/appl/sosui/report.rb - SOSUI report class</a>
    <li><a href="#module-Bio-label-Example">Example</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Ftargetp-2Freport.rb+-+TargetP+report+class">bio/appl/targetp/report.rb - TargetP report class</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-Example">Example</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fappl-2Ftcoffee.rb+-+T-Coffee+application+wrapper+class">bio/appl/tcoffee.rb - T-Coffee application wrapper class</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fcompat-2Freferences.rb+-+Obsoleted+References+class">bio/compat/references.rb - Obsoleted References class</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-bio-2Fdata-2Faa.rb+-+Amino+Acids">bio/data/aa.rb - Amino Acids</a>
    <li><a href="#module-Bio-label-bio-2Fdata-2Fcodontable.rb+-+Codon+Table">bio/data/codontable.rb - Codon Table</a>
    <li><a href="#module-Bio-label-Data+source">Data source</a>
    <li><a href="#module-Bio-label-Examples">Examples</a>
    <li><a href="#module-Bio-label-bio-2Fdata-2Fna.rb+-+Nucleic+Acids">bio/data/na.rb - Nucleic Acids</a>
    <li><a href="#module-Bio-label-Synopsis">Synopsis</a>
    <li><a href="#module-Bio-label-Usage">Usage</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Ffasta-2Fdefline.rb+-+FASTA+defline+parser+class">bio/db/fasta/defline.rb - FASTA defline parser class</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-Examples">Examples</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fkegg-2Fcommon.rb+-+Common+methods+for+KEGG+database+classes">bio/db/kegg/common.rb - Common methods for KEGG database classes</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fkegg-2Fgenes.rb+-+KEGG-2FGENES+database+class">bio/db/kegg/genes.rb - KEGG/GENES database class</a>
    <li><a href="#module-Bio-label-KEGG+GENES+parser">KEGG GENES parser</a>
    <li><a href="#module-Bio-label-Examples">Examples</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fkegg-2Fkeggtab.rb+-+KEGG+keggtab+class">bio/db/kegg/keggtab.rb - KEGG keggtab class</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fchain.rb+-+chain+class+for+PDB">bio/db/pdb/chain.rb - chain class for PDB</a>
    <li><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AChain">Bio::PDB::Chain</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fchemicalcomponent.rb+-+PDB+Chemical+Component+Dictionary+parser">bio/db/pdb/chemicalcomponent.rb - PDB Chemical Component Dictionary parser</a>
    <li><a href="#module-Bio-label-About+Bio-3A-3APDB-3A-3AChemicalComponent">About Bio::PDB::ChemicalComponent</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fmodel.rb+-+model+class+for+PDB">bio/db/pdb/model.rb - model class for PDB</a>
    <li><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AModel">Bio::PDB::Model</a>
    <li><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fresidue.rb+-+residue+class+for+PDB">bio/db/pdb/residue.rb - residue class for PDB</a>
    <li><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AResidue">Bio::PDB::Residue</a>
    <li><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AHeterogen">Bio::PDB::Heterogen</a>
    <li><a href="#module-Bio-label-bio-2Ffeature.rb+-+Features-2FFeature+class+-28GenBank+Feature+table-29">bio/feature.rb - Features/Feature class (GenBank Feature table)</a>
    <li><a href="#module-Bio-label-bio-2Fio-2Fflatfile.rb+-+flatfile+access+wrapper+class">bio/io/flatfile.rb - flatfile access wrapper class</a>
    <li><a href="#module-Bio-label-bio-2Flocation.rb+-+Locations-2FLocation+class+-28GenBank+location+format-29">bio/location.rb - Locations/Location class (GenBank location format)</a>
    <li><a href="#module-Bio-label-bio-2Fsequence.rb+-+biological+sequence+class">bio/sequence.rb - biological sequence class</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Faa.rb+-+amino+acid+sequence+class">bio/sequence/aa.rb - amino acid sequence class</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fadapter.rb+-+Bio-3A-3ASequence+adapter+helper+module">bio/sequence/adapter.rb - Bio::Sequence adapter helper module</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fcommon.rb+-+common+methods+for+biological+sequence">bio/sequence/common.rb - common methods for biological sequence</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fcompat.rb+-+methods+for+backward+compatibility">bio/sequence/compat.rb - methods for backward compatibility</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fdblink.rb+-+sequence+ID+with+database+name">bio/sequence/dblink.rb - sequence ID with database name</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fgeneric.rb+-+generic+sequence+class+to+store+an+intact+string">bio/sequence/generic.rb - generic sequence class to store an intact string</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fna.rb+-+nucleic+acid+sequence+class">bio/sequence/na.rb - nucleic acid sequence class</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fquality_score.rb+-+Sequence+quality+score+manipulation+modules">bio/sequence/quality_score.rb - Sequence quality score manipulation modules</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fsequence-2Fsequence_masker.rb+-+Sequence+masking+helper+methods">bio/sequence/sequence_masker.rb - Sequence masking helper methods</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-bio-2Ftree-2Foutput.rb+-+Phylogenetic+tree+formatter">bio/tree/output.rb - Phylogenetic tree formatter</a>
    <li><a href="#module-Bio-label-Description">Description</a>
    <li><a href="#module-Bio-label-References">References</a>
    <li><a href="#module-Bio-label-bio-2Fversion.rb+-+BioRuby+version+information">bio/version.rb - BioRuby version information</a>
  </ul>
</div>


  <div id="class-metadata">
    
    
    
    
    
  </div>
</nav>

<main role="main" aria-labelledby="module-Bio">
  <h1 id="module-Bio" class="module">
    module Bio
  </h1>

  <section class="description">
    
<h1 id="module-Bio-label-bio.rb+-+Loading+all+BioRuby+modules">bio.rb - Loading all BioRuby modules<span><a href="#module-Bio-label-bio.rb+-+Loading+all+BioRuby+modules">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001-2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<p>+</p>

<h1 id="module-Bio-label-bio-2Fappl-2Fbl2seq-2Freport.rb+-+bl2seq+-28BLAST+2+sequences-29+parser">bio/appl/bl2seq/report.rb - bl2seq (BLAST 2 sequences) parser<span><a href="#module-Bio-label-bio-2Fappl-2Fbl2seq-2Freport.rb+-+bl2seq+-28BLAST+2+sequences-29+parser">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2005 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p><a href="Bio/Blast/Bl2seq/Report.html"><code>Bio::Blast::Bl2seq::Report</code></a> is a <a href="Bio/NCBI.html"><code>NCBI</code></a> bl2seq (BLAST 2 sequences) output parser.</p>

<h1 id="module-Bio-label-Acknowledgements">Acknowledgements<span><a href="#module-Bio-label-Acknowledgements">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Thanks to Tomoaki NISHIYAMA &lt;tomoakin __at__ kenroku.kanazawa-u.ac.jp&gt;  for providing bl2seq parser patches based on lib/bio/appl/blast/format0.rb.</p>

<h1 id="module-Bio-label-bio-2Fappl-2Fblast-2Fformat8.rb+-+BLAST+tab-delimited+output+-28-m+8-29+parser">bio/appl/blast/format8.rb - BLAST tab-delimited output (-m 8) parser<span><a href="#module-Bio-label-bio-2Fappl-2Fblast-2Fformat8.rb+-+BLAST+tab-delimited+output+-28-m+8-29+parser">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2002, 2003, 2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="module-Bio-label-Note">Note<span><a href="#module-Bio-label-Note">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>This file is automatically loaded by bio/appl/blast/report.rb</p>

<h1 id="module-Bio-label-bio-2Fappl-2Fblast-2Fremote.rb+-+remote+BLAST+wrapper+basic+module">bio/appl/blast/remote.rb - remote BLAST wrapper basic module<span><a href="#module-Bio-label-bio-2Fappl-2Fblast-2Fremote.rb+-+remote+BLAST+wrapper+basic+module">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008  Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fappl-2Fblast-2Fwublast.rb+-+WU-BLAST+default+output+parser">bio/appl/blast/wublast.rb - WU-BLAST default output parser<span><a href="#module-Bio-label-bio-2Fappl-2Fblast-2Fwublast.rb+-+WU-BLAST+default+output+parser">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003, 2008 Naohisa GOTO &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>WU-BLAST default output parser.</p>

<p>The parser is still incomplete and may contain many bugs, because I didn&#39;t have WU-BLAST license. It was tested under web-based WU-BLAST results and obsolete version downloaded from <a href="http://blast.wustl.edu">blast.wustl.edu</a>/ .</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://blast.wustl.edu">blast.wustl.edu</a>/</p>
</li><li>
<p><a href="http://www.ebi.ac.uk/blast2">www.ebi.ac.uk/blast2</a>/</p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fappl-2Femboss.rb+-+EMBOSS+wrapper">bio/appl/emboss.rb - <a href="Bio/EMBOSS.html"><code>EMBOSS</code></a> wrapper<span><a href="#module-Bio-label-bio-2Fappl-2Femboss.rb+-+EMBOSS+wrapper">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2002, 2005 Toshiaki Katayama&lt;k@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2006       Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: emboss.rb,v 1.9 2008/01/10 03:51:06 ngoto Exp $</p>

<p>+</p>

<h1 id="module-Bio-label-bio-2Fappl-2Fgcg-2Fseq.rb+-+GCG+sequence+file+format+class+-28.seq-2F.pep+file-29">bio/appl/gcg/seq.rb - <a href="Bio/GCG.html"><code>GCG</code></a> sequence file format class (.seq/.pep file)<span><a href="#module-Bio-label-bio-2Fappl-2Fgcg-2Fseq.rb+-+GCG+sequence+file+format+class+-28.seq-2F.pep+file-29">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003, 2006 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: seq.rb,v 1.3 2007/04/05 23:35:39 trevor Exp $</p>

<h1 id="module-Bio-label-About+Bio-3A-3AGCG-3A-3ASeq">About <a href="Bio/GCG/Seq.html"><code>Bio::GCG::Seq</code></a><span><a href="#module-Bio-label-About+Bio-3A-3AGCG-3A-3ASeq">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Please refer document of <a href="Bio/GCG/Seq.html"><code>Bio::GCG::Seq</code></a>.</p>

<h1 id="module-Bio-label-bio-2Fappl-2Fiprscan-2Freport.rb+-+a+class+for+iprscan+output.">bio/appl/iprscan/report.rb - a class for iprscan output.<span><a href="#module-Bio-label-bio-2Fappl-2Fiprscan-2Freport.rb+-+a+class+for+iprscan+output.">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Mitsuteru C. Nakao &lt;mn@kazusa.or.jp&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Report+classes+for+the+iprscan+program.">Report classes for the iprscan program.<span><a href="#module-Bio-label-Report+classes+for+the+iprscan+program.">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h1 id="module-Bio-label-bio-2Fappl-2Fmeme-2Fmotif.rb+-+Class+to+represent+a+sequence+motif">bio/appl/meme/motif.rb - Class to represent a sequence motif<span><a href="#module-Bio-label-bio-2Fappl-2Fmeme-2Fmotif.rb+-+Class+to+represent+a+sequence+motif">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Adam Kraut &lt;adamnkraut@gmail.com&gt;,</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>This file contains a minimal class to represent meme motifs</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://meme.sdsc.edu/meme/intro.html">meme.sdsc.edu/meme/intro.html</a></p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fappl-2Fmuscle.rb+-+MUSCLE+application+wrapper+class">bio/appl/muscle.rb - MUSCLE application wrapper class<span><a href="#module-Bio-label-bio-2Fappl-2Fmuscle.rb+-+MUSCLE+application+wrapper+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: muscle.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Muscle.html"><code>Bio::Muscle</code></a> is a wrapper class to execute MUSCLE.</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://www.drive5.com/muscle">www.drive5.com/muscle</a>/</p>
</li><li>
<p>Edgar R.C. MUSCLE: multiple sequence alignment with high accuracy and high throughput. Nucleic Acids Res. 32: 1792-1797, 2004.</p>
</li><li>
<p>Edgar, R.C. MUSCLE: a multiple sequence alignment method with reduced time and space complexity. BMC Bioinformatics 5: 113, 2004.</p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fappl-2Fphylip-2Falignment.rb+-+phylip+multiple+alignment+format+parser">bio/appl/phylip/alignment.rb - phylip multiple alignment format parser<span><a href="#module-Bio-label-bio-2Fappl-2Fphylip-2Falignment.rb+-+phylip+multiple+alignment+format+parser">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 GOTO Naohisa &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: alignment.rb,v 1.2 2007/04/05 23:35:40 trevor Exp $</pre>

<h1 id="module-Bio-label-About+Bio-3A-3APhylip-3A-3APhylipFormat">About <a href="Bio/Phylip/PhylipFormat.html"><code>Bio::Phylip::PhylipFormat</code></a><span><a href="#module-Bio-label-About+Bio-3A-3APhylip-3A-3APhylipFormat">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Please refer document of <a href="Bio/Phylip/PhylipFormat.html"><code>Bio::Phylip::PhylipFormat</code></a> class.</p>

<h1 id="module-Bio-label-bio-2Fappl-2Fprobcons.rb+-+ProbCons+application+wrapper+class">bio/appl/probcons.rb - ProbCons application wrapper class<span><a href="#module-Bio-label-bio-2Fappl-2Fprobcons.rb+-+ProbCons+application+wrapper+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: probcons.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Probcons.html"><code>Bio::Probcons</code></a> is a wrapper class to execute ProbCons (Probabilistic Consistency-based Multiple <a href="Bio/Alignment.html"><code>Alignment</code></a> of Amino Acid Sequences).</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://probcons.stanford.edu">probcons.stanford.edu</a>/</p>
</li><li>
<p>Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. ProbCons: Probabilistic Consistency-based Multiple <a href="Bio/Sequence.html"><code>Sequence</code></a> <a href="Bio/Alignment.html"><code>Alignment</code></a>. Genome Research 15: 330-340, 2005.</p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fappl-2Fsim4-2Freport.rb+-+sim4+result+parser">bio/appl/sim4/report.rb - sim4 result parser<span><a href="#module-Bio-label-bio-2Fappl-2Fsim4-2Freport.rb+-+sim4+result+parser">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004 GOTO Naohisa &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>The sim4 report parser classes.</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Florea, L., et al., A Computer program for aligning a cDNA sequence with a genomic DNA sequence, Genome Research, 8, 967–974, 1998. <a href="http://www.genome.org/cgi/content/abstract/8/9/967">www.genome.org/cgi/content/abstract/8/9/967</a></p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fappl-2Fsosui-2Freport.rb+-+SOSUI+report+class">bio/appl/sosui/report.rb - <a href="Bio/SOSUI.html"><code>SOSUI</code></a> report class<span><a href="#module-Bio-label-bio-2Fappl-2Fsosui-2Freport.rb+-+SOSUI+report+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003  Mitsuteru C. Nakao &lt;n@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Example">Example<span><a href="#module-Bio-label-Example">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://bp.nuap.nagoya-u.ac.jp/sosui">bp.nuap.nagoya-u.ac.jp/sosui</a>/</p>
</li><li>
<p><a href="http://bp.nuap.nagoya-u.ac.jp/sosui/sosui_submit.html">bp.nuap.nagoya-u.ac.jp/sosui/sosui_submit.html</a></p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fappl-2Ftargetp-2Freport.rb+-+TargetP+report+class">bio/appl/targetp/report.rb - <a href="Bio/TargetP.html"><code>TargetP</code></a> report class<span><a href="#module-Bio-label-bio-2Fappl-2Ftargetp-2Freport.rb+-+TargetP+report+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003  Mitsuteru C. Nakao &lt;n@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p><a href="Bio/TargetP.html"><code>TargetP</code></a> class for <a href="http://www.cbs.dtu.dk/services/TargetP">www.cbs.dtu.dk/services/TargetP</a>/</p>

<h2 id="module-Bio-label-Example">Example<span><a href="#module-Bio-label-Example">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h1 id="module-Bio-label-bio-2Fappl-2Ftcoffee.rb+-+T-Coffee+application+wrapper+class">bio/appl/tcoffee.rb - T-Coffee application wrapper class<span><a href="#module-Bio-label-bio-2Fappl-2Ftcoffee.rb+-+T-Coffee+application+wrapper+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: tcoffee.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Tcoffee.html"><code>Bio::Tcoffee</code></a> is a wrapper class to execute T-Coffee.</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://www.tcoffee.org/Projects_home_page/t_coffee_home_page.html">www.tcoffee.org/Projects_home_page/t_coffee_home_page.html</a></p>
</li><li>
<p>Notredame, C., Higgins, D.G. and Heringa, J. T-Coffee: A novel method for fast and accurate multiple sequence alignment. J. Mol. Biol. 302: 205-217, 2000.</p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fcompat-2Freferences.rb+-+Obsoleted+References+class">bio/compat/references.rb - Obsoleted <a href="Bio/References.html"><code>References</code></a> class<span><a href="#module-Bio-label-bio-2Fcompat-2Freferences.rb+-+Obsoleted+References+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum &lt;ryan@raaum.org&gt;, Jan Aerts &lt;jandot@bioruby.org&gt;, Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: references.rb,v 1.1.2.1 2008/03/04 10:07:49 ngoto Exp $</p>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>The <a href="Bio/References.html"><code>Bio::References</code></a> class was obsoleted after BioRuby 1.2.1. To keep compatibility, some wrapper methods are provided in this file. As the compatibility methods (and <a href="Bio/References.html"><code>Bio::References</code></a>) will soon be removed, Please change your code not to use <a href="Bio/References.html"><code>Bio::References</code></a>.</p>

<p>Note that <a href="Bio/Reference.html"><code>Bio::Reference</code></a> is different from <a href="Bio/References.html"><code>Bio::References</code></a>. <a href="Bio/Reference.html"><code>Bio::Reference</code></a> still exists for storing a reference information in sequence entries.</p>

<h1 id="module-Bio-label-bio-2Fdata-2Faa.rb+-+Amino+Acids">bio/data/aa.rb - Amino Acids<span><a href="#module-Bio-label-bio-2Fdata-2Faa.rb+-+Amino+Acids">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h1 id="module-Bio-label-bio-2Fdata-2Fcodontable.rb+-+Codon+Table">bio/data/codontable.rb - Codon Table<span><a href="#module-Bio-label-bio-2Fdata-2Fcodontable.rb+-+Codon+Table">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2004 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Data+source">Data source<span><a href="#module-Bio-label-Data+source">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Data in this class is converted from the NCBI&#39;s genetic codes page.</p>

<pre>* ((&lt;URL:http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=t&gt;))</pre>

<h3 id="module-Bio-label-Examples">Examples<span><a href="#module-Bio-label-Examples">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Obtain a codon table No.1 – Standard (Eukaryote)</p>

<pre class="ruby"><span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-value">1</span>]
</pre>

<p>Obtain a copy of the codon table No.1 to modify.  In this example, reassign a seleno cystein (&#39;U&#39;) to the &#39;tga&#39; codon.</p>

<pre class="ruby"><span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>.<span class="ruby-identifier">copy</span>(<span class="ruby-value">1</span>)
<span class="ruby-identifier">table</span>[<span class="ruby-string">&#39;tga&#39;</span>] = <span class="ruby-string">&#39;U&#39;</span>
</pre>

<p>Create a new codon table by your own from the Hash which contains pairs of codon and amino acid.  You can also define the table name in the second argument.</p>

<pre>hash = { &#39;ttt&#39; =&gt; &#39;F&#39;, &#39;ttc&#39; =&gt; &#39;ttc&#39;, ... }
table = Bio::CodonTable.new(hash, &quot;my codon table&quot;)</pre>

<p>Obtain a translated amino acid by codon.</p>

<pre class="ruby"><span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-value">1</span>]
<span class="ruby-identifier">table</span>[<span class="ruby-string">&#39;ttt&#39;</span>]  <span class="ruby-comment"># =&gt; F</span>
</pre>

<p>Reverse translation of a amino acid into a list of relevant codons.</p>

<pre class="ruby"><span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-value">1</span>]
<span class="ruby-identifier">table</span>.<span class="ruby-identifier">revtrans</span>(<span class="ruby-string">&quot;A&quot;</span>)  <span class="ruby-comment"># =&gt; [&quot;gcg&quot;, &quot;gct&quot;, &quot;gca&quot;, &quot;gcc&quot;]</span>
</pre>

<h1 id="module-Bio-label-bio-2Fdata-2Fna.rb+-+Nucleic+Acids">bio/data/na.rb - Nucleic Acids<span><a href="#module-Bio-label-bio-2Fdata-2Fna.rb+-+Nucleic+Acids">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="module-Bio-label-Synopsis">Synopsis<span><a href="#module-Bio-label-Synopsis">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p><a href="Bio/NucleicAcid.html"><code>Bio::NucleicAcid</code></a> class contains data related to nucleic acids.</p>

<h2 id="module-Bio-label-Usage">Usage<span><a href="#module-Bio-label-Usage">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Examples:</p>

<pre class="ruby"><span class="ruby-identifier">require</span> <span class="ruby-string">&#39;bio&#39;</span>

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;### na = Bio::NucleicAcid.new&quot;</span>
<span class="ruby-identifier">na</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">new</span>

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;# na.to_re(&#39;yrwskmbdhvnatgc&#39;)&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">na</span>.<span class="ruby-identifier">to_re</span>(<span class="ruby-string">&#39;yrwskmbdhvnatgc&#39;</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;# Bio::NucleicAcid.to_re(&#39;yrwskmbdhvnatgc&#39;)&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">to_re</span>(<span class="ruby-string">&#39;yrwskmbdhvnatgc&#39;</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;# na.weight(&#39;A&#39;)&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">na</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">&#39;A&#39;</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;# Bio::NucleicAcid.weight(&#39;A&#39;)&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">&#39;A&#39;</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;# na.weight(&#39;atgc&#39;)&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">na</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">&#39;atgc&#39;</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;# Bio::NucleicAcid.weight(&#39;atgc&#39;)&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
</pre>

<p>require &#39;bio/sequence&#39;</p>

<h1 id="module-Bio-label-bio-2Fdb-2Ffasta-2Fdefline.rb+-+FASTA+defline+parser+class">bio/db/fasta/defline.rb - FASTA defline parser class<span><a href="#module-Bio-label-bio-2Fdb-2Ffasta-2Fdefline.rb+-+FASTA+defline+parser+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2002 GOTO Naohisa &lt;ngoto@gen-info.osaka-u.ac.jp&gt;, Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p><a href="Bio/FastaDefline.html"><code>Bio::FastaDefline</code></a> is a parser class for definition line (defline) of the FASTA format.</p>

<h2 id="module-Bio-label-Examples">Examples<span><a href="#module-Bio-label-Examples">&para;</a> <a href="#top">&uarr;</a></span></h2>

<pre>rub = Bio::FastaDefline.new(&#39;&gt;gi|671595|emb|CAA85678.1| rubisco large subunit [Perovskia abrotanoides]&#39;)
rub.entry_id       ==&gt; &#39;gi|671595&#39;
rub.get(&#39;emb&#39;)     ==&gt; &#39;CAA85678.1&#39;
rub.emb            ==&gt; &#39;CAA85678.1&#39;
rub.gi             ==&gt; &#39;671595&#39;
rub.accession      ==&gt; &#39;CAA85678&#39;
rub.accessions     ==&gt; [ &#39;CAA85678&#39; ]
rub.acc_version    ==&gt; &#39;CAA85678.1&#39;
rub.locus          ==&gt; nil
rub.list_ids       ==&gt; [[&quot;gi&quot;, &quot;671595&quot;],
                        [&quot;emb&quot;, &quot;CAA85678.1&quot;, nil],
                        [&quot;Perovskia abrotanoides&quot;]]

ckr = Bio::FastaDefline.new(&quot;&gt;gi|2495000|sp|Q63931|CCKR_CAVPO CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)\001gi|2147182|pir||I51898 cholecystokinin A receptor - guinea pig\001gi|544724|gb|AAB29504.1| cholecystokinin A receptor; CCK-A receptor [Cavia]&quot;)
ckr.entry_id      ==&gt; &quot;gi|2495000&quot;
ckr.sp            ==&gt; &quot;CCKR_CAVPO&quot;
ckr.pir           ==&gt; &quot;I51898&quot;
ckr.gb            ==&gt; &quot;AAB29504.1&quot;
ckr.gi            ==&gt; &quot;2495000&quot;
ckr.accession     ==&gt; &quot;AAB29504&quot;
ckr.accessions    ==&gt; [&quot;Q63931&quot;, &quot;AAB29504&quot;]
ckr.acc_version   ==&gt; &quot;AAB29504.1&quot;
ckr.locus         ==&gt; nil
ckr.description   ==&gt;
  &quot;CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)&quot;
ckr.descriptions  ==&gt;
  [&quot;CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)&quot;,
   &quot;cholecystokinin A receptor - guinea pig&quot;,
   &quot;cholecystokinin A receptor; CCK-A receptor [Cavia]&quot;]
ckr.words         ==&gt; 
  [&quot;cavia&quot;, &quot;cck-a&quot;, &quot;cck-ar&quot;, &quot;cholecystokinin&quot;, &quot;guinea&quot;, &quot;pig&quot;,
   &quot;receptor&quot;, &quot;type&quot;]
ckr.id_strings    ==&gt;
  [&quot;2495000&quot;, &quot;Q63931&quot;, &quot;CCKR_CAVPO&quot;, &quot;2147182&quot;, &quot;I51898&quot;,
   &quot;544724&quot;, &quot;AAB29504.1&quot;, &quot;Cavia&quot;]
ckr.list_ids      ==&gt;
  [[&quot;gi&quot;, &quot;2495000&quot;], [&quot;sp&quot;, &quot;Q63931&quot;, &quot;CCKR_CAVPO&quot;],
   [&quot;gi&quot;, &quot;2147182&quot;], [&quot;pir&quot;, nil, &quot;I51898&quot;], [&quot;gi&quot;, &quot;544724&quot;],
   [&quot;gb&quot;, &quot;AAB29504.1&quot;, nil], [&quot;Cavia&quot;]]</pre>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>FASTA format (WikiPedia) <a href="http://en.wikipedia.org/wiki/FASTA_format">en.wikipedia.org/wiki/FASTA_format</a></p>
</li><li>
<p><a href="Bio/Fasta.html"><code>Fasta</code></a> format description (<a href="Bio/NCBI.html"><code>NCBI</code></a>) <a href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml">www.ncbi.nlm.nih.gov/BLAST/fasta.shtml</a></p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fdb-2Fkegg-2Fcommon.rb+-+Common+methods+for+KEGG+database+classes">bio/db/kegg/common.rb - Common methods for <a href="Bio/KEGG.html"><code>KEGG</code></a> database classes<span><a href="#module-Bio-label-bio-2Fdb-2Fkegg-2Fcommon.rb+-+Common+methods+for+KEGG+database+classes">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001-2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2003 Masumi Itoh &lt;m@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2009 Kozo Nishida &lt;kozo-ni@is.naist.jp&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Note that the modules in this file are intended to be Bio::KEGG::*  internal use only.</p>

<p>This file contains modules that implement methods commonly used from <a href="Bio/KEGG.html"><code>KEGG</code></a> database parser classes.</p>

<h1 id="module-Bio-label-bio-2Fdb-2Fkegg-2Fgenes.rb+-+KEGG-2FGENES+database+class">bio/db/kegg/genes.rb - KEGG/GENES database class<span><a href="#module-Bio-label-bio-2Fdb-2Fkegg-2Fgenes.rb+-+KEGG-2FGENES+database+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2002, 2006, 2010 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="module-Bio-label-KEGG+GENES+parser"><a href="Bio/KEGG.html"><code>KEGG</code></a> GENES parser<span><a href="#module-Bio-label-KEGG+GENES+parser">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>See <a href="http://www.genome.jp/kegg/genes.html">www.genome.jp/kegg/genes.html</a></p>

<h3 id="module-Bio-label-Examples">Examples<span><a href="#module-Bio-label-Examples">&para;</a> <a href="#top">&uarr;</a></span></h3>

<pre class="ruby"><span class="ruby-identifier">require</span> <span class="ruby-string">&#39;bio/io/fetch&#39;</span>
<span class="ruby-identifier">entry_string</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Fetch</span>.<span class="ruby-identifier">query</span>(<span class="ruby-string">&#39;genes&#39;</span>, <span class="ruby-string">&#39;b0002&#39;</span>)

<span class="ruby-identifier">entry</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">GENES</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">entry_string</span>)

<span class="ruby-comment"># ENTRY</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">entry</span>       <span class="ruby-comment"># =&gt; Hash</span>

<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">entry_id</span>    <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">division</span>    <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">organism</span>    <span class="ruby-comment"># =&gt; String</span>

<span class="ruby-comment"># NAME</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">name</span>        <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">names</span>       <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># DEFINITION</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">definition</span>  <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">eclinks</span>     <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># PATHWAY</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">pathway</span>     <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">pathways</span>    <span class="ruby-comment"># =&gt; Hash</span>

<span class="ruby-comment"># POSITION</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">position</span>    <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">chromosome</span>  <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">gbposition</span>  <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">locations</span>   <span class="ruby-comment"># =&gt; Bio::Locations</span>

<span class="ruby-comment"># MOTIF</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">motifs</span>      <span class="ruby-comment"># =&gt; Hash of Array</span>

<span class="ruby-comment"># DBLINKS</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">dblinks</span>     <span class="ruby-comment"># =&gt; Hash of Array</span>

<span class="ruby-comment"># STRUCTURE</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">structure</span>   <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># CODON_USAGE</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">codon_usage</span> <span class="ruby-comment"># =&gt; Hash</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">cu_list</span>     <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># AASEQ</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">aaseq</span>       <span class="ruby-comment"># =&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">aalen</span>       <span class="ruby-comment"># =&gt; Fixnum</span>

<span class="ruby-comment"># NTSEQ</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">ntseq</span>       <span class="ruby-comment"># =&gt; Bio::Sequence::NA</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">naseq</span>       <span class="ruby-comment"># =&gt; Bio::Sequence::NA</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">ntlen</span>       <span class="ruby-comment"># =&gt; Fixnum</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">nalen</span>       <span class="ruby-comment"># =&gt; Fixnum</span>
</pre>

<h1 id="module-Bio-label-bio-2Fdb-2Fkegg-2Fkeggtab.rb+-+KEGG+keggtab+class">bio/db/kegg/keggtab.rb - <a href="Bio/KEGG.html"><code>KEGG</code></a> keggtab class<span><a href="#module-Bio-label-bio-2Fdb-2Fkegg-2Fkeggtab.rb+-+KEGG+keggtab+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001 Mitsuteru C. Nakao &lt;n@bioruby.org&gt; Copyright (C) 2003, 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: keggtab.rb,v 1.10 2007/04/05 23:35:41 trevor Exp $</pre>

<p>bio/db/lasergene.rb - Interface for DNAStar <a href="Bio/Lasergene.html"><code>Lasergene</code></a> sequence file format</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2007 Center for Biomedical Research Informatics, University of Minnesota (<a href="http://cbri.umn.edu">cbri.umn.edu</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>definition of the <a href="Bio/PDB.html"><code>PDB</code></a> class</p>

<h1 id="module-Bio-label-bio-2Fdb-2Fpdb-2Fchain.rb+-+chain+class+for+PDB">bio/db/pdb/chain.rb - chain class for <a href="Bio/PDB.html"><code>PDB</code></a><span><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fchain.rb+-+chain+class+for+PDB">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004, 2006 Alex Gutteridge &lt;alexg@ebi.ac.uk&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-Bio-3A-3APDB-3A-3AChain"><a href="Bio/PDB/Chain.html"><code>Bio::PDB::Chain</code></a><span><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AChain">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Please refer <a href="Bio/PDB/Chain.html"><code>Bio::PDB::Chain</code></a>.</p>

<h1 id="module-Bio-label-bio-2Fdb-2Fpdb-2Fchemicalcomponent.rb+-+PDB+Chemical+Component+Dictionary+parser">bio/db/pdb/chemicalcomponent.rb - <a href="Bio/PDB.html"><code>PDB</code></a> Chemical Component Dictionary parser<span><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fchemicalcomponent.rb+-+PDB+Chemical+Component+Dictionary+parser">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 GOTO Naohisa &lt;ngoto@gen-info.osaka-u.ac.jp&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-About+Bio-3A-3APDB-3A-3AChemicalComponent">About <a href="Bio/PDB/ChemicalComponent.html"><code>Bio::PDB::ChemicalComponent</code></a><span><a href="#module-Bio-label-About+Bio-3A-3APDB-3A-3AChemicalComponent">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Please refer <a href="Bio/PDB/ChemicalComponent.html"><code>Bio::PDB::ChemicalComponent</code></a>.</p>

<h1 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h1>
<ul><li>
<p>((&lt;URL:<a href="http://deposit.pdb.org/cc_dict_tut.html">deposit.pdb.org/cc_dict_tut.html</a>&gt;))</p>
</li><li>
<p><a href="http://deposit.pdb.org/het_dictionary.txt">deposit.pdb.org/het_dictionary.txt</a></p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fdb-2Fpdb-2Fmodel.rb+-+model+class+for+PDB">bio/db/pdb/model.rb - model class for <a href="Bio/PDB.html"><code>PDB</code></a><span><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fmodel.rb+-+model+class+for+PDB">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004, 2006 Alex Gutteridge &lt;alexg@ebi.ac.uk&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-Bio-3A-3APDB-3A-3AModel"><a href="Bio/PDB/Model.html"><code>Bio::PDB::Model</code></a><span><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AModel">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Please refer <a href="Bio/PDB/Model.html"><code>Bio::PDB::Model</code></a>.</p>

<h1 id="module-Bio-label-bio-2Fdb-2Fpdb-2Fresidue.rb+-+residue+class+for+PDB">bio/db/pdb/residue.rb - residue class for <a href="Bio/PDB.html"><code>PDB</code></a><span><a href="#module-Bio-label-bio-2Fdb-2Fpdb-2Fresidue.rb+-+residue+class+for+PDB">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004, 2006 Alex Gutteridge &lt;alexg@ebi.ac.uk&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-Bio-3A-3APDB-3A-3AResidue"><a href="Bio/PDB/Residue.html"><code>Bio::PDB::Residue</code></a><span><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AResidue">&para;</a> <a href="#top">&uarr;</a></span></h1>

<h1 id="module-Bio-label-Bio-3A-3APDB-3A-3AHeterogen"><a href="Bio/PDB/Heterogen.html"><code>Bio::PDB::Heterogen</code></a><span><a href="#module-Bio-label-Bio-3A-3APDB-3A-3AHeterogen">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>bio/db/soft.rb - Interface for <a href="Bio/SOFT.html"><code>SOFT</code></a> formatted files</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<h1 id="module-Bio-label-bio-2Ffeature.rb+-+Features-2FFeature+class+-28GenBank+Feature+table-29">bio/feature.rb - Features/Feature class (<a href="Bio/GenBank.html"><code>GenBank</code></a> <a href="Bio/Feature.html"><code>Feature</code></a> table)<span><a href="#module-Bio-label-bio-2Ffeature.rb+-+Features-2FFeature+class+-28GenBank+Feature+table-29">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright © 2002, 2005  Toshiaki Katayama &lt;k@bioruby.org&gt;</p>

<pre>2006        Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;</pre>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fio-2Fflatfile.rb+-+flatfile+access+wrapper+class">bio/io/flatfile.rb - flatfile access wrapper class<span><a href="#module-Bio-label-bio-2Fio-2Fflatfile.rb+-+flatfile+access+wrapper+class">&para;</a> <a href="#top">&uarr;</a></span></h1>

<pre>Copyright (C) 2001-2006 Naohisa Goto &lt;ng@bioruby.org&gt;</pre>
<dl class="rdoc-list note-list"><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p><a href="Bio/FlatFile.html"><code>Bio::FlatFile</code></a> is a helper and wrapper class to read a biological data file. It acts like a IO object. It can automatically detect data format, and users do not need to tell the class what the data is.</p>

<h1 id="module-Bio-label-bio-2Flocation.rb+-+Locations-2FLocation+class+-28GenBank+location+format-29">bio/location.rb - Locations/Location class (<a href="Bio/GenBank.html"><code>GenBank</code></a> location format)<span><a href="#module-Bio-label-bio-2Flocation.rb+-+Locations-2FLocation+class+-28GenBank+location+format-29">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>

<pre>2006       Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;
2008       Naohisa Goto &lt;ng@bioruby.org&gt;</pre>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h1 id="module-Bio-label-bio-2Fsequence.rb+-+biological+sequence+class">bio/sequence.rb - biological sequence class<span><a href="#module-Bio-label-bio-2Fsequence.rb+-+biological+sequence+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2000-2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Yoshinori K. Okuji &lt;okuji@enbug.org&gt;, Naohisa Goto &lt;ng@bioruby.org&gt;, Ryan Raaum &lt;ryan@raaum.org&gt;, Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Faa.rb+-+amino+acid+sequence+class">bio/sequence/aa.rb - amino acid sequence class<span><a href="#module-Bio-label-bio-2Fsequence-2Faa.rb+-+amino+acid+sequence+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum &lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fadapter.rb+-+Bio-3A-3ASequence+adapter+helper+module">bio/sequence/adapter.rb - <a href="Bio/Sequence.html"><code>Bio::Sequence</code></a> adapter helper module<span><a href="#module-Bio-label-bio-2Fsequence-2Fadapter.rb+-+Bio-3A-3ASequence+adapter+helper+module">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Naohisa Goto &lt;ng@bioruby.org&gt;,</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fcommon.rb+-+common+methods+for+biological+sequence">bio/sequence/common.rb - common methods for biological sequence<span><a href="#module-Bio-label-bio-2Fsequence-2Fcommon.rb+-+common+methods+for+biological+sequence">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum &lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fcompat.rb+-+methods+for+backward+compatibility">bio/sequence/compat.rb - methods for backward compatibility<span><a href="#module-Bio-label-bio-2Fsequence-2Fcompat.rb+-+methods+for+backward+compatibility">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum &lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fdblink.rb+-+sequence+ID+with+database+name">bio/sequence/dblink.rb - sequence ID with database name<span><a href="#module-Bio-label-bio-2Fsequence-2Fdblink.rb+-+sequence+ID+with+database+name">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fgeneric.rb+-+generic+sequence+class+to+store+an+intact+string">bio/sequence/generic.rb - generic sequence class to store an intact string<span><a href="#module-Bio-label-bio-2Fsequence-2Fgeneric.rb+-+generic+sequence+class+to+store+an+intact+string">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fna.rb+-+nucleic+acid+sequence+class">bio/sequence/na.rb - nucleic acid sequence class<span><a href="#module-Bio-label-bio-2Fsequence-2Fna.rb+-+nucleic+acid+sequence+class">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum &lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fsequence-2Fquality_score.rb+-+Sequence+quality+score+manipulation+modules">bio/sequence/quality_score.rb - <a href="Bio/Sequence.html"><code>Sequence</code></a> quality score manipulation modules<span><a href="#module-Bio-label-bio-2Fsequence-2Fquality_score.rb+-+Sequence+quality+score+manipulation+modules">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2009 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p><a href="Bio/Sequence.html"><code>Sequence</code></a> quality score manipulation modules, mainly used by <a href="Bio/Fastq.html"><code>Bio::Fastq</code></a> and related classes.</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>FASTQ format specification <a href="http://maq.sourceforge.net/fastq.shtml">maq.sourceforge.net/fastq.shtml</a></p>
</li></ul>

<h1 id="module-Bio-label-bio-2Fsequence-2Fsequence_masker.rb+-+Sequence+masking+helper+methods">bio/sequence/sequence_masker.rb - <a href="Bio/Sequence.html"><code>Sequence</code></a> masking helper methods<span><a href="#module-Bio-label-bio-2Fsequence-2Fsequence_masker.rb+-+Sequence+masking+helper+methods">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2010 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p><a href="Bio/Sequence/SequenceMasker.html"><code>Bio::Sequence::SequenceMasker</code></a> is a mix-in module to provide helpful methods for masking a sequence.</p>

<p>For details, see documentation of <a href="Bio/Sequence/SequenceMasker.html"><code>Bio::Sequence::SequenceMasker</code></a>.</p>

<h1 id="module-Bio-label-bio-2Ftree-2Foutput.rb+-+Phylogenetic+tree+formatter">bio/tree/output.rb - Phylogenetic tree formatter<span><a href="#module-Bio-label-bio-2Ftree-2Foutput.rb+-+Phylogenetic+tree+formatter">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004-2006 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="module-Bio-label-Description">Description<span><a href="#module-Bio-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>This file contains formatter of <a href="Bio/Newick.html"><code>Newick</code></a>, NHX and <a href="Bio/Phylip.html"><code>Phylip</code></a> distance matrix.</p>

<h2 id="module-Bio-label-References"><a href="Bio/References.html"><code>References</code></a><span><a href="#module-Bio-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="http://evolution.genetics.washington.edu/phylip/newick_doc.html">evolution.genetics.washington.edu/phylip/newick_doc.html</a></p>
</li><li>
<p><a href="http://www.phylosoft.org/forester/NHX.html">www.phylosoft.org/forester/NHX.html</a></p>
</li></ul>

<p>bio/util/color_scheme.rb - Popular color codings for nucleic and amino acids</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>bio/util/contingency_table.rb - Statistical contingency table analysis for aligned sequences</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>bio/util/restriction_enzyme.rb - Digests DNA based on restriction enzyme cut patterns</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/analysis.rb - Does the work of fragmenting the DNA from the enzymes</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/cut_symbol.rb - Defines the symbol used to mark a cut in an enzyme sequence</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/dense_int_array.rb - Internal data storage for <a href="Bio/RestrictionEnzyme/Range/SequenceRange.html"><code>Bio::RestrictionEnzyme::Range::SequenceRange</code></a></p>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2011 Naohisa Goto &lt;ng@bioruby.org&gt; Tomoaki NISHIYAMA</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded.rb - DoubleStranded restriction enzyme sequence</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/aligned_strands.rb - Align two SingleStrand objects</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_location_pair.rb - Stores a cut location pair in 0-based index notation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_location_pair_in_enzyme_notation.rb - Inherits from DoubleStranded::CutLocationPair</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_locations.rb - Contains an Array of CutLocationPair objects</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_locations_in_enzyme_notation.rb - Inherits from DoubleStrand::CutLocations</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/cut_range.rb - Abstract base class for HorizontalCutRange and VerticalCutRange</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/cut_ranges.rb - Container for many CutRange objects or CutRange child objects.</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/horizontal_cut_range.rb - </p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/sequence_range.rb - A defined range over a nucleotide sequence</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/sequence_range/calculated_cuts.rb - </p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/sequence_range/fragment.rb - </p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/analysis/fragments.rb - </p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/vertical_cut_range.rb - </p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/single_strand/cut_locations_in_enzyme_notation.rb - The cut locations, in enzyme notation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/single_strand_complement.rb - Single strand restriction enzyme sequence in complement orientation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/sorted_num_array.rb - Internal data storage for <a href="Bio/RestrictionEnzyme/Range/SequenceRange.html"><code>Bio::RestrictionEnzyme::Range::SequenceRange</code></a></p>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2011 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/string_formatting.rb - Useful functions for string manipulation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright © 2005-2007 Midwinter Laboratories, LLC (<a href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="module-Bio-label-bio-2Fversion.rb+-+BioRuby+version+information">bio/version.rb - BioRuby version information<span><a href="#module-Bio-label-bio-2Fversion.rb+-+BioRuby+version+information">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001-2012 Toshiaki Katayama &lt;k@bioruby.org&gt;, Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

  </section>

  <section id="5Buntitled-5D" class="documentation-section">


    <section class="constants-list">
      <header>
        <h3>Constants</h3>
      </header>
      <dl>
        <dt id="BIORUBY_EXTRA_VERSION">BIORUBY_EXTRA_VERSION
        <dd><p>Extra version specifier (String or nil). Existance of the value indicates development version.</p>
<dl class="rdoc-list note-list"><dt>nil             
<dd>
<p>Release version.</p>
</dd><dt>“.pre           
<dd>
<p>Pre-release version.</p>
</dd></dl>

<p>References: <a href="https://guides.rubygems.org/patterns/#prerelease-gems">guides.rubygems.org/patterns/#prerelease-gems</a></p>
        <dt id="BIORUBY_VERSION">BIORUBY_VERSION
        <dd><p>BioRuby version (Array containing Integer)</p>
        <dt id="BIORUBY_VERSION_ID">BIORUBY_VERSION_ID
        <dd><p>Version identifier, including extra version string (String) Unlike <a href="Bio.html#BIORUBY_VERSION"><code>BIORUBY_VERSION</code></a>, it is not comparable.</p>
        <dt id="SPTR">SPTR
        <dd><p><a href="Bio.html#SPTR"><code>Bio::SPTR</code></a> is changed to an alias of <a href="Bio/UniProtKB.html"><code>Bio::UniProtKB</code></a>. Please use <a href="Bio/UniProtKB.html"><code>Bio::UniProtKB</code></a>. <a href="Bio.html#SPTR"><code>Bio::SPTR</code></a> may be deprecated in the future.</p>
        <dt id="UniProt">UniProt
        <dd><p><a href="Bio/UniProtKB.html"><code>Bio::UniProt</code></a> is changed to an alias of <a href="Bio/UniProtKB.html"><code>Bio::UniProtKB</code></a>. Please use <a href="Bio/UniProtKB.html"><code>Bio::UniProtKB</code></a>. <a href="Bio/UniProtKB.html"><code>Bio::UniProt</code></a> may be deprecated in the future.</p>

<p>Note that <a href="Bio.html#SPTR"><code>Bio::SPTR</code></a> have been renamed to <a href="Bio/UniProtKB.html"><code>Bio::UniProtKB</code></a> and is also an alias of <a href="Bio/UniProtKB.html"><code>Bio::UniProtKB</code></a>.</p>
      </dl>
    </section>



  </section>
</main>


<footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.3.2.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

