<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />

  <title>Module: Bio</title>

  <link rel="stylesheet" href="./rdoc.css" type="text/css" media="screen" />

  <script src="./js/jquery.js" type="text/javascript" charset="utf-8"></script>
  <script src="./js/thickbox-compressed.js" type="text/javascript" charset="utf-8"></script>
  <script src="./js/quicksearch.js" type="text/javascript" charset="utf-8"></script>
  <script src="./js/darkfish.js" type="text/javascript" charset="utf-8"></script>

</head>
<body id="top" class="module">

  <div id="metadata">
    <div id="home-metadata">
      <div id="home-section" class="section">
        <h3 class="section-header">
          <a href="./index.html">Home</a>
          <a href="./index.html#classes">Classes</a>
          <a href="./index.html#methods">Methods</a>
        </h3>
      </div>
    </div>

    <div id="file-metadata">
      <div id="file-list-section" class="section">
        <h3 class="section-header">In Files</h3>
        <div class="section-body">
          <ul>
          
            <li><a href="./lib/bio_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio.rb">lib/bio.rb</a></li>
          
            <li><a href="./lib/bio/alignment_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/alignment.rb">lib/bio/alignment.rb</a></li>
          
            <li><a href="./lib/bio/appl/bl2seq/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/bl2seq/report.rb">lib/bio/appl/bl2seq/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast.rb">lib/bio/appl/blast.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/format0_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/format0.rb">lib/bio/appl/blast/format0.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/format8_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/format8.rb">lib/bio/appl/blast/format8.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/report.rb">lib/bio/appl/blast/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/rexml_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/rexml.rb">lib/bio/appl/blast/rexml.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/rpsblast_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/rpsblast.rb">lib/bio/appl/blast/rpsblast.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/wublast_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/wublast.rb">lib/bio/appl/blast/wublast.rb</a></li>
          
            <li><a href="./lib/bio/appl/blast/xmlparser_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blast/xmlparser.rb">lib/bio/appl/blast/xmlparser.rb</a></li>
          
            <li><a href="./lib/bio/appl/blat/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/blat/report.rb">lib/bio/appl/blat/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/clustalw_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/clustalw.rb">lib/bio/appl/clustalw.rb</a></li>
          
            <li><a href="./lib/bio/appl/clustalw/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/clustalw/report.rb">lib/bio/appl/clustalw/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/emboss_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/emboss.rb">lib/bio/appl/emboss.rb</a></li>
          
            <li><a href="./lib/bio/appl/fasta_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/fasta.rb">lib/bio/appl/fasta.rb</a></li>
          
            <li><a href="./lib/bio/appl/fasta/format10_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/fasta/format10.rb">lib/bio/appl/fasta/format10.rb</a></li>
          
            <li><a href="./lib/bio/appl/gcg/msf_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/gcg/msf.rb">lib/bio/appl/gcg/msf.rb</a></li>
          
            <li><a href="./lib/bio/appl/gcg/seq_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/gcg/seq.rb">lib/bio/appl/gcg/seq.rb</a></li>
          
            <li><a href="./lib/bio/appl/genscan/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/genscan/report.rb">lib/bio/appl/genscan/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/hmmer_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/hmmer.rb">lib/bio/appl/hmmer.rb</a></li>
          
            <li><a href="./lib/bio/appl/hmmer/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/hmmer/report.rb">lib/bio/appl/hmmer/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/iprscan/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/iprscan/report.rb">lib/bio/appl/iprscan/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/mafft_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/mafft.rb">lib/bio/appl/mafft.rb</a></li>
          
            <li><a href="./lib/bio/appl/mafft/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/mafft/report.rb">lib/bio/appl/mafft/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/meme/mast_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/meme/mast.rb">lib/bio/appl/meme/mast.rb</a></li>
          
            <li><a href="./lib/bio/appl/meme/mast/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/meme/mast/report.rb">lib/bio/appl/meme/mast/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/meme/motif_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/meme/motif.rb">lib/bio/appl/meme/motif.rb</a></li>
          
            <li><a href="./lib/bio/appl/muscle_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/muscle.rb">lib/bio/appl/muscle.rb</a></li>
          
            <li><a href="./lib/bio/appl/paml/codeml_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/paml/codeml.rb">lib/bio/appl/paml/codeml.rb</a></li>
          
            <li><a href="./lib/bio/appl/paml/common_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/paml/common.rb">lib/bio/appl/paml/common.rb</a></li>
          
            <li><a href="./lib/bio/appl/phylip/alignment_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/phylip/alignment.rb">lib/bio/appl/phylip/alignment.rb</a></li>
          
            <li><a href="./lib/bio/appl/phylip/distance_matrix_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/phylip/distance_matrix.rb">lib/bio/appl/phylip/distance_matrix.rb</a></li>
          
            <li><a href="./lib/bio/appl/probcons_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/probcons.rb">lib/bio/appl/probcons.rb</a></li>
          
            <li><a href="./lib/bio/appl/psort_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/psort.rb">lib/bio/appl/psort.rb</a></li>
          
            <li><a href="./lib/bio/appl/psort/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/psort/report.rb">lib/bio/appl/psort/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/pts1_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/pts1.rb">lib/bio/appl/pts1.rb</a></li>
          
            <li><a href="./lib/bio/appl/sim4_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/sim4.rb">lib/bio/appl/sim4.rb</a></li>
          
            <li><a href="./lib/bio/appl/sim4/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/sim4/report.rb">lib/bio/appl/sim4/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/sosui/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/sosui/report.rb">lib/bio/appl/sosui/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/spidey/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/spidey/report.rb">lib/bio/appl/spidey/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/targetp/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/targetp/report.rb">lib/bio/appl/targetp/report.rb</a></li>
          
            <li><a href="./lib/bio/appl/tcoffee_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/tcoffee.rb">lib/bio/appl/tcoffee.rb</a></li>
          
            <li><a href="./lib/bio/appl/tmhmm/report_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/appl/tmhmm/report.rb">lib/bio/appl/tmhmm/report.rb</a></li>
          
            <li><a href="./lib/bio/command_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/command.rb">lib/bio/command.rb</a></li>
          
            <li><a href="./lib/bio/compat/features_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/compat/features.rb">lib/bio/compat/features.rb</a></li>
          
            <li><a href="./lib/bio/compat/references_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/compat/references.rb">lib/bio/compat/references.rb</a></li>
          
            <li><a href="./lib/bio/data/aa_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/data/aa.rb">lib/bio/data/aa.rb</a></li>
          
            <li><a href="./lib/bio/data/codontable_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/data/codontable.rb">lib/bio/data/codontable.rb</a></li>
          
            <li><a href="./lib/bio/data/na_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/data/na.rb">lib/bio/data/na.rb</a></li>
          
            <li><a href="./lib/bio/db_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db.rb">lib/bio/db.rb</a></li>
          
            <li><a href="./lib/bio/db/aaindex_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/aaindex.rb">lib/bio/db/aaindex.rb</a></li>
          
            <li><a href="./lib/bio/db/biosql/sequence_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/biosql/sequence.rb">lib/bio/db/biosql/sequence.rb</a></li>
          
            <li><a href="./lib/bio/db/embl/common_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/common.rb">lib/bio/db/embl/common.rb</a></li>
          
            <li><a href="./lib/bio/db/embl/embl_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/embl.rb">lib/bio/db/embl/embl.rb</a></li>
          
            <li><a href="./lib/bio/db/embl/sptr_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/sptr.rb">lib/bio/db/embl/sptr.rb</a></li>
          
            <li><a href="./lib/bio/db/embl/swissprot_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/swissprot.rb">lib/bio/db/embl/swissprot.rb</a></li>
          
            <li><a href="./lib/bio/db/embl/trembl_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/trembl.rb">lib/bio/db/embl/trembl.rb</a></li>
          
            <li><a href="./lib/bio/db/embl/uniprot_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/uniprot.rb">lib/bio/db/embl/uniprot.rb</a></li>
          
            <li><a href="./lib/bio/db/fantom_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/fantom.rb">lib/bio/db/fantom.rb</a></li>
          
            <li><a href="./lib/bio/db/fasta_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/fasta.rb">lib/bio/db/fasta.rb</a></li>
          
            <li><a href="./lib/bio/db/fasta/defline_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/fasta/defline.rb">lib/bio/db/fasta/defline.rb</a></li>
          
            <li><a href="./lib/bio/db/fasta/qual_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/fasta/qual.rb">lib/bio/db/fasta/qual.rb</a></li>
          
            <li><a href="./lib/bio/db/fastq_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/fastq.rb">lib/bio/db/fastq.rb</a></li>
          
            <li><a href="./lib/bio/db/genbank/common_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/genbank/common.rb">lib/bio/db/genbank/common.rb</a></li>
          
            <li><a href="./lib/bio/db/genbank/ddbj_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/genbank/ddbj.rb">lib/bio/db/genbank/ddbj.rb</a></li>
          
            <li><a href="./lib/bio/db/genbank/genbank_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/genbank/genbank.rb">lib/bio/db/genbank/genbank.rb</a></li>
          
            <li><a href="./lib/bio/db/genbank/genpept_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/genbank/genpept.rb">lib/bio/db/genbank/genpept.rb</a></li>
          
            <li><a href="./lib/bio/db/genbank/refseq_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/genbank/refseq.rb">lib/bio/db/genbank/refseq.rb</a></li>
          
            <li><a href="./lib/bio/db/gff_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/gff.rb">lib/bio/db/gff.rb</a></li>
          
            <li><a href="./lib/bio/db/go_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/go.rb">lib/bio/db/go.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/brite_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/brite.rb">lib/bio/db/kegg/brite.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/common_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/common.rb">lib/bio/db/kegg/common.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/compound_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/compound.rb">lib/bio/db/kegg/compound.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/drug_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/drug.rb">lib/bio/db/kegg/drug.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/enzyme_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/enzyme.rb">lib/bio/db/kegg/enzyme.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/expression_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/expression.rb">lib/bio/db/kegg/expression.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/genes_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/genes.rb">lib/bio/db/kegg/genes.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/genome_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/genome.rb">lib/bio/db/kegg/genome.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/glycan_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/glycan.rb">lib/bio/db/kegg/glycan.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/keggtab_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/keggtab.rb">lib/bio/db/kegg/keggtab.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/kgml_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/kgml.rb">lib/bio/db/kegg/kgml.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/module_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/module.rb">lib/bio/db/kegg/module.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/orthology_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/orthology.rb">lib/bio/db/kegg/orthology.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/pathway_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/pathway.rb">lib/bio/db/kegg/pathway.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/reaction_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/reaction.rb">lib/bio/db/kegg/reaction.rb</a></li>
          
            <li><a href="./lib/bio/db/kegg/taxonomy_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/kegg/taxonomy.rb">lib/bio/db/kegg/taxonomy.rb</a></li>
          
            <li><a href="./lib/bio/db/lasergene_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/lasergene.rb">lib/bio/db/lasergene.rb</a></li>
          
            <li><a href="./lib/bio/db/litdb_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/litdb.rb">lib/bio/db/litdb.rb</a></li>
          
            <li><a href="./lib/bio/db/medline_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/medline.rb">lib/bio/db/medline.rb</a></li>
          
            <li><a href="./lib/bio/db/nbrf_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/nbrf.rb">lib/bio/db/nbrf.rb</a></li>
          
            <li><a href="./lib/bio/db/newick_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/newick.rb">lib/bio/db/newick.rb</a></li>
          
            <li><a href="./lib/bio/db/nexus_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/nexus.rb">lib/bio/db/nexus.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb.rb">lib/bio/db/pdb.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/atom_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/atom.rb">lib/bio/db/pdb/atom.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/chain_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/chain.rb">lib/bio/db/pdb/chain.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/chemicalcomponent_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/chemicalcomponent.rb">lib/bio/db/pdb/chemicalcomponent.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/model_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/model.rb">lib/bio/db/pdb/model.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/pdb_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/pdb.rb">lib/bio/db/pdb/pdb.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/residue_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/residue.rb">lib/bio/db/pdb/residue.rb</a></li>
          
            <li><a href="./lib/bio/db/pdb/utils_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/pdb/utils.rb">lib/bio/db/pdb/utils.rb</a></li>
          
            <li><a href="./lib/bio/db/phyloxml/phyloxml_elements_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/phyloxml/phyloxml_elements.rb">lib/bio/db/phyloxml/phyloxml_elements.rb</a></li>
          
            <li><a href="./lib/bio/db/phyloxml/phyloxml_parser_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/phyloxml/phyloxml_parser.rb">lib/bio/db/phyloxml/phyloxml_parser.rb</a></li>
          
            <li><a href="./lib/bio/db/phyloxml/phyloxml_writer_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/phyloxml/phyloxml_writer.rb">lib/bio/db/phyloxml/phyloxml_writer.rb</a></li>
          
            <li><a href="./lib/bio/db/prosite_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/prosite.rb">lib/bio/db/prosite.rb</a></li>
          
            <li><a href="./lib/bio/db/rebase_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/rebase.rb">lib/bio/db/rebase.rb</a></li>
          
            <li><a href="./lib/bio/db/sanger_chromatogram/abif_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/sanger_chromatogram/abif.rb">lib/bio/db/sanger_chromatogram/abif.rb</a></li>
          
            <li><a href="./lib/bio/db/sanger_chromatogram/chromatogram_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/sanger_chromatogram/chromatogram.rb">lib/bio/db/sanger_chromatogram/chromatogram.rb</a></li>
          
            <li><a href="./lib/bio/db/sanger_chromatogram/scf_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/sanger_chromatogram/scf.rb">lib/bio/db/sanger_chromatogram/scf.rb</a></li>
          
            <li><a href="./lib/bio/db/soft_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/soft.rb">lib/bio/db/soft.rb</a></li>
          
            <li><a href="./lib/bio/db/transfac_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/transfac.rb">lib/bio/db/transfac.rb</a></li>
          
            <li><a href="./lib/bio/feature_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/feature.rb">lib/bio/feature.rb</a></li>
          
            <li><a href="./lib/bio/io/biosql/ar-biosql_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/biosql/ar-biosql.rb">lib/bio/io/biosql/ar-biosql.rb</a></li>
          
            <li><a href="./lib/bio/io/biosql/biosql_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/biosql/biosql.rb">lib/bio/io/biosql/biosql.rb</a></li>
          
            <li><a href="./lib/bio/io/das_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/das.rb">lib/bio/io/das.rb</a></li>
          
            <li><a href="./lib/bio/io/dbget_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/dbget.rb">lib/bio/io/dbget.rb</a></li>
          
            <li><a href="./lib/bio/io/ddbjrest_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/ddbjrest.rb">lib/bio/io/ddbjrest.rb</a></li>
          
            <li><a href="./lib/bio/io/ddbjxml_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/ddbjxml.rb">lib/bio/io/ddbjxml.rb</a></li>
          
            <li><a href="./lib/bio/io/ebisoap_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/ebisoap.rb">lib/bio/io/ebisoap.rb</a></li>
          
            <li><a href="./lib/bio/io/ensembl_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/ensembl.rb">lib/bio/io/ensembl.rb</a></li>
          
            <li><a href="./lib/bio/io/fastacmd_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/fastacmd.rb">lib/bio/io/fastacmd.rb</a></li>
          
            <li><a href="./lib/bio/io/fetch_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/fetch.rb">lib/bio/io/fetch.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile.rb">lib/bio/io/flatfile.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile/autodetection_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/autodetection.rb">lib/bio/io/flatfile/autodetection.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile/bdb_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/bdb.rb">lib/bio/io/flatfile/bdb.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile/buffer_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/buffer.rb">lib/bio/io/flatfile/buffer.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile/index_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/index.rb">lib/bio/io/flatfile/index.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile/indexer_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/indexer.rb">lib/bio/io/flatfile/indexer.rb</a></li>
          
            <li><a href="./lib/bio/io/flatfile/splitter_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/splitter.rb">lib/bio/io/flatfile/splitter.rb</a></li>
          
            <li><a href="./lib/bio/io/higet_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/higet.rb">lib/bio/io/higet.rb</a></li>
          
            <li><a href="./lib/bio/io/hinv_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/hinv.rb">lib/bio/io/hinv.rb</a></li>
          
            <li><a href="./lib/bio/io/keggapi_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/keggapi.rb">lib/bio/io/keggapi.rb</a></li>
          
            <li><a href="./lib/bio/io/ncbirest_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/ncbirest.rb">lib/bio/io/ncbirest.rb</a></li>
          
            <li><a href="./lib/bio/io/ncbisoap_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/ncbisoap.rb">lib/bio/io/ncbisoap.rb</a></li>
          
            <li><a href="./lib/bio/io/pubmed_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/pubmed.rb">lib/bio/io/pubmed.rb</a></li>
          
            <li><a href="./lib/bio/io/registry_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/registry.rb">lib/bio/io/registry.rb</a></li>
          
            <li><a href="./lib/bio/io/soapwsdl_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/soapwsdl.rb">lib/bio/io/soapwsdl.rb</a></li>
          
            <li><a href="./lib/bio/io/sql_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/sql.rb">lib/bio/io/sql.rb</a></li>
          
            <li><a href="./lib/bio/io/togows_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/togows.rb">lib/bio/io/togows.rb</a></li>
          
            <li><a href="./lib/bio/location_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/location.rb">lib/bio/location.rb</a></li>
          
            <li><a href="./lib/bio/map_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/map.rb">lib/bio/map.rb</a></li>
          
            <li><a href="./lib/bio/pathway_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/pathway.rb">lib/bio/pathway.rb</a></li>
          
            <li><a href="./lib/bio/reference_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/reference.rb">lib/bio/reference.rb</a></li>
          
            <li><a href="./lib/bio/sequence_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence.rb">lib/bio/sequence.rb</a></li>
          
            <li><a href="./lib/bio/sequence/aa_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/aa.rb">lib/bio/sequence/aa.rb</a></li>
          
            <li><a href="./lib/bio/sequence/common_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/common.rb">lib/bio/sequence/common.rb</a></li>
          
            <li><a href="./lib/bio/sequence/compat_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/compat.rb">lib/bio/sequence/compat.rb</a></li>
          
            <li><a href="./lib/bio/sequence/format_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/format.rb">lib/bio/sequence/format.rb</a></li>
          
            <li><a href="./lib/bio/sequence/generic_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/generic.rb">lib/bio/sequence/generic.rb</a></li>
          
            <li><a href="./lib/bio/sequence/na_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/na.rb">lib/bio/sequence/na.rb</a></li>
          
            <li><a href="./lib/bio/sequence/quality_score_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/quality_score.rb">lib/bio/sequence/quality_score.rb</a></li>
          
            <li><a href="./lib/bio/sequence/sequence_masker_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/sequence/sequence_masker.rb">lib/bio/sequence/sequence_masker.rb</a></li>
          
            <li><a href="./lib/bio/shell/plugin/das_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/shell/plugin/das.rb">lib/bio/shell/plugin/das.rb</a></li>
          
            <li><a href="./lib/bio/tree_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/tree.rb">lib/bio/tree.rb</a></li>
          
            <li><a href="./lib/bio/util/color_scheme_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/color_scheme.rb">lib/bio/util/color_scheme.rb</a></li>
          
            <li><a href="./lib/bio/util/contingency_table_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/contingency_table.rb">lib/bio/util/contingency_table.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme.rb">lib/bio/util/restriction_enzyme.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/analysis_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/analysis.rb">lib/bio/util/restriction_enzyme/analysis.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/analysis_basic_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/analysis_basic.rb">lib/bio/util/restriction_enzyme/analysis_basic.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/cut_symbol_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/cut_symbol.rb">lib/bio/util/restriction_enzyme/cut_symbol.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/dense_int_array_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/dense_int_array.rb">lib/bio/util/restriction_enzyme/dense_int_array.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/double_stranded_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/double_stranded.rb">lib/bio/util/restriction_enzyme/double_stranded.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/double_stranded/aligned_strands_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/double_stranded/aligned_strands.rb">lib/bio/util/restriction_enzyme/double_stranded/aligned_strands.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair.rb">lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair_in_enzyme_notation_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair_in_enzyme_notation.rb">lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair_in_enzyme_notation.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/double_stranded/cut_locations_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/double_stranded/cut_locations.rb">lib/bio/util/restriction_enzyme/double_stranded/cut_locations.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/double_stranded/cut_locations_in_enzyme_notation_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/double_stranded/cut_locations_in_enzyme_notation.rb">lib/bio/util/restriction_enzyme/double_stranded/cut_locations_in_enzyme_notation.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/cut_range_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/cut_range.rb">lib/bio/util/restriction_enzyme/range/cut_range.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/cut_ranges_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/cut_ranges.rb">lib/bio/util/restriction_enzyme/range/cut_ranges.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/horizontal_cut_range_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/horizontal_cut_range.rb">lib/bio/util/restriction_enzyme/range/horizontal_cut_range.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/sequence_range_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/sequence_range.rb">lib/bio/util/restriction_enzyme/range/sequence_range.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/sequence_range/calculated_cuts_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/sequence_range/calculated_cuts.rb">lib/bio/util/restriction_enzyme/range/sequence_range/calculated_cuts.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/sequence_range/fragment_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/sequence_range/fragment.rb">lib/bio/util/restriction_enzyme/range/sequence_range/fragment.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/sequence_range/fragments_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/sequence_range/fragments.rb">lib/bio/util/restriction_enzyme/range/sequence_range/fragments.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/range/vertical_cut_range_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/range/vertical_cut_range.rb">lib/bio/util/restriction_enzyme/range/vertical_cut_range.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/single_strand_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/single_strand.rb">lib/bio/util/restriction_enzyme/single_strand.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/single_strand/cut_locations_in_enzyme_notation_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/single_strand/cut_locations_in_enzyme_notation.rb">lib/bio/util/restriction_enzyme/single_strand/cut_locations_in_enzyme_notation.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/single_strand_complement_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/single_strand_complement.rb">lib/bio/util/restriction_enzyme/single_strand_complement.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/sorted_num_array_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/sorted_num_array.rb">lib/bio/util/restriction_enzyme/sorted_num_array.rb</a></li>
          
            <li><a href="./lib/bio/util/restriction_enzyme/string_formatting_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/restriction_enzyme/string_formatting.rb">lib/bio/util/restriction_enzyme/string_formatting.rb</a></li>
          
            <li><a href="./lib/bio/util/sirna_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/util/sirna.rb">lib/bio/util/sirna.rb</a></li>
          
            <li><a href="./lib/bio/version_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/version.rb">lib/bio/version.rb</a></li>
          
          </ul>
        </div>
      </div>

      
    </div>

    <div id="class-metadata">
      

      

      
      <!-- Namespace Contents -->
      <div id="namespace-list-section" class="section">
        <h3 class="section-header">Namespace</h3>
        <ul class="link-list">
          
          <li><span class="type">MODULE</span> <a href="Bio/Alignment.html">Bio::Alignment</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/ColorScheme.html">Bio::ColorScheme</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/Command.html">Bio::Command</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/FANTOM.html">Bio::FANTOM</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/GCG.html">Bio::GCG</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/Map.html">Bio::Map</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/Meme.html">Bio::Meme</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/PAML.html">Bio::PAML</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/Phylip.html">Bio::Phylip</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/PhyloXML.html">Bio::PhyloXML</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/Shell.html">Bio::Shell</a></li>
          
          <li><span class="type">MODULE</span> <a href="Bio/TogoWS.html">Bio::TogoWS</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/AAindex.html">Bio::AAindex</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/AAindex1.html">Bio::AAindex1</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/AAindex2.html">Bio::AAindex2</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Abif.html">Bio::Abif</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/AminoAcid.html">Bio::AminoAcid</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Blast.html">Bio::Blast</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Blat.html">Bio::Blat</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/ClustalW.html">Bio::ClustalW</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/CodonTable.html">Bio::CodonTable</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/ContingencyTable.html">Bio::ContingencyTable</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/DAS.html">Bio::DAS</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/DB.html">Bio::DB</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/DBGET.html">Bio::DBGET</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/DDBJ.html">Bio::DDBJ</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/EBI.html">Bio::EBI</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/EMBL.html">Bio::EMBL</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/EMBLDB.html">Bio::EMBLDB</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/EMBOSS.html">Bio::EMBOSS</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Ensembl.html">Bio::Ensembl</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Fasta.html">Bio::Fasta</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/FastaDefline.html">Bio::FastaDefline</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/FastaFormat.html">Bio::FastaFormat</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Fastq.html">Bio::Fastq</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Feature.html">Bio::Feature</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Features.html">Bio::Features</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Fetch.html">Bio::Fetch</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/FlatFile.html">Bio::FlatFile</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/FlatFileIndex.html">Bio::FlatFileIndex</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/GFF.html">Bio::GFF</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/GO.html">Bio::GO</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/GenBank.html">Bio::GenBank</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/GenPept.html">Bio::GenPept</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Genscan.html">Bio::Genscan</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/HGC.html">Bio::HGC</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/HMMER.html">Bio::HMMER</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Hinv.html">Bio::Hinv</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Iprscan.html">Bio::Iprscan</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/KEGG.html">Bio::KEGG</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/KEGGDB.html">Bio::KEGGDB</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/LITDB.html">Bio::LITDB</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Lasergene.html">Bio::Lasergene</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Location.html">Bio::Location</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Locations.html">Bio::Locations</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/MAFFT.html">Bio::MAFFT</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/MEDLINE.html">Bio::MEDLINE</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Muscle.html">Bio::Muscle</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/NBRF.html">Bio::NBRF</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/NCBI.html">Bio::NCBI</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/NCBIDB.html">Bio::NCBIDB</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Newick.html">Bio::Newick</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Nexus.html">Bio::Nexus</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/NucleicAcid.html">Bio::NucleicAcid</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/PDB.html">Bio::PDB</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/PROSITE.html">Bio::PROSITE</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/PSORT.html">Bio::PSORT</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/PTS1.html">Bio::PTS1</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Pathway.html">Bio::Pathway</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Probcons.html">Bio::Probcons</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/PubMed.html">Bio::PubMed</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/REBASE.html">Bio::REBASE</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/RefSeq.html">Bio::RefSeq</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Reference.html">Bio::Reference</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/References.html">Bio::References</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Registry.html">Bio::Registry</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SOAPWSDL.html">Bio::SOAPWSDL</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SOFT.html">Bio::SOFT</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SOSUI.html">Bio::SOSUI</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SPTR.html">Bio::SPTR</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SQL.html">Bio::SQL</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SangerChromatogram.html">Bio::SangerChromatogram</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Scf.html">Bio::Scf</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Sequence.html">Bio::Sequence</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SiRNA.html">Bio::SiRNA</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Sim4.html">Bio::Sim4</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Spidey.html">Bio::Spidey</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/SwissProt.html">Bio::SwissProt</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/TMHMM.html">Bio::TMHMM</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/TRANSFAC.html">Bio::TRANSFAC</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/TargetP.html">Bio::TargetP</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Taxonomy.html">Bio::Taxonomy</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Tcoffee.html">Bio::Tcoffee</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/TrEMBL.html">Bio::TrEMBL</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/Tree.html">Bio::Tree</a></li>
          
          <li><span class="type">CLASS</span> <a href="Bio/UniProt.html">Bio::UniProt</a></li>
          
        </ul>
      </div>
      

      
      <!-- Method Quickref -->
      <div id="method-list-section" class="section">
        <h3 class="section-header">Methods</h3>
        <ul class="link-list">
          
          <li><a href="#method-c-method_missing">::method_missing</a></li>
          
        </ul>
      </div>
      

      
    </div>

    <div id="project-metadata">
      
      
      <div id="fileindex-section" class="section project-section">
        <h3 class="section-header">Files</h3>
        <ul>
        
          <li class="file"><a href="./ChangeLog.html">ChangeLog</a></li>
        
          <li class="file"><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a></li>
        
          <li class="file"><a href="./README_rdoc.html">README.rdoc</a></li>
        
          <li class="file"><a href="./README_DEV_rdoc.html">README_DEV.rdoc</a></li>
        
          <li class="file"><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a></li>
        
          <li class="file"><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3.rdoc</a></li>
        
          <li class="file"><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0.rdoc</a></li>
        
          <li class="file"><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1.rdoc</a></li>
        
        </ul>
      </div>
      

      <div id="classindex-section" class="section project-section">
        <h3 class="section-header">Class/Module Index
          <span class="search-toggle"><img src="./images/find.png"
            height="16" width="16" alt="[+]"
            title="show/hide quicksearch" /></span></h3>
        <form action="#" method="get" accept-charset="utf-8" class="initially-hidden">
        <fieldset>
          <legend>Quicksearch</legend>
          <input type="text" name="quicksearch" value=""
            class="quicksearch-field" />
        </fieldset>
        </form>

        <ul class="link-list">
        
          <li><a href="./Bio.html">Bio</a></li>
        
          <li><a href="./Bio/AAindex.html">Bio::AAindex</a></li>
        
          <li><a href="./Bio/AAindex1.html">Bio::AAindex1</a></li>
        
          <li><a href="./Bio/AAindex2.html">Bio::AAindex2</a></li>
        
          <li><a href="./Bio/Abif.html">Bio::Abif</a></li>
        
          <li><a href="./Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a></li>
        
          <li><a href="./Bio/Alignment.html">Bio::Alignment</a></li>
        
          <li><a href="./Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a></li>
        
          <li><a href="./Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a></li>
        
          <li><a href="./Bio/Alignment/GAP.html">Bio::Alignment::GAP</a></li>
        
          <li><a href="./Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a></li>
        
          <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a></li>
        
          <li><a href="./Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a></li>
        
          <li><a href="./Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a></li>
        
          <li><a href="./Bio/Alignment/Output.html">Bio::Alignment::Output</a></li>
        
          <li><a href="./Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a></li>
        
          <li><a href="./Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a></li>
        
          <li><a href="./Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a></li>
        
          <li><a href="./Bio/Alignment/Site.html">Bio::Alignment::Site</a></li>
        
          <li><a href="./Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a></li>
        
          <li><a href="./Bio/AminoAcid.html">Bio::AminoAcid</a></li>
        
          <li><a href="./Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a></li>
        
          <li><a href="./Bio/Blast.html">Bio::Blast</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Default.html">Bio::Blast::Default</a></li>
        
          <li><a href="./Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a></li>
        
          <li><a href="./Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a></li>
        
          <li><a href="./Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a></li>
        
          <li><a href="./Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a></li>
        
          <li><a href="./Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Remote.html">Bio::Blast::Remote</a></li>
        
          <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a></li>
        
          <li><a href="./Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a></li>
        
          <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a></li>
        
          <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a></li>
        
          <li><a href="./Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a></li>
        
          <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a></li>
        
          <li><a href="./Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a></li>
        
          <li><a href="./Bio/Blast/Report.html">Bio::Blast::Report</a></li>
        
          <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a></li>
        
          <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="./Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a></li>
        
          <li><a href="./Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a></li>
        
          <li><a href="./Bio/Blast/WU.html">Bio::Blast::WU</a></li>
        
          <li><a href="./Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a></li>
        
          <li><a href="./Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a></li>
        
          <li><a href="./Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a></li>
        
          <li><a href="./Bio/Blat.html">Bio::Blat</a></li>
        
          <li><a href="./Bio/Blat/Report.html">Bio::Blat::Report</a></li>
        
          <li><a href="./Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a></li>
        
          <li><a href="./Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a></li>
        
          <li><a href="./Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a></li>
        
          <li><a href="./Bio/ClustalW.html">Bio::ClustalW</a></li>
        
          <li><a href="./Bio/ClustalW/Report.html">Bio::ClustalW::Report</a></li>
        
          <li><a href="./Bio/CodonTable.html">Bio::CodonTable</a></li>
        
          <li><a href="./Bio/ColorScheme.html">Bio::ColorScheme</a></li>
        
          <li><a href="./Bio/Command.html">Bio::Command</a></li>
        
          <li><a href="./Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a></li>
        
          <li><a href="./Bio/ContingencyTable.html">Bio::ContingencyTable</a></li>
        
          <li><a href="./Bio/DAS.html">Bio::DAS</a></li>
        
          <li><a href="./Bio/DAS/DNA.html">Bio::DAS::DNA</a></li>
        
          <li><a href="./Bio/DAS/DSN.html">Bio::DAS::DSN</a></li>
        
          <li><a href="./Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a></li>
        
          <li><a href="./Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a></li>
        
          <li><a href="./Bio/DAS/GFF.html">Bio::DAS::GFF</a></li>
        
          <li><a href="./Bio/DAS/GROUP.html">Bio::DAS::GROUP</a></li>
        
          <li><a href="./Bio/DAS/LINK.html">Bio::DAS::LINK</a></li>
        
          <li><a href="./Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a></li>
        
          <li><a href="./Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a></li>
        
          <li><a href="./Bio/DAS/TARGET.html">Bio::DAS::TARGET</a></li>
        
          <li><a href="./Bio/DAS/TYPE.html">Bio::DAS::TYPE</a></li>
        
          <li><a href="./Bio/DAS/TYPES.html">Bio::DAS::TYPES</a></li>
        
          <li><a href="./Bio/DB.html">Bio::DB</a></li>
        
          <li><a href="./Bio/DBGET.html">Bio::DBGET</a></li>
        
          <li><a href="./Bio/DDBJ.html">Bio::DDBJ</a></li>
        
          <li><a href="./Bio/DDBJ/REST.html">Bio::DDBJ::REST</a></li>
        
          <li><a href="./Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a></li>
        
          <li><a href="./Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a></li>
        
          <li><a href="./Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a></li>
        
          <li><a href="./Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a></li>
        
          <li><a href="./Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a></li>
        
          <li><a href="./Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a></li>
        
          <li><a href="./Bio/DDBJ/XML.html">Bio::DDBJ::XML</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a></li>
        
          <li><a href="./Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a></li>
        
          <li><a href="./Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a></li>
        
          <li><a href="./Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a></li>
        
          <li><a href="./Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a></li>
        
          <li><a href="./Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a></li>
        
          <li><a href="./Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a></li>
        
          <li><a href="./Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a></li>
        
          <li><a href="./Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a></li>
        
          <li><a href="./Bio/EBI.html">Bio::EBI</a></li>
        
          <li><a href="./Bio/EBI/SOAP.html">Bio::EBI::SOAP</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a></li>
        
          <li><a href="./Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a></li>
        
          <li><a href="./Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a></li>
        
          <li><a href="./Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a></li>
        
          <li><a href="./Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a></li>
        
          <li><a href="./Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a></li>
        
          <li><a href="./Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a></li>
        
          <li><a href="./Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a></li>
        
          <li><a href="./Bio/EMBL.html">Bio::EMBL</a></li>
        
          <li><a href="./Bio/EMBLDB.html">Bio::EMBLDB</a></li>
        
          <li><a href="./Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a></li>
        
          <li><a href="./Bio/EMBOSS.html">Bio::EMBOSS</a></li>
        
          <li><a href="./Bio/Ensembl.html">Bio::Ensembl</a></li>
        
          <li><a href="./Bio/Ensembl/Base.html">Bio::Ensembl::Base</a></li>
        
          <li><a href="./Bio/Ensembl/Human.html">Bio::Ensembl::Human</a></li>
        
          <li><a href="./Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a></li>
        
          <li><a href="./Bio/FANTOM.html">Bio::FANTOM</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a></li>
        
          <li><a href="./Bio/Fasta.html">Bio::Fasta</a></li>
        
          <li><a href="./Bio/Fasta/Report.html">Bio::Fasta::Report</a></li>
        
          <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a></li>
        
          <li><a href="./Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a></li>
        
          <li><a href="./Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a></li>
        
          <li><a href="./Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a></li>
        
          <li><a href="./Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a></li>
        
          <li><a href="./Bio/FastaDefline.html">Bio::FastaDefline</a></li>
        
          <li><a href="./Bio/FastaFormat.html">Bio::FastaFormat</a></li>
        
          <li><a href="./Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a></li>
        
          <li><a href="./Bio/Fastq.html">Bio::Fastq</a></li>
        
          <li><a href="./Bio/Fastq/Error.html">Bio::Fastq::Error</a></li>
        
          <li><a href="./Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a></li>
        
          <li><a href="./Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a></li>
        
          <li><a href="./Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a></li>
        
          <li><a href="./Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a></li>
        
          <li><a href="./Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a></li>
        
          <li><a href="./Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a></li>
        
          <li><a href="./Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a></li>
        
          <li><a href="./Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a></li>
        
          <li><a href="./Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a></li>
        
          <li><a href="./Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a></li>
        
          <li><a href="./Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a></li>
        
          <li><a href="./Bio/Feature.html">Bio::Feature</a></li>
        
          <li><a href="./Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a></li>
        
          <li><a href="./Bio/Features.html">Bio::Features</a></li>
        
          <li><a href="./Bio/Fetch.html">Bio::Fetch</a></li>
        
          <li><a href="./Bio/FlatFile.html">Bio::FlatFile</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a></li>
        
          <li><a href="./Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a></li>
        
          <li><a href="./Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a></li>
        
          <li><a href="./Bio/FlatFileIndex.html">Bio::FlatFileIndex</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a></li>
        
          <li><a href="./Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a></li>
        
          <li><a href="./Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a></li>
        
          <li><a href="./Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a></li>
        
          <li><a href="./Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a></li>
        
          <li><a href="./Bio/GCG.html">Bio::GCG</a></li>
        
          <li><a href="./Bio/GCG/Msf.html">Bio::GCG::Msf</a></li>
        
          <li><a href="./Bio/GCG/Seq.html">Bio::GCG::Seq</a></li>
        
          <li><a href="./Bio/GFF.html">Bio::GFF</a></li>
        
          <li><a href="./Bio/GFF/GFF2.html">Bio::GFF::GFF2</a></li>
        
          <li><a href="./Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a></li>
        
          <li><a href="./Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a></li>
        
          <li><a href="./Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a></li>
        
          <li><a href="./Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a></li>
        
          <li><a href="./Bio/GFF/GFF3.html">Bio::GFF::GFF3</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a></li>
        
          <li><a href="./Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a></li>
        
          <li><a href="./Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a></li>
        
          <li><a href="./Bio/GFF/Record.html">Bio::GFF::Record</a></li>
        
          <li><a href="./Bio/GO.html">Bio::GO</a></li>
        
          <li><a href="./Bio/GO/External2go.html">Bio::GO::External2go</a></li>
        
          <li><a href="./Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a></li>
        
          <li><a href="./Bio/GO/Ontology.html">Bio::GO::Ontology</a></li>
        
          <li><a href="./Bio/GenBank.html">Bio::GenBank</a></li>
        
          <li><a href="./Bio/GenBank/Locus.html">Bio::GenBank::Locus</a></li>
        
          <li><a href="./Bio/GenPept.html">Bio::GenPept</a></li>
        
          <li><a href="./Bio/GenPept/Locus.html">Bio::GenPept::Locus</a></li>
        
          <li><a href="./Bio/Genscan.html">Bio::Genscan</a></li>
        
          <li><a href="./Bio/Genscan/Report.html">Bio::Genscan::Report</a></li>
        
          <li><a href="./Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a></li>
        
          <li><a href="./Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a></li>
        
          <li><a href="./Bio/HGC.html">Bio::HGC</a></li>
        
          <li><a href="./Bio/HGC/HiGet.html">Bio::HGC::HiGet</a></li>
        
          <li><a href="./Bio/HMMER.html">Bio::HMMER</a></li>
        
          <li><a href="./Bio/HMMER/Report.html">Bio::HMMER::Report</a></li>
        
          <li><a href="./Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a></li>
        
          <li><a href="./Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a></li>
        
          <li><a href="./Bio/Hinv.html">Bio::Hinv</a></li>
        
          <li><a href="./Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a></li>
        
          <li><a href="./Bio/Hinv/Common.html">Bio::Hinv::Common</a></li>
        
          <li><a href="./Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a></li>
        
          <li><a href="./Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a></li>
        
          <li><a href="./Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a></li>
        
          <li><a href="./Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a></li>
        
          <li><a href="./Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a></li>
        
          <li><a href="./Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a></li>
        
          <li><a href="./Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a></li>
        
          <li><a href="./Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a></li>
        
          <li><a href="./Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a></li>
        
          <li><a href="./Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a></li>
        
          <li><a href="./Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a></li>
        
          <li><a href="./Bio/Iprscan.html">Bio::Iprscan</a></li>
        
          <li><a href="./Bio/Iprscan/Report.html">Bio::Iprscan::Report</a></li>
        
          <li><a href="./Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a></li>
        
          <li><a href="./Bio/KEGG.html">Bio::KEGG</a></li>
        
          <li><a href="./Bio/KEGG/API.html">Bio::KEGG::API</a></li>
        
          <li><a href="./Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a></li>
        
          <li><a href="./Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a></li>
        
          <li><a href="./Bio/KEGG/Common.html">Bio::KEGG::Common</a></li>
        
          <li><a href="./Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a></li>
        
          <li><a href="./Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a></li>
        
          <li><a href="./Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a></li>
        
          <li><a href="./Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a></li>
        
          <li><a href="./Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a></li>
        
          <li><a href="./Bio/KEGG/GENES.html">Bio::KEGG::GENES</a></li>
        
          <li><a href="./Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a></li>
        
          <li><a href="./Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a></li>
        
          <li><a href="./Bio/KEGG/KGML.html">Bio::KEGG::KGML</a></li>
        
          <li><a href="./Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a></li>
        
          <li><a href="./Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a></li>
        
          <li><a href="./Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a></li>
        
          <li><a href="./Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a></li>
        
          <li><a href="./Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a></li>
        
          <li><a href="./Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a></li>
        
          <li><a href="./Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a></li>
        
          <li><a href="./Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a></li>
        
          <li><a href="./Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a></li>
        
          <li><a href="./Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a></li>
        
          <li><a href="./Bio/KEGGDB.html">Bio::KEGGDB</a></li>
        
          <li><a href="./Bio/LITDB.html">Bio::LITDB</a></li>
        
          <li><a href="./Bio/Lasergene.html">Bio::Lasergene</a></li>
        
          <li><a href="./Bio/Location.html">Bio::Location</a></li>
        
          <li><a href="./Bio/Locations.html">Bio::Locations</a></li>
        
          <li><a href="./Bio/MAFFT.html">Bio::MAFFT</a></li>
        
          <li><a href="./Bio/MAFFT/Report.html">Bio::MAFFT::Report</a></li>
        
          <li><a href="./Bio/MEDLINE.html">Bio::MEDLINE</a></li>
        
          <li><a href="./Bio/Map.html">Bio::Map</a></li>
        
          <li><a href="./Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a></li>
        
          <li><a href="./Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a></li>
        
          <li><a href="./Bio/Map/Mapping.html">Bio::Map::Mapping</a></li>
        
          <li><a href="./Bio/Map/Marker.html">Bio::Map::Marker</a></li>
        
          <li><a href="./Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a></li>
        
          <li><a href="./Bio/Meme.html">Bio::Meme</a></li>
        
          <li><a href="./Bio/Meme/Mast.html">Bio::Meme::Mast</a></li>
        
          <li><a href="./Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a></li>
        
          <li><a href="./Bio/Meme/Motif.html">Bio::Meme::Motif</a></li>
        
          <li><a href="./Bio/Muscle.html">Bio::Muscle</a></li>
        
          <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a></li>
        
          <li><a href="./Bio/NBRF.html">Bio::NBRF</a></li>
        
          <li><a href="./Bio/NCBI.html">Bio::NCBI</a></li>
        
          <li><a href="./Bio/NCBI/REST.html">Bio::NCBI::REST</a></li>
        
          <li><a href="./Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a></li>
        
          <li><a href="./Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a></li>
        
          <li><a href="./Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a></li>
        
          <li><a href="./Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a></li>
        
          <li><a href="./Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a></li>
        
          <li><a href="./Bio/NCBIDB.html">Bio::NCBIDB</a></li>
        
          <li><a href="./Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a></li>
        
          <li><a href="./Bio/Newick.html">Bio::Newick</a></li>
        
          <li><a href="./Bio/Newick/ParseError.html">Bio::Newick::ParseError</a></li>
        
          <li><a href="./Bio/Nexus.html">Bio::Nexus</a></li>
        
          <li><a href="./Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a></li>
        
          <li><a href="./Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a></li>
        
          <li><a href="./Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a></li>
        
          <li><a href="./Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a></li>
        
          <li><a href="./Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a></li>
        
          <li><a href="./Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a></li>
        
          <li><a href="./Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a></li>
        
          <li><a href="./Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a></li>
        
          <li><a href="./Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a></li>
        
          <li><a href="./Bio/Nexus/Util.html">Bio::Nexus::Util</a></li>
        
          <li><a href="./Bio/NucleicAcid.html">Bio::NucleicAcid</a></li>
        
          <li><a href="./Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a></li>
        
          <li><a href="./Bio/PAML.html">Bio::PAML</a></li>
        
          <li><a href="./Bio/PAML/Baseml.html">Bio::PAML::Baseml</a></li>
        
          <li><a href="./Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a></li>
        
          <li><a href="./Bio/PAML/Codeml.html">Bio::PAML::Codeml</a></li>
        
          <li><a href="./Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a></li>
        
          <li><a href="./Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a></li>
        
          <li><a href="./Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a></li>
        
          <li><a href="./Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a></li>
        
          <li><a href="./Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a></li>
        
          <li><a href="./Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a></li>
        
          <li><a href="./Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a></li>
        
          <li><a href="./Bio/PAML/Common.html">Bio::PAML::Common</a></li>
        
          <li><a href="./Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a></li>
        
          <li><a href="./Bio/PAML/Yn00.html">Bio::PAML::Yn00</a></li>
        
          <li><a href="./Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a></li>
        
          <li><a href="./Bio/PDB.html">Bio::PDB</a></li>
        
          <li><a href="./Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a></li>
        
          <li><a href="./Bio/PDB/Chain.html">Bio::PDB::Chain</a></li>
        
          <li><a href="./Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a></li>
        
          <li><a href="./Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a></li>
        
          <li><a href="./Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a></li>
        
          <li><a href="./Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a></li>
        
          <li><a href="./Bio/PDB/DataType.html">Bio::PDB::DataType</a></li>
        
          <li><a href="./Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a></li>
        
          <li><a href="./Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a></li>
        
          <li><a href="./Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a></li>
        
          <li><a href="./Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a></li>
        
          <li><a href="./Bio/PDB/Model.html">Bio::PDB::Model</a></li>
        
          <li><a href="./Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a></li>
        
          <li><a href="./Bio/PDB/Record.html">Bio::PDB::Record</a></li>
        
          <li><a href="./Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a></li>
        
          <li><a href="./Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a></li>
        
          <li><a href="./Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a></li>
        
          <li><a href="./Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a></li>
        
          <li><a href="./Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a></li>
        
          <li><a href="./Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a></li>
        
          <li><a href="./Bio/PDB/Residue.html">Bio::PDB::Residue</a></li>
        
          <li><a href="./Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a></li>
        
          <li><a href="./Bio/PDB/Utils.html">Bio::PDB::Utils</a></li>
        
          <li><a href="./Bio/PROSITE.html">Bio::PROSITE</a></li>
        
          <li><a href="./Bio/PSORT.html">Bio::PSORT</a></li>
        
          <li><a href="./Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a></li>
        
          <li><a href="./Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a></li>
        
          <li><a href="./Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a></li>
        
          <li><a href="./Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a></li>
        
          <li><a href="./Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a></li>
        
          <li><a href="./Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a></li>
        
          <li><a href="./Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a></li>
        
          <li><a href="./Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a></li>
        
          <li><a href="./Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a></li>
        
          <li><a href="./Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a></li>
        
          <li><a href="./Bio/PTS1.html">Bio::PTS1</a></li>
        
          <li><a href="./Bio/PTS1/Report.html">Bio::PTS1::Report</a></li>
        
          <li><a href="./Bio/Pathway.html">Bio::Pathway</a></li>
        
          <li><a href="./Bio/Phylip.html">Bio::Phylip</a></li>
        
          <li><a href="./Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a></li>
        
          <li><a href="./Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></li>
        
          <li><a href="./Bio/PhyloXML.html">Bio::PhyloXML</a></li>
        
          <li><a href="./Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a></li>
        
          <li><a href="./Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a></li>
        
          <li><a href="./Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a></li>
        
          <li><a href="./Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a></li>
        
          <li><a href="./Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a></li>
        
          <li><a href="./Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a></li>
        
          <li><a href="./Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a></li>
        
          <li><a href="./Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a></li>
        
          <li><a href="./Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a></li>
        
          <li><a href="./Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a></li>
        
          <li><a href="./Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a></li>
        
          <li><a href="./Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a></li>
        
          <li><a href="./Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a></li>
        
          <li><a href="./Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a></li>
        
          <li><a href="./Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a></li>
        
          <li><a href="./Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a></li>
        
          <li><a href="./Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a></li>
        
          <li><a href="./Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a></li>
        
          <li><a href="./Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a></li>
        
          <li><a href="./Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a></li>
        
          <li><a href="./Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a></li>
        
          <li><a href="./Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a></li>
        
          <li><a href="./Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a></li>
        
          <li><a href="./Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a></li>
        
          <li><a href="./Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a></li>
        
          <li><a href="./Bio/Probcons.html">Bio::Probcons</a></li>
        
          <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a></li>
        
          <li><a href="./Bio/PubMed.html">Bio::PubMed</a></li>
        
          <li><a href="./Bio/REBASE.html">Bio::REBASE</a></li>
        
          <li><a href="./Bio/RefSeq.html">Bio::RefSeq</a></li>
        
          <li><a href="./Bio/Reference.html">Bio::Reference</a></li>
        
          <li><a href="./Bio/References.html">Bio::References</a></li>
        
          <li><a href="./Bio/Registry.html">Bio::Registry</a></li>
        
          <li><a href="./Bio/Registry/DB.html">Bio::Registry::DB</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a></li>
        
          <li><a href="./Bio/SOAPWSDL.html">Bio::SOAPWSDL</a></li>
        
          <li><a href="./Bio/SOFT.html">Bio::SOFT</a></li>
        
          <li><a href="./Bio/SOSUI.html">Bio::SOSUI</a></li>
        
          <li><a href="./Bio/SOSUI/Report.html">Bio::SOSUI::Report</a></li>
        
          <li><a href="./Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a></li>
        
          <li><a href="./Bio/SPTR.html">Bio::SPTR</a></li>
        
          <li><a href="./Bio/SQL.html">Bio::SQL</a></li>
        
          <li><a href="./Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a></li>
        
          <li><a href="./Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a></li>
        
          <li><a href="./Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a></li>
        
          <li><a href="./Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a></li>
        
          <li><a href="./Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a></li>
        
          <li><a href="./Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a></li>
        
          <li><a href="./Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a></li>
        
          <li><a href="./Bio/SQL/Comment.html">Bio::SQL::Comment</a></li>
        
          <li><a href="./Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a></li>
        
          <li><a href="./Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a></li>
        
          <li><a href="./Bio/SQL/Location.html">Bio::SQL::Location</a></li>
        
          <li><a href="./Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/Ontology.html">Bio::SQL::Ontology</a></li>
        
          <li><a href="./Bio/SQL/Reference.html">Bio::SQL::Reference</a></li>
        
          <li><a href="./Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a></li>
        
          <li><a href="./Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a></li>
        
          <li><a href="./Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a></li>
        
          <li><a href="./Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a></li>
        
          <li><a href="./Bio/SQL/Sequence.html">Bio::SQL::Sequence</a></li>
        
          <li><a href="./Bio/SQL/Taxon.html">Bio::SQL::Taxon</a></li>
        
          <li><a href="./Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a></li>
        
          <li><a href="./Bio/SQL/Term.html">Bio::SQL::Term</a></li>
        
          <li><a href="./Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a></li>
        
          <li><a href="./Bio/SQL/TermPath.html">Bio::SQL::TermPath</a></li>
        
          <li><a href="./Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a></li>
        
          <li><a href="./Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a></li>
        
          <li><a href="./Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a></li>
        
          <li><a href="./Bio/SangerChromatogram.html">Bio::SangerChromatogram</a></li>
        
          <li><a href="./Bio/Scf.html">Bio::Scf</a></li>
        
          <li><a href="./Bio/Sequence.html">Bio::Sequence</a></li>
        
          <li><a href="./Bio/Sequence/AA.html">Bio::Sequence::AA</a></li>
        
          <li><a href="./Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a></li>
        
          <li><a href="./Bio/Sequence/Common.html">Bio::Sequence::Common</a></li>
        
          <li><a href="./Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a></li>
        
          <li><a href="./Bio/Sequence/Format.html">Bio::Sequence::Format</a></li>
        
          <li><a href="./Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a></li>
        
          <li><a href="./Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a></li>
        
          <li><a href="./Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a></li>
        
          <li><a href="./Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a></li>
        
          <li><a href="./Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a></li>
        
          <li><a href="./Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a></li>
        
          <li><a href="./Bio/Sequence/NA.html">Bio::Sequence::NA</a></li>
        
          <li><a href="./Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a></li>
        
          <li><a href="./Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a></li>
        
          <li><a href="./Bio/Shell.html">Bio::Shell</a></li>
        
          <li><a href="./Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a></li>
        
          <li><a href="./Bio/Shell/Core.html">Bio::Shell::Core</a></li>
        
          <li><a href="./Bio/Shell/Demo.html">Bio::Shell::Demo</a></li>
        
          <li><a href="./Bio/Shell/Ghost.html">Bio::Shell::Ghost</a></li>
        
          <li><a href="./Bio/Shell/Irb.html">Bio::Shell::Irb</a></li>
        
          <li><a href="./Bio/Shell/Private.html">Bio::Shell::Private</a></li>
        
          <li><a href="./Bio/Shell/Script.html">Bio::Shell::Script</a></li>
        
          <li><a href="./Bio/Shell/Setup.html">Bio::Shell::Setup</a></li>
        
          <li><a href="./Bio/Shell/Web.html">Bio::Shell::Web</a></li>
        
          <li><a href="./Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a></li>
        
          <li><a href="./Bio/SiRNA.html">Bio::SiRNA</a></li>
        
          <li><a href="./Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a></li>
        
          <li><a href="./Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a></li>
        
          <li><a href="./Bio/Sim4.html">Bio::Sim4</a></li>
        
          <li><a href="./Bio/Sim4/Report.html">Bio::Sim4::Report</a></li>
        
          <li><a href="./Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a></li>
        
          <li><a href="./Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a></li>
        
          <li><a href="./Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a></li>
        
          <li><a href="./Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a></li>
        
          <li><a href="./Bio/Spidey.html">Bio::Spidey</a></li>
        
          <li><a href="./Bio/Spidey/Report.html">Bio::Spidey::Report</a></li>
        
          <li><a href="./Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a></li>
        
          <li><a href="./Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a></li>
        
          <li><a href="./Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a></li>
        
          <li><a href="./Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a></li>
        
          <li><a href="./Bio/SwissProt.html">Bio::SwissProt</a></li>
        
          <li><a href="./Bio/TMHMM.html">Bio::TMHMM</a></li>
        
          <li><a href="./Bio/TMHMM/Report.html">Bio::TMHMM::Report</a></li>
        
          <li><a href="./Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a></li>
        
          <li><a href="./Bio/TRANSFAC.html">Bio::TRANSFAC</a></li>
        
          <li><a href="./Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a></li>
        
          <li><a href="./Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a></li>
        
          <li><a href="./Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a></li>
        
          <li><a href="./Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a></li>
        
          <li><a href="./Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a></li>
        
          <li><a href="./Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a></li>
        
          <li><a href="./Bio/TargetP.html">Bio::TargetP</a></li>
        
          <li><a href="./Bio/TargetP/Report.html">Bio::TargetP::Report</a></li>
        
          <li><a href="./Bio/Taxonomy.html">Bio::Taxonomy</a></li>
        
          <li><a href="./Bio/Tcoffee.html">Bio::Tcoffee</a></li>
        
          <li><a href="./Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a></li>
        
          <li><a href="./Bio/TogoWS.html">Bio::TogoWS</a></li>
        
          <li><a href="./Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a></li>
        
          <li><a href="./Bio/TogoWS/REST.html">Bio::TogoWS::REST</a></li>
        
          <li><a href="./Bio/TrEMBL.html">Bio::TrEMBL</a></li>
        
          <li><a href="./Bio/Tree.html">Bio::Tree</a></li>
        
          <li><a href="./Bio/Tree/Edge.html">Bio::Tree::Edge</a></li>
        
          <li><a href="./Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a></li>
        
          <li><a href="./Bio/Tree/Node.html">Bio::Tree::Node</a></li>
        
          <li><a href="./Bio/UniProt.html">Bio::UniProt</a></li>
        
          <li><a href="./BiorubyController.html">BiorubyController</a></li>
        
          <li><a href="./BiorubyGenerator.html">BiorubyGenerator</a></li>
        
          <li><a href="./BiorubyHelper.html">BiorubyHelper</a></li>
        
          <li><a href="./Hash.html">Hash</a></li>
        
          <li><a href="./Object.html">Object</a></li>
        
          <li><a href="./String.html">String</a></li>
        
        </ul>
        <div id="no-class-search-results" style="display: none;">No matching classes.</div>
      </div>

      
    </div>
  </div>

  <div id="documentation">
    <h1 class="module">Bio</h1>

    <div id="description" class="description">
      
<h1>bio.rb - Loading all BioRuby modules</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001-2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>
<hr style="height: 1px">

<p>+</p>
<hr style="height: 1px">

<h1>bio/appl/blast/format8.rb - BLAST tab-delimited output (-m 8) parser</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2002, 2003, 2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>

<h2>Note</h2>

<p>This file is automatically loaded by bio/appl/blast/report.rb</p>
<hr style="height: 1px">

<h1>bio/appl/emboss.rb - <a href="Bio/EMBOSS.html">EMBOSS</a> wrapper</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2002, 2005 Toshiaki Katayama&lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006       Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id: emboss.rb,v 1.9 2008/01/10 03:51:06 ngoto Exp $</p>
<hr style="height: 1px">

<p>+</p>
<hr style="height: 1px">

<h1>bio/appl/gcg/seq.rb - <a href="Bio/GCG.html">GCG</a> sequence file format class (.seq/.pep file)</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2003, 2006 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id: seq.rb,v 1.3 2007/04/05 23:35:39 trevor Exp $</p>

<h1>About <a href="Bio/GCG/Seq.html">Bio::GCG::Seq</a></h1>

<p>Please refer document of <a href="Bio/GCG/Seq.html">Bio::GCG::Seq</a>.</p>
<hr style="height: 1px">

<h1>bio/appl/iprscan/report.rb - a class for iprscan output.</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006 Mitsuteru C. Nakao &lt;mn@kazusa.or.jp&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: report.rb,v 1.9 2007/07/18 11:11:57 nakao Exp $</pre>

<h2>Report classes for the iprscan program.</h2>
<hr style="height: 1px">

<h1>bio/appl/meme/motif.rb - Class to represent a sequence motif</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2008 Adam Kraut &lt;adamnkraut@gmail.com&gt;,</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<h2>Description</h2>

<p>This file contains a minimal class to represent meme motifs</p>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://meme.sdsc.edu/meme/intro.html">meme.sdsc.edu/meme/intro.html</a></p>
</li></ul>
<hr style="height: 1px">

<h1>bio/appl/muscle.rb - MUSCLE application wrapper class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of
Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: muscle.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Muscle.html">Bio::Muscle</a> is a wrapper class to execute
MUSCLE.</p>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a href="http://www.drive5.com/muscle/">www.drive5.com/muscle/</a></p>
</li><li>
<p>Edgar R.C. MUSCLE: multiple sequence alignment with high accuracy and high
throughput. Nucleic Acids Res. 32: 1792-1797, 2004.</p>
</li><li>
<p>Edgar, R.C. MUSCLE: a multiple sequence alignment method with reduced time
and space complexity. BMC Bioinformatics 5: 113, 2004.</p>
</li></ul>
<hr style="height: 1px">

<h1>bio/appl/phylip/alignment.rb - phylip multiple alignment format parser</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006 GOTO Naohisa &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: alignment.rb,v 1.2 2007/04/05 23:35:40 trevor Exp $</pre>

<h1>About <a href="Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></h1>

<p>Please refer document of <a
href="Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a> class.</p>
<hr style="height: 1px">

<h1>bio/appl/probcons.rb - ProbCons application wrapper class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of
Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: probcons.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Probcons.html">Bio::Probcons</a> is a wrapper class to execute
ProbCons (Probabilistic Consistency-based Multiple <a
href="Bio/Alignment.html">Alignment</a> of Amino Acid Sequences).</p>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a href="http://probcons.stanford.edu/">probcons.stanford.edu/</a></p>
</li><li>
<p>Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. ProbCons:
Probabilistic Consistency-based Multiple <a
href="Bio/Sequence.html">Sequence</a> <a
href="Bio/Alignment.html">Alignment</a>. Genome Research 15: 330-340, 2005.</p>
</li></ul>
<hr style="height: 1px">

<h1>bio/appl/sim4/report.rb - sim4 result parser</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2004 GOTO Naohisa &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>

<p>The sim4 report parser classes.</p>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p>Florea, L., et al., A Computer program for aligning a cDNA sequence with a
genomic DNA sequence, Genome Research, 8, 967--974, 1998. <a
href="http://www.genome.org/cgi/content/abstract/8/9/967">www.genome.org/cgi/content/abstract/8/9/967</a></p>
</li></ul>
<hr style="height: 1px">

<h1>bio/appl/sosui/report.rb - <a href="Bio/SOSUI.html">SOSUI</a> report class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2003  Mitsuteru C. Nakao &lt;n@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>

<h2>Example</h2>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://bp.nuap.nagoya-u.ac.jp/sosui/">bp.nuap.nagoya-u.ac.jp/sosui/</a></p>
</li><li>
<p><a
href="http://bp.nuap.nagoya-u.ac.jp/sosui/sosui_submit.html">bp.nuap.nagoya-u.ac.jp/sosui/sosui_submit.html</a></p>
</li></ul>
<hr style="height: 1px">

<h1>bio/appl/targetp/report.rb - <a href="Bio/TargetP.html">TargetP</a> report class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2003  Mitsuteru C. Nakao &lt;n@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>

<h2>Description</h2>

<p><a href="Bio/TargetP.html">TargetP</a> class for <a
href="http://www.cbs.dtu.dk/services/TargetP/">www.cbs.dtu.dk/services/TargetP/</a></p>

<h2>Example</h2>

<h2><a href="Bio/References.html">References</a></h2>
<hr style="height: 1px">

<h1>bio/appl/tcoffee.rb - T-Coffee application wrapper class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of
Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: tcoffee.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Tcoffee.html">Bio::Tcoffee</a> is a wrapper class to execute
T-Coffee.</p>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://www.tcoffee.org/Projects_home_page/t_coffee_home_page.html">www.tcoffee.org/Projects_home_page/t_coffee_home_page.html</a></p>
</li><li>
<p>Notredame, C., Higgins, D.G. and Heringa, J. T-Coffee: A novel method for
fast and accurate multiple sequence alignment. J. Mol. Biol. 302: 205-217,
2000.</p>
</li></ul>
<hr style="height: 1px">

<h1>bio/compat/references.rb - Obsoleted <a href="Bio/References.html">References</a> class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2008 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;, Jan Aerts &lt;jandot@bioruby.org&gt;, Naohisa Goto
&lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id: references.rb,v 1.1.2.1 2008/03/04 10:07:49 ngoto Exp $</p>

<h2>Description</h2>

<p>The <a href="Bio/References.html">Bio::References</a> class was obsoleted
after BioRuby 1.2.1. To keep compatibility, some wrapper methods are
provided in this file. As the compatibility methods (and <a
href="Bio/References.html">Bio::References</a>) will soon be removed,
Please change your code not to use <a
href="Bio/References.html">Bio::References</a>.</p>

<p>Note that <a href="Bio/Reference.html">Bio::Reference</a> is different from
<a href="Bio/References.html">Bio::References</a>. <a
href="Bio/Reference.html">Bio::Reference</a> still exists for storing a
reference information in sequence entries.</p>
<hr style="height: 1px">

<h1>bio/data/aa.rb - Amino Acids</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>
<hr style="height: 1px">

<h1>bio/data/codontable.rb - Codon Table</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2004 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>

<h2>Data source</h2>

<p>Data in this class is converted from the NCBI's genetic codes page.</p>

<pre>* ((&lt;URL:http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=t&gt;))</pre>

<h3>Examples</h3>

<p>Obtain a codon table No.1 -- Standard (Eukaryote)</p>

<pre>table = Bio::CodonTable[1]</pre>

<p>Obtain a copy of the codon table No.1 to modify.  In this example, reassign
a seleno cystein ('U') to the 'tga' codon.</p>

<pre>table = Bio::CodonTable.copy(1)
table['tga'] = 'U'</pre>

<p>Create a new codon table by your own from the <a href="Hash.html">Hash</a>
which contains pairs of codon and amino acid.  You can also define the
table name in the second argument.</p>

<pre>hash = { 'ttt' =&gt; 'F', 'ttc' =&gt; 'ttc', ... }
table = Bio::CodonTable.new(hash, &quot;my codon table&quot;)</pre>

<p>Obtain a translated amino acid by codon.</p>

<pre>table = Bio::CodonTable[1]
table['ttt']  # =&gt; F</pre>

<p>Reverse translation of a amino acid into a list of relevant codons.</p>

<pre>table = Bio::CodonTable[1]
table.revtrans(&quot;A&quot;)  # =&gt; [&quot;gcg&quot;, &quot;gct&quot;, &quot;gca&quot;, &quot;gcc&quot;]</pre>
<hr style="height: 1px">

<h1>bio/data/na.rb - Nucleic Acids</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>

<h2>Synopsis</h2>

<p><a href="Bio/NucleicAcid.html">Bio::NucleicAcid</a> class contains data
related to nucleic acids.</p>

<h2>Usage</h2>

<p>Examples:</p>

<pre>require 'bio'

puts &quot;### na = Bio::NucleicAcid.new&quot;
na = Bio::NucleicAcid.new

puts &quot;# na.to_re('yrwskmbdhvnatgc')&quot;
p na.to_re('yrwskmbdhvnatgc')

puts &quot;# Bio::NucleicAcid.to_re('yrwskmbdhvnatgc')&quot;
p Bio::NucleicAcid.to_re('yrwskmbdhvnatgc')

puts &quot;# na.weight('A')&quot;
p na.weight('A')

puts &quot;# Bio::NucleicAcid.weight('A')&quot;
p Bio::NucleicAcid.weight('A')

puts &quot;# na.weight('atgc')&quot;
p na.weight('atgc')

puts &quot;# Bio::NucleicAcid.weight('atgc')&quot;
p Bio::NucleicAcid.weight('atgc')</pre>
<hr style="height: 1px">

<p>TODO save on db reading from a genbank or embl object</p>
<hr style="height: 1px">

<p>require 'bio/sequence'</p>
<hr style="height: 1px">

<h1>bio/db/fasta/defline.rb - FASTA defline parser class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2002 GOTO Naohisa &lt;ngoto@gen-info.osaka-u.ac.jp&gt;,
Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id: defline.rb,v 1.1.2.1 2008/06/20 13:22:32 ngoto Exp $</p>

<h2>Description</h2>

<p><a href="Bio/FastaDefline.html">Bio::FastaDefline</a> is a parser class for
definition line (defline) of the FASTA format.</p>

<h2>Examples</h2>

<pre>rub = Bio::FastaDefline.new('&gt;gi|671595|emb|CAA85678.1| rubisco large subunit [Perovskia abrotanoides]')
rub.entry_id       ==&gt; 'gi|671595'
rub.get('emb')     ==&gt; 'CAA85678.1'
rub.emb            ==&gt; 'CAA85678.1'
rub.gi             ==&gt; '671595'
rub.accession      ==&gt; 'CAA85678'
rub.accessions     ==&gt; [ 'CAA85678' ]
rub.acc_version    ==&gt; 'CAA85678.1'
rub.locus          ==&gt; nil
rub.list_ids       ==&gt; [[&quot;gi&quot;, &quot;671595&quot;],
                        [&quot;emb&quot;, &quot;CAA85678.1&quot;, nil],
                        [&quot;Perovskia abrotanoides&quot;]]

ckr = Bio::FastaDefline.new(&quot;&gt;gi|2495000|sp|Q63931|CCKR_CAVPO CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)\001gi|2147182|pir||I51898 cholecystokinin A receptor - guinea pig\001gi|544724|gb|AAB29504.1| cholecystokinin A receptor; CCK-A receptor [Cavia]&quot;)
ckr.entry_id      ==&gt; &quot;gi|2495000&quot;
ckr.sp            ==&gt; &quot;CCKR_CAVPO&quot;
ckr.pir           ==&gt; &quot;I51898&quot;
ckr.gb            ==&gt; &quot;AAB29504.1&quot;
ckr.gi            ==&gt; &quot;2495000&quot;
ckr.accession     ==&gt; &quot;AAB29504&quot;
ckr.accessions    ==&gt; [&quot;Q63931&quot;, &quot;AAB29504&quot;]
ckr.acc_version   ==&gt; &quot;AAB29504.1&quot;
ckr.locus         ==&gt; nil
ckr.description   ==&gt;
  &quot;CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)&quot;
ckr.descriptions  ==&gt;
  [&quot;CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)&quot;,
   &quot;cholecystokinin A receptor - guinea pig&quot;,
   &quot;cholecystokinin A receptor; CCK-A receptor [Cavia]&quot;]
ckr.words         ==&gt; 
  [&quot;cavia&quot;, &quot;cck-a&quot;, &quot;cck-ar&quot;, &quot;cholecystokinin&quot;, &quot;guinea&quot;, &quot;pig&quot;,
   &quot;receptor&quot;, &quot;type&quot;]
ckr.id_strings    ==&gt;
  [&quot;2495000&quot;, &quot;Q63931&quot;, &quot;CCKR_CAVPO&quot;, &quot;2147182&quot;, &quot;I51898&quot;,
   &quot;544724&quot;, &quot;AAB29504.1&quot;, &quot;Cavia&quot;]
ckr.list_ids      ==&gt;
  [[&quot;gi&quot;, &quot;2495000&quot;], [&quot;sp&quot;, &quot;Q63931&quot;, &quot;CCKR_CAVPO&quot;],
   [&quot;gi&quot;, &quot;2147182&quot;], [&quot;pir&quot;, nil, &quot;I51898&quot;], [&quot;gi&quot;, &quot;544724&quot;],
   [&quot;gb&quot;, &quot;AAB29504.1&quot;, nil], [&quot;Cavia&quot;]]</pre>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p>FASTA format (WikiPedia) <a
href="http://en.wikipedia.org/wiki/FASTA_format">en.wikipedia.org/wiki/FASTA_format</a></p>
</li><li>
<p><a href="Bio/Fasta.html">Fasta</a> format description (<a
href="Bio/NCBI.html">NCBI</a>) <a
href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml">www.ncbi.nlm.nih.gov/BLAST/fasta.shtml</a></p>
</li></ul>
<hr style="height: 1px">

<h1>bio/db/kegg/common.rb - Common methods for <a href="Bio/KEGG.html">KEGG</a> database classes</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001-2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2003 Masumi Itoh &lt;m@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2009 Kozo Nishida &lt;kozo-ni@is.naist.jp&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<h2>Description</h2>

<p>Note that the modules in this file are intended to be Bio::KEGG::* 
internal use only.</p>

<p>This file contains modules that implement methods commonly used from <a
href="Bio/KEGG.html">KEGG</a> database parser classes.</p>
<hr style="height: 1px">

<h1>bio/db/kegg/genes.rb - KEGG/GENES database class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2002, 2006, 2010 Toshiaki Katayama
&lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>

<h2><a href="Bio/KEGG.html">KEGG</a> GENES parser</h2>

<p>See <a
href="http://www.genome.jp/kegg/genes.html">www.genome.jp/kegg/genes.html</a></p>

<h3>Examples</h3>

<pre>require 'bio/io/fetch'
entry_string = Bio::Fetch.query('genes', 'b0002')

entry = Bio::KEGG::GENES.new(entry_string)

# ENTRY
p entry.entry       # =&gt; Hash

p entry.entry_id    # =&gt; String
p entry.division    # =&gt; String
p entry.organism    # =&gt; String

# NAME
p entry.name        # =&gt; String
p entry.names       # =&gt; Array

# DEFINITION
p entry.definition  # =&gt; String
p entry.eclinks     # =&gt; Array

# PATHWAY
p entry.pathway     # =&gt; String
p entry.pathways    # =&gt; Hash

# POSITION
p entry.position    # =&gt; String
p entry.chromosome  # =&gt; String
p entry.gbposition  # =&gt; String
p entry.locations   # =&gt; Bio::Locations

# MOTIF
p entry.motifs      # =&gt; Hash of Array

# DBLINKS
p entry.dblinks     # =&gt; Hash of Array

# STRUCTURE
p entry.structure   # =&gt; Array

# CODON_USAGE
p entry.codon_usage # =&gt; Hash
p entry.cu_list     # =&gt; Array

# AASEQ
p entry.aaseq       # =&gt; Bio::Sequence::AA
p entry.aalen       # =&gt; Fixnum

# NTSEQ
p entry.ntseq       # =&gt; Bio::Sequence::NA
p entry.naseq       # =&gt; Bio::Sequence::NA
p entry.ntlen       # =&gt; Fixnum
p entry.nalen       # =&gt; Fixnum</pre>
<hr style="height: 1px">

<h1>bio/db/kegg/keggtab.rb - <a href="Bio/KEGG.html">KEGG</a> keggtab class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001 Mitsuteru C. Nakao &lt;n@bioruby.org&gt; Copyright (C)
2003, 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: keggtab.rb,v 1.10 2007/04/05 23:35:41 trevor Exp $</pre>
<hr style="height: 1px">

<h1>bio/db/kegg/taxonomy.rb - <a href="Bio/KEGG.html">KEGG</a> taxonomy parser class</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>
<hr style="height: 1px">

<p>bio/db/lasergene.rb - Interface for DNAStar <a
href="Bio/Lasergene.html">Lasergene</a> sequence file format</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Author</p></td>
<td>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (c) 2007 Center for Biomedical Research Informatics, University
of Minnesota (<a href="http://cbri.umn.edu">cbri.umn.edu</a>)</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>
<hr style="height: 1px">

<p>definition of the <a href="Bio/PDB.html">PDB</a> class</p>
<hr style="height: 1px">

<p>Autoload definition</p>
<hr style="height: 1px">

<p>bio/db/soft.rb - Interface for <a href="Bio/SOFT.html">SOFT</a> formatted
files</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Author</p></td>
<td>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (c) 2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>
<hr style="height: 1px">

<h1>bio/io/flatfile.rb - flatfile access wrapper class</h1>

<pre>Copyright (C) 2001-2006 Naohisa Goto &lt;ng@bioruby.org&gt;</pre>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>

<p><a href="Bio/FlatFile.html">Bio::FlatFile</a> is a helper and wrapper class
to read a biological data file. It acts like a IO object. It can
automatically detect data format, and users do not need to tell the class
what the data is.</p>
<hr style="height: 1px">

<p>Ok Hilmar gives to me some clarification 1) "EMBL/GenBank/SwissProt" name
in term table, is only a convention assuming data loaded by genbank embl
ans swissprot formats.</p>

<pre>If your features come from others ways for example blast or alignment ... whatever.. the user as to take care about the source.</pre>

<p>TODO: 1) source_term_id =&gt; surce_term and check before if the source
term is present or not and the level, the root should always be something
"EMBL/GenBank/SwissProt" or contestualized. 2) Into DummyBase class delete
connection there and use Bio::ArSQL.establish_connection which reads info
from a yml file. 3) Chk <a href="Bio/Locations.html">Locations</a> in
Biofeatures ArSQL</p>
<hr style="height: 1px">

<h1>bio/location.rb - Locations/Location class (<a href="Bio/GenBank.html">GenBank</a> location format)</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>

<pre>2006       Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;
2008       Naohisa Goto &lt;ng@bioruby.org&gt;</pre>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>
<hr style="height: 1px">

<h1>bio/sequence/common.rb - common methods for biological sequence</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>
<hr style="height: 1px">

<h1>bio/sequence/compat.rb - methods for backward compatibility</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id:$</p>
<hr style="height: 1px">

<h1>bio/sequence/quality_score.rb - <a href="Bio/Sequence.html">Sequence</a> quality score manipulation modules</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2009 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<h2>Description</h2>

<p><a href="Bio/Sequence.html">Sequence</a> quality score manipulation
modules, mainly used by <a href="Bio/Fastq.html">Bio::Fastq</a> and related
classes.</p>

<h2><a href="Bio/References.html">References</a></h2>
<ul><li>
<p>FASTQ format specification <a
href="http://maq.sourceforge.net/fastq.shtml">maq.sourceforge.net/fastq.shtml</a></p>
</li></ul>
<hr style="height: 1px">

<h1>bio/sequence/sequence_masker.rb - <a href="Bio/Sequence.html">Sequence</a> masking helper methods</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2010 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<h2>Description</h2>

<p><a
href="Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
is a mix-in module to provide helpful methods for masking a sequence.</p>

<p>For details, see documentation of <a
href="Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>.</p>
<hr style="height: 1px">

<h1>bio/shell/plugin/keggdas.rb - plugin for <a href="Bio/KEGG.html">KEGG</a> <a href="Bio/DAS.html">DAS</a></h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<p>$Id: das.rb,v 1.3 2007/04/05 23:35:41 trevor Exp $</p>
<hr style="height: 1px">

<p>bio/util/color_scheme.rb - Popular color codings for nucleic and amino
acids</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Author</p></td>
<td>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (c) 2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>
<hr style="height: 1px">

<p>bio/util/contingency_table.rb - Statistical contingency table analysis for
aligned sequences</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Author</p></td>
<td>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (c) 2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>
<hr style="height: 1px">

<p>bio/util/restriction_enzyme.rb - Digests DNA based on restriction enzyme
cut patterns</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Author</p></td>
<td>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (c) 2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id:$</pre>
<hr style="height: 1px">

<p>bio/util/restriction_enzyme/cut_symbol.rb - Defines the symbol used to mark
a cut in an enzyme sequence</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Author</p></td>
<td>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (c) 2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

<pre>$Id: cut_symbol.rb,v 1.6 2007/07/16 19:28:48 k Exp $</pre>
<hr style="height: 1px">

<p>bio/util/restriction_enzyme/dense_int_array.rb - Internal data storage for
<a
href="Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2011 Naohisa Goto &lt;ng@bioruby.org&gt; Tomoaki NISHIYAMA</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>
<hr style="height: 1px">

<p>bio/util/restriction_enzyme/sorted_num_array.rb - Internal data storage for
<a
href="Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2011 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>
<hr style="height: 1px">

<h1>bio/version.rb - BioRuby version information</h1>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>Copyright</p></td>
<td>
<p>Copyright (C) 2001-2009 Toshiaki Katayama &lt;k@bioruby.org&gt;, Naohisa
Goto &lt;ng@bioruby.org&gt;</p>
</td></tr><tr><td class="rdoc-term"><p>License</p></td>
<td>
<p>The Ruby License</p>
</td></tr></table>

    </div><!-- description -->

    
    
    
    <div id="5Buntitled-5D" class="documentation-section">
      

      

      
      <!-- Constants -->
      <div id="constants-list" class="section">
        <h3 class="section-header">Constants</h3>
        <dl>
        
          <dt><a name="BIORUBY_EXTRA_VERSION">BIORUBY_EXTRA_VERSION</a></dt>
          
          <dd class="description"><p>Extra version specifier (<a href="String.html">String</a> or nil).
Existance of the value indicates pre-release version or modified version.</p></dd>
          
        
          <dt><a name="BIORUBY_VERSION">BIORUBY_VERSION</a></dt>
          
          <dd class="description"><p>BioRuby version (Array containing Integer)</p></dd>
          
        
          <dt><a name="BIORUBY_VERSION_ID">BIORUBY_VERSION_ID</a></dt>
          
          <dd class="description"><p>Version identifier, including extra version string (<a
href="String.html">String</a>) Unlike <a
href="Bio.html#BIORUBY_VERSION">BIORUBY_VERSION</a>, it is not comparable.</p></dd>
          
        
        </dl>
      </div>
      

      

      <!-- Methods -->
      
      <div id="public-class-method-details" class="method-section section">
        <h3 class="section-header">Public Class Methods</h3>

      
        <div id="method_missing-method" class="method-detail ">
          <a name="method-c-method_missing"></a>

          
          <div class="method-heading">
            <span class="method-name">method_missing</span><span
              class="method-args">(*args)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Provide BioRuby shell ‘command’ also as ‘Bio.command’ (like ChemRuby)</p>
            

            
            <div class="method-source-code" id="method_missing-source">
<pre>
<span class="ruby-comment"># File lib/bio.rb, line 309</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">method_missing</span>(*<span class="ruby-identifier">args</span>)
  <span class="ruby-identifier">require</span> <span class="ruby-string">'bio/shell'</span>
  <span class="ruby-identifier">extend</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Shell</span>
  <span class="ruby-identifier">public_class_method</span>(*<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Shell</span>.<span class="ruby-identifier">private_instance_methods</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-constant">Bio</span>.<span class="ruby-identifier">respond_to?</span>(<span class="ruby-identifier">args</span>.<span class="ruby-identifier">first</span>)
    <span class="ruby-constant">Bio</span>.<span class="ruby-identifier">send</span>(*<span class="ruby-identifier">args</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">raise</span> <span class="ruby-constant">NameError</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- method_missing-source -->
            
          </div>

          

          
        </div><!-- method_missing-method -->

      
      </div><!-- public-class-method-details -->
    
    </div><!-- 5Buntitled-5D -->
  

  </div><!-- documentation -->

  <div id="validator-badges">
    <p><small><a href="http://validator.w3.org/check/referer">[Validate]</a></small></p>
    <p><small>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish
      Rdoc Generator</a> 2</small>.</p>
  </div>

</body>
</html>

