<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>ChangeLog - BioRuby API documentation</title>

<link type="text/css" media="screen" href="./rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "./";
</script>

<script type="text/javascript" charset="utf-8" src="./js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/darkfish.js"></script>


<body class="file">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="./index.html">Home</a>
    <a href="./table_of_contents.html#classes">Classes</a>
    <a href="./table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="./ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="./LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="./README_rdoc.html">README</a>
  
    <li class="file"><a href="./README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="./Bio.html">Bio</a>
  
    <li><a href="./Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="./Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="./Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="./Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="./Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="./Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="./Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="./Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="./Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="./Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="./Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="./Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="./Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="./Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="./Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="./Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="./Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="./Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="./Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="./Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="./Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="./Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="./Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="./Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="./Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="./Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="./Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="./Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="./Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="./Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="./Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="./Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="./Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="./Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="./Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="./Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="./Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="./Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="./Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="./Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="./Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="./Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="./Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="./Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="./Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="./Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="./Bio/Command.html">Bio::Command</a>
  
    <li><a href="./Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="./Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="./Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="./Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="./Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="./Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="./Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="./Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="./Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="./Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="./Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="./Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="./Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="./Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="./Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="./Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="./Bio/DB.html">Bio::DB</a>
  
    <li><a href="./Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="./Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="./Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="./Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="./Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="./Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="./Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="./Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="./Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="./Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="./Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="./Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="./Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="./Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="./Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="./Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="./Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="./Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="./Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="./Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="./Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="./Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="./Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="./Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="./Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="./Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="./Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="./Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="./Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="./Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="./Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="./Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="./Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="./Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="./Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="./Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="./Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="./Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="./Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="./Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="./Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="./Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="./Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="./Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="./Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="./Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="./Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="./Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="./Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="./Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="./Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="./Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="./Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="./Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="./Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="./Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="./Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="./Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="./Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="./Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="./Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="./Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="./Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="./Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="./Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="./Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="./Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="./Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="./Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="./Bio/Features.html">Bio::Features</a>
  
    <li><a href="./Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="./Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="./Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="./Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="./Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="./Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="./Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="./Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="./Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="./Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="./Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="./Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="./Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="./Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="./Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="./Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="./Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="./Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="./Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="./Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="./Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="./Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="./Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="./Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="./Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="./Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="./Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="./Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="./Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="./Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="./Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="./Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="./Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="./Bio/GO.html">Bio::GO</a>
  
    <li><a href="./Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="./Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="./Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="./Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="./Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="./Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="./Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="./Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="./Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="./Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="./Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="./Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="./Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="./Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="./Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="./Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="./Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="./Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="./Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="./Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="./Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="./Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="./Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="./Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="./Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="./Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="./Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="./Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="./Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="./Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="./Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="./Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="./Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="./Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="./Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="./Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="./Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="./Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="./Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="./Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="./Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="./Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="./Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="./Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="./Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="./Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="./Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="./Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="./Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="./Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="./Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="./Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="./Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="./Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="./Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="./Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="./Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="./Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="./Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="./Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="./Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="./Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="./Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="./Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="./Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="./Bio/Location.html">Bio::Location</a>
  
    <li><a href="./Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="./Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="./Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="./Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="./Bio/Map.html">Bio::Map</a>
  
    <li><a href="./Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="./Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="./Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="./Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="./Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="./Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="./Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="./Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="./Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="./Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="./Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="./Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="./Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="./Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="./Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="./Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="./Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="./Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="./Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="./Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="./Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="./Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="./Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="./Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="./Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="./Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="./Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="./Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="./Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="./Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="./Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="./Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="./Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="./Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="./Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="./Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="./Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="./Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="./Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="./Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="./Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="./Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="./Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="./Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="./Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="./Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="./Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="./Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="./Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="./Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="./Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="./Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="./Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="./Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="./Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="./Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="./Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="./Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="./Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="./Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="./Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="./Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="./Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="./Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="./Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="./Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="./Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="./Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="./Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="./Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="./Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="./Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="./Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="./Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="./Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="./Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="./Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="./Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="./Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="./Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="./Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="./Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="./Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="./Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="./Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="./Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="./Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="./Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="./Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="./Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="./Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="./Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="./Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="./Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="./Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="./Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="./Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="./Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="./Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="./Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="./Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="./Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="./Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="./Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="./Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="./Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="./Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="./Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="./Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="./Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="./Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="./Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="./Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="./Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="./Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="./Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="./Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="./Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="./Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="./Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="./Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="./Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="./Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="./Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="./Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="./Bio/References.html">Bio::References</a>
  
    <li><a href="./Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="./Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="./Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="./Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="./Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="./Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="./Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="./Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="./Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="./Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="./Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="./Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="./Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="./Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="./Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="./Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="./Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="./Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="./Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="./Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="./Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="./Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="./Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="./Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="./Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="./Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="./Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="./Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="./Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="./Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="./Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="./Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="./Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="./Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="./Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="./Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="./Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="./Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="./Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="./Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="./Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="./Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="./Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="./Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="./Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="./Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="./Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="./Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="./Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="./Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="./Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="./Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="./Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="./Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="./Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="./Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="./Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="./Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="./Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="./Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="./Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="./Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="./Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="./Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="./Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="./Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="./Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="./Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="./Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="./Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="./Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="./Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="./Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="./Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="./Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="./Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="./Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="./Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="./Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="./Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="./Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="./Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="./Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="./Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="./Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="./Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="./Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="./Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="./Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="./Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="./Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="./Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="./Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="./Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="./Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="./Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="./Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="./Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="./Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="./Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="./Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="./Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="./Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="./Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="./Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="./Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="./Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="./Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="./Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="./Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="./Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="./Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="./Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="./Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="./Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="./BiorubyController.html">BiorubyController</a>
  
    <li><a href="./BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="./BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="./Hash.html">Hash</a>
  
    <li><a href="./Object.html">Object</a>
  
    <li><a href="./String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation" class="description">
  
<p>commit 51ab2dec144c99a14ca9009c7b589b500f1cad5f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 22 00:12:47 2012 +0900</p>

<pre>   Preparation to re-release BioRuby 1.4.3

ChangeLog |   22 ++++++++++++++++++++++
1 files changed, 22 insertions(+), 0 deletions(-)</pre>

<p>commit 5ff159d12252393ff04afe52b59a315d15c63d18 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 22 00:00:40 2012 +0900</p>

<pre>   Bug fix: bin/bioruby failed to save object

    * Bug fix: bin/bioruby: Failed to save object with error message
      &quot;can't convert Symbol into String&quot; on Ruby 1.9.

RELEASE_NOTES.rdoc    |    2 ++
lib/bio/shell/core.rb |    1 +
2 files changed, 3 insertions(+), 0 deletions(-)</pre>

<p>commit 74c6ce09413e7ddde1431d74e10cc9c4cdbb95ba Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 21 22:35:18 2012 +0900</p>

<pre>   BioRuby 1.4.3 is released.

ChangeLog |   21 +++++++++++++++++++++
1 files changed, 21 insertions(+), 0 deletions(-)</pre>

<p>commit 61af85b6cfc7bb1f3668ed68232113eb0751e7ea Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 21 22:33:30 2012 +0900</p>

<pre>   preparation for BioRuby 1.4.3 release version

bioruby.gemspec    |    2 +-
lib/bio/version.rb |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 1ec68beac42a06e9ef0a9c953650ef4d599e4e65 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 21 20:53:04 2012 +0900</p>

<pre>   ChangeLog modified; release candidate version 1.4.3-rc2

ChangeLog          | 1353 ++++++++++++++++++++++++++++++++++++++++++++++++++++
bioruby.gemspec    |    2 +-
lib/bio/version.rb |    4 +-
3 files changed, 1356 insertions(+), 3 deletions(-)</pre>

<p>commit e0d570b237a8b96ae0c1e7b1ad72c7333be07c52 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Aug 20 20:35:58 2012 +0900</p>

<pre>   version changed to 1.4.3-rc1

bioruby.gemspec    |    3 ++-
lib/bio/version.rb |    2 +-
2 files changed, 3 insertions(+), 2 deletions(-)</pre>

<p>commit 511c81ba67f7b8dc9cff85cf68db654d2feaf52e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Aug 20 20:17:14 2012 +0900</p>

<pre>   document JRUBY-5678 (resolved) and related issue with the workaround.

KNOWN_ISSUES.rdoc  |    9 +++++++++
RELEASE_NOTES.rdoc |    9 +++++++++
2 files changed, 18 insertions(+), 0 deletions(-)</pre>

<p>commit 2fdd7a3b3555a33dead31181c9526af22f24916f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Aug 20 19:44:39 2012 +0900</p>

<pre>   update recommended Ruby versions and the year in copyright lines

README.rdoc |    7 +++----
1 files changed, 3 insertions(+), 4 deletions(-)</pre>

<p>commit b156227749e5ada74330e837c9ce48a16e6a6a2f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Aug 20 19:16:25 2012 +0900</p>

<pre>   Bug fix: Bio::EMBL#os raises error, with incompatible change

    * Bug fix: Bio::EMBL#os raises error. The bug is reported by
      Marc P. Hoeppner in the BioRuby mailing list
      (https://redmine.open-bio.org/issues/3294).
    * Incompatible change: Bio::EMBL#os no longer splits the content with
      comma, and it no longer raises error even if the OS line is not in
      the &quot;Genus species (name)&quot; format. The changes may affect the parsing
      of old EMBL files which contain two or more species names in an OS line.
    * Unit tests are modified to catch up the above incompatible changes.

RELEASE_NOTES.rdoc                       |   14 ++++++
lib/bio/db/embl/embl.rb                  |   74 ++++++++++++++++++++++++++++++
test/unit/bio/db/embl/test_embl.rb       |    9 +---
test/unit/bio/db/embl/test_embl_rel89.rb |    9 +---
4 files changed, 92 insertions(+), 14 deletions(-)</pre>

<p>commit 31c8b4cb6ce2364aacee8137ddec3aa5f7d2d0d8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Aug 20 19:04:50 2012 +0900</p>

<pre>   Workaround for jruby-1.7.0.preview2 bugs JRUBY-6195, JRUBY-6818

    * Workaroud for jruby-1.7.0.preview2 bugs JRUBY-6195 and JRUBY-6818.
    * Refactoring of call_command_popen: split _call_command_popen_ruby18
      and _call_command_popen_ruby19, add _call_command_popen_jruby19.
      Note that _call_command_popen_jruby19 will be removed in the future
      after the bugs are fixed.

lib/bio/command.rb |   98 ++++++++++++++++++++++++++++++++++++++++++++++-----
1 files changed, 88 insertions(+), 10 deletions(-)</pre>

<p>commit 05f51fa2e871e71c2b20559eb05e456768a4f7d6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Aug 18 00:32:31 2012 +0900</p>

<pre>   New default etc/bioinformatics/seqdatabase.ini

    * New default etc/bioinformatics/seqdatabase.ini, with currently
      available services.

etc/bioinformatics/seqdatabase.ini |   27 +++++++++++++++++++++++++++
1 files changed, 27 insertions(+), 0 deletions(-)
create mode 100644 etc/bioinformatics/seqdatabase.ini</pre>

<p>commit a4264cc3667b98289c09efc7ccba9c8e86f6d89c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Aug 18 00:31:10 2012 +0900</p>

<pre>   etc/bioinformatics/seqdatabase.ini is moved to sample/

etc/bioinformatics/seqdatabase.ini |  210 ------------------------------------
sample/seqdatabase.ini             |  210 ++++++++++++++++++++++++++++++++++++
2 files changed, 210 insertions(+), 210 deletions(-)
delete mode 100644 etc/bioinformatics/seqdatabase.ini
create mode 100644 sample/seqdatabase.ini</pre>

<p>commit 04b7a27b557576f5325b3ee420262922ab66ca3b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Aug 18 00:30:38 2012 +0900</p>

<pre>   known issue about http://bioruby.org/cgi-bin/biofetch.rb server down

KNOWN_ISSUES.rdoc |    9 +++++++++
1 files changed, 9 insertions(+), 0 deletions(-)</pre>

<p>commit 4a8193f7b91ff703c8f3dc6e6a6ae0c981a404e6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 17 23:45:41 2012 +0900</p>

<pre>   Update descriptions about JRuby and Rubinius bugs

KNOWN_ISSUES.rdoc  |   14 ++++++++++----
RELEASE_NOTES.rdoc |   14 ++++++++++----
2 files changed, 20 insertions(+), 8 deletions(-)</pre>

<p>commit a2d8dd8ccebde84e91f82c59e531cc08fbf0f3fe Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 17 17:19:22 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Remove</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">suffix</span> .<span class="ruby-identifier">rb</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">require</span>, <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">potential</span> <span class="ruby-identifier">multiple</span> <span class="ruby-identifier">loading</span>.

<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">fasta</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_defline</span>.<span class="ruby-identifier">rb</span>        <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">genbank</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_genpept</span>.<span class="ruby-identifier">rb</span>      <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_drug</span>.<span class="ruby-identifier">rb</span>            <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_genome</span>.<span class="ruby-identifier">rb</span>          <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_glycan</span>.<span class="ruby-identifier">rb</span>          <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_restriction_enzyme</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-value">6</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">6</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">6</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 1d2e8b02db3699c2cd4f4890abc078ffd2b503aa Author: Ben J. Woodcroft
&lt;donttrustben near gmail.com&gt; Date:   Wed Aug 8 09:41:20 2012 +1000</p>

<pre>   fill in missing piece of documentation in FastaFormat

lib/bio/db/fasta.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 83bf09d4d81803c8d06e0d45ca25e7c09016161c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 8 00:08:26 2012 +0900</p>

<pre>   RELEASE_NOTE.rdoc modified to reflect recent changes

RELEASE_NOTES.rdoc |  107 ++++++++++++++++++++++++++++++++++++++++++++-------
1 files changed, 92 insertions(+), 15 deletions(-)</pre>

<p>commit c3afb1eb98cf8777ee021624c3d2eab92b3543f2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 8 00:06:09 2012 +0900</p>

<pre>   Descriptions about JRuby, Rubinius, DDBJ Web API, SOAP4R etc.

KNOWN_ISSUES.rdoc |   45 +++++++++++++++++++++++++++++++++++++++++++--
1 files changed, 43 insertions(+), 2 deletions(-)</pre>

<p>commit 01da7401a011aa519c43a021f89f6e7f769b4649 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 7 23:55:09 2012 +0900</p>

<pre>   regenerate bioruby.gemspec with rake regemspec

bioruby.gemspec |    3 +--
1 files changed, 1 insertions(+), 2 deletions(-)</pre>

<p>commit 9f70c27d9b75408fddae8384a2a09715b959dcb5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 7 23:51:56 2012 +0900</p>

<pre>   improve documentation; version changed to 1.4.3-pre1

lib/bio/version.rb |   13 +++++++++++--
1 files changed, 11 insertions(+), 2 deletions(-)</pre>

<p>commit c11f12c8aa56b8509cd082f3478e96374210e5d7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 7 23:31:41 2012 +0900</p>

<pre>   Remove autorequire which have been deprecated

bioruby.gemspec.erb |    1 -
1 files changed, 0 insertions(+), 1 deletions(-)</pre>

<p>commit 7792b092033d2c819f2bcad0e206f27608481db5 Author: Ben J Woodcroft
&lt;donttrustben@gmail.com&gt; Date:   Mon Aug 6 09:40:55 2012 +1000</p>

<pre>   flesh out FastaFormat documentation

lib/bio/db/fasta.rb         |  102 ++++++++++++++++++++++++-------------------
lib/bio/db/fasta/defline.rb |    2 +-
2 files changed, 58 insertions(+), 46 deletions(-)</pre>

<p>commit 9a2fe67c247cdc7c9ddc9f8b8de771515ba76ac1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 3 22:36:12 2012 +0900</p>

<pre>   .travis.yml: restructure matrix, add allow_failures lines

    * Add allow_failures lines
    * Restructure matrix: remove many exclude lines and add some
      include lines.
    * When running jruby, Set TMPDIR to avoid known issue about
      FileUtils#remove_entry_secure.

.travis.yml |   52 ++++++++++++++++++----------------------------------
1 files changed, 18 insertions(+), 34 deletions(-)</pre>

<p>commit 553fd102c533c42675f93895557e3e00d36fd3e7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 3 22:05:39 2012 +0900</p>

<pre>   Improve tests for BLAST &quot;-m 8&quot; tabular format parser

test/unit/bio/appl/blast/test_report.rb |  119 +++++++++++++++++++++++++++++++
1 files changed, 119 insertions(+), 0 deletions(-)</pre>

<p>commit 3e1c062dbc168bd558ca8408a6da115aa570f3a7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 3 22:05:07 2012 +0900</p>

<pre>   Improve test and suppress warning: assigned but unused variable

test/unit/bio/io/flatfile/test_buffer.rb |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 7e29ce1f050e9e5b23299372d8ddfae781447dc3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 3 22:02:21 2012 +0900</p>

<pre>   Improve test and suppress warning: assigned but unused variable

test/unit/bio/db/test_newick.rb |    2 ++
1 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit 1053b62069df74f336934e4ed0f3f217e4ad3312 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 27 13:56:53 2012 +0900</p>

<pre>   Suppress warnings: shadowing outer local variable

    * Suppress warnings: shadowing outer local variable.
      Thanks to Andrew Grimm: https://github.com/bioruby/bioruby/pull/64

lib/bio/db/gff.rb |    8 ++++----
1 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit e55794f65b3fb45c99e61d45220fe42f718426a3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 25 23:29:17 2012 +0900</p>

<pre>   Suppress warnings in lib/bio/alignment.rb:2322

    * A space is inserted to suppress warnings in lib/bio/alignment.rb:2322.
      * warning: :' after local variable is interpreted as binary operator
      * warning: even though it seems like symbol literal

lib/bio/alignment.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 174a38ea8c4ecea70724bf6ec8e72b2e4259853b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 25 23:12:51 2012 +0900</p>

<pre>   Modified to follow changes of GenomeNet BLAST site

lib/bio/appl/blast/genomenet.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 93e24935840dcdec76984313719700134d69daf2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 25 15:21:32 2012 +0900</p>

<pre>   suppress warnings: instance variable @comment not initialized

lib/bio/db/gff.rb |    6 ++++--
1 files changed, 4 insertions(+), 2 deletions(-)</pre>

<p>commit 0ad3818fedb707a26e849877bde1f8dab006b848 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 25 00:54:02 2012 +0900</p>

<pre>   suppress warnings: URI.escape/URI.unescape is obsolete

lib/bio/db/gff.rb |   39 +++++++++++++++++++++++++++++++++------
1 files changed, 33 insertions(+), 6 deletions(-)</pre>

<p>commit 1263938742e7eeedb4a877aff7314e304320eca9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jul 23 21:15:52 2012 +0900</p>

<pre>   Added link to blastall options reference

    * Added link to blastall options reference.
      Thanks to Gareth Rees who sent a pull request.
      (https://github.com/bioruby/bioruby/pull/49)

lib/bio/appl/blast/genomenet.rb |    5 +++++
1 files changed, 5 insertions(+), 0 deletions(-)</pre>

<p>commit 2ec5f4fd5abd0db7ec79ab3a9fd4adde7c9384a8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jul 23 17:26:45 2012 +0900</p>

<pre>   Next bioruby release version will be 1.4.3.

RELEASE_NOTES.rdoc |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 6cf1318507a5d82bb93acdfe33e96723a2e742fc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jul 23 17:25:35 2012 +0900</p>

<pre>   fix typo

README.rdoc |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 2fd71cac315affe6e4d90b03dadac782f11553a5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jul 23 17:21:57 2012 +0900</p>

<pre>   Bug fix: Genomenet remote blast: catch up changes of the server

lib/bio/appl/blast/genomenet.rb |   33 +++++++++++++++++++++++----------
1 files changed, 23 insertions(+), 10 deletions(-)</pre>

<p>commit 69d9717da11b2fe81a8f840bbafcc5fbb0dbe688 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 20 11:24:37 2012 +0900</p>

<pre>   regenerate bioruby.gemspec with rake regemspec

bioruby.gemspec |    2 ++
1 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit 9683da186579dbfa5da1bb1a32edc49cfdc026b8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 18 23:19:33 2012 +0900</p>

<pre>   Incompatible changes in Bio::KEGG::KGML are documented.

    * Incompatible changes in Bio::KEGG::KGML are documented.
    * Next BioRuby release version will be 1.4.3.

RELEASE_NOTES.rdoc |   44 +++++++++++++++++++++++++++++++++++++++++---
1 files changed, 41 insertions(+), 3 deletions(-)</pre>

<p>commit 6cab377ae760d1abfda06caafe4a04ecd549e21d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 18 22:56:00 2012 +0900</p>

<pre>   Incompatible changes: Bio::KEGG::KGML::Reaction#substrates, products

    * Incompatible changes: Bio::KEGG::KGML::Reaction#substrates and
      Bio::KEGG::KGML::Reaction#products are changed to return an array
      containing Bio::KEGG::KGML::Substrate and Bio::KEGG::KGML::Product
      objects, respectively. The aim of these changes are to store ID
      of substrates and products that were thrown away in the previous
      versions.

lib/bio/db/kegg/kgml.rb            |   48 ++++++++++++++--
test/unit/bio/db/kegg/test_kgml.rb |  104 +++++++++++++++++++++++++++++++++++-
2 files changed, 144 insertions(+), 8 deletions(-)</pre>

<p>commit 3cb1e09709d3c6b934028e28f9cafed149c9c751 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 18 22:16:46 2012 +0900</p>

<pre>   Bio::KEGG::KGML#parse_* :use new attribute names

    * In Bio::KEGG::KGML#parse_* (private methods) new attribute method
      names should be used instead of deprecated old names.

lib/bio/db/kegg/kgml.rb |   18 +++++++++---------
1 files changed, 9 insertions(+), 9 deletions(-)</pre>

<p>commit c5ef981db6add98dc6778cd9809aff38a7071593 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 18 22:14:33 2012 +0900</p>

<pre>   modified documentation for Bio::KEGG::KGML

lib/bio/db/kegg/kgml.rb |   73 +++++++++++++++++++++++++++--------------------
1 files changed, 42 insertions(+), 31 deletions(-)</pre>

<p>commit 5416b84eaa37b5abf15f905586a5eee65c4026f0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 18 15:01:58 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">New</span> <span class="ruby-keyword">class</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Graphics</span> <span class="ruby-identifier">with</span> <span class="ruby-identifier">tests</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span>

    * <span class="ruby-constant">New</span> <span class="ruby-keyword">class</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Graphics</span> <span class="ruby-keyword">for</span> <span class="ruby-identifier">storing</span> <span class="ruby-identifier">a</span> <span class="ruby-identifier">graphics</span> <span class="ruby-identifier">element</span>.
      <span class="ruby-constant">This</span> <span class="ruby-identifier">fixes</span> <span class="ruby-identifier">https</span>:<span class="ruby-operator">/</span><span class="ruby-regexp">%rgithub.com/</span><span class="ruby-identifier">bioruby</span><span class="ruby-operator">/</span><span class="ruby-identifier">bioruby</span><span class="ruby-operator">/</span><span class="ruby-identifier">issues</span><span class="ruby-operator">/</span><span class="ruby-value">51</span>.
    * <span class="ruby-constant">Unit</span> <span class="ruby-identifier">tests</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">added</span> <span class="ruby-identifier">with</span> <span class="ruby-identifier">mock</span> <span class="ruby-identifier">test</span> <span class="ruby-identifier">data</span>.
    * <span class="ruby-constant">Improve</span> <span class="ruby-identifier">rdoc</span> <span class="ruby-identifier">documentation</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span>.
    * <span class="ruby-constant">New</span> <span class="ruby-identifier">method</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Reaction</span><span class="ruby-comment">#id</span>
    * <span class="ruby-constant">Attribute</span> <span class="ruby-identifier">methods</span> <span class="ruby-identifier">that</span> <span class="ruby-identifier">were</span> <span class="ruby-identifier">different</span> <span class="ruby-identifier">from</span> <span class="ruby-identifier">the</span> <span class="ruby-constant">KGML</span> <span class="ruby-identifier">attribute</span>
      <span class="ruby-identifier">names</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">renamed</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">names</span> <span class="ruby-identifier">of</span> <span class="ruby-identifier">the</span> <span class="ruby-constant">KGML</span> <span class="ruby-identifier">attribute</span> <span class="ruby-identifier">names</span>.
      <span class="ruby-constant">Old</span> <span class="ruby-identifier">method</span> <span class="ruby-identifier">names</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">deprecated</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">changed</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">aliases</span>
      <span class="ruby-keyword">and</span> <span class="ruby-identifier">will</span> <span class="ruby-identifier">be</span> <span class="ruby-identifier">removed</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">future</span>.
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#id (old name: entry_id)</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#type (old name: category)</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#entry1 (old name: node1)</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#entry2 (old name: node2)</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#type (old name: rel)</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Reaction</span><span class="ruby-comment">#name (old name: entry_id)</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Reaction</span><span class="ruby-comment">#type (old name: direction)</span>
    * <span class="ruby-constant">Following</span> <span class="ruby-identifier">attribute</span> <span class="ruby-identifier">methods</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">deprecated</span> <span class="ruby-identifier">because</span> <span class="ruby-identifier">two</span> <span class="ruby-keyword">or</span> <span class="ruby-identifier">more</span>
      <span class="ruby-identifier">graphics</span> <span class="ruby-identifier">elements</span> <span class="ruby-identifier">may</span> <span class="ruby-identifier">exist</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">an</span> <span class="ruby-identifier">entry</span> <span class="ruby-identifier">element</span>. <span class="ruby-constant">They</span> <span class="ruby-identifier">will</span> <span class="ruby-identifier">be</span>
      <span class="ruby-identifier">removed</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">future</span>.
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#label</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#shape</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#x</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#y</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#width</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#height</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#fgcolor</span>
       * <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">KGML</span><span class="ruby-operator">::</span><span class="ruby-constant">Entry</span><span class="ruby-comment">#bgcolor</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">kgml</span>.<span class="ruby-identifier">rb</span>            <span class="ruby-operator">|</span>  <span class="ruby-value">321</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">data</span><span class="ruby-operator">/</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">/</span><span class="ruby-identifier">test</span>.<span class="ruby-identifier">kgml</span>           <span class="ruby-operator">|</span>   <span class="ruby-value">37</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_kgml</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>  <span class="ruby-value">922</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-value">3</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">1223</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">57</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
<span class="ruby-identifier">create</span> <span class="ruby-identifier">mode</span> <span class="ruby-value">100644</span> <span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">data</span><span class="ruby-operator">/</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">/</span><span class="ruby-identifier">test</span>.<span class="ruby-identifier">kgml</span>
<span class="ruby-identifier">create</span> <span class="ruby-identifier">mode</span> <span class="ruby-value">100644</span> <span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_kgml</span>.<span class="ruby-identifier">rb</span>
</pre>

<p>commit e5478363ef6969ec14c4e09c2bd7c6d27c12cf5b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Jul 17 22:23:28 2012 +0900</p>

<pre>   rdoc documentation for Bio::KEGG::KGML

lib/bio/db/kegg/kgml.rb |  166 ++++++++++++++++++++++++++++++++++++++++++++---
1 files changed, 157 insertions(+), 9 deletions(-)</pre>

<p>commit 4a97e7034cae835b3bbc8ef918b9c6c48910dec5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 11 15:16:49 2012 +0900</p>

<pre>   autoload should not be used for libraries outside bio

lib/bio/db/kegg/kgml.rb |    3 +--
1 files changed, 1 insertions(+), 2 deletions(-)</pre>

<p>commit 338d4cd9913d70041349c5201f80f7a65e7135a6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 6 00:50:01 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">remove</span> <span class="ruby-identifier">unnecessary</span> <span class="ruby-identifier">require</span> <span class="ruby-string">&quot;bio/db&quot;</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span>.<span class="ruby-identifier">rb</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">1</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">4</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 87c806a480fcacb0fc610c9669de19e4cb661a9c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 6 00:47:20 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-identifier">about</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">PDB</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">atom</span>.<span class="ruby-identifier">rb</span>              <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">chain</span>.<span class="ruby-identifier">rb</span>             <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">chemicalcomponent</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">model</span>.<span class="ruby-identifier">rb</span>             <span class="ruby-operator">|</span>    <span class="ruby-value">4</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span>.<span class="ruby-identifier">rb</span>               <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">residue</span>.<span class="ruby-identifier">rb</span>           <span class="ruby-operator">|</span>    <span class="ruby-value">4</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">pdb</span><span class="ruby-operator">/</span><span class="ruby-identifier">utils</span>.<span class="ruby-identifier">rb</span>             <span class="ruby-operator">|</span>   <span class="ruby-value">11</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">7</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">21</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">16</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 874f35c3930506fa029b419aa84677d1fea6681a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 6 00:24:24 2012 +0900</p>

<pre>   regenerate bioruby.gemspec with rake regemspec

bioruby.gemspec |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 090d4edb5698135f87df450a963ef35a307349c4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 6 00:19:54 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Tree</span> <span class="ruby-identifier">output</span> (<span class="ruby-identifier">formatter</span>) <span class="ruby-identifier">methods</span> <span class="ruby-identifier">moved</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">tree</span><span class="ruby-operator">/</span><span class="ruby-identifier">output</span>.<span class="ruby-identifier">rb</span>

    * <span class="ruby-constant">To</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-identifier">about</span> <span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">tree</span>, <span class="ruby-identifier">phylogenetic</span> <span class="ruby-identifier">tree</span> <span class="ruby-identifier">output</span>
     (<span class="ruby-identifier">formatter</span>) <span class="ruby-identifier">methods</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">moved</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">tree</span><span class="ruby-operator">/</span><span class="ruby-identifier">output</span>.<span class="ruby-identifier">rb</span>.

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">newick</span>.<span class="ruby-identifier">rb</span>   <span class="ruby-operator">|</span>  <span class="ruby-value">244</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">tree</span>.<span class="ruby-identifier">rb</span>        <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">+</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">tree</span><span class="ruby-operator">/</span><span class="ruby-identifier">output</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>  <span class="ruby-value">264</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-value">3</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">265</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">246</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
<span class="ruby-identifier">create</span> <span class="ruby-identifier">mode</span> <span class="ruby-value">100644</span> <span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">tree</span><span class="ruby-operator">/</span><span class="ruby-identifier">output</span>.<span class="ruby-identifier">rb</span>
</pre>

<p>commit b3d12b63097a5141b029bbfb3690870cd1935a60 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jul 6 00:18:44 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Blast</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">bl2seq</span><span class="ruby-operator">/</span><span class="ruby-identifier">report</span>.<span class="ruby-identifier">rb</span>     <span class="ruby-operator">|</span>    <span class="ruby-value">6</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">ddbj</span>.<span class="ruby-identifier">rb</span>        <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">format0</span>.<span class="ruby-identifier">rb</span>     <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">genomenet</span>.<span class="ruby-identifier">rb</span>   <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">ncbioptions</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>   <span class="ruby-value">11</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">remote</span>.<span class="ruby-identifier">rb</span>      <span class="ruby-operator">|</span>   <span class="ruby-value">11</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">report</span>.<span class="ruby-identifier">rb</span>      <span class="ruby-operator">|</span>   <span class="ruby-value">16</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">rpsblast</span>.<span class="ruby-identifier">rb</span>    <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">wublast</span>.<span class="ruby-identifier">rb</span>     <span class="ruby-operator">|</span>    <span class="ruby-value">6</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">9</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">36</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">27</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 8f6c906c7b0d65b93ebf0a1e1307259e6eab8465 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 23:29:42 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">remove</span> <span class="ruby-identifier">old</span> <span class="ruby-identifier">require</span> <span class="ruby-identifier">lines</span> <span class="ruby-identifier">that</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">commented</span> <span class="ruby-identifier">out</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">appl</span><span class="ruby-operator">/</span><span class="ruby-identifier">blast</span><span class="ruby-operator">/</span><span class="ruby-identifier">format0</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">0</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">5</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit c632fbf2d0320860eadfacb196d51d80ed3a2b34 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 23:16:49 2012 +0900</p>

<pre>   Remove old workaround of strscan.so for Ruby 1.7 or earlier

lib/bio/appl/blast/format0.rb |   18 +-----------------
1 files changed, 1 insertions(+), 17 deletions(-)</pre>

<p>commit c81dce87f53d3ea7c7d2335e077fa609f2737779 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 23:03:40 2012 +0900</p>

<pre>   .travis.yml: include ruby 1.9.2 test

.travis.yml |    2 ++
1 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit 34709d114089c722b5da796028ffb91021761fdd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 23:00:37 2012 +0900</p>

<pre>   Remove old comment lines

lib/bio/sequence/format.rb |    6 ------
1 files changed, 0 insertions(+), 6 deletions(-)</pre>

<p>commit e0d5ed61e0101e2e72ad024dccd58c8c90def2b9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 22:42:17 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Finalizer</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Command</span><span class="ruby-operator">::</span><span class="ruby-constant">Tmpdir</span> <span class="ruby-identifier">is</span> <span class="ruby-identifier">changed</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">suppress</span> <span class="ruby-identifier">test</span> <span class="ruby-identifier">failure</span>

    * <span class="ruby-constant">New</span> <span class="ruby-keyword">class</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Command</span><span class="ruby-operator">::</span><span class="ruby-constant">Tmpdir</span><span class="ruby-operator">::</span><span class="ruby-constant">Remover</span> <span class="ruby-keyword">for</span> <span class="ruby-identifier">removing</span> <span class="ruby-identifier">temporary</span>
      <span class="ruby-identifier">directory</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">finilizer</span>. <span class="ruby-constant">This</span> <span class="ruby-keyword">class</span> <span class="ruby-identifier">is</span> <span class="ruby-constant">BioRuby</span> <span class="ruby-identifier">internal</span> <span class="ruby-identifier">use</span> <span class="ruby-identifier">only</span>.
      <span class="ruby-constant">Users</span> <span class="ruby-identifier">should</span> <span class="ruby-keyword">not</span> <span class="ruby-identifier">use</span> <span class="ruby-identifier">this</span> <span class="ruby-keyword">class</span>.
    * <span class="ruby-constant">Finalizer</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Command</span><span class="ruby-operator">::</span><span class="ruby-constant">Tmpdir</span> <span class="ruby-identifier">is</span> <span class="ruby-identifier">changed</span> <span class="ruby-identifier">from</span> <span class="ruby-identifier">a</span> <span class="ruby-constant">Proc</span> <span class="ruby-identifier">object</span>
      <span class="ruby-identifier">to</span> <span class="ruby-identifier">an</span> <span class="ruby-identifier">instance</span> <span class="ruby-identifier">of</span> <span class="ruby-identifier">the</span> <span class="ruby-constant">Remover</span> <span class="ruby-keyword">class</span>.
    * <span class="ruby-constant">Test</span> <span class="ruby-identifier">failure</span> <span class="ruby-identifier">fix</span><span class="ruby-operator">:</span> <span class="ruby-constant">In</span> <span class="ruby-identifier">some</span> <span class="ruby-identifier">environment</span>, <span class="ruby-identifier">with</span> <span class="ruby-constant">Ruby</span> <span class="ruby-value">1.9</span><span class="ruby-value">.2</span>,
      <span class="ruby-identifier">test_output_embl</span>(<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">FuncTestSequenceOutputEMBL</span>) <span class="ruby-identifier">was</span> <span class="ruby-identifier">failed</span> <span class="ruby-identifier">with</span>
      <span class="ruby-node">&quot;&lt;#&lt;ArgumentError: wrong number of arguments (1 for 0)&gt;&quot;</span> <span class="ruby-identifier">that</span> <span class="ruby-identifier">was</span>
      <span class="ruby-identifier">raised</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">finalizer</span> <span class="ruby-identifier">callback</span> <span class="ruby-identifier">of</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Command</span><span class="ruby-operator">::</span><span class="ruby-constant">Tmpdir</span>. <span class="ruby-constant">This</span>
      <span class="ruby-identifier">commit</span> <span class="ruby-identifier">fixes</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">problem</span>.

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">command</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>   <span class="ruby-value">56</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">30</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">26</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit cca98d1378ce66d6db84cc9c1beadd39ed0e0fee Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 22:21:34 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-keyword">and</span> <span class="ruby-constant">JRuby</span> <span class="ruby-identifier">autoload</span> <span class="ruby-identifier">bug</span>

    * <span class="ruby-string">&quot;require&quot;</span> <span class="ruby-identifier">lines</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">modified</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span>.
    * <span class="ruby-constant">In</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">that</span> <span class="ruby-identifier">would</span> <span class="ruby-identifier">be</span> <span class="ruby-identifier">required</span> <span class="ruby-identifier">directly</span> <span class="ruby-identifier">from</span> <span class="ruby-identifier">outside</span> <span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span>
      (<span class="ruby-identifier">aa</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">adapter</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">common</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">compat</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">dblink</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">generic</span>.<span class="ruby-identifier">rb</span>,
      <span class="ruby-identifier">na</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">quality_score</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">sequence_masker</span>.<span class="ruby-identifier">rb</span>), <span class="ruby-identifier">because</span> <span class="ruby-identifier">of</span> <span class="ruby-identifier">avoiding</span>
      <span class="ruby-identifier">potential</span> <span class="ruby-identifier">mismatch</span> <span class="ruby-identifier">of</span> <span class="ruby-identifier">superclass</span> <span class="ruby-keyword">and</span><span class="ruby-regexp">%ror lack of some methods,
      bio/</span><span class="ruby-identifier">equence</span>.<span class="ruby-identifier">rb</span> <span class="ruby-identifier">is</span> <span class="ruby-identifier">required</span> <span class="ruby-keyword">when</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span> <span class="ruby-identifier">is</span> <span class="ruby-keyword">not</span> <span class="ruby-identifier">defined</span>.
    * <span class="ruby-identifier">workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-constant">JRuby</span> <span class="ruby-identifier">autoload</span> <span class="ruby-identifier">bug</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">rb</span>                 <span class="ruby-operator">|</span>   <span class="ruby-value">10</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">aa</span>.<span class="ruby-identifier">rb</span>              <span class="ruby-operator">|</span>    <span class="ruby-value">8</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">adapter</span>.<span class="ruby-identifier">rb</span>         <span class="ruby-operator">|</span>   <span class="ruby-value">12</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">common</span>.<span class="ruby-identifier">rb</span>          <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">compat</span>.<span class="ruby-identifier">rb</span>          <span class="ruby-operator">|</span>    <span class="ruby-value">9</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">dblink</span>.<span class="ruby-identifier">rb</span>          <span class="ruby-operator">|</span>   <span class="ruby-value">11</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">generic</span>.<span class="ruby-identifier">rb</span>         <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">na</span>.<span class="ruby-identifier">rb</span>              <span class="ruby-operator">|</span>   <span class="ruby-value">10</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">quality_score</span>.<span class="ruby-identifier">rb</span>   <span class="ruby-operator">|</span>    <span class="ruby-value">2</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence_masker</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-value">10</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">37</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">37</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit d2915c33ae7f330837688195a58c1e60fe78402a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 21:04:28 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-keyword">in</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span>

    * <span class="ruby-constant">Workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-keyword">in</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span>
    * <span class="ruby-constant">Special</span> <span class="ruby-identifier">care</span> <span class="ruby-identifier">was</span> <span class="ruby-identifier">needed</span> <span class="ruby-keyword">for</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Analysis</span>
      <span class="ruby-identifier">because</span> <span class="ruby-identifier">its</span> <span class="ruby-identifier">method</span> <span class="ruby-identifier">definitions</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">divided</span> <span class="ruby-identifier">into</span> <span class="ruby-identifier">two</span> <span class="ruby-identifier">files</span><span class="ruby-operator">:</span>
      <span class="ruby-identifier">analysis</span>.<span class="ruby-identifier">rb</span>, <span class="ruby-identifier">analysis_basic</span>.<span class="ruby-identifier">rb</span>.

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">analysis</span>.<span class="ruby-identifier">rb</span>        <span class="ruby-operator">|</span>   <span class="ruby-value">13</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">analysis_basic</span>.<span class="ruby-identifier">rb</span>  <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">cut_symbol</span>.<span class="ruby-identifier">rb</span>      <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">dense_int_array</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">double_stranded</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rdouble_stranded/</span><span class="ruby-identifier">aligned_strands</span>.<span class="ruby-identifier">rb</span>             <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rdouble_stranded/</span><span class="ruby-identifier">cut_location_pair</span>.<span class="ruby-identifier">rb</span>           <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rcut_location_pair_in_enzyme_notation.rb        |    7 +++----
.../</span><span class="ruby-identifier">double_stranded</span><span class="ruby-operator">/</span><span class="ruby-identifier">cut_locations</span>.<span class="ruby-identifier">rb</span>               <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rcut_locations_in_enzyme_notation.rb            |    7 +++----
lib/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">range</span><span class="ruby-operator">/</span><span class="ruby-identifier">cut_range</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rutil/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">range</span><span class="ruby-operator">/</span><span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">rb</span>    <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rrange/</span><span class="ruby-identifier">horizontal_cut_range</span>.<span class="ruby-identifier">rb</span>                  <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rrestriction_enzyme/</span><span class="ruby-identifier">range</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">rb</span>     <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rrange/</span><span class="ruby-identifier">equence_range</span><span class="ruby-operator">/</span><span class="ruby-identifier">calculated_cuts</span>.<span class="ruby-identifier">rb</span>        <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rrange/</span><span class="ruby-identifier">equence_range</span><span class="ruby-operator">/</span><span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">rb</span>               <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rrange/</span><span class="ruby-identifier">equence_range</span><span class="ruby-operator">/</span><span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">rb</span>              <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rrestriction_enzyme/</span><span class="ruby-identifier">range</span><span class="ruby-operator">/</span><span class="ruby-identifier">vertical_cut_range</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">util</span><span class="ruby-operator">/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">single_strand</span>.<span class="ruby-identifier">rb</span>   <span class="ruby-operator">|</span>    <span class="ruby-value">6</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rcut_locations_in_enzyme_notation.rb            |    7 +++----
.../</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">single_strand_complement</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rutil/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">sorted_num_array</span>.<span class="ruby-identifier">rb</span>    <span class="ruby-operator">|</span>    <span class="ruby-value">3</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-operator">...</span><span class="ruby-regexp">%rutil/</span><span class="ruby-identifier">restriction_enzyme</span><span class="ruby-operator">/</span><span class="ruby-identifier">string_formatting</span>.<span class="ruby-identifier">rb</span>   <span class="ruby-operator">|</span>    <span class="ruby-value">7</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">23</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">75</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">81</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 7df4843288ffde6d7132a5651fe978301f8ebd2b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jul 5 20:18:08 2012 +0900</p>

<pre>   workaround to avoid JRuby autoload bug

lib/bio/util/restriction_enzyme.rb |    4 +---
1 files changed, 1 insertions(+), 3 deletions(-)</pre>

<p>commit 97d95f2b400006d4229a7ce69d7d8a5cdce42764 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 4 22:00:27 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">changed</span> <span class="ruby-identifier">require</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">autoload</span> <span class="ruby-keyword">for</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">workaround</span> <span class="ruby-identifier">of</span> <span class="ruby-constant">JRuby</span> <span class="ruby-identifier">autoload</span> <span class="ruby-identifier">bug</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">feature</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">2</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">3</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 530b82a45731c2a71a110826341be425de1271e0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 4 22:00:06 2012 +0900</p>

<pre>   workaround to avoid JRuby autoload bug

lib/bio/sequence/common.rb |    4 +---
1 files changed, 1 insertions(+), 3 deletions(-)</pre>

<p>commit 8614f31b36fb93d6e49d109268d646ff3032cd1a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jul 4 21:28:52 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-constant">JRuby</span> <span class="ruby-identifier">autoload</span> <span class="ruby-identifier">bug</span>

    * <span class="ruby-constant">Workaround</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">avoid</span> <span class="ruby-constant">JRuby</span> <span class="ruby-identifier">autoload</span> <span class="ruby-identifier">bug</span>.
    * <span class="ruby-constant">Changed</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">require</span> <span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span>.<span class="ruby-identifier">rb</span> <span class="ruby-identifier">because</span> <span class="ruby-identifier">it</span> <span class="ruby-identifier">is</span> <span class="ruby-identifier">always</span> <span class="ruby-identifier">loaded</span>.

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">kegg</span><span class="ruby-operator">/</span><span class="ruby-identifier">genes</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">6</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">3</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">3</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit ea500006ed56857139c858bdfeb98773e5ca541e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 21:36:35 2012 +0900</p>

<pre>   Rakefile: use own mktmpdir

Rakefile |   59 +++++++++++++++++++++++++++++++++++++++++++----------------
1 files changed, 43 insertions(+), 16 deletions(-)</pre>

<p>commit 452fadcab61083dcb9d01ee05d300eae5cb23fee Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 20:37:59 2012 +0900</p>

<pre>   .travis.yml: remove &quot;rake regemspec&quot; from after_install

.travis.yml |    2 --
1 files changed, 0 insertions(+), 2 deletions(-)</pre>

<p>commit 3fad822af3d7e558a58b71fd8ec2a7061b49f9f2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 20:36:59 2012 +0900</p>

<pre>   regenerate bioruby.gemspec with rake regemspec

bioruby.gemspec |    2 ++
1 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit ea6e96fc654c797664b118a6326a84e4f9b1a8a3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 20:35:49 2012 +0900</p>

<pre>   print message when doing Dir.chdir

Rakefile |   17 +++++++++++------
1 files changed, 11 insertions(+), 6 deletions(-)</pre>

<p>commit c2fcd5e8cc71da38dc3c6d1f8c8d0233e47398b3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 20:28:41 2012 +0900</p>

<pre>   In tar-install, removed dependency to regemspec

Rakefile |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 67a7e83d516aab5d60f8263525b359be8b0ffc0b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 20:23:24 2012 +0900</p>

<pre>   Rakefile: give up using Dir.mktmpdir because of JRuby's behavior

    * Rakefile: give up using Dir.mktmpdir because of JRuby's behavior
      that may be related with http://jira.codehaus.org/browse/JRUBY-5678

Rakefile |   61 ++++++++++++++++++++++++++++++++++++++++++++++---------------
1 files changed, 46 insertions(+), 15 deletions(-)</pre>

<p>commit cff098034a338bbe9579d6c7b4380c7132a38ef5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 19:23:57 2012 +0900</p>

<pre>   gem-integration-test, gem-install and gem-install-nodoc are removed

    * gem-integration-test, gem-install and gem-install-nodoc are removed
      because they are useless with Bundler

Rakefile |   13 -------------
1 files changed, 0 insertions(+), 13 deletions(-)</pre>

<p>commit d5c054265af4f80318cbfa5a5bbdee6125219de2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 18:10:05 2012 +0900</p>

<pre>   .travis.yml: .gemspec is needed to install local gem

.travis.yml                 |    1 +
gemfiles/prepare-gemspec.rb |   25 +++++++++++++++++++++++++
2 files changed, 26 insertions(+), 0 deletions(-)
create mode 100644 gemfiles/prepare-gemspec.rb</pre>

<p>commit 05b6172123f42a1d8d46668d8a3d5f698c371704 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 17:51:43 2012 +0900</p>

<pre>   remove 1.9.2; add tar/gem integration tests

    * Remove ruby version 1.9.2 from matrix for reducing builds
    * Add tar/gem integration tests
    * Add a new helper script gemfiles/modify-Gemfile.rb,
      modifying gemfile when running gem integration test.
    * Remove jruby version comments

.travis.yml                |   26 +++++++++++++++++---------
gemfiles/modify-Gemfile.rb |   28 ++++++++++++++++++++++++++++
2 files changed, 45 insertions(+), 9 deletions(-)
create mode 100644 gemfiles/modify-Gemfile.rb</pre>

<p>commit 6813f91893e7ddc3000047357c9ed2dafb32a722 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 28 17:06:28 2012 +0900</p>

<pre>   descriptions are modified for danger operations

Rakefile |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit a209688952c922d9ba45c227874990bccd3da7c0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jun 25 23:25:51 2012 +0900</p>

<pre>   regenerate bioruby.gemspec with rake regemspec

bioruby.gemspec |    5 +++++
1 files changed, 5 insertions(+), 0 deletions(-)</pre>

<p>commit 8f6459497be0e9ca7dc3eb2eb9606e42d97ad60c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jun 25 21:01:06 2012 +0900</p>

<pre>   rake tasks added and default task is changed

    * New tasks:
      * gem-install: build gem and install it
      * gem-install-nodoc: build gem and install it with --no-ri --no-rdoc.
      * gem-test: test installed bioruby gem installed with gem-install
        (or gem-install-nodoc)
      * gem-integration-test: build gem, install and run test (with --no-ri
        --no-rdoc)
      * tar-install: DANGER: build tar and install by using setup.rb
      * installed-test: test installed bioruby
      * tar-integration-test: DANGER: build tar, install and run test
      * see-env: see BIORUBY_RAKE_DEFAULT_TASK environment variable and
        invoke the specified task. If the variable did not exist, it
        invokes &quot;test&quot; which is previously the default task. It is added
        for selecting task on Travis-ci. It is not recommended to invoke
        the task explicitly by hand.
    * Default task is changed from &quot;test&quot; to &quot;see-env&quot;.

Rakefile |  110 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++--
1 files changed, 107 insertions(+), 3 deletions(-)</pre>

<p>commit 3b400042cd361e1ab6d0fb0d8c8cce14a6c2ae10 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jun 25 20:58:13 2012 +0900</p>

<pre>   BIORUBY_TEST_LIB is always added on the top of $LOAD_PATH

    * When BIORUBY_TEST_LIB is specified, the specified directory name
      is always added on the top of $LOAD_PATH even if it is already
      included in the middle of $LOAD_PATH.

test/bioruby_test_helper.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 848304b6f90310f8fa15c80ba06655ae5cae5053 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Jun 25 20:42:07 2012 +0900</p>

<pre>   New env BIORUBY_TEST_GEM and BIORUBY_TEST_LIB behavior changed

    * New environment variable BIORUBY_TEST_GEM for testing installed
      bio-X.X.X gem. Version number can be specified.
      Example with version number:
       % env BIORUBY_TEST_GEM=1.4.2.5000 ruby test/runner.rb
      Example without version number:
       % env BIORUBY_TEST_GEM=&quot;&quot; ruby test/runner.rb
    * When BIORUBY_TEST_LIB is empty, it no longer add an empty string to
      $LOAD_PATH. Moreover, when BIORUBY_TEST_GEM is set, the variable is
      ignored.

test/bioruby_test_helper.rb |   49 ++++++++++++++++++++++++++++++++----------
1 files changed, 37 insertions(+), 12 deletions(-)</pre>

<p>commit 9453a6773c24f866698370195fd8e767443a38b9 Author: Tomoaki NISHIYAMA
&lt;tomoakin@kenroku.kanazawa-u.ac.jp&gt; Date:   Fri Jun 1 18:06:40 2012
+0900</p>

<pre>   broader FASTQ file recognition

    * Because PacBio RS sequencer may produce kilobases long reads and
      read buffer size (default 31 lines) for file format detection
      may not be sufficient to find the second id line starting with &quot;+&quot;,
      the regular expression for FASTQ is truncated only to check the
      first id line starting with &quot;@&quot;.
    * Test code is added.

lib/bio/io/flatfile/autodetection.rb            |    2 +-
test/unit/bio/io/flatfile/test_autodetection.rb |    6 ++++++
2 files changed, 7 insertions(+), 1 deletions(-)</pre>

<p>commit 120e780c023cba06b83899c2f8a17c8fc1de4faa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jun 8 15:36:29 2012 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Retry</span> <span class="ruby-identifier">sequence</span> <span class="ruby-identifier">randomize</span> <span class="ruby-identifier">test</span> <span class="ruby-identifier">up</span> <span class="ruby-identifier">to</span> <span class="ruby-value">10</span> <span class="ruby-identifier">times</span> <span class="ruby-keyword">when</span> <span class="ruby-identifier">fails</span>

    * <span class="ruby-constant">To</span> <span class="ruby-identifier">suppress</span> <span class="ruby-identifier">rare</span> <span class="ruby-identifier">failure</span> <span class="ruby-identifier">of</span> <span class="ruby-identifier">chi</span><span class="ruby-operator">-</span><span class="ruby-identifier">square</span> <span class="ruby-identifier">equiprobability</span> <span class="ruby-identifier">tests</span> <span class="ruby-keyword">for</span>
      <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">Common</span><span class="ruby-comment">#randomize, test code changed to retry</span>
      <span class="ruby-identifier">up</span> <span class="ruby-identifier">to</span> <span class="ruby-value">10</span> <span class="ruby-identifier">times</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">chi</span><span class="ruby-operator">-</span><span class="ruby-identifier">square</span> <span class="ruby-identifier">test</span> <span class="ruby-identifier">fails</span>.  <span class="ruby-constant">The</span> <span class="ruby-identifier">assertion</span> <span class="ruby-identifier">fails</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">chi</span><span class="ruby-operator">-</span><span class="ruby-identifier">square</span> <span class="ruby-identifier">test</span> <span class="ruby-identifier">fails</span> <span class="ruby-value">10</span> <span class="ruby-identifier">consecutive</span> <span class="ruby-identifier">times</span>, <span class="ruby-keyword">and</span> <span class="ruby-identifier">this</span>
      <span class="ruby-identifier">strongly</span> <span class="ruby-identifier">suggests</span> <span class="ruby-identifier">bugs</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">codes</span> <span class="ruby-keyword">or</span> <span class="ruby-keyword">in</span> <span class="ruby-identifier">the</span> <span class="ruby-identifier">random</span> <span class="ruby-identifier">number</span> <span class="ruby-identifier">generator</span>.
    * <span class="ruby-constant">The</span> <span class="ruby-identifier">chi</span><span class="ruby-operator">-</span><span class="ruby-identifier">square</span> <span class="ruby-identifier">equiprobability</span> <span class="ruby-identifier">tests</span> <span class="ruby-identifier">are</span> <span class="ruby-identifier">separated</span> <span class="ruby-identifier">into</span> <span class="ruby-identifier">a</span> <span class="ruby-identifier">new</span>
      <span class="ruby-identifier">test</span> <span class="ruby-keyword">class</span>.

<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">unit</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_common</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>   <span class="ruby-value">40</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">36</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">4</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 20dde52f7da784d4d9ac551957700cd96e842ef6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat May 19 18:14:19 2012 +0900</p>

<pre>   libxml-ruby is disabled because of build error on Travis-ci

gemfiles/Gemfile.travis-jruby1.8 |    3 ++-
gemfiles/Gemfile.travis-jruby1.9 |    3 ++-
2 files changed, 4 insertions(+), 2 deletions(-)</pre>

<p>commit 3c5c1cc277d30737815c7e44a2abbb308f5324b0 Author: Clayton Wheeler
&lt;cswh@umich.edu&gt; Date:   Mon May 14 21:48:41 2012 -0400</p>

<pre>   Use libxml-ruby instead of libxml-jruby to fix JRuby test failures.

   The travis-ci Gemfiles currently call for libxml-jruby; this appears
   not to support the same API as libxml-ruby, resulting in several tests
   in test/unit/bio/db/test_phyloxml.rb failing with &quot;NameError:
   uninitialized constant LibXMLJRuby::XML::Parser::Options&quot;. Switching
   to the C libxml-ruby library allows these tests to pass under JRuby in
   1.8 mode.

   JRuby in 1.9 mode still fails a few PhyloXML tests due to
   https://jira.codehaus.org/browse/JRUBY-6662.

gemfiles/Gemfile.travis-jruby1.8 |    2 +-
gemfiles/Gemfile.travis-jruby1.9 |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 01a618242d67f0d00fe681dfd85e68bb393513fc Author: Clayton Wheeler
&lt;cswh@umich.edu&gt; Date:   Thu May 10 23:13:56 2012 -0400</p>

<pre>   test_tree.rb: to use %f instead of %g to prevent odd behavior.

test/unit/bio/test_tree.rb |   22 +++++++++++-----------
1 files changed, 11 insertions(+), 11 deletions(-)</pre>

<p>commit 5e80e4394bf2a5e4ee472fe84ab76239b293e1b5 Author: Clayton Wheeler
&lt;cswh@umich.edu&gt; Date:   Thu May 10 23:04:55 2012 -0400</p>

<pre>   Fixed spurious JRuby failures in test_tree.rb due to floating point differences.

test/unit/bio/test_tree.rb |   14 +++++++-------
1 files changed, 7 insertions(+), 7 deletions(-)</pre>

<p>commit 459d4da894e9a9b9db0d793e3711dc45bae2089b Author: Artem Tarasov
&lt;lomereiter@gmail.com&gt; Date:   Thu May 10 16:23:13 2012 +0400</p>

<pre>   Test bug fix: order of hash keys are not guaranteed

    * Test bug fix: Bio::TestSOFT#test_dataset: order of hash keys are
      not guaranteed.

test/unit/bio/db/test_soft.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 7e730691d6ec597a610dc0d4665db3598fcfde59 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu May 10 00:06:19 2012 +0900</p>

<pre class="ruby">   <span class="ruby-identifier">removed</span> <span class="ruby-identifier">potential</span> <span class="ruby-identifier">circular</span> <span class="ruby-identifier">require</span> <span class="ruby-identifier">about</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">Format</span>

<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">embl</span><span class="ruby-operator">/</span><span class="ruby-identifier">format_embl</span>.<span class="ruby-identifier">rb</span>       <span class="ruby-operator">|</span>    <span class="ruby-value">4</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">fasta</span><span class="ruby-operator">/</span><span class="ruby-identifier">format_fasta</span>.<span class="ruby-identifier">rb</span>     <span class="ruby-operator">|</span>    <span class="ruby-value">4</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">fasta</span><span class="ruby-operator">/</span><span class="ruby-identifier">format_qual</span>.<span class="ruby-identifier">rb</span>      <span class="ruby-operator">|</span>    <span class="ruby-value">5</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">fastq</span><span class="ruby-operator">/</span><span class="ruby-identifier">format_fastq</span>.<span class="ruby-identifier">rb</span>     <span class="ruby-operator">|</span>    <span class="ruby-value">1</span> <span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">db</span><span class="ruby-operator">/</span><span class="ruby-identifier">genbank</span><span class="ruby-operator">/</span><span class="ruby-identifier">format_genbank</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>    <span class="ruby-value">4</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">lib</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">sequence</span><span class="ruby-operator">/</span><span class="ruby-identifier">format_raw</span>.<span class="ruby-identifier">rb</span>       <span class="ruby-operator">|</span>    <span class="ruby-value">4</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">6</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">0</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">22</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit f1c398fdc3488bd18bd13ac864920ce6db4dab9e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 9 15:54:20 2012 +0900</p>

<pre>   .travis.yml: comment out apt-get lines

    * .travis.yml: comment out apt-get lines because libxml2-dev
      and libexpat1-dev are already installed.

.travis.yml |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit bc5ef4959e51f4a199d9f740b07812e9b8216255 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 9 15:47:11 2012 +0900</p>

<pre>   travis-ci: comment out soap4r-ruby1.9 in Gemfile because of error

    * travis-ci: soap4r-ruby1.9 gem in Gemfile.travis-ruby1.9 and
      Gemfile.travis-jruby1.9 is commented out because of an error
      &quot;uninitialized constant XML::SaxParser&quot;.

gemfiles/Gemfile.travis-jruby1.9 |    4 +++-
gemfiles/Gemfile.travis-ruby1.9  |    4 +++-
2 files changed, 6 insertions(+), 2 deletions(-)</pre>

<p>commit 7e8153c09660c31d6286c1924680b8c5073a10b6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue May 1 18:11:09 2012 +0900</p>

<pre>   config files for Travis CI continuous integration service

.travis.yml                      |   73 ++++++++++++++++++++++++++++++++++++++
gemfiles/Gemfile.travis-jruby1.8 |    6 +++
gemfiles/Gemfile.travis-jruby1.9 |    7 ++++
gemfiles/Gemfile.travis-ruby1.8  |    7 ++++
gemfiles/Gemfile.travis-ruby1.9  |    8 ++++
5 files changed, 101 insertions(+), 0 deletions(-)
create mode 100644 .travis.yml
create mode 100644 gemfiles/Gemfile.travis-jruby1.8
create mode 100644 gemfiles/Gemfile.travis-jruby1.9
create mode 100644 gemfiles/Gemfile.travis-ruby1.8
create mode 100644 gemfiles/Gemfile.travis-ruby1.9</pre>

<p>commit f1ecae7763648cb735a885ddb6c46d71c59b0694 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Mar 23 01:36:59 2012 +0900</p>

<pre>   Test bug fix: tests affected by the bug of Bio::NucleicAcid.to_re(&quot;s&quot;)

test/unit/bio/data/test_na.rb     |    2 +-
test/unit/bio/sequence/test_na.rb |    2 +-
test/unit/bio/test_sequence.rb    |    4 ++--
3 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit 3fd9384b1b59140a929c81dcc4b07cb3c2e47525 Author: Trevor Wennblom
&lt;trevor@well.com&gt; Date:   Sat Feb 25 15:26:27 2012 -0600</p>

<pre>   Bug fix: Bio::NucleicAcid.to_re(&quot;s&quot;) typo

lib/bio/data/na.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit c552aa3a6773139b14ae95e79e0fb43a2f91c6fb Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jan 12 22:24:37 2012 +0900</p>

<pre>   Bug fix: GenomeNet BLAST server URI changed.

    * Bug fix: GenomeNet BLAST server URI changed.
      Reported by joaocardoso via GitHub.
      ( https://github.com/bioruby/bioruby/issues/44 )

lib/bio/appl/blast/genomenet.rb |    3 ++-
1 files changed, 2 insertions(+), 1 deletions(-)</pre>

<p>commit f33abf9bbd90c3c1e320f06447fdb54ffd094c5d Author: peterjc
&lt;googlemail at p.j.a.cock.com&gt; Date:   Fri Nov 25 11:20:08 2011 +0000</p>

<pre class="ruby">   <span class="ruby-constant">Mark</span> <span class="ruby-identifier">echoarg2</span>.<span class="ruby-identifier">bat</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">echoarg2</span>.<span class="ruby-identifier">sh</span> <span class="ruby-identifier">as</span> <span class="ruby-identifier">world</span> <span class="ruby-identifier">executable</span>

<span class="ruby-value">0</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">0</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">0</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
<span class="ruby-identifier">mode</span> <span class="ruby-identifier">change</span> <span class="ruby-value">100644</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">100755</span> <span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">data</span><span class="ruby-operator">/</span><span class="ruby-identifier">command</span><span class="ruby-operator">/</span><span class="ruby-identifier">echoarg2</span>.<span class="ruby-identifier">bat</span>
<span class="ruby-identifier">mode</span> <span class="ruby-identifier">change</span> <span class="ruby-value">100644</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">100755</span> <span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">data</span><span class="ruby-operator">/</span><span class="ruby-identifier">command</span><span class="ruby-operator">/</span><span class="ruby-identifier">echoarg2</span>.<span class="ruby-identifier">sh</span>
</pre>

<p>commit d2d66f833d0b20647e8d761d2a240b99b206eaa8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 24 13:32:37 2011 +0900</p>

<pre>   Bug fix: rake aborted without git

bioruby.gemspec.erb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit c2139739988ef731d61bf1a8cdba2dc5c48393bd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 24 13:07:10 2011 +0900</p>

<pre>   regenerate bioruby.gemspec with rake regemspec.

bioruby.gemspec |   18 ++++++++++--------
1 files changed, 10 insertions(+), 8 deletions(-)</pre>

<p>commit 6213b45d28bfea2cc8c838813b524d48c369266b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 24 13:05:07 2011 +0900</p>

<pre class="ruby">   <span class="ruby-constant">Added</span> <span class="ruby-identifier">workaround</span> <span class="ruby-keyword">for</span> <span class="ruby-identifier">changes</span> <span class="ruby-identifier">of</span> <span class="ruby-identifier">a</span> <span class="ruby-keyword">module</span> <span class="ruby-identifier">name</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">file</span> <span class="ruby-identifier">names</span> <span class="ruby-identifier">to</span> <span class="ruby-identifier">require</span>.

<span class="ruby-constant">Rakefile</span> <span class="ruby-operator">|</span>   <span class="ruby-value">21</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-value">1</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">19</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">2</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
</pre>

<p>commit 39f847cf8d453476275361078b831da43d400816 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 24 12:08:47 2011 +0900</p>

<pre>   Use binary mode to open files.

Rakefile |    6 ++++--
1 files changed, 4 insertions(+), 2 deletions(-)</pre>

<p>commit 688779e71a27e861fb01e07f816384561b8cfe45 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 24 11:49:30 2011 +0900</p>

<pre>   Rakefile: new tasks: test-all to run all tests, etc.

    * Rakefile: new tasks: test-all to run all tests, and test-network
      to run tests in test/network.

Rakefile |   10 ++++++++++
1 files changed, 10 insertions(+), 0 deletions(-)</pre>

<p>commit 53719535defcb0fefb3cf8bebe3fad6716bf7de2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 24 11:28:38 2011 +0900</p>

<pre>   test/runner.rb: Run tests only in test/unit and test/functional.

test/runner.rb |   22 ++++++++++++++++------
1 files changed, 16 insertions(+), 6 deletions(-)</pre>

<p>commit fb9ee403db6b447aee73ebb7f12ff5a5b73d6c52 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 23 20:36:36 2011 +0900</p>

<pre class="ruby">   <span class="ruby-constant">A</span> <span class="ruby-identifier">test</span> <span class="ruby-keyword">class</span> <span class="ruby-identifier">using</span> <span class="ruby-identifier">network</span> <span class="ruby-identifier">connection</span> <span class="ruby-identifier">is</span> <span class="ruby-identifier">moved</span> <span class="ruby-identifier">under</span> <span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">network</span><span class="ruby-operator">/</span>.

<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">functional</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_command</span>.<span class="ruby-identifier">rb</span> <span class="ruby-operator">|</span>   <span class="ruby-value">16</span> <span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span><span class="ruby-operator">-</span>
<span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">network</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_command</span>.<span class="ruby-identifier">rb</span>    <span class="ruby-operator">|</span>   <span class="ruby-value">35</span> <span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span><span class="ruby-operator">+</span>
<span class="ruby-value">2</span> <span class="ruby-identifier">files</span> <span class="ruby-identifier">changed</span>, <span class="ruby-value">35</span> <span class="ruby-identifier">insertions</span>(<span class="ruby-operator">+</span>), <span class="ruby-value">16</span> <span class="ruby-identifier">deletions</span>(<span class="ruby-operator">-</span>)
<span class="ruby-identifier">create</span> <span class="ruby-identifier">mode</span> <span class="ruby-value">100644</span> <span class="ruby-identifier">test</span><span class="ruby-operator">/</span><span class="ruby-identifier">network</span><span class="ruby-operator">/</span><span class="ruby-identifier">bio</span><span class="ruby-operator">/</span><span class="ruby-identifier">test_command</span>.<span class="ruby-identifier">rb</span>
</pre>

<p>commit a6dda2215aa686a9ca4af7484aa190f726d51e69 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 23 20:28:58 2011 +0900</p>

<pre>   Tests using network connections are moved to test/network/

    * Tests using network connections are moved to test/network/.
      * renamed: test/functional/bio/appl -&gt; test/network/bio/appl
      * renamed: test/functional/bio/io   -&gt; test/network/bio/io

test/functional/bio/appl/blast/test_remote.rb |   93 ---------
test/functional/bio/appl/test_blast.rb        |   61 ------
test/functional/bio/appl/test_pts1.rb         |  117 -----------
test/functional/bio/io/test_ddbjrest.rb       |   47 -----
test/functional/bio/io/test_ensembl.rb        |  230 ---------------------
test/functional/bio/io/test_pubmed.rb         |  135 -------------
test/functional/bio/io/test_soapwsdl.rb       |   53 -----
test/functional/bio/io/test_togows.rb         |  268 -------------------------
test/network/bio/appl/blast/test_remote.rb    |   93 +++++++++
test/network/bio/appl/test_blast.rb           |   61 ++++++
test/network/bio/appl/test_pts1.rb            |  117 +++++++++++
test/network/bio/io/test_ddbjrest.rb          |   47 +++++
test/network/bio/io/test_ensembl.rb           |  230 +++++++++++++++++++++
test/network/bio/io/test_pubmed.rb            |  135 +++++++++++++
test/network/bio/io/test_soapwsdl.rb          |   53 +++++
test/network/bio/io/test_togows.rb            |  268 +++++++++++++++++++++++++
16 files changed, 1004 insertions(+), 1004 deletions(-)
delete mode 100644 test/functional/bio/appl/blast/test_remote.rb
delete mode 100644 test/functional/bio/appl/test_blast.rb
delete mode 100644 test/functional/bio/appl/test_pts1.rb
delete mode 100644 test/functional/bio/io/test_ddbjrest.rb
delete mode 100644 test/functional/bio/io/test_ensembl.rb
delete mode 100644 test/functional/bio/io/test_pubmed.rb
delete mode 100644 test/functional/bio/io/test_soapwsdl.rb
delete mode 100644 test/functional/bio/io/test_togows.rb
create mode 100644 test/network/bio/appl/blast/test_remote.rb
create mode 100644 test/network/bio/appl/test_blast.rb
create mode 100644 test/network/bio/appl/test_pts1.rb
create mode 100644 test/network/bio/io/test_ddbjrest.rb
create mode 100644 test/network/bio/io/test_ensembl.rb
create mode 100644 test/network/bio/io/test_pubmed.rb
create mode 100644 test/network/bio/io/test_soapwsdl.rb
create mode 100644 test/network/bio/io/test_togows.rb</pre>

<p>commit ec747aa33d06e08a6469dfd330360161d1b0f8e2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 23 15:03:08 2011 +0900</p>

<pre>   Test bug fix: use binmode to disable CR/LF conversion (fail on Windows)

test/unit/bio/appl/blast/test_rpsblast.rb |    1 +
test/unit/bio/io/flatfile/test_buffer.rb  |    1 +
2 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit 07ce32da009baa2c4e81f6d96f45e3dac49da183 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 23 14:47:33 2011 +0900</p>

<pre>   Test bug fix: Read Sanger chromatogram files with binary mode

    * Test bug fix: Read Sanger chromatogram files with binary mode.
      Fix error/failure on Windows due to default text mode reading.

test/unit/bio/db/sanger_chromatogram/test_abif.rb |    3 ++-
test/unit/bio/db/sanger_chromatogram/test_scf.rb  |    6 ++++--
2 files changed, 6 insertions(+), 3 deletions(-)</pre>

<p>commit 20d9068643214e3482d18c36028e50b3c9109755 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 23 14:17:25 2011 +0900</p>

<pre>   Incompatible change: Bio::FlatFile.open and auto use binary mode

    * Incompatible change: Bio::FlatFile.open and auto use binary mode
      (binmode) unless text mode option is explicitly given.

RELEASE_NOTES.rdoc                       |    7 ++
lib/bio/io/flatfile/buffer.rb            |   84 ++++++++++++++++++
test/unit/bio/io/flatfile/test_buffer.rb |  139 ++++++++++++++++++++++++++++++
3 files changed, 230 insertions(+), 0 deletions(-)</pre>

<p>commit 48bd150a6180d59879872bd85dd95c7ddf1a19c0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 22 17:32:23 2011 +0900</p>

<pre>   Test bug fix: fixed incomplete Windows platform detection.

test/unit/bio/test_command.rb |   13 +++++++++----
1 files changed, 9 insertions(+), 4 deletions(-)</pre>

<p>commit d499bcee7956b1a0a4c04aeb106e50a0839167b0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 22 16:15:05 2011 +0900</p>

<pre>   FuncTestCommandCall is changed to test various command-lines.

    * New file test/data/command/echoarg2.sh shell script, which acts
      like echoarg2.bat for Windows.
    * FuncTestCommandCall is changed to test various command-lines.

test/data/command/echoarg2.sh       |    4 ++
test/functional/bio/test_command.rb |   70 +++++++++++++++++++++++++++++------
2 files changed, 62 insertions(+), 12 deletions(-)
create mode 100644 test/data/command/echoarg2.sh</pre>

<p>commit d45e311c09ad2f4116770dd903f81e652a63ca2a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 22 14:21:34 2011 +0900</p>

<pre>   Test bug fix: Opened files should be closed.

    * Test bug fix: Opened files should be closed.  When finalizing writer
      tests, temporary files are not properly closed after verify reading,
      and removing the temporary files raise erro on Windows.

test/unit/bio/db/test_phyloxml_writer.rb |   24 +++++++++++++++---------
1 files changed, 15 insertions(+), 9 deletions(-)</pre>

<p>commit a9022c61b98746e98a83f1cfd902e0e6b11c7bbb Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 22 13:55:15 2011 +0900</p>

<pre>   New method Bio::PhyloXML::Parser#closed?, and Bio::PhyloXML::Parser.open with block.

    * New method Bio::PhyloXML::Parser#closed? to check if it is closed
      or not.
    * Bio::PhyloXML::Parser.open and open_uri now can get a block.
      When a block is given, a Bio::PhyloXML::Parser object is passed
      to the block as an argument. When the block terminates, the object
      is closed.
    * Added tests about the above changes.

lib/bio/db/phyloxml/phyloxml_parser.rb |   57 +++++++++++++++++++++++++++++---
test/unit/bio/db/test_phyloxml.rb      |   56 ++++++++++++++++++++++++++++++-
2 files changed, 106 insertions(+), 7 deletions(-)</pre>

<p>commit 893cbe6ca993eca08427074059c2ba03621ea889 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Nov 5 00:49:10 2011 +0900</p>

<pre>   Ruby 1.9 should be fully supported, and optional requirements are revised.

README.rdoc |   48 +++++++++++++++++++++++++++++++++---------------
1 files changed, 33 insertions(+), 15 deletions(-)</pre>

<p>commit 38b1715c2d6bad39560e0846781ca903b1c16eda Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 4 22:12:38 2011 +0900</p>

<pre>   Added REFERENCE.

README.rdoc |   12 ++++++++++++
1 files changed, 12 insertions(+), 0 deletions(-)</pre>

<p>commit 9a766cd17236bbe1e28d6972001dd5e3ed596123 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 4 21:39:20 2011 +0900</p>

<pre>   Removed &quot;setup.rb test&quot; and added about running tests.

README.rdoc |   39 ++++++++++++++++++++++++++++++++++-----
1 files changed, 34 insertions(+), 5 deletions(-)</pre>

<p>commit 39737179b06366e1d5acf2e5ac930e41b3a4ee38 Author: Pjotr Prins
&lt;thebird at pjotr.public01.nl&gt; Date:   Fri Oct 14 08:58:01 2011 +0200</p>

<pre>   Tutorial: added info on biogems

doc/Tutorial.rd      |   16 ++++++++++++++++
doc/Tutorial.rd.html |   23 +++++++++++++++--------
2 files changed, 31 insertions(+), 8 deletions(-)</pre>

<p>commit e84400c5e9e94d95d6a8d3c4b72388b94d204766 Author: Pjotr Prins
&lt;thebird at pjotr.public01.nl&gt; Date:   Fri Oct 14 08:49:41 2011 +0200</p>

<pre>   Tutorial: small updates

doc/Tutorial.rd      |    8 +++++---
doc/Tutorial.rd.html |    9 +++++----
2 files changed, 10 insertions(+), 7 deletions(-)</pre>

<p>commit 9fe07345b3b7be890d5baad9a51f0752af5e0ac4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Sep 13 23:05:39 2011 +0900</p>

<pre>   README_DEV.rdoc: added git tips and policies, etc.

    * Added Git tips about sending a patch or a pull request.
    * Added Git management policies for the blessed repository.
    * Added some coding styles.
    * Added descriptions about Ruby versions and OS.

README_DEV.rdoc |   95 +++++++++++++++++++++++++++++++++++++++++++++++++++++-
1 files changed, 93 insertions(+), 2 deletions(-)</pre>

<p>commit 3c952c4a782501b21f36ece5bcab672dab12fc6d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Sep 13 13:21:20 2011 +0900</p>

<pre>   README.rdoc: for release notes and changelog, about sample files.

README.rdoc |   10 +++++++++-
1 files changed, 9 insertions(+), 1 deletions(-)</pre>

<p>commit fba9a6c0f1f79dd567ca54ba085b6258ac8efb31 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Sep 13 13:20:05 2011 +0900</p>

<pre>   RELEASE_NOTES.rdoc: mentioned about removal of rdoc.zsh.

RELEASE_NOTES.rdoc |    6 +++++-
1 files changed, 5 insertions(+), 1 deletions(-)</pre>

<p>commit 685b6bb7b98083e1b50e73baf4e7fa71bc9a39fa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 21:23:34 2011 +0900</p>

<pre>   bioruby.gemspec.erb: LEGAL is added to rdoc files

    * bioruby.gemspec.erb: LEGAL is added to rdoc files.
    * bioruby.gemspec is updated by &quot;rake regemspec&quot;.

bioruby.gemspec     |    9 ++++++---
bioruby.gemspec.erb |    6 +++++-
2 files changed, 11 insertions(+), 4 deletions(-)</pre>

<p>commit 414a6331f40fc99f554042e9a031689ea6d76da4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 20:54:06 2011 +0900</p>

<pre>   deleted rdoc.zsh which is obsolete and unused

    * Deleted rdoc.zsh which is obsolete and unused.
      To generate rdoc html, &quot;rake rdoc&quot; or &quot;rake rerdoc&quot;.
      See &quot;rake -T&quot; for more information.

rdoc.zsh |    8 --------
1 files changed, 0 insertions(+), 8 deletions(-)
delete mode 100644 rdoc.zsh</pre>

<p>commit 272d9106cec43b0f219edd92a6f7bd3f9875a761 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 20:35:47 2011 +0900</p>

<pre>   Added new ChangeLog, showing changes after 1.4.2 release.

    * Added new ChangeLog, showing changes after 1.4.2 release.
      For the changes before 1.4.2, see doc/ChangeLog-before-1.4.2.
      For the changes before 1.3.1, see doc/ChangeLog-before-1.3.1.

ChangeLog |   64 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 64 insertions(+), 0 deletions(-)
create mode 100644 ChangeLog</pre>

<p>commit 941493378f9884978c81d5f63ee4ed5c175d4bea Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 20:28:28 2011 +0900</p>

<pre>   Rakefile: add new task :rechangelog to update ChangeLog using git log.

    * Rakefile: add new task :rechangelog to update ChangeLog using
      git log. Note that the tag name (currently 1.4.2) is hardcoded
      in Rakefile.

Rakefile |    9 +++++++++
1 files changed, 9 insertions(+), 0 deletions(-)</pre>

<p>commit 1c89e6546223c3c05ea79b8ade4b493580851efa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 20:24:49 2011 +0900</p>

<pre>   renamed ChangeLog to doc/ChangeLog-before-1.4.2

ChangeLog                  | 5013 --------------------------------------------
doc/ChangeLog-before-1.4.2 | 5013 ++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 5013 insertions(+), 5013 deletions(-)
delete mode 100644 ChangeLog
create mode 100644 doc/ChangeLog-before-1.4.2</pre>

<p>commit 2233fbada55034bd16fb5b9c642292b4b6ccca83 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 20:22:49 2011 +0900</p>

<pre>   ChangeLog updated: add log about 1.4.2 release

ChangeLog |    9 +++++++++
1 files changed, 9 insertions(+), 0 deletions(-)</pre>

<p>commit 1c02ab0488e4097a2cf5c16180c3179c78e3d572 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 19:40:54 2011 +0900</p>

<pre>   New RELEASE_NOTES.rdoc for next release version.

RELEASE_NOTES.rdoc |   47 +++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 47 insertions(+), 0 deletions(-)
create mode 100644 RELEASE_NOTES.rdoc</pre>

<p>commit 4e63e69e98c0c440ec476ef3407fcc8fd2411056 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Sep 12 19:32:48 2011 +0900</p>

<pre>   renamed RELEASE_NOTES.rdoc to doc/RELEASE_NOTES-1.4.2.rdoc

RELEASE_NOTES.rdoc           |  132 ------------------------------------------
doc/RELEASE_NOTES-1.4.2.rdoc |  132 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 132 insertions(+), 132 deletions(-)
delete mode 100644 RELEASE_NOTES.rdoc
create mode 100644 doc/RELEASE_NOTES-1.4.2.rdoc</pre>

<p>commit 9c5c8cafc3ec372ef80aa20d01d13034f94d5af2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Sep 2 12:02:41 2011 +0900</p>

<pre>   Bio::BIORUBY_EXTRA_VERSION set to &quot;.5000&quot; (unstable version).

lib/bio/version.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

</div>



<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

