<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />

  <title>File: ChangeLog [BioRuby API documentation]</title>

  <link type="text/css" media="screen" href="./rdoc.css" rel="stylesheet" />

  <script src="./js/jquery.js" type="text/javascript"
    charset="utf-8"></script>
  <script src="./js/thickbox-compressed.js" type="text/javascript"
    charset="utf-8"></script>
  <script src="./js/quicksearch.js" type="text/javascript"
    charset="utf-8"></script>
  <script src="./js/darkfish.js" type="text/javascript"
    charset="utf-8"></script>
</head>

<body class="file">
  <div id="metadata">
    <div id="home-metadata">
      <div id="home-section" class="section">
        <h3 class="section-header">
          <a href="./index.html">Home</a>
          <a href="./index.html#classes">Classes</a>
          <a href="./index.html#methods">Methods</a>
        </h3>
      </div>
    </div>

    <div id="project-metadata">
      
      
      <div id="fileindex-section" class="section project-section">
        <h3 class="section-header">Files</h3>
        <ul>
        
          <li class="file"><a href="./ChangeLog.html">ChangeLog</a></li>
        
          <li class="file"><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a></li>
        
          <li class="file"><a href="./README_rdoc.html">README.rdoc</a></li>
        
          <li class="file"><a href="./README_DEV_rdoc.html">README_DEV.rdoc</a></li>
        
          <li class="file"><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a></li>
        
          <li class="file"><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3.rdoc</a></li>
        
          <li class="file"><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0.rdoc</a></li>
        
          <li class="file"><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1.rdoc</a></li>
        
        </ul>
      </div>
      

      <div id="classindex-section" class="section project-section">
        <h3 class="section-header">Class Index
          <span class="search-toggle"><img src="./images/find.png"
            height="16" width="16" alt="[+]"
            title="show/hide quicksearch" /></span></h3>
        <form action="#" method="get" accept-charset="utf-8" class="initially-hidden">
        <fieldset>
          <legend>Quicksearch</legend>
          <input type="text" name="quicksearch" value=""
            class="quicksearch-field" />
        </fieldset>
        </form>

        <ul class="link-list">
        
          <li><a href="./Bio.html">Bio</a></li>
        
          <li><a href="./Bio/AAindex.html">Bio::AAindex</a></li>
        
          <li><a href="./Bio/AAindex1.html">Bio::AAindex1</a></li>
        
          <li><a href="./Bio/AAindex2.html">Bio::AAindex2</a></li>
        
          <li><a href="./Bio/Abif.html">Bio::Abif</a></li>
        
          <li><a href="./Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a></li>
        
          <li><a href="./Bio/Alignment.html">Bio::Alignment</a></li>
        
          <li><a href="./Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a></li>
        
          <li><a href="./Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a></li>
        
          <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a></li>
        
          <li><a href="./Bio/Alignment/GAP.html">Bio::Alignment::GAP</a></li>
        
          <li><a href="./Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a></li>
        
          <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a></li>
        
          <li><a href="./Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a></li>
        
          <li><a href="./Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a></li>
        
          <li><a href="./Bio/Alignment/Output.html">Bio::Alignment::Output</a></li>
        
          <li><a href="./Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a></li>
        
          <li><a href="./Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a></li>
        
          <li><a href="./Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a></li>
        
          <li><a href="./Bio/Alignment/Site.html">Bio::Alignment::Site</a></li>
        
          <li><a href="./Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a></li>
        
          <li><a href="./Bio/AminoAcid.html">Bio::AminoAcid</a></li>
        
          <li><a href="./Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a></li>
        
          <li><a href="./Bio/Blast.html">Bio::Blast</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Default.html">Bio::Blast::Default</a></li>
        
          <li><a href="./Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a></li>
        
          <li><a href="./Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a></li>
        
          <li><a href="./Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a></li>
        
          <li><a href="./Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a></li>
        
          <li><a href="./Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="./Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Remote.html">Bio::Blast::Remote</a></li>
        
          <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a></li>
        
          <li><a href="./Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a></li>
        
          <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a></li>
        
          <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a></li>
        
          <li><a href="./Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a></li>
        
          <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a></li>
        
          <li><a href="./Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a></li>
        
          <li><a href="./Bio/Blast/Report.html">Bio::Blast::Report</a></li>
        
          <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a></li>
        
          <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="./Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a></li>
        
          <li><a href="./Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a></li>
        
          <li><a href="./Bio/Blast/WU.html">Bio::Blast::WU</a></li>
        
          <li><a href="./Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a></li>
        
          <li><a href="./Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a></li>
        
          <li><a href="./Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a></li>
        
          <li><a href="./Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a></li>
        
          <li><a href="./Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a></li>
        
          <li><a href="./Bio/Blat.html">Bio::Blat</a></li>
        
          <li><a href="./Bio/Blat/Report.html">Bio::Blat::Report</a></li>
        
          <li><a href="./Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a></li>
        
          <li><a href="./Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a></li>
        
          <li><a href="./Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a></li>
        
          <li><a href="./Bio/ClustalW.html">Bio::ClustalW</a></li>
        
          <li><a href="./Bio/ClustalW/Report.html">Bio::ClustalW::Report</a></li>
        
          <li><a href="./Bio/CodonTable.html">Bio::CodonTable</a></li>
        
          <li><a href="./Bio/ColorScheme.html">Bio::ColorScheme</a></li>
        
          <li><a href="./Bio/Command.html">Bio::Command</a></li>
        
          <li><a href="./Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a></li>
        
          <li><a href="./Bio/ContingencyTable.html">Bio::ContingencyTable</a></li>
        
          <li><a href="./Bio/DAS.html">Bio::DAS</a></li>
        
          <li><a href="./Bio/DAS/DNA.html">Bio::DAS::DNA</a></li>
        
          <li><a href="./Bio/DAS/DSN.html">Bio::DAS::DSN</a></li>
        
          <li><a href="./Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a></li>
        
          <li><a href="./Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a></li>
        
          <li><a href="./Bio/DAS/GFF.html">Bio::DAS::GFF</a></li>
        
          <li><a href="./Bio/DAS/GROUP.html">Bio::DAS::GROUP</a></li>
        
          <li><a href="./Bio/DAS/LINK.html">Bio::DAS::LINK</a></li>
        
          <li><a href="./Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a></li>
        
          <li><a href="./Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a></li>
        
          <li><a href="./Bio/DAS/TARGET.html">Bio::DAS::TARGET</a></li>
        
          <li><a href="./Bio/DAS/TYPE.html">Bio::DAS::TYPE</a></li>
        
          <li><a href="./Bio/DAS/TYPES.html">Bio::DAS::TYPES</a></li>
        
          <li><a href="./Bio/DB.html">Bio::DB</a></li>
        
          <li><a href="./Bio/DBGET.html">Bio::DBGET</a></li>
        
          <li><a href="./Bio/DDBJ.html">Bio::DDBJ</a></li>
        
          <li><a href="./Bio/DDBJ/REST.html">Bio::DDBJ::REST</a></li>
        
          <li><a href="./Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a></li>
        
          <li><a href="./Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a></li>
        
          <li><a href="./Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a></li>
        
          <li><a href="./Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a></li>
        
          <li><a href="./Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a></li>
        
          <li><a href="./Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a></li>
        
          <li><a href="./Bio/DDBJ/XML.html">Bio::DDBJ::XML</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a></li>
        
          <li><a href="./Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a></li>
        
          <li><a href="./Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a></li>
        
          <li><a href="./Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a></li>
        
          <li><a href="./Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a></li>
        
          <li><a href="./Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a></li>
        
          <li><a href="./Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a></li>
        
          <li><a href="./Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a></li>
        
          <li><a href="./Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a></li>
        
          <li><a href="./Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a></li>
        
          <li><a href="./Bio/EBI.html">Bio::EBI</a></li>
        
          <li><a href="./Bio/EBI/SOAP.html">Bio::EBI::SOAP</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a></li>
        
          <li><a href="./Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a></li>
        
          <li><a href="./Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a></li>
        
          <li><a href="./Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a></li>
        
          <li><a href="./Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a></li>
        
          <li><a href="./Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a></li>
        
          <li><a href="./Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a></li>
        
          <li><a href="./Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a></li>
        
          <li><a href="./Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a></li>
        
          <li><a href="./Bio/EMBL.html">Bio::EMBL</a></li>
        
          <li><a href="./Bio/EMBLDB.html">Bio::EMBLDB</a></li>
        
          <li><a href="./Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a></li>
        
          <li><a href="./Bio/EMBOSS.html">Bio::EMBOSS</a></li>
        
          <li><a href="./Bio/Ensembl.html">Bio::Ensembl</a></li>
        
          <li><a href="./Bio/Ensembl/Base.html">Bio::Ensembl::Base</a></li>
        
          <li><a href="./Bio/Ensembl/Human.html">Bio::Ensembl::Human</a></li>
        
          <li><a href="./Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a></li>
        
          <li><a href="./Bio/FANTOM.html">Bio::FANTOM</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a></li>
        
          <li><a href="./Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a></li>
        
          <li><a href="./Bio/Fasta.html">Bio::Fasta</a></li>
        
          <li><a href="./Bio/Fasta/Report.html">Bio::Fasta::Report</a></li>
        
          <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a></li>
        
          <li><a href="./Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a></li>
        
          <li><a href="./Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a></li>
        
          <li><a href="./Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a></li>
        
          <li><a href="./Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a></li>
        
          <li><a href="./Bio/FastaDefline.html">Bio::FastaDefline</a></li>
        
          <li><a href="./Bio/FastaFormat.html">Bio::FastaFormat</a></li>
        
          <li><a href="./Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a></li>
        
          <li><a href="./Bio/Fastq.html">Bio::Fastq</a></li>
        
          <li><a href="./Bio/Fastq/Error.html">Bio::Fastq::Error</a></li>
        
          <li><a href="./Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a></li>
        
          <li><a href="./Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a></li>
        
          <li><a href="./Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a></li>
        
          <li><a href="./Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a></li>
        
          <li><a href="./Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a></li>
        
          <li><a href="./Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a></li>
        
          <li><a href="./Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a></li>
        
          <li><a href="./Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a></li>
        
          <li><a href="./Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a></li>
        
          <li><a href="./Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a></li>
        
          <li><a href="./Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a></li>
        
          <li><a href="./Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a></li>
        
          <li><a href="./Bio/Feature.html">Bio::Feature</a></li>
        
          <li><a href="./Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a></li>
        
          <li><a href="./Bio/Features.html">Bio::Features</a></li>
        
          <li><a href="./Bio/Fetch.html">Bio::Fetch</a></li>
        
          <li><a href="./Bio/FlatFile.html">Bio::FlatFile</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a></li>
        
          <li><a href="./Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a></li>
        
          <li><a href="./Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a></li>
        
          <li><a href="./Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a></li>
        
          <li><a href="./Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a></li>
        
          <li><a href="./Bio/FlatFileIndex.html">Bio::FlatFileIndex</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a></li>
        
          <li><a href="./Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a></li>
        
          <li><a href="./Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a></li>
        
          <li><a href="./Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a></li>
        
          <li><a href="./Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a></li>
        
          <li><a href="./Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a></li>
        
          <li><a href="./Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a></li>
        
          <li><a href="./Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a></li>
        
          <li><a href="./Bio/GCG.html">Bio::GCG</a></li>
        
          <li><a href="./Bio/GCG/Msf.html">Bio::GCG::Msf</a></li>
        
          <li><a href="./Bio/GCG/Seq.html">Bio::GCG::Seq</a></li>
        
          <li><a href="./Bio/GFF.html">Bio::GFF</a></li>
        
          <li><a href="./Bio/GFF/GFF2.html">Bio::GFF::GFF2</a></li>
        
          <li><a href="./Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a></li>
        
          <li><a href="./Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a></li>
        
          <li><a href="./Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a></li>
        
          <li><a href="./Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a></li>
        
          <li><a href="./Bio/GFF/GFF3.html">Bio::GFF::GFF3</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a></li>
        
          <li><a href="./Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a></li>
        
          <li><a href="./Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a></li>
        
          <li><a href="./Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a></li>
        
          <li><a href="./Bio/GFF/Record.html">Bio::GFF::Record</a></li>
        
          <li><a href="./Bio/GO.html">Bio::GO</a></li>
        
          <li><a href="./Bio/GO/External2go.html">Bio::GO::External2go</a></li>
        
          <li><a href="./Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a></li>
        
          <li><a href="./Bio/GO/Ontology.html">Bio::GO::Ontology</a></li>
        
          <li><a href="./Bio/GenBank.html">Bio::GenBank</a></li>
        
          <li><a href="./Bio/GenBank/Locus.html">Bio::GenBank::Locus</a></li>
        
          <li><a href="./Bio/GenPept.html">Bio::GenPept</a></li>
        
          <li><a href="./Bio/GenPept/Locus.html">Bio::GenPept::Locus</a></li>
        
          <li><a href="./Bio/Genscan.html">Bio::Genscan</a></li>
        
          <li><a href="./Bio/Genscan/Report.html">Bio::Genscan::Report</a></li>
        
          <li><a href="./Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a></li>
        
          <li><a href="./Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a></li>
        
          <li><a href="./Bio/HGC.html">Bio::HGC</a></li>
        
          <li><a href="./Bio/HGC/HiGet.html">Bio::HGC::HiGet</a></li>
        
          <li><a href="./Bio/HMMER.html">Bio::HMMER</a></li>
        
          <li><a href="./Bio/HMMER/Report.html">Bio::HMMER::Report</a></li>
        
          <li><a href="./Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a></li>
        
          <li><a href="./Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a></li>
        
          <li><a href="./Bio/Hinv.html">Bio::Hinv</a></li>
        
          <li><a href="./Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a></li>
        
          <li><a href="./Bio/Hinv/Common.html">Bio::Hinv::Common</a></li>
        
          <li><a href="./Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a></li>
        
          <li><a href="./Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a></li>
        
          <li><a href="./Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a></li>
        
          <li><a href="./Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a></li>
        
          <li><a href="./Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a></li>
        
          <li><a href="./Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a></li>
        
          <li><a href="./Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a></li>
        
          <li><a href="./Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a></li>
        
          <li><a href="./Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a></li>
        
          <li><a href="./Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a></li>
        
          <li><a href="./Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a></li>
        
          <li><a href="./Bio/Iprscan.html">Bio::Iprscan</a></li>
        
          <li><a href="./Bio/Iprscan/Report.html">Bio::Iprscan::Report</a></li>
        
          <li><a href="./Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a></li>
        
          <li><a href="./Bio/KEGG.html">Bio::KEGG</a></li>
        
          <li><a href="./Bio/KEGG/API.html">Bio::KEGG::API</a></li>
        
          <li><a href="./Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a></li>
        
          <li><a href="./Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a></li>
        
          <li><a href="./Bio/KEGG/Common.html">Bio::KEGG::Common</a></li>
        
          <li><a href="./Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a></li>
        
          <li><a href="./Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a></li>
        
          <li><a href="./Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a></li>
        
          <li><a href="./Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a></li>
        
          <li><a href="./Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a></li>
        
          <li><a href="./Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a></li>
        
          <li><a href="./Bio/KEGG/GENES.html">Bio::KEGG::GENES</a></li>
        
          <li><a href="./Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a></li>
        
          <li><a href="./Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a></li>
        
          <li><a href="./Bio/KEGG/KGML.html">Bio::KEGG::KGML</a></li>
        
          <li><a href="./Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a></li>
        
          <li><a href="./Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a></li>
        
          <li><a href="./Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a></li>
        
          <li><a href="./Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a></li>
        
          <li><a href="./Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a></li>
        
          <li><a href="./Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a></li>
        
          <li><a href="./Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a></li>
        
          <li><a href="./Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a></li>
        
          <li><a href="./Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a></li>
        
          <li><a href="./Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a></li>
        
          <li><a href="./Bio/KEGGDB.html">Bio::KEGGDB</a></li>
        
          <li><a href="./Bio/LITDB.html">Bio::LITDB</a></li>
        
          <li><a href="./Bio/Lasergene.html">Bio::Lasergene</a></li>
        
          <li><a href="./Bio/Location.html">Bio::Location</a></li>
        
          <li><a href="./Bio/Locations.html">Bio::Locations</a></li>
        
          <li><a href="./Bio/MAFFT.html">Bio::MAFFT</a></li>
        
          <li><a href="./Bio/MAFFT/Report.html">Bio::MAFFT::Report</a></li>
        
          <li><a href="./Bio/MEDLINE.html">Bio::MEDLINE</a></li>
        
          <li><a href="./Bio/Map.html">Bio::Map</a></li>
        
          <li><a href="./Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a></li>
        
          <li><a href="./Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a></li>
        
          <li><a href="./Bio/Map/Mapping.html">Bio::Map::Mapping</a></li>
        
          <li><a href="./Bio/Map/Marker.html">Bio::Map::Marker</a></li>
        
          <li><a href="./Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a></li>
        
          <li><a href="./Bio/Meme.html">Bio::Meme</a></li>
        
          <li><a href="./Bio/Meme/Mast.html">Bio::Meme::Mast</a></li>
        
          <li><a href="./Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a></li>
        
          <li><a href="./Bio/Meme/Motif.html">Bio::Meme::Motif</a></li>
        
          <li><a href="./Bio/Muscle.html">Bio::Muscle</a></li>
        
          <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a></li>
        
          <li><a href="./Bio/NBRF.html">Bio::NBRF</a></li>
        
          <li><a href="./Bio/NCBI.html">Bio::NCBI</a></li>
        
          <li><a href="./Bio/NCBI/REST.html">Bio::NCBI::REST</a></li>
        
          <li><a href="./Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a></li>
        
          <li><a href="./Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a></li>
        
          <li><a href="./Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a></li>
        
          <li><a href="./Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a></li>
        
          <li><a href="./Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a></li>
        
          <li><a href="./Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a></li>
        
          <li><a href="./Bio/NCBIDB.html">Bio::NCBIDB</a></li>
        
          <li><a href="./Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a></li>
        
          <li><a href="./Bio/Newick.html">Bio::Newick</a></li>
        
          <li><a href="./Bio/Newick/ParseError.html">Bio::Newick::ParseError</a></li>
        
          <li><a href="./Bio/Nexus.html">Bio::Nexus</a></li>
        
          <li><a href="./Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a></li>
        
          <li><a href="./Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a></li>
        
          <li><a href="./Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a></li>
        
          <li><a href="./Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a></li>
        
          <li><a href="./Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a></li>
        
          <li><a href="./Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a></li>
        
          <li><a href="./Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a></li>
        
          <li><a href="./Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a></li>
        
          <li><a href="./Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a></li>
        
          <li><a href="./Bio/Nexus/Util.html">Bio::Nexus::Util</a></li>
        
          <li><a href="./Bio/NucleicAcid.html">Bio::NucleicAcid</a></li>
        
          <li><a href="./Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a></li>
        
          <li><a href="./Bio/PAML.html">Bio::PAML</a></li>
        
          <li><a href="./Bio/PAML/Baseml.html">Bio::PAML::Baseml</a></li>
        
          <li><a href="./Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a></li>
        
          <li><a href="./Bio/PAML/Codeml.html">Bio::PAML::Codeml</a></li>
        
          <li><a href="./Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a></li>
        
          <li><a href="./Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a></li>
        
          <li><a href="./Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a></li>
        
          <li><a href="./Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a></li>
        
          <li><a href="./Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a></li>
        
          <li><a href="./Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a></li>
        
          <li><a href="./Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a></li>
        
          <li><a href="./Bio/PAML/Common.html">Bio::PAML::Common</a></li>
        
          <li><a href="./Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a></li>
        
          <li><a href="./Bio/PAML/Yn00.html">Bio::PAML::Yn00</a></li>
        
          <li><a href="./Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a></li>
        
          <li><a href="./Bio/PDB.html">Bio::PDB</a></li>
        
          <li><a href="./Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a></li>
        
          <li><a href="./Bio/PDB/Chain.html">Bio::PDB::Chain</a></li>
        
          <li><a href="./Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a></li>
        
          <li><a href="./Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a></li>
        
          <li><a href="./Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a></li>
        
          <li><a href="./Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a></li>
        
          <li><a href="./Bio/PDB/DataType.html">Bio::PDB::DataType</a></li>
        
          <li><a href="./Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a></li>
        
          <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a></li>
        
          <li><a href="./Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a></li>
        
          <li><a href="./Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a></li>
        
          <li><a href="./Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a></li>
        
          <li><a href="./Bio/PDB/Model.html">Bio::PDB::Model</a></li>
        
          <li><a href="./Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a></li>
        
          <li><a href="./Bio/PDB/Record.html">Bio::PDB::Record</a></li>
        
          <li><a href="./Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a></li>
        
          <li><a href="./Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a></li>
        
          <li><a href="./Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a></li>
        
          <li><a href="./Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a></li>
        
          <li><a href="./Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a></li>
        
          <li><a href="./Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a></li>
        
          <li><a href="./Bio/PDB/Residue.html">Bio::PDB::Residue</a></li>
        
          <li><a href="./Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a></li>
        
          <li><a href="./Bio/PDB/Utils.html">Bio::PDB::Utils</a></li>
        
          <li><a href="./Bio/PROSITE.html">Bio::PROSITE</a></li>
        
          <li><a href="./Bio/PSORT.html">Bio::PSORT</a></li>
        
          <li><a href="./Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a></li>
        
          <li><a href="./Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a></li>
        
          <li><a href="./Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a></li>
        
          <li><a href="./Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a></li>
        
          <li><a href="./Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a></li>
        
          <li><a href="./Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a></li>
        
          <li><a href="./Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a></li>
        
          <li><a href="./Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a></li>
        
          <li><a href="./Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a></li>
        
          <li><a href="./Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a></li>
        
          <li><a href="./Bio/PTS1.html">Bio::PTS1</a></li>
        
          <li><a href="./Bio/PTS1/Report.html">Bio::PTS1::Report</a></li>
        
          <li><a href="./Bio/Pathway.html">Bio::Pathway</a></li>
        
          <li><a href="./Bio/Phylip.html">Bio::Phylip</a></li>
        
          <li><a href="./Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a></li>
        
          <li><a href="./Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></li>
        
          <li><a href="./Bio/PhyloXML.html">Bio::PhyloXML</a></li>
        
          <li><a href="./Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a></li>
        
          <li><a href="./Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a></li>
        
          <li><a href="./Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a></li>
        
          <li><a href="./Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a></li>
        
          <li><a href="./Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a></li>
        
          <li><a href="./Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a></li>
        
          <li><a href="./Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a></li>
        
          <li><a href="./Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a></li>
        
          <li><a href="./Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a></li>
        
          <li><a href="./Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a></li>
        
          <li><a href="./Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a></li>
        
          <li><a href="./Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a></li>
        
          <li><a href="./Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a></li>
        
          <li><a href="./Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a></li>
        
          <li><a href="./Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a></li>
        
          <li><a href="./Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a></li>
        
          <li><a href="./Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a></li>
        
          <li><a href="./Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a></li>
        
          <li><a href="./Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a></li>
        
          <li><a href="./Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a></li>
        
          <li><a href="./Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a></li>
        
          <li><a href="./Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a></li>
        
          <li><a href="./Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a></li>
        
          <li><a href="./Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a></li>
        
          <li><a href="./Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a></li>
        
          <li><a href="./Bio/Probcons.html">Bio::Probcons</a></li>
        
          <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a></li>
        
          <li><a href="./Bio/PubMed.html">Bio::PubMed</a></li>
        
          <li><a href="./Bio/REBASE.html">Bio::REBASE</a></li>
        
          <li><a href="./Bio/RefSeq.html">Bio::RefSeq</a></li>
        
          <li><a href="./Bio/Reference.html">Bio::Reference</a></li>
        
          <li><a href="./Bio/References.html">Bio::References</a></li>
        
          <li><a href="./Bio/Registry.html">Bio::Registry</a></li>
        
          <li><a href="./Bio/Registry/DB.html">Bio::Registry::DB</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a></li>
        
          <li><a href="./Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a></li>
        
          <li><a href="./Bio/SOAPWSDL.html">Bio::SOAPWSDL</a></li>
        
          <li><a href="./Bio/SOFT.html">Bio::SOFT</a></li>
        
          <li><a href="./Bio/SOSUI.html">Bio::SOSUI</a></li>
        
          <li><a href="./Bio/SOSUI/Report.html">Bio::SOSUI::Report</a></li>
        
          <li><a href="./Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a></li>
        
          <li><a href="./Bio/SPTR.html">Bio::SPTR</a></li>
        
          <li><a href="./Bio/SQL.html">Bio::SQL</a></li>
        
          <li><a href="./Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a></li>
        
          <li><a href="./Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a></li>
        
          <li><a href="./Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a></li>
        
          <li><a href="./Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a></li>
        
          <li><a href="./Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a></li>
        
          <li><a href="./Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a></li>
        
          <li><a href="./Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a></li>
        
          <li><a href="./Bio/SQL/Comment.html">Bio::SQL::Comment</a></li>
        
          <li><a href="./Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a></li>
        
          <li><a href="./Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a></li>
        
          <li><a href="./Bio/SQL/Location.html">Bio::SQL::Location</a></li>
        
          <li><a href="./Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/Ontology.html">Bio::SQL::Ontology</a></li>
        
          <li><a href="./Bio/SQL/Reference.html">Bio::SQL::Reference</a></li>
        
          <li><a href="./Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a></li>
        
          <li><a href="./Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a></li>
        
          <li><a href="./Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a></li>
        
          <li><a href="./Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a></li>
        
          <li><a href="./Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a></li>
        
          <li><a href="./Bio/SQL/Sequence.html">Bio::SQL::Sequence</a></li>
        
          <li><a href="./Bio/SQL/Taxon.html">Bio::SQL::Taxon</a></li>
        
          <li><a href="./Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a></li>
        
          <li><a href="./Bio/SQL/Term.html">Bio::SQL::Term</a></li>
        
          <li><a href="./Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a></li>
        
          <li><a href="./Bio/SQL/TermPath.html">Bio::SQL::TermPath</a></li>
        
          <li><a href="./Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a></li>
        
          <li><a href="./Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a></li>
        
          <li><a href="./Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a></li>
        
          <li><a href="./Bio/SangerChromatogram.html">Bio::SangerChromatogram</a></li>
        
          <li><a href="./Bio/Scf.html">Bio::Scf</a></li>
        
          <li><a href="./Bio/Sequence.html">Bio::Sequence</a></li>
        
          <li><a href="./Bio/Sequence/AA.html">Bio::Sequence::AA</a></li>
        
          <li><a href="./Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a></li>
        
          <li><a href="./Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a></li>
        
          <li><a href="./Bio/Sequence/Common.html">Bio::Sequence::Common</a></li>
        
          <li><a href="./Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a></li>
        
          <li><a href="./Bio/Sequence/Format.html">Bio::Sequence::Format</a></li>
        
          <li><a href="./Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a></li>
        
          <li><a href="./Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a></li>
        
          <li><a href="./Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a></li>
        
          <li><a href="./Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a></li>
        
          <li><a href="./Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a></li>
        
          <li><a href="./Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a></li>
        
          <li><a href="./Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a></li>
        
          <li><a href="./Bio/Sequence/NA.html">Bio::Sequence::NA</a></li>
        
          <li><a href="./Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a></li>
        
          <li><a href="./Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a></li>
        
          <li><a href="./Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a></li>
        
          <li><a href="./Bio/Shell.html">Bio::Shell</a></li>
        
          <li><a href="./Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a></li>
        
          <li><a href="./Bio/Shell/Core.html">Bio::Shell::Core</a></li>
        
          <li><a href="./Bio/Shell/Demo.html">Bio::Shell::Demo</a></li>
        
          <li><a href="./Bio/Shell/Ghost.html">Bio::Shell::Ghost</a></li>
        
          <li><a href="./Bio/Shell/Irb.html">Bio::Shell::Irb</a></li>
        
          <li><a href="./Bio/Shell/Private.html">Bio::Shell::Private</a></li>
        
          <li><a href="./Bio/Shell/Script.html">Bio::Shell::Script</a></li>
        
          <li><a href="./Bio/Shell/Setup.html">Bio::Shell::Setup</a></li>
        
          <li><a href="./Bio/Shell/Web.html">Bio::Shell::Web</a></li>
        
          <li><a href="./Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a></li>
        
          <li><a href="./Bio/SiRNA.html">Bio::SiRNA</a></li>
        
          <li><a href="./Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a></li>
        
          <li><a href="./Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a></li>
        
          <li><a href="./Bio/Sim4.html">Bio::Sim4</a></li>
        
          <li><a href="./Bio/Sim4/Report.html">Bio::Sim4::Report</a></li>
        
          <li><a href="./Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a></li>
        
          <li><a href="./Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a></li>
        
          <li><a href="./Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a></li>
        
          <li><a href="./Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a></li>
        
          <li><a href="./Bio/Spidey.html">Bio::Spidey</a></li>
        
          <li><a href="./Bio/Spidey/Report.html">Bio::Spidey::Report</a></li>
        
          <li><a href="./Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a></li>
        
          <li><a href="./Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a></li>
        
          <li><a href="./Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a></li>
        
          <li><a href="./Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a></li>
        
          <li><a href="./Bio/SwissProt.html">Bio::SwissProt</a></li>
        
          <li><a href="./Bio/TMHMM.html">Bio::TMHMM</a></li>
        
          <li><a href="./Bio/TMHMM/Report.html">Bio::TMHMM::Report</a></li>
        
          <li><a href="./Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a></li>
        
          <li><a href="./Bio/TRANSFAC.html">Bio::TRANSFAC</a></li>
        
          <li><a href="./Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a></li>
        
          <li><a href="./Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a></li>
        
          <li><a href="./Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a></li>
        
          <li><a href="./Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a></li>
        
          <li><a href="./Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a></li>
        
          <li><a href="./Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a></li>
        
          <li><a href="./Bio/TargetP.html">Bio::TargetP</a></li>
        
          <li><a href="./Bio/TargetP/Report.html">Bio::TargetP::Report</a></li>
        
          <li><a href="./Bio/Taxonomy.html">Bio::Taxonomy</a></li>
        
          <li><a href="./Bio/Tcoffee.html">Bio::Tcoffee</a></li>
        
          <li><a href="./Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a></li>
        
          <li><a href="./Bio/TogoWS.html">Bio::TogoWS</a></li>
        
          <li><a href="./Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a></li>
        
          <li><a href="./Bio/TogoWS/REST.html">Bio::TogoWS::REST</a></li>
        
          <li><a href="./Bio/TrEMBL.html">Bio::TrEMBL</a></li>
        
          <li><a href="./Bio/Tree.html">Bio::Tree</a></li>
        
          <li><a href="./Bio/Tree/Edge.html">Bio::Tree::Edge</a></li>
        
          <li><a href="./Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a></li>
        
          <li><a href="./Bio/Tree/Node.html">Bio::Tree::Node</a></li>
        
          <li><a href="./Bio/UniProt.html">Bio::UniProt</a></li>
        
          <li><a href="./BiorubyController.html">BiorubyController</a></li>
        
          <li><a href="./BiorubyGenerator.html">BiorubyGenerator</a></li>
        
          <li><a href="./BiorubyHelper.html">BiorubyHelper</a></li>
        
          <li><a href="./Hash.html">Hash</a></li>
        
          <li><a href="./Object.html">Object</a></li>
        
          <li><a href="./String.html">String</a></li>
        
        </ul>
        <div id="no-class-search-results" style="display: none;">No matching classes.</div>
      </div>

      
    </div>
  </div>

  <div id="documentation">
    
<p>commit 3acc1e098839cacbe85b5c23367ab14e0c4fe3ea Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 15:01:49 2011 +0900</p>

<pre>   Preparation for bioruby-1.4.2 release.

bioruby.gemspec    |    2 +-
lib/bio/version.rb |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit bf69125192fa01ae3495e094e7ef1b5e895954ad Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 14:42:02 2011 +0900</p>

<pre>   updated bioruby.gemspec

bioruby.gemspec |    3 +++
1 files changed, 3 insertions(+), 0 deletions(-)</pre>

<p>commit e0a3ead917812199c6a0e495f3afa6a636bbf0c5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 14:39:54 2011 +0900</p>

<pre>   Added PLUGIN section to README.rdoc, and some changes made.

README.rdoc |   20 +++++++++++++++++---
1 files changed, 17 insertions(+), 3 deletions(-)</pre>

<p>commit 1da0a1ce6eddcef8f8fc811b0a2cf8d58f880642 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 14:38:13 2011 +0900</p>

<pre>   updated doc/Tutorial.rd.html

doc/Tutorial.rd.html |   41 ++++++++++++++++++++---------------------
1 files changed, 20 insertions(+), 21 deletions(-)</pre>

<p>commit a8b90367b830e58b397536be3dada10cdde97aab Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 14:36:24 2011 +0900</p>

<pre>   Removed sections contain obsolete (404 Not Found) URL in Tutorial.rd.

doc/Tutorial.rd |   12 ------------
1 files changed, 0 insertions(+), 12 deletions(-)</pre>

<p>commit e17546cb90a012cd1f51674ceb4c8da5dd516bdf Author: Michael O’Keefe
&lt;okeefm@rpi.edu&gt; Date:   Tue Aug 23 20:15:44 2011 -0400</p>

<pre>   Updated tutorial

    * Updated tutorial
      (original commit id: 7b9108657961cf2354278e04971c32059b3ed4e2
       and some preceding commits)

doc/Tutorial.rd |   55 ++++++++++++++++++++++++++++++++-----------------------
1 files changed, 32 insertions(+), 23 deletions(-)</pre>

<p>commit de8a394129c752a0b9a5975a73c5eb582d9681d3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 13:24:27 2011 +0900</p>

<pre>   fix typo and change order of lines

RELEASE_NOTES.rdoc |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 1cf2a11199655e4c9f5fc49c5a588b99c18ab7ca Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 13:16:11 2011 +0900</p>

<pre>   RELEASE_NOTE.rdoc modified to reflect recent changes

RELEASE_NOTES.rdoc |   14 ++++++++++++++
1 files changed, 14 insertions(+), 0 deletions(-)</pre>

<p>commit b44871a5866eeb2d379f080b39b09693c9e9e3cc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 13:15:14 2011 +0900</p>

<pre>   In BioRuby Shell, getent() fails when EMBOSS seqret does not found.

lib/bio/shell/plugin/entry.rb |   10 +++++++---
1 files changed, 7 insertions(+), 3 deletions(-)</pre>

<p>commit 179e7506b008a220d5dd42ce1a6c7ce589c3fcda Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 26 12:26:52 2011 +0900</p>

<pre>   New methods Bio::NCBI::REST::EFetch.nucleotide and protein

    * New methods Bio::NCBI::REST::EFetch.nucleotide and protein,
      to get data from &quot;nucleotide&quot; and &quot;protein&quot; database respectively.
      Because NCBI changed not to accept &quot;gb&quot; format for the database
      &quot;sequence&quot;, the two new methods are added for convenience.
    * In BioRuby Shell, efetch method uses the above new methods.

lib/bio/io/ncbirest.rb           |  122 +++++++++++++++++++++++++++++++++++++-
lib/bio/shell/plugin/ncbirest.rb |    6 ++-
2 files changed, 126 insertions(+), 2 deletions(-)</pre>

<p>commit 99b31379bb41c7cad34c1e7dc00f802da37de1cd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Aug 25 19:03:43 2011 +0900</p>

<pre>   New method Bio::Fastq#to_s

    * New method Bio::Fastq#to_s.
      Thanks to Tomoaki NISHIYAMA who wrote a patch.
      (https://github.com/bioruby/bioruby/pull/37)

lib/bio/db/fastq.rb            |   14 ++++++++++++++
test/unit/bio/db/test_fastq.rb |   14 ++++++++++++++
2 files changed, 28 insertions(+), 0 deletions(-)</pre>

<p>commit 8ab772b37850c3874b55cf37d091046394cda5bd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Aug 25 15:23:00 2011 +0900</p>

<pre>   RELEASE_NOTES.rdoc changed to reflect recent changes.

RELEASE_NOTES.rdoc |   16 ++++++++++++++++
1 files changed, 16 insertions(+), 0 deletions(-)</pre>

<p>commit 8db6abdcc81db6a58bdd99e7f8d410b1a74496b1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Aug 25 14:28:42 2011 +0900</p>

<pre>   A test connecting to DDBJ BLAST web service is enabled.

test/functional/bio/appl/test_blast.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 121ad93c0c1f018ee389972ac5e5e8cc395f00d1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Aug 25 14:15:23 2011 +0900</p>

<pre>   Bio::DDBJ::REST::*. new classes for DDBJ REST web services.

    * Bio::DDBJ::REST::*: new classes for DDBJ REST web services (WABI).
      Currently not all services are covered. (lib/bio/io/ddbjrest.rb)
    * autoload of the above (lib/bio/db/genbank/ddbj.rb, lib/bio.rb)
    * Tests for the above (but still incomplete)
      (test/functional/bio/io/test_ddbjrest.rb)
    * Remote BLAST using DDBJ server now uses REST interface instead
      of SOAP, for Ruby 1.9.x support.
      (lib/bio/appl/blast/ddbj.rb)

lib/bio.rb                              |    1 +
lib/bio/appl/blast/ddbj.rb              |   33 +---
lib/bio/db/genbank/ddbj.rb              |    3 +-
lib/bio/io/ddbjrest.rb                  |  344 +++++++++++++++++++++++++++++++
test/functional/bio/io/test_ddbjrest.rb |   47 +++++
5 files changed, 399 insertions(+), 29 deletions(-)
create mode 100644 lib/bio/io/ddbjrest.rb
create mode 100644 test/functional/bio/io/test_ddbjrest.rb</pre>

<p>commit 7e8ba7c1388204daa5245d2128d01f6f40298185 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Aug 18 00:08:51 2011 +0900</p>

<pre>   In Fastq formatter, default width value changed to nil

    * In Bio::Sequence#output(:fastq) (Fastq output formatter),
      default width value is changed from 70 to nil, which means
      &quot;without wrapping&quot;.  close [Feature #3191]
      (https://redmine.open-bio.org/issues/3191)

RELEASE_NOTES.rdoc               |    8 ++++++--
lib/bio/db/fastq/format_fastq.rb |    4 ++--
test/unit/bio/db/test_fastq.rb   |   12 ++++++++++++
3 files changed, 20 insertions(+), 4 deletions(-)</pre>

<p>commit 0fb65211519febff18413c589fe7af753ee2e61d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 17 22:02:03 2011 +0900</p>

<pre>   Bug fix: Bio::SPTR follow-up of UniProtKB format changes

    * Bug fix: Bio::SPTR follow-up of UniProtKB format changes.
    * Tests are added about the fix.
    * Bug fix: Bio::SPTR#cc_web_resource should be private.
    * Incompatible changes in Bio::SPTR#cc(&quot;WEB RESOURCE&quot;) is
      documented in RELEASE_NOTES.rdoc.
    * KNOWN_ISSUES.rdoc: description about incompleteness of the fix.
    * Thanks to Nicholas Letourneau who reports the issue.
      (https://github.com/bioruby/bioruby/pull/36)

KNOWN_ISSUES.rdoc                              |    5 +
RELEASE_NOTES.rdoc                             |   20 ++-
lib/bio/db/embl/sptr.rb                        |  214 +++++++++++++++++++++---
test/unit/bio/db/embl/test_sptr.rb             |   12 +-
test/unit/bio/db/embl/test_uniprot_new_part.rb |  208 +++++++++++++++++++++++
5 files changed, 430 insertions(+), 29 deletions(-)
create mode 100644 test/unit/bio/db/embl/test_uniprot_new_part.rb</pre>

<p>commit 0d066ab6b8fc19f1cf6e66e07c2065775739cccd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Aug 13 00:58:51 2011 +0900</p>

<pre>   preparation for release: alpha test version 1.4.2-alpha1

bioruby.gemspec    |   23 +++++++++++++++++++++--
lib/bio/version.rb |    4 ++--
2 files changed, 23 insertions(+), 4 deletions(-)</pre>

<p>commit 55ece17775f5d24cf62f93d54ded5dc6eed53584 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 12 21:57:25 2011 +0900</p>

<pre>   Test bug fix: use sort command in PATH

    * Test bug fix: FuncTestCommandQuery: use sort command in PATH.
      Thanks to Tomoaki Nishiyama who reports the issue.
      (https://github.com/bioruby/bioruby/pull/13)

test/functional/bio/test_command.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 2f464aae016387cd50031f9d9664e78e220e2d01 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 12 20:37:18 2011 +0900</p>

<pre>   RELEASE_NOTES.rdoc is updated following recent changes.

RELEASE_NOTES.rdoc |   21 ++++++++++++++-------
1 files changed, 14 insertions(+), 7 deletions(-)</pre>

<p>commit d1a193684afdfd4c632ef75a978d4f3680d1bdf3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Aug 12 20:30:53 2011 +0900</p>

<pre>   README.rdoc: changed required Ruby version etc.

    * README.rdoc: now Ruby 1.8.6 or later is required.
    * README.rdoc: removed old obsolete descriptions.
    * README.rdoc: modified about RubyGems.
    * KNOWN_ISSUES.rdoc: moved descriptions about older RubyGems and CVS
      from README.rdoc.
    * KNOWN_ISSUES.rdoc: modified about end-of-life Ruby versions.

KNOWN_ISSUES.rdoc |   40 ++++++++++++++++++++++++++++++----
README.rdoc       |   61 ++++++++++++++++++++--------------------------------
2 files changed, 59 insertions(+), 42 deletions(-)</pre>

<p>commit b5cbdc6ab7e81aae4db9aeb708fac57ffbce5636 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Jul 16 00:12:17 2011 +0900</p>

<pre>   Added topics for the release notes

RELEASE_NOTES.rdoc |   39 ++++++++++++++++++++++++++++++++++++++-
1 files changed, 38 insertions(+), 1 deletions(-)</pre>

<p>commit f062b5f37a6d8ad35b5b10c942fd61e1a4d37e08 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Jul 2 01:05:42 2011 +0900</p>

<pre>   Speedup of Bio::RestrictionEnzyme::Analysis.cut.

    * Speedup of Bio::RestrictionEnzyme::Analysis.cut. The new code
      is 50 to 80 fold faster than the previous code when cutting
      1Mbp sequence running on Ruby 1.9.2p180.
    * Thanks to Tomoaki NISHIYAMA who wrote the first version of the
      patch. Thanks to ray1729 (on GitHub) who reports the issue.
      (https://github.com/bioruby/bioruby/issues/10)

lib/bio/util/restriction_enzyme.rb                 |    3 +
.../restriction_enzyme/range/sequence_range.rb     |   14 ++--
.../range/sequence_range/calculated_cuts.rb        |   75 +++++++++++++++-----
.../range/sequence_range/fragment.rb               |    4 +-
4 files changed, 69 insertions(+), 27 deletions(-)</pre>

<p>commit 735379421d9d6b7ceb06b91dcfcca6d5ff841236 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Jul 2 00:59:58 2011 +0900</p>

<pre>   New classes (for internal use only) for restriction enzyme classes

    * New classes Bio::RestrictionEnzyme::SortedNumArray and
      Bio::RestrictionEnzyme::DenseIntArray. Both of them are for
      Bio::RestrictionEnzyme internal use only. They will be used
      for the speedup of restriction enzyme analysis.

lib/bio/util/restriction_enzyme/dense_int_array.rb |  195 ++++++++++++++
.../util/restriction_enzyme/sorted_num_array.rb    |  219 +++++++++++++++
.../restriction_enzyme/test_dense_int_array.rb     |  201 ++++++++++++++
.../restriction_enzyme/test_sorted_num_array.rb    |  281 ++++++++++++++++++++
4 files changed, 896 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/util/restriction_enzyme/dense_int_array.rb
create mode 100644 lib/bio/util/restriction_enzyme/sorted_num_array.rb
create mode 100644 test/unit/bio/util/restriction_enzyme/test_dense_int_array.rb
create mode 100644 test/unit/bio/util/restriction_enzyme/test_sorted_num_array.rb</pre>

<p>commit 6cbb0c230d1a0bf3125c3b0fdb9ec3333d9564f8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jun 30 20:47:26 2011 +0900</p>

<pre>   A sample benchmark script for Bio::RestrictionEnzyme::Analysis.cut

sample/test_restriction_enzyme_long.rb | 4403 ++++++++++++++++++++++++++++++++
1 files changed, 4403 insertions(+), 0 deletions(-)
create mode 100644 sample/test_restriction_enzyme_long.rb</pre>

<p>commit 413442bd7424f837c73d8170ced8e01a01f87a59 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue May 24 23:26:41 2011 +0900</p>

<pre>   Added a test for Bio::FastaFormat#entry_overrun etc.

    * Added a test for Bio::FastaFormat#entry_overrun.
    * Removed a void test class.

test/unit/bio/db/test_fasta.rb |   24 ++++++++++++------------
1 files changed, 12 insertions(+), 12 deletions(-)</pre>

<p>commit b74020ff9b5c9fc8531c584898a329987008870e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue May 24 22:21:17 2011 +0900</p>

<pre>   Bug fix: Bio::FastaFormat#query passes nil to the given factory

    * Bug fix: Bio::FastaFormat#query passes nil to the given factory
      object. Thanks to Philipp Comans who reports the bug.
      (https://github.com/bioruby/bioruby/issues/35)
    * Test method for Bio::FastaFormat#query is added.

lib/bio/db/fasta.rb            |    2 +-
test/unit/bio/db/test_fasta.rb |   22 ++++++++++++++++++++++
2 files changed, 23 insertions(+), 1 deletions(-)</pre>

<p>commit 80e49373e0e9013442680ba33499be80c58471db Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue May 17 22:33:56 2011 +0900</p>

<pre>   Changed database name in the example.

    * Changed database name in the example.
      Thanks to Philipp Comans who reports the issue.

lib/bio/appl/blast.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 7427d1f1355a6c190c6bf8522978e462dea64134 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu May 12 22:15:37 2011 +0900</p>

<pre>   Bug fix: changed GenomeNet remote BLAST URL.

    * Bug fix: changed GenomeNet remote BLAST host name and path.
      Thanks to Philipp Comans who reports the bug.
      ( https://github.com/bioruby/bioruby/issues/34 )

lib/bio/appl/blast/genomenet.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit c1c231b0a17c06ec042534245ed903e0256a59ed Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue May 10 20:57:17 2011 +0900</p>

<pre>   updated doc/Tutorial.rd.html

doc/Tutorial.rd.html |   34 ++++++++++++++++++----------------
1 files changed, 18 insertions(+), 16 deletions(-)</pre>

<p>commit 5261c926cae8dac890d7d0380e84f2eb88912417 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Thu May 5 12:07:54 2011 +0200</p>

<pre>   Tutorial: Fixed URL and the intro

doc/Tutorial.rd |   34 ++++++++++++++++++++--------------
1 files changed, 20 insertions(+), 14 deletions(-)</pre>

<p>commit 71de394053376f4759d705c52e6f16eca3da9d62 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Wed Mar 9 10:26:53 2011 +0100</p>

<pre>   Tutorial: Added a commnet for rubydoctest, changed Ruby version

    * Added a comment for rubydoctest
    * Changed example Ruby version representation
    * This is part of commit ba5b9c2d29223860252451110a99d4ff0250395d
      and modified to merge with the current HEAD.

doc/Tutorial.rd |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 1d27153065b8e8595a470b2201961b0a39bf8ca1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Apr 28 23:58:57 2011 +0900</p>

<pre>   updated doc/Tutorial.rd.html

doc/Tutorial.rd.html |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit ae9beff3bc43db3724a292b10a214583d9fbc111 Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Wed Apr 6 11:46:54 2011 -0400</p>

<pre>   Updated through the section on Homology searching with BLAST

doc/Tutorial.rd      |   64 ++++++++++++++++++++--------------------------
doc/Tutorial.rd.html |   68 ++++++++++++++++++++++----------------------------
2 files changed, 58 insertions(+), 74 deletions(-)</pre>

<p>commit 971da799b16628a927abd7dd6c218994506f8fd8 Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Thu Mar 24 18:29:20 2011 -0400</p>

<pre>   Updated the html file generated from the RDoc

doc/Tutorial.rd.html |  224 ++++++++++++++++++++++++++------------------------
1 files changed, 117 insertions(+), 107 deletions(-)</pre>

<p>commit c6afd7eeed121926b56d300cb4170b5024f29eb0 Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Thu Mar 24 18:08:17 2011 -0400</p>

<pre>   Finished updating the tutorial

doc/Tutorial.rd |   16 ++++++++--------
1 files changed, 8 insertions(+), 8 deletions(-)</pre>

<p>commit 7349ac550ec03e2c5266496297becbdb3f4e0edd Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Thu Mar 24 15:56:37 2011 -0400</p>

<pre>   Edited tutorial up through the extra stuff section

doc/Tutorial.rd |   28 ++++++++++++++--------------
1 files changed, 14 insertions(+), 14 deletions(-)</pre>

<p>commit 32ba3b15ad00d02b12ef2b44636505e23caaf620 Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Thu Mar 24 15:26:48 2011 -0400</p>

<pre>   Updated tutorial up through BioSQL

doc/Tutorial.rd |  172 ++++++++++++++++++-------------------------------------
1 files changed, 57 insertions(+), 115 deletions(-)</pre>

<p>commit 249580edb49a13545708fdcb559104217e37f162 Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Thu Mar 24 12:09:03 2011 -0400</p>

<pre>   Updated tutorial through the section on alignments

doc/Tutorial.rd |   40 +++++++++++++++++++++++++++++++++-------
1 files changed, 33 insertions(+), 7 deletions(-)</pre>

<p>commit 54f7b54044bb245ec5953dc7426f1c434b41f24f Author: Michael O’Keefe
&lt;okeemf@rpi.edu&gt; Date:   Thu Mar 24 11:51:23 2011 -0400</p>

<pre>   Updated the tutorial (mostly grammar fixes) up until GenBank

doc/Tutorial.rd |   31 +++++++++++++++----------------
1 files changed, 15 insertions(+), 16 deletions(-)</pre>

<p>commit f046a52081a8af0e9afbf65fd2673c29689be769 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 8 12:58:50 2011 +0900</p>

<pre>   Added a test protein sequence data for BLAST test.

test/data/fasta/EFTU_BACSU.fasta |    8 ++++++++
1 files changed, 8 insertions(+), 0 deletions(-)
create mode 100644 test/data/fasta/EFTU_BACSU.fasta</pre>

<p>commit f61a5f4bdde16fa051f43cbe3efef4570b415a6a Author: Anthony Underwood
&lt;email2ants@gmail.com&gt; Date:   Mon Jan 31 12:44:55 2011 +0000</p>

<pre>   Bug fix: GenBank sequence output should format date as 27-JAN-2011

    * Bug fix: GenBank sequence output should format date as 27-JAN-2011
      rather than 2011-01-27 as specified by offical GenBank specs.

lib/bio/db/genbank/format_genbank.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit be144e75a059058ab000a55d7bf535597e7e2617 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Feb 3 20:28:03 2011 +0900</p>

<pre>   Added tests for remote BLAST execution via GenomeNet and DDBJ.

    * Added tests for remote BLAST execution via GenomeNet and DDBJ.
      Currently, a test for DDBJ BLAST web API is disabled because
      it takes relatively long time.
    * Tests to retrieve remote BLAST database information for GenomeNet
      and DDBJ servers are also added.

test/functional/bio/appl/blast/test_remote.rb |   93 +++++++++++++++++++++++++
test/functional/bio/appl/test_blast.rb        |   61 ++++++++++++++++
2 files changed, 154 insertions(+), 0 deletions(-)
create mode 100644 test/functional/bio/appl/blast/test_remote.rb
create mode 100644 test/functional/bio/appl/test_blast.rb</pre>

<p>commit 67314f1f1a248954c030f7ffe048faf862bf07d2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Feb 3 20:19:11 2011 +0900</p>

<pre>   Updated _parse_databases following the changes in the DDBJ server

    * Updated _parse_databases following the changes in the DDBJ server.
      Changed to use (NT) or (AA) in the tail of each description.
      Thanks to DDBJ to improve their web service API.

lib/bio/appl/blast/ddbj.rb |   29 ++++++++++++++++++++---------
1 files changed, 20 insertions(+), 9 deletions(-)</pre>

<p>commit d6aad2f4cc53c1227c86b6b573644cca15c9ed82 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Feb 2 00:02:32 2011 +0900</p>

<pre>   Release notes for the next release is added.

RELEASE_NOTES.rdoc |   38 ++++++++++++++++++++++++++++++++++++++
1 files changed, 38 insertions(+), 0 deletions(-)
create mode 100644 RELEASE_NOTES.rdoc</pre>

<p>commit b4a30cc8ac9472b9e1c2a298afc624d0229c64c9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 1 23:33:18 2011 +0900</p>

<pre>   Bug fix: Execution failure due to the changes of DDBJ BLAST server

lib/bio/appl/blast/ddbj.rb |    4 +++-
1 files changed, 3 insertions(+), 1 deletions(-)</pre>

<p>commit d30cb5975febd8b526088612c4fb9689a6cc46ba Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 1 23:01:34 2011 +0900</p>

<pre>   Support for database &quot;mine-aa&quot; and &quot;mine-nt&quot; with KEGG organism codes

    * Added support for database &quot;mine-aa&quot; and &quot;mine-nt&quot; combined with
      KEGG organism codes. When database name starts with mine-aa or
      mine-nt, space-separated list of KEGG organism codes can be given.
      For example, &quot;mine-aa eco bsu hsa&quot;.

lib/bio/appl/blast/genomenet.rb |   11 +++++++++++
1 files changed, 11 insertions(+), 0 deletions(-)</pre>

<p>commit abcba798ccf57894dcd570a6578ef78db30a3e25 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 1 22:20:02 2011 +0900</p>

<pre>   RELEASE_NOTES.rdoc is renamed to doc/RELEASE_NOTES-1.4.1.rdoc

RELEASE_NOTES.rdoc           |  104 ------------------------------------------
doc/RELEASE_NOTES-1.4.1.rdoc |  104 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 104 insertions(+), 104 deletions(-)
delete mode 100644 RELEASE_NOTES.rdoc
create mode 100644 doc/RELEASE_NOTES-1.4.1.rdoc</pre>

<p>commit 8719cf4e06fc8a8cd0564aeb0b95372a7a0bcefb Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 1 22:07:32 2011 +0900</p>

<pre>   Bug: Options &quot;-v&quot; and &quot;-b&quot; should be used for the limit of hits.

    * Bug: Options &quot;-v&quot; and &quot;-b&quot; should be used for the limit of hits,
      and &quot;-V&quot; and &quot;-B&quot; should not be used for the purpose.

lib/bio/appl/blast/genomenet.rb |    8 ++++----
1 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit 974b640badae9837fe9fc173d690b27c9b045454 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 1 20:34:50 2011 +0900</p>

<pre>   Bug fix: Workaround for a change in the GenomeNet BLAST site.

   * Bug fix: Workaround for a change in the GenomeNet BLAST site.
     Thanks to the anonymous reporter. The patch was originally made
     by Toshiaki Katyama.

lib/bio/appl/blast/genomenet.rb |    7 +++----
1 files changed, 3 insertions(+), 4 deletions(-)</pre>

<p>commit 001d3e3570c77185cece4aed1be5be2ed6f94f7e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jan 6 23:39:19 2011 +0900</p>

<pre>   Added tests to check the previous Bio::Reference#endnote fix.

test/unit/bio/test_reference.rb |   30 ++++++++++++++++++++++++++++++
1 files changed, 30 insertions(+), 0 deletions(-)</pre>

<p>commit e1cd766abe24dbcc08a42103127c75ad0ab929aa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jan 6 23:07:35 2011 +0900</p>

<pre>   Bio::Reference#pubmed_url is updated to follow recent NCBI changes.

lib/bio/reference.rb            |    5 ++---
test/unit/bio/test_reference.rb |    5 +++++
2 files changed, 7 insertions(+), 3 deletions(-)</pre>

<p>commit 48024313a7568a38f4291618708541ae1dac312c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jan 6 22:56:37 2011 +0900</p>

<pre>   Bug fix: Bio::Reference#endnote fails when url is not set

    * Bug fix: Bio::Reference#endnote fails when url is not set.
      Thanks to Warren Kibbe who reports the bug.
      (https://github.com/bioruby/bioruby/issues#issue/15)

lib/bio/reference.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 577278a95340abfa32d3e67415d3a10bc74b82c0 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Dec 17 12:16:31 2010 +0100</p>

<pre>   Bug fix: In Bio::MEDLINE#reference, doi field should be filled.

    * Bug fix: In Bio::MEDLINE#reference, doi field should be filled.
      (https://github.com/bioruby/bioruby/issues#issue/29)

lib/bio/db/medline.rb            |    1 +
test/unit/bio/db/test_medline.rb |    1 +
2 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit daa20c85681576d3bfbdc8f87580a4b6227b122c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jan 6 20:25:03 2011 +0900</p>

<pre>   Bug fix: Bio::Newick#reparse failure

    * Bug fix: Bio::Newick#reparse failure.
      Thanks to jdudley who reports the bug.
      (https://github.com/bioruby/bioruby/issues#issue/28)
    * Tests are added to confirm the bug fix.

lib/bio/db/newick.rb            |    4 +++-
test/unit/bio/db/test_newick.rb |   12 ++++++++++++
2 files changed, 15 insertions(+), 1 deletions(-)</pre>

<p>commit 16117aefdf57ac3ae16b5568f462f7b919ef005f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Jan 6 20:14:18 2011 +0900</p>

<pre>   Use setup for the preparation of adding more test methods.

test/unit/bio/db/test_newick.rb |   14 ++++++++++----
1 files changed, 10 insertions(+), 4 deletions(-)</pre>

<p>commit e5dc5896a5c6249e2a6cb03d63a3c2ade36b67e7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 21:07:13 2010 +0900</p>

<pre>   Ruby 1.9 support: suppressed warning &quot;mismatched indentations&quot;

test/unit/bio/db/pdb/test_pdb.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 1bce41c7ebed46ac6cf433b047fe6a4c3a538089 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 21:05:06 2010 +0900</p>

<pre>   Ruby 1.9 support: Suppressed warning &quot;shadowing outer local variable&quot;

lib/bio/db/pdb/residue.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 0f31727769833ccf9d6891ae192da1bc180223e0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 21:04:07 2010 +0900</p>

<pre>   Ruby 1.9 support: Suppressed warning &quot;shadowing outer local variable&quot;

lib/bio/location.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 70c1135b171fcf47dd9dc1bc396d15d1c3acfa62 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 21:02:57 2010 +0900</p>

<pre>   Ruby 1.9 support: Suppressed warning &quot;shadowing outer local variable&quot;

lib/bio/db/pdb/pdb.rb |   14 +++++++-------
1 files changed, 7 insertions(+), 7 deletions(-)</pre>

<p>commit a77e4ab78211a85aa052ca6645a2051a4f3b76d8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 20:42:22 2010 +0900</p>

<pre>   Ruby 1.9 support: use Array#join instead of Array#to_s

lib/bio/db/pdb/pdb.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 734c2e54613e3ed5efd95e1212feab8f014d5f19 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 20:38:11 2010 +0900</p>

<pre>   Changed to use assert_instance_of

    * Changed to use assert_instance_of(klass, obj) instead of
      assert_equal(klass, obj.class).

test/unit/bio/db/pdb/test_pdb.rb |   16 ++++++++--------
1 files changed, 8 insertions(+), 8 deletions(-)</pre>

<p>commit 490286018c8ce314da441c646ea9c5fb3f765c95 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 20:24:36 2010 +0900</p>

<pre>   Float::EPSILON was too small for the delta tolerance.

test/unit/bio/db/pdb/test_pdb.rb |   20 ++++++++++----------
1 files changed, 10 insertions(+), 10 deletions(-)</pre>

<p>commit e92d225cadabe63fe23c7c32a4d1d50a371366cc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 20:16:30 2010 +0900</p>

<pre>   Ruby 1.9 support: use assert_in_delta

test/unit/bio/db/pdb/test_pdb.rb |   24 +++++++++++++++---------
1 files changed, 15 insertions(+), 9 deletions(-)</pre>

<p>commit 41452971a132ef55de3486022962fa2c333b4c85 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 19 13:19:39 2010 +0900</p>

<pre>   Fixed Object#id problem and suppressed warning messages.

    * Changed not to call nil.id (==4) invoked from chain.id.
    * Suppressed warning message about useless use of a variable.
    * Suppressed waring about conflict of IDs when testing
      addResidue, addLigand and addChain methods.

test/unit/bio/db/pdb/test_pdb.rb |  119 +++++++++++++++++++-------------------
1 files changed, 59 insertions(+), 60 deletions(-)</pre>

<p>commit b4af5826f77002933de9d3c2ddfcc5a7cb5629db Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 17 22:26:26 2010 +0900</p>

<pre>   Adjusted copyright line

test/unit/bio/db/pdb/test_pdb.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 06fd989072b2287a3accbb60684b8a029bfc0ac3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 17 21:54:05 2010 +0900</p>

<pre>   A module name is changed to avoid potential name conflict.

    * A module name is changed to avoid potential name conflict.
    * Removed a Shift_JIS character (Zenkaku space) in a comment line.

test/unit/bio/db/pdb/test_pdb.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 09007f93abe8c9c5e7561f082b55ca307a7d4a1e Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jul 15 21:06:28 2010 +0900</p>

<pre>   Added more unit tests for Bio::PDB

    * Added more unit tests for Bio::PDB.
    * This is part of combinations of the 13 commits:
      * 555f7b49a43e7c35c82cd48b199af96ca93d4179
      * 2b5c87a9ada248597e0517e22191bb4c88be2a42
      * a16e24fa35debdcacd11cf36fdf0e60fe82b3684
      * e3607c0f7154a4156fd53ed17470aa3628cd2586
      * 4e74c9107f09c5eb3fc7fc9ec38d9d773fe89367
      * 605fb0a222f70eeaa1e576d31af484a9a6a2ac27
      * 2c8b2b5496fee04b886bfcbd11fb99948816cc28
      * 202cf2b1b57fbcac215aa135cf6343af6a49d2ef
      * f13c070c763c9035451c3639e6e29c9a156947cd
      * 843378e608bd1ef27a713d9be2d50f0f56915b0b
      * a9078b8a586b66d8026af612d9a5999c6c77de33
      * f0174a8ca3ee136271c51047fce12597d3fbb58c
      * 6675fd930718e41ad009f469b8167f81c9b2ad52

test/unit/bio/db/pdb/test_pdb.rb | 3281 +++++++++++++++++++++++++++++++++++++-
1 files changed, 3276 insertions(+), 5 deletions(-)</pre>

<p>commit 2f6a1d29b14d89ac39b408582c9865ad06560ae1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Nov 6 00:41:21 2010 +0900</p>

<pre>   Adjusted test data file path, required files and header descriptions.

test/unit/bio/db/test_litdb.rb |    8 ++++----
1 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit d3394e69c98b3be63c8287af84dc530830fb977a Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Fri Jun 18 17:11:25 2010 +0900</p>

<pre>   added unit test for Bio::LITDB with a sample file

test/data/litdb/1717226.litdb  |   13 +++++
test/unit/bio/db/test_litdb.rb |   95 ++++++++++++++++++++++++++++++++++++++++
2 files changed, 108 insertions(+), 0 deletions(-)
create mode 100644 test/data/litdb/1717226.litdb
create mode 100644 test/unit/bio/db/test_litdb.rb</pre>

<p>commit cef1d2c824f138fe268d20eaa9ffd85223c85ef9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 5 23:54:31 2010 +0900</p>

<pre>   Adjusted test data file path, required files and header descriptions.

test/unit/bio/db/test_nbrf.rb |    8 ++++----
1 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit e892e7f9b9b4daadeff44c9e479e6f51f02e383e Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Fri Jun 25 13:20:46 2010 +0900</p>

<pre>   Added unit tests for Bio::NBRF with test data.

    * Added unit tests for Bio::NBRF with test data.
    * This is part of combinations of the two commits:
      * 53873a82182e072e738da20381dcb2bfd8bc9e96
        (Modified the unit test for Bio::NBRF)
      * 4675cf85aa9c0b4de9f527f9c6bb80804fdaaaa9
        (Modified Bio::TestNBRF and Bio::TestTRANSFAC.)

test/data/pir/CRAB_ANAPL.pir  |    6 +++
test/unit/bio/db/test_nbrf.rb |   82 +++++++++++++++++++++++++++++++++++++++++
2 files changed, 88 insertions(+), 0 deletions(-)
create mode 100644 test/data/pir/CRAB_ANAPL.pir
create mode 100644 test/unit/bio/db/test_nbrf.rb</pre>

<p>commit 4922d5151138312d5a09ac60a06419c23978ba3c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 5 23:07:13 2010 +0900</p>

<pre>   Mock class for testing is moved under the test class

    * Mock class for testing is moved under the test class, to avoid
      potential name conflicts.

test/unit/bio/db/genbank/test_common.rb |   25 ++++++++++++-------------
1 files changed, 12 insertions(+), 13 deletions(-)</pre>

<p>commit 8f630700bc6dd8c183d08291c66c665394873586 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 5 23:01:32 2010 +0900</p>

<pre>   Adjusted header descriptions.

test/unit/bio/db/genbank/test_common.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit ab3d6384ca721fb6004efb5988461eecefad4d6b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 5 22:58:21 2010 +0900</p>

<pre>   Adjusted test data file path and header descriptions.

test/unit/bio/db/genbank/test_genpept.rb |    8 ++++----
1 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit 1d24aecfac9dbc190cdc3eef0956451cc88cfe4f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 5 22:54:19 2010 +0900</p>

<pre>   Adjusted test data file path, required files and header descriptions.

test/unit/bio/db/genbank/test_genbank.rb |   12 ++++++++----
1 files changed, 8 insertions(+), 4 deletions(-)</pre>

<p>commit 09275d8661f3d49a7e40be59a086ba33659b2448 Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jun 17 22:08:15 2010 +0900</p>

<pre>   Added unit tests for Bio::GenPept newly.

    * Added unit tests for Bio::GenPept newly.
    * This is part of the commit 8e46ff42b627791f259033d5a20c1610e32cfa62
      (Added unit tests for NBRF and GenPept newly.)

test/data/genbank/CAA35997.gp            |   48 ++++++++++++++++++
test/unit/bio/db/genbank/test_genpept.rb |   81 ++++++++++++++++++++++++++++++
2 files changed, 129 insertions(+), 0 deletions(-)
create mode 100644 test/data/genbank/CAA35997.gp
create mode 100644 test/unit/bio/db/genbank/test_genpept.rb</pre>

<p>commit 2cde22cb358a2b7ec8197866fe35a0b46ebf9b00 Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jun 24 18:52:37 2010 +0900</p>

<pre>   Added unit tests for Bio::NCBIDB::Common

    * Added unit tests for Bio::NCBIDB::Common.
    * This is part of combination of the 4 commits:
      * 7da8d557e8ee53da9d93c6fadfd0d8f493977c81
        (added test/unit/bio/db/genbank/test_common.rb newly)
      * 2b5c87a9ada248597e0517e22191bb4c88be2a42
        (Modified a few lines of Bio::NCBIDB::TestCommon,
         Bio::TestPDBRecord and Bio::TestPDB)
      * 10c043535dd7bf5b9682b4060183f494742c53df
        (Modified unit test for Bio::GenBank::Common)
      * 0af08fb988e08948a54e33273861b5460b7f6b2d
        (Modified the unit test for Bio::GenBank)

test/unit/bio/db/genbank/test_common.rb |  275 +++++++++++++++++++++++++++++++
1 files changed, 275 insertions(+), 0 deletions(-)
create mode 100644 test/unit/bio/db/genbank/test_common.rb</pre>

<p>commit f775d9b7f7deda2e30d4196d4cf507b59936a654 Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Sat Jun 26 17:54:32 2010 +0900</p>

<pre>   Added unit tests for Bio::GenBank with test data.

    * Added unit tests for Bio::GenBank with test data.
    * This is part of combination of the two commits:
      * 555f7b49a43e7c35c82cd48b199af96ca93d4179
        (added test_genbank.rb and test_go.rb with the test files.
        modified test_pdb.rb)
      * a46f895bf378ce08143ff031ddda302f970c270a
        (Modified Bio::GenBank and Bio::Nexus)

test/data/genbank/SCU49845.gb            |  167 +++++++++++++
test/unit/bio/db/genbank/test_genbank.rb |  397 ++++++++++++++++++++++++++++++
2 files changed, 564 insertions(+), 0 deletions(-)
create mode 100644 test/data/genbank/SCU49845.gb
create mode 100644 test/unit/bio/db/genbank/test_genbank.rb</pre>

<p>commit 33621c1f4c16173efd05861759f577b2c4733a53 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 29 16:45:22 2010 +0900</p>

<pre>   Bio::BIORUBY_EXTRA_VERSION is changed to &quot;.5000&quot;.

bioruby.gemspec    |    2 +-
lib/bio/version.rb |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit cfb2c744f3762689077f5bf2092f715d25e066ed Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 22 13:02:03 2010 +0900</p>

<pre>   BioRuby 1.4.1 is released.

ChangeLog |   55 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 55 insertions(+), 0 deletions(-)</pre>

<p>commit 92cfda14c08b270ed1beca33153125141f88510e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 22 13:00:09 2010 +0900</p>

<pre>   Preparation for bioruby-1.4.1 release.

bioruby.gemspec    |    2 +-
lib/bio/version.rb |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit d7999539392bba617b041e3120b5b2d785301f24 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 22 10:27:02 2010 +0900</p>

<pre>   Newly added issue is copied from KNOWN_ISSUES.rdoc to the release note.

RELEASE_NOTES.rdoc |   15 +++++++++++++++
1 files changed, 15 insertions(+), 0 deletions(-)</pre>

<p>commit a9f287658441038a4e9bb220502523de039417f9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 22 10:26:44 2010 +0900</p>

<pre>   updated description of an issue

KNOWN_ISSUES.rdoc |   12 +++++++-----
1 files changed, 7 insertions(+), 5 deletions(-)</pre>

<p>commit bb946d1c97d1eb0de62c8b509bbfb02d67efffeb Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 23:17:25 2010 +0900</p>

<pre>   Added an issue about command-line string escaping on Windows with Ruby 1.9.

KNOWN_ISSUES.rdoc |    8 ++++++++
1 files changed, 8 insertions(+), 0 deletions(-)</pre>

<p>commit fe7d26516cc6b9a3cf8c16e6f8204a4d5eb5e5ae Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 20:34:32 2010 +0900</p>

<pre>   Added descriptions.

RELEASE_NOTES.rdoc |   52 ++++++++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 52 insertions(+), 0 deletions(-)</pre>

<p>commit fd5da3b47ebce1df46922f20d013439faef483e9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 18:27:44 2010 +0900</p>

<pre>   ChangeLog is updated.

ChangeLog | 1657 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 1657 insertions(+), 0 deletions(-)</pre>

<p>commit fab16977d23bb3a5fdfc976eece14dfdabdcac4d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 18:07:43 2010 +0900</p>

<pre>   preparation for release candidate 1.4.1-rc1

bioruby.gemspec    |   40 ++++++++++++++++++++++++++++++++++++++--
lib/bio/version.rb |    4 ++--
2 files changed, 40 insertions(+), 4 deletions(-)</pre>

<p>commit 119cc3bf5582735a5df574450ec685fd2f989b5d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 18:05:13 2010 +0900</p>

<pre>   Temporarily removed for packaging new version. It will be reverted later.

doc/howtos/sequence_codon.txt |   38 --------------------------------------
1 files changed, 0 insertions(+), 38 deletions(-)
delete mode 100644 doc/howtos/sequence_codon.txt</pre>

<p>commit 1b1b3752e3c98a29caf837bfc12c1ed79a04dba2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 16:48:43 2010 +0900</p>

<pre>   Fixed typo, reported by Tomoaki NISHIYAMA.

KNOWN_ISSUES.rdoc |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 47ed7e5eaca4a261ef0fd4f76909c930e52aadd5 Merge: c002142 548cb58
Author: Naohisa Goto &lt;ng@bioruby.org&gt; Date:   Thu Oct 21 16:17:59
2010 +0900</p>

<pre>Merge branch 'test-defline-by-jtprince'</pre>

<p>commit 548cb58aaad06bb9161e09f7b4ae45729898ca5e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 16:16:28 2010 +0900</p>

<pre>   adjusted filename in header

test/unit/bio/db/fasta/test_defline_misc.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 95be260708ef21be7848a5d4b7c494cc6bb3d81f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 16:14:58 2010 +0900</p>

<pre>   Renamed to test_defline_misc.rb to resolve the file name conflict.

test/unit/bio/db/fasta/test_defline.rb      |  490 ---------------------------
test/unit/bio/db/fasta/test_defline_misc.rb |  490 +++++++++++++++++++++++++++
2 files changed, 490 insertions(+), 490 deletions(-)
delete mode 100644 test/unit/bio/db/fasta/test_defline.rb
create mode 100644 test/unit/bio/db/fasta/test_defline_misc.rb</pre>

<p>commit 1e7628e2c396330743d4904b100d62d2c2773bf0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 16:11:14 2010 +0900</p>

<pre>   Test bug fix: mistake in test_get method in two classes.

test/unit/bio/db/fasta/test_defline.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit c479f56f14fb531e31c7e5fdd02f6c934ac468fa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 16:06:46 2010 +0900</p>

<pre>   Test bug fix: test classes should inherit Test::Unit::TestCase.

test/unit/bio/db/fasta/test_defline.rb |   10 +++++-----
1 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit 0e8ea46e5a239df5c1da3c63e602376c04191ef4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 15:55:02 2010 +0900</p>

<pre>   Bug fix: syntax error in Ruby 1.8.7 due to a comma.

test/unit/bio/db/fasta/test_defline.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 62a2c1d7c47fbef7a7e7c4f1c079f98fa74e5099 Author: John Prince
&lt;jtprince@gmail.com&gt; Date:   Tue Oct 19 11:20:16 2010 -0600</p>

<pre>   added individual unit tests for Bio::FastaDefline

test/unit/bio/db/fasta/test_defline.rb |  490 ++++++++++++++++++++++++++++++++
1 files changed, 490 insertions(+), 0 deletions(-)
create mode 100644 test/unit/bio/db/fasta/test_defline.rb</pre>

<p>commit c002142cdb478b0ad08b7bd5e3331c7b643222f1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 15:36:58 2010 +0900</p>

<pre>   Adjusted file paths and the copyright line.

test/unit/bio/db/fasta/test_defline.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit a5818c5f8ae07e4ec4bdcc2229df9a59bded63f0 Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jun 17 11:54:15 2010 +0900</p>

<pre>   Newly added unit tests for Bio::FastaDefline

    * Newly added unit tests for Bio::FastaDefline.
    * This is part of combination of the two commits:
      bd2452caf0768e7000d19d462465b1772e3c030b
      (modified test file for Bio::FastaDefline)
      cae1b6c00cdb9044cb0dfb4db58e6acfe9b7d246
      (Added test/unit/bio/db/fasta/test_defline.rb and
       test/unit/bio/db/kegg/test_kgml.rb with the sample file newly.)

test/unit/bio/db/fasta/test_defline.rb |  160 ++++++++++++++++++++++++++++++++
1 files changed, 160 insertions(+), 0 deletions(-)
create mode 100644 test/unit/bio/db/fasta/test_defline.rb</pre>

<p>commit 4addb906df442adf4ed20275428070b651abbf07 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 15:08:46 2010 +0900</p>

<pre>   Added note for a dead link, updated a URL, and added a new reference.

lib/bio/db/fasta/defline.rb |    8 ++++++--
1 files changed, 6 insertions(+), 2 deletions(-)</pre>

<p>commit e636f123adf28688748cc5bbbc6e0c817358d475 Author: John Prince
&lt;jtprince@gmail.com&gt; Date:   Thu Oct 14 15:38:07 2010 -0600</p>

<pre>   included TREMBL prefix to list of NSIDs (tr|)

    * Included TREMBL prefix to list of NSIDs (tr|).
      This is a standard prefix found in UniprotKB FASTA files.

lib/bio/db/fasta/defline.rb |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 5277eb0b5376a0dc217dc051c49993c505956400 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 14:16:03 2010 +0900</p>

<pre>   Bug fix: Bio::ClustalW::Report#get_sequence may fail

    * Bug fix: Bio::ClustalW::Report#get_sequence may fail when the second
      argument of Bio::ClustalW::Report.new is specified.

lib/bio/appl/clustalw/report.rb |    5 +++--
1 files changed, 3 insertions(+), 2 deletions(-)</pre>

<p>commit 81b9238abb643573a4051dc0f10c4f9a2cff40fa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 14:21:52 2010 +0900</p>

<pre>   Added a test class to test the second argument of Bio::ClustalW::Report.new.

test/unit/bio/appl/clustalw/test_report.rb |   19 +++++++++++++++++++
1 files changed, 19 insertions(+), 0 deletions(-)</pre>

<p>commit 3e9b149aec91585732a34efaa960c96bcec2eef8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 13:22:38 2010 +0900</p>

<pre>   Ruby 1.9.2 support: defined Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands#initialize

    * Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands#initialize
      is explicitly defined, due to the behavior change of argument number
      check in the default initialize method in Ruby 1.9.2.

.../double_stranded/aligned_strands.rb             |    5 +++++
1 files changed, 5 insertions(+), 0 deletions(-)</pre>

<p>commit cfe31c02d4bd0d97e588d25dc30188da6be81e85 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 11:49:08 2010 +0900</p>

<pre>   Ruby 1.9.2 support: assert_in_delta for a float value.

test/unit/bio/db/test_aaindex.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 3f5d2ccb9ac8bc44febc88d441f47aeddf7f12ff Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 11:41:38 2010 +0900</p>

<pre>   Ruby 1.9.2 support: using assert_in_delta for float values.

    * Ruby 1.9.2 support: using assert_in_delta for float values.
      The patch is written by Tomoaki NISHIYAMA during BH2010.10.

test/unit/bio/util/test_contingency_table.rb |   14 +++++++-------
1 files changed, 7 insertions(+), 7 deletions(-)</pre>

<p>commit f357929bc5dcf8295b0a11a09b4025e3592d9eda Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 11:16:37 2010 +0900</p>

<pre>   Small changes for README.rdoc.

README.rdoc |    5 +++--
1 files changed, 3 insertions(+), 2 deletions(-)</pre>

<p>commit c6c567fe9602ae8d7d343a5773f51d8aa22c8876 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 21 11:12:17 2010 +0900</p>

<pre>   Shows message when running &quot;ruby setup.rb test&quot; with Ruby1.9.

setup.rb |    5 +++++
1 files changed, 5 insertions(+), 0 deletions(-)</pre>

<p>commit 9b66463c4150a679e63289e0cee3c4d1200c7d0f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 20 17:53:36 2010 +0900</p>

<pre>   Added description about incompatible the change in Bio::AAindex2.

RELEASE_NOTES.rdoc |    6 ++++++
1 files changed, 6 insertions(+), 0 deletions(-)</pre>

<p>commit 327ea878d4e15b99711d8121a54698da29d4b0aa Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 20 17:35:53 2010 +0900</p>

<pre>   Changed the expected return values in the unit tests, following the last change to Bio::AAindex2.

    * Changed the expected return values in the unit tests,
    following the last change to Bio::AAindex2.
    * The patch is written by Tomoaki NISHIYAMA during BH2010.10.

test/unit/bio/db/test_aaindex.rb |   15 ++++++++-------
1 files changed, 8 insertions(+), 7 deletions(-)</pre>

<p>commit 31963b43daab2801087f5f6d23b04e357bb7b1e2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 20 17:32:26 2010 +0900</p>

<pre>   Ruby 1.9.2 support: Incompatible change: the symmetric elements for triangular matrix should be copied

    * Ruby 1.9.2 support: Incompatible change: the symmetric elements
      for triangular matrix should be copied. The patch is written by
      Tomoaki NISHIYAMA during BH2010.10.

lib/bio/db/aaindex.rb |   12 +++++++++++-
1 files changed, 11 insertions(+), 1 deletions(-)</pre>

<p>commit e8a1d65984781466eff9d5a262f18cb1c3e01056 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 20 16:10:51 2010 +0900</p>

<pre>   Test bug fix: confusion between assert and assert_equal

    * Test bug fix: the assert should be assert_equal. The bug was
      found with Ruby 1.9.2-p0.

test/unit/bio/db/embl/test_sptr.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit feb2cda47beab91e2fc3dddf99d5cc1cacf3fbae Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 20 14:59:40 2010 +0900</p>

<pre>   Test bug fix: confusion between assert and assert_equal, and apparently wrong expected values.

    * Test bug fix: the assert should be assert_equal. The bug was found
      with Ruby 1.9.2-p0.
    * In the test_rates_hundred_and_fiftieth_position method, the index
      for @example_rates and the expected value of the third assertion
      were apparently wrong.
    * Reported by Tomoaki NISHIYAMA during BH2010.10.

test/unit/bio/appl/paml/codeml/test_rates.rb |   12 ++++++------
1 files changed, 6 insertions(+), 6 deletions(-)</pre>

<p>commit ffc03a11a4ef7b36ea78de58d4c8d4e9259093c4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 16 01:06:36 2010 +0900</p>

<pre>   Tests for Bio::KEGG::PATHWAY are improved with new test data.

test/data/KEGG/ec00072.pathway        |   23 +
test/data/KEGG/hsa00790.pathway       |   59 ++
test/data/KEGG/ko00312.pathway        |   16 +
test/data/KEGG/rn00250.pathway        |  114 ++++
test/unit/bio/db/kegg/test_pathway.rb | 1055 +++++++++++++++++++++++++++++++++
5 files changed, 1267 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/ec00072.pathway
create mode 100644 test/data/KEGG/hsa00790.pathway
create mode 100644 test/data/KEGG/ko00312.pathway
create mode 100644 test/data/KEGG/rn00250.pathway</pre>

<p>commit 1e1d974c2c72ddf5a45e41c6f2510729fb65a4ad Author: Toshiaki Katayama
&lt;k@bioruby.org&gt; Date:   Tue Jul 20 11:46:52 2010 +0900</p>

<pre>   Added methods for parsing KEGG PATHWAY fields.

    * Added methods for parsing KEGG PATHWAY fields.
    * This is part of commit e2abe5764f3ded91c82689245f19a0412d3a7afb
      and modified to merge with the current HEAD (original commit
      message: Changes for TogoWS).

lib/bio/db/kegg/pathway.rb |  146 +++++++++++++++++++++++++++++++++++++++++++-
1 files changed, 145 insertions(+), 1 deletions(-)</pre>

<p>commit 957c8ee630538a8c49c52339cb3c0364e5328378 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 16 00:50:42 2010 +0900</p>

<pre>   Private method strings_as_hash is moved to Bio::KEGG::Common::StringsAsHash.

lib/bio/db/kegg/common.rb |   18 ++++++++++++++++++
lib/bio/db/kegg/module.rb |   19 +++++--------------
2 files changed, 23 insertions(+), 14 deletions(-)</pre>

<p>commit b7fd85382bccceaa29958d5daf98ca0a513e5a9a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 15 22:01:05 2010 +0900</p>

<pre>   Renamed Bio::KEGG::*#pathway_modules to modules, etc.

    * Renamed following methods in Bio::KEGG::ORTHOLOGY and
      Bio::KEGG:PATHWAY classes: pathway_modules to modules,
      pathway_modules_as_strings to modules_as_strings, and
      pathway_modules_as_hash to modules_as_hash.
    * Unit tests are also modified.

lib/bio/db/kegg/common.rb               |   18 +++++++++---------
lib/bio/db/kegg/orthology.rb            |   10 +++++-----
lib/bio/db/kegg/pathway.rb              |   10 +++++-----
test/unit/bio/db/kegg/test_orthology.rb |   12 ++++++------
test/unit/bio/db/kegg/test_pathway.rb   |   20 ++++++++++----------
5 files changed, 35 insertions(+), 35 deletions(-)</pre>

<p>commit 02aea9f18ff6e3079309a76d04d02ea1f2902e7b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 15 00:20:48 2010 +0900</p>

<pre>   Modified tests for Bio::KEGG::GENES following the changes of the class.

test/unit/bio/db/kegg/test_genes.rb |   30 +++++++++++++++++++++++++++++-
1 files changed, 29 insertions(+), 1 deletions(-)</pre>

<p>commit f4b45ea629734ecff820483475d83fef6cbe068e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Oct 14 23:57:18 2010 +0900</p>

<pre>   Reverted Bio::KEGG::GENES#genes, gene and motif methods and modified.

    * Reverted Bio::KEGG::GENES#genes, gene and motif methods which are
      removed in the last commit. To avoid code duplication, they are
      also modified to use other methods, and RDoc is added about the
      deprecation or change of the methods.
    * Modified RDoc.

lib/bio/db/kegg/genes.rb |   32 +++++++++++++++++++++++++++++++-
1 files changed, 31 insertions(+), 1 deletions(-)</pre>

<p>commit dd987911cb4a84e23565bb37707611d054c22101 Author: Toshiaki Katayama
&lt;k@bioruby.org&gt; Date:   Tue Jul 20 11:46:52 2010 +0900</p>

<pre>   New methods Bio::KEGG::GENES#keggclass etc.

    * New methods and aliases are added: Bio::KEGG::GENES#keggclass,
      keggclasses, names_as_array, names, motifs_as_strings,
      motifs_as_hash, motifs.
    * Removed Bio::KEGG::GENES#genes, gene and motif methods.
    * Added a comment about deprecation of CODON_USAGE lines.
    * This is part of commit e2abe5764f3ded91c82689245f19a0412d3a7afb
      (original commit message: Changes for TogoWS).

lib/bio/db/kegg/genes.rb |   41 ++++++++++++++++++++++++++---------------
1 files changed, 26 insertions(+), 15 deletions(-)</pre>

<p>commit b2575f5acfeca269c93a35baa3809fdac17a7271 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 13 23:12:33 2010 +0900</p>

<pre>   Release notes for the upcoming release version.

RELEASE_NOTES.rdoc |   31 +++++++++++++++++++++++++++++++
1 files changed, 31 insertions(+), 0 deletions(-)
create mode 100644 RELEASE_NOTES.rdoc</pre>

<p>commit 83992875c45a1fdd54d042c923dee51119026e49 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 13 23:11:21 2010 +0900</p>

<pre>   Renamed RELEASE_NOTES.rdoc to doc/RELEASE_NOTES-1.4.0.rdoc.

RELEASE_NOTES.rdoc           |  167 ------------------------------------------
doc/RELEASE_NOTES-1.4.0.rdoc |  167 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 167 insertions(+), 167 deletions(-)
delete mode 100644 RELEASE_NOTES.rdoc
create mode 100644 doc/RELEASE_NOTES-1.4.0.rdoc</pre>

<p>commit f649629eb6216aeabbd2020bcac9b7f870b12395 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 13 21:58:58 2010 +0900</p>

<pre>   Added acknowledgement to Kozo Nishida for KEGG parsers.

RELEASE_NOTES.rdoc |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 379f177edc0f95dee1ec0c2d2cf679c27918e41b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 8 16:31:58 2010 +0900</p>

<pre>   Fixed a variable name mistake in Bio::Command, and English grammer fix.

    * Fixed a variable name mistake in Bio::Command#no_fork?.
    * English grammer fix for comments. Thanks to Andrew Grimm who
      reports the fix.

lib/bio/command.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 1344c27c90438d8c8840ee507d0ab43224f89054 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 6 23:47:26 2010 +0900</p>

<pre>   Bug fix: fork(2) is called on platforms that do not support it.

    * Bug fix: fork(2) is called on platforms that do not support it.
      Thanks to Andrew Grimm who reports the bug
      (fork() is called on platforms that do not support it;
      http://github.com/bioruby/bioruby/issues#issue/6).
    * Bio::Command#call_command and query_command can now fall back
      into using popen when fork(2) is not implemented.
    * Detection of Windows platform is improved. The idea of the code
      is taken from Redmine's platform.rb.

lib/bio/command.rb                  |   98 +++++++++++++++++++++++++++++------
test/functional/bio/test_command.rb |    9 +--
2 files changed, 84 insertions(+), 23 deletions(-)</pre>

<p>commit 0bfa1c3a8d7b8d03919d54a2a241ca96a79bad83 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 6 15:49:51 2010 +0900</p>

<pre>   Bug fix: Bio::MEDLINE#reference is changed not to put empty values

    * Bug fix: Bio::MEDLINE#reference is changed not to put empty values
      in the returned Bio::Reference object. I think the original bahavior
      is a bug. This is an incompatible change but the effect is very small.

lib/bio/db/medline.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 8e0bc03d79a1f20743c29f0a44e273d362eaa2cd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 6 15:39:34 2010 +0900</p>

<pre>   Bug fix: Bio::MEDLINE#initialize should handle continuation of lines.

    * Bug fix: Bio::MEDLINE#initialize should handle continuation of lines.
      Thanks to Steven Bedrick who reports the bug
      (Bio::MEDLINE#initialize handles multi-line MeSH terms incorrectly;
      http://github.com/bioruby/bioruby/issues#issue/7).

lib/bio/db/medline.rb |    6 +++++-
1 files changed, 5 insertions(+), 1 deletions(-)</pre>

<p>commit 728de78b438108e44066a7ce7490632c81108fb6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Oct 6 15:29:10 2010 +0900</p>

<pre>   Added unit tests for Bio::MEDLINE with test data.

    * Added unit tests for Bio::MEDLINE with test data. The data is
      taken from NCBI and the abstract was removed to avoid possible
      copyright problem. The choice of the data (PMID: 20146148) is
      suggested by Steven Bedrick in a bug report (Bio::MEDLINE#initialize
      handles multi-line MeSH terms incorrectly).

test/data/medline/20146148_modified.medline |   54 ++++++++++
test/unit/bio/db/test_medline.rb            |  148 +++++++++++++++++++++++++++
2 files changed, 202 insertions(+), 0 deletions(-)
create mode 100644 test/data/medline/20146148_modified.medline</pre>

<p>commit 930095817ce60793ac909a4d01731d1f97bc4fa5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Sep 29 20:51:12 2010 +0900</p>

<pre>   Bug fix: NoMethodError in Bio::Tree#collect_edge!

    * Bug fix: NoMethodError in Bio::Tree#collect_edge!.
      Thanks to Kazuhiro Hayashi who reports the bug.

lib/bio/tree.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit c7927ec4743ddc4ec4501790bbed097b69f616e7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Sep 29 20:49:44 2010 +0900</p>

<pre>   Modified and improved tests for Bio::Tree.

test/unit/bio/test_tree.rb |  393 +++++++++++++++++++++++++++-----------------
1 files changed, 240 insertions(+), 153 deletions(-)</pre>

<p>commit 0161148c9b4d9ea404af92b4baf8241239a283de Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Fri Jul 16 00:09:39 2010 +0900</p>

<pre>   Modified unit tests for Bio::Tree

    * Modified unit tests for Bio::Tree.
    * This is part of combination of the two commits:
      * 6675fd930718e41ad009f469b8167f81c9b2ad52
        (Modified unit tests and classes)
      * a6dc63ffe3460ea8d8980b3d6c641356881e0862
        (Modified unit test for Bio::Tree)

test/unit/bio/test_tree.rb |  174 +++++++++++++++++++++++++++++++++++++++++++-
1 files changed, 173 insertions(+), 1 deletions(-)</pre>

<p>commit 31ded691a9329e45fe563e5f70138648d3b30bbf Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jul 15 21:06:28 2010 +0900</p>

<pre>   Bug fix: Bio::Tree#remove_edge_if did not work.

    * Bug fix: Bio::Tree#remove_edge_if did not work.
    * This is part of commit 6675fd930718e41ad009f469b8167f81c9b2ad52
      (original commit message: Modified unit tests and classes)
      and removed a comment line.

lib/bio/tree.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit d0a3af23c74004688a8fc0b5be3d09f7144e33a1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Sep 22 19:23:50 2010 +0900</p>

<pre>   Renamed test/data/go/part_of_* to avoid possible confusion

    * Renamed test/data/go/part_of_* to selected_* to avoid possible
      confusion: The word &quot;part_of&quot; is a keyword in Gene Ontology.

test/data/go/part_of_component.ontology    |   12 ----------
test/data/go/part_of_gene_association.sgd  |   31 ----------------------------
test/data/go/part_of_wikipedia2go          |   13 -----------
test/data/go/selected_component.ontology   |   12 ++++++++++
test/data/go/selected_gene_association.sgd |   31 ++++++++++++++++++++++++++++
test/data/go/selected_wikipedia2go         |   13 +++++++++++
test/unit/bio/db/test_go.rb                |    6 ++--
7 files changed, 59 insertions(+), 59 deletions(-)
delete mode 100644 test/data/go/part_of_component.ontology
delete mode 100644 test/data/go/part_of_gene_association.sgd
delete mode 100644 test/data/go/part_of_wikipedia2go
create mode 100644 test/data/go/selected_component.ontology
create mode 100644 test/data/go/selected_gene_association.sgd
create mode 100644 test/data/go/selected_wikipedia2go</pre>

<p>commit e4f82da52402f8175bd92b50209b09bc83bfddd6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Sep 22 19:21:36 2010 +0900</p>

<pre>   Removed unused test/data/go/wikipedia2go.txt.

test/data/go/wikipedia2go.txt |  728 -----------------------------------------
1 files changed, 0 insertions(+), 728 deletions(-)
delete mode 100644 test/data/go/wikipedia2go.txt</pre>

<p>commit 5003fd53b0a3852fa23b76ad6ec8e9e76d5850fc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Sep 16 22:37:31 2010 +0900</p>

<pre>   Adjusted test data file paths and header lines in test_go.rb.

    * Adjusted test data file paths.
    * Adjusted copyright and description in the header.

test/unit/bio/db/test_go.rb |   26 +++++++++++++++++---------
1 files changed, 17 insertions(+), 9 deletions(-)</pre>

<p>commit 540cb7ab27e79634f5436476cce51cc20ca0f70f Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jul 15 21:06:28 2010 +0900</p>

<pre>   Added tests for Bio::GO classes.

    * Added tests for Bio::GO classes.
    * This is part of combination of the three commits:
      * 555f7b49a43e7c35c82cd48b199af96ca93d4179
        (added test_genbank.rb and test_go.rb with the test files.
         modified test_pdb.rb)
      * e966f17546427b8ad39cb9942807ceb8a068d746
        (modified test/unit/bio/db/test_go.rb and added the test files
         for each GO class)
      * 6675fd930718e41ad009f469b8167f81c9b2ad52
        (Modified unit tests and classes)

test/unit/bio/db/test_go.rb |  163 +++++++++++++++++++++++++++++++++++++++++++
1 files changed, 163 insertions(+), 0 deletions(-)
create mode 100644 test/unit/bio/db/test_go.rb</pre>

<p>commit 5ff01f7dfbc3661d8c66b44874a2ba4ff2f96b56 Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Fri Jun 11 21:02:29 2010 +0900</p>

<pre>   Added test data for Bio::GO classes.

    * Added test data for Bio::GO classes.
    * This is part of combination of the three commits:
      * 555f7b49a43e7c35c82cd48b199af96ca93d4179
        (added test_genbank.rb and test_go.rb with the test files.
         modified test_pdb.rb)
      * e966f17546427b8ad39cb9942807ceb8a068d746
        (modified test/unit/bio/db/test_go.rb and added the test files
         for each GO class)
      * 6675fd930718e41ad009f469b8167f81c9b2ad52
        (Modified unit tests and classes)
    * License for the test data is the public domain.
      ( http://wiki.geneontology.org/index.php/Legal_FAQ )

test/data/go/part_of_component.ontology   |   12 +
test/data/go/part_of_gene_association.sgd |   31 ++
test/data/go/part_of_wikipedia2go         |   13 +
test/data/go/wikipedia2go.txt             |  728 +++++++++++++++++++++++++++++
4 files changed, 784 insertions(+), 0 deletions(-)
create mode 100644 test/data/go/part_of_component.ontology
create mode 100644 test/data/go/part_of_gene_association.sgd
create mode 100644 test/data/go/part_of_wikipedia2go
create mode 100644 test/data/go/wikipedia2go.txt</pre>

<p>commit d4210673a1a696bfb02c93b7743e60dea1a5fcc8 Author: Kazuhiro Hayashi
&lt;k.hayashi.info@gmail.com&gt; Date:   Thu Jul 15 21:06:28 2010 +0900</p>

<pre>   Bug fix: Typo and missing field in Bio::GO::GeneAssociation#to_str.

    * Bug fix: Typo and missing field in Bio::GO::GeneAssociation#to_str.
    * This is part of commit 6675fd930718e41ad009f469b8167f81c9b2ad52
      (original commit message: Modified unit tests and classes)
      and modified. The bug is also reported by Ralf Stephan
      ([BioRuby] [PATCH] GO annotations fixes and improvements).

lib/bio/db/go.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit acab0bb4a4e0f970f8f6be3aea2c371f63a49fa7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 25 22:58:42 2010 +0900</p>

<pre>   Database names used in tests are changed, following the change of TogoWS.

    * Database names used in tests are changed, following the change of
      TogoWS: &quot;gene&quot; to &quot;kegg-genes&quot; and &quot;enzyme&quot; to &quot;kegg-enzyme&quot;.

test/functional/bio/io/test_togows.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 1fb1f1cc5ca3edb42de03874b3527ce0cf0de294 Author: Toshiaki Katayama
&lt;k@bioruby.org&gt; Date:   Tue Jul 20 11:46:52 2010 +0900</p>

<pre>   Database name used in tests is changed, following the change of TogoWS.

    * The database name &quot;genbank&quot; is changed to &quot;nucleotide&quot;, following
      the change in TogoWS.
    * This is part of commit e2abe5764f3ded91c82689245f19a0412d3a7afb.
      (Original commit message: Changes for TogoWS)

test/functional/bio/io/test_togows.rb |   10 +++++-----
1 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit 1db4b8011f4fee158aeb78ec2d76c76688714788 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 11 23:33:49 2010 +0900</p>

<pre>   New method Bio::Fastq#mask for masking low score regions.

    * New method Bio::Fastq#mask for masking low score regions is added
      with unit tests. This method is implemented as a shortcut of
      Bio::Sequence#mask_with_quality_score method.

lib/bio/db/fastq.rb            |   15 +++++++++++++++
test/unit/bio/db/test_fastq.rb |   36 ++++++++++++++++++++++++++++++++++++
2 files changed, 51 insertions(+), 0 deletions(-)</pre>

<p>commit 72b47b2391a01c5f4214fd188abe0857cd3ed166 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 11 23:04:57 2010 +0900</p>

<pre>   New module Bio::Sequence::SequenceMasker to help masking a sequence.

    * New module Bio::Sequence::SequenceMasker to help masking a sequence.
      The module is only expected to be included in Bio::Sequence.
      In the future, methods in this module might be moved to
      Bio::Sequence or other module and this module might be removed.
    * Unit tests for Bio::Sequence::SequenceMasker are also added.

lib/bio/sequence.rb                            |    2 +
lib/bio/sequence/sequence_masker.rb            |   95 +++++++++++++
test/unit/bio/sequence/test_sequence_masker.rb |  169 ++++++++++++++++++++++++
3 files changed, 266 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/sequence/sequence_masker.rb
create mode 100644 test/unit/bio/sequence/test_sequence_masker.rb</pre>

<p>commit a2b21fa31c87fc47ae375380fb34958460414107 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Aug 11 22:59:46 2010 +0900</p>

<pre>   New method Bio::Sequence#output_fasta, a replacement for to_fasta.

    * New method Bio::Sequence#output_fasta, a replacement for
      Bio::Sequence#to_fasta. This is also implemented as a shortcut of
      Bio::Sequence#output(:fasta).

lib/bio/sequence/format.rb |   14 ++++++++++++++
1 files changed, 14 insertions(+), 0 deletions(-)</pre>

<p>commit d139a1e3e7f77317102eaa24649515541761a212 Author: Toshiaki Katayama
&lt;k@bioruby.org&gt; Date:   Tue Jul 20 11:46:52 2010 +0900</p>

<pre>   File format autodetection for Bio::KEGG::PATHWAY and Bio::KEGG::MODULE.

    * Added file format autodetection for Bio::KEGG::PATHWAY and
      Bio::KEGG::MODULE.
    * This is part of commit e2abe5764f3ded91c82689245f19a0412d3a7afb.
      (Original commit message: Changes for TogoWS)

lib/bio/io/flatfile/autodetection.rb |    6 ++++++
1 files changed, 6 insertions(+), 0 deletions(-)</pre>

<p>commit 920d92c13b44921a3f58ddbd8566e7a90dd59996 Author: Toshiaki Katayama
&lt;k@bioruby.org&gt; Date:   Tue Jul 20 11:46:52 2010 +0900</p>

<pre>   Added autoload for Bio::KEGG::PATHWAY and Bio::KEGG::MODULE.

    * Added autoload for Bio::KEGG::PATHWAY and Bio::KEGG::MODULE.
    * This is part of commit e2abe5764f3ded91c82689245f19a0412d3a7afb.
      (Original commit message: Changes for TogoWS)

lib/bio.rb |    2 ++
1 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit bf342c28e0c75c9b48770144f421dd12babd9d0e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 17 23:19:47 2010 +0900</p>

<pre>   Unit tests for Bio::KEGG::MODULE is modified and improved.

test/unit/bio/db/kegg/test_module.rb |  194 ++++++++++++++++++++++++++++++----
1 files changed, 173 insertions(+), 21 deletions(-)</pre>

<p>commit 1742568a4f27e75a19441e4a4437ca3f1c0251f8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 17 23:15:44 2010 +0900</p>

<pre>   In Bio::KEGG::MODULE, an internal-only method is changed to private.

lib/bio/db/kegg/module.rb |    4 ++++
1 files changed, 4 insertions(+), 0 deletions(-)</pre>

<p>commit 8f5ff66cca678ac6be75a7dda1ff840ac3111f42 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 17 23:10:39 2010 +0900</p>

<pre>   Removed unused comments.

lib/bio/db/kegg/module.rb |   32 --------------------------------
1 files changed, 0 insertions(+), 32 deletions(-)</pre>

<p>commit f9d23fb32eeb15dc57580e25653d3f2fff5fa1dc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Aug 17 22:59:18 2010 +0900</p>

<pre>   Reverted Bio::KEGG::MODULE#keggclass.

    * Reverted Bio::KEGG::MODULE#keggclass.
    * Removed keggclasses and keggclasses_as_array methods, because
      they are inconsistent with Bio::KEGG::ORTHOLOGY#keggclasses.

lib/bio/db/kegg/module.rb |    6 +-----
1 files changed, 1 insertions(+), 5 deletions(-)</pre>

<p>commit 94188d23ad843c7cb998c99f46371e540ce457dc Author: Toshiaki Katayama
&lt;k@bioruby.org&gt; Date:   Tue Jul 20 11:41:50 2010 +0900</p>

<pre>   For Bio::KEGG::MODULE, methods are added and modified.

    * For Bio::KEGG::MODULE, methods are added and modified.
      * New methods: definition, etc.
      * Removed methods: pathway, orthologies, keggclass, etc.
      * Changed methods: reactions, compounds, etc.
    * (Original commit message: Changes for TogoWS)

lib/bio/db/kegg/module.rb |  136 +++++++++++++++++++++++++++++++++++++++++---
1 files changed, 126 insertions(+), 10 deletions(-)</pre>

<p>commit 92efc03707a49fa0b2c02e7b2f8b53749a75ad59 Author: Kozo Nishida
&lt;kozo-ni@cg04.naist.jp&gt; Date:   Thu Feb 4 22:59:20 2010 +0900</p>

<pre>   New class Bio::KEGG::MODULE, parser for KEGG MODULE (Pathway Module).

lib/bio/db/kegg/module.rb            |   83 ++++++++++++++++++++++++++++++
test/data/KEGG/M00118.module         |   44 ++++++++++++++++
test/unit/bio/db/kegg/test_module.rb |   94 ++++++++++++++++++++++++++++++++++
3 files changed, 221 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/kegg/module.rb
create mode 100644 test/data/KEGG/M00118.module
create mode 100644 test/unit/bio/db/kegg/test_module.rb</pre>

<p>commit b7c75cc6023d5dc9096111fde99a6e89db2e4bdc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 12 01:13:52 2010 +0900</p>

<pre>   Improvement of tests for Bio::KEGG::ORTHOLOGY using updated test data.

test/unit/bio/db/kegg/test_orthology.rb |   95 +++++++++++++++++++++++++++++++
1 files changed, 95 insertions(+), 0 deletions(-)</pre>

<p>commit f61c232371f6f673044960b0626486b2e8e160b8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 12 01:11:16 2010 +0900</p>

<pre>   Updated test data K02338.orthology to follow KEGG format changes.

test/data/KEGG/K02338.orthology |  232 ++++++++++++++++++++++++++++++---------
1 files changed, 180 insertions(+), 52 deletions(-)</pre>

<p>commit 2aa060f42263392877683a47ef9bd744ef4de7f8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 12 01:03:18 2010 +0900</p>

<pre>   Incompatible change of Bio::KEGG::ORTHOLOGY#pathways, and added new methods

    * Incompatible change of Bio::KEGG::ORTHOLOGY#pathways due to the
      changes of KEGG ORTHOLOGY format changes: Because PATHWAY field
      is added, the method is changed to return a hash. The pathway
      method of old behavior is renamed to pathways_in_keggclass
      for compatibility.
    * New methods are added to Bio::KEGG::ORTHOLOGY: references,
      pathways_as_strings, pathways_as_hash, pathway_modules,
      pathway_modules_as_hash, pathway_modules_as_strings.

lib/bio/db/kegg/orthology.rb |   41 ++++++++++++++++++++++++++++++++++++++++-
1 files changed, 40 insertions(+), 1 deletions(-)</pre>

<p>commit 2e6754f2598f66f29afb573c3dc83592089b411c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 12 00:59:26 2010 +0900</p>

<pre>   Changed to use Bio::KEGG::Common::PathwayModulesAsHash.

lib/bio/db/kegg/pathway.rb |   25 ++++++++-----------------
1 files changed, 8 insertions(+), 17 deletions(-)</pre>

<p>commit 527920da990f4374e20333d6852b810ea73ead02 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed May 12 00:55:10 2010 +0900</p>

<pre>   New module Bio::KEGG::Common::PathwayModulesAsHash (internal use only)

    * New module Bio::KEGG::Common::PathwayModulesAsHash is added,
      based on Bio::KEGG::PATHWAY#pathway_modules_as_hash method.
      Note that the method is Bio::KEGG::* internal use only.

lib/bio/db/kegg/common.rb |   22 ++++++++++++++++++++++
1 files changed, 22 insertions(+), 0 deletions(-)</pre>

<p>commit 36041377dbafce642180eb1c664ee36ef21d3bfb Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Mar 19 23:25:56 2010 +0900</p>

<pre>   New method Bio::KEGG::PATHWAY#references.

    * New method Bio::KEGG::PATHWAY#references.
    * Additional unit tests for Bio::KEGG::PATHWAY with test data.

lib/bio/db/kegg/pathway.rb            |    8 ++
test/data/KEGG/map00030.pathway       |   37 ++++++++++
test/unit/bio/db/kegg/test_pathway.rb |  120 ++++++++++++++++++++++++++++++++-
3 files changed, 162 insertions(+), 3 deletions(-)
create mode 100644 test/data/KEGG/map00030.pathway</pre>

<p>commit d743892d298786eb9e88e2a51ac9f7774785848f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Mar 19 00:06:20 2010 +0900</p>

<pre>   Improvement of Bio::KEGG::Common::References#references.

    * Improvement of Bio::KEGG::Common::References#references:
      added support for parsing &quot;journal (year)&quot; style.

lib/bio/db/kegg/common.rb |    4 ++++
1 files changed, 4 insertions(+), 0 deletions(-)</pre>

<p>commit 35807ae22c9ad9a3ce37ed5c655d1c080f8d2334 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Mar 19 00:02:10 2010 +0900</p>

<pre>   Implementation of Bio::KEGG::GENOME#references is moved.

    * Implementation of Bio::KEGG::GENOME#references is moved to
      Bio::KEGG::Common::References#references, which will be
      shared with Bio::KEGG::Pathway and other classes.

lib/bio/db/kegg/common.rb |   61 +++++++++++++++++++++++++++++++++++++++++++-
lib/bio/db/kegg/genome.rb |   62 +++++---------------------------------------
2 files changed, 67 insertions(+), 56 deletions(-)</pre>

<p>commit 263c37a07203b87e3b33d35adef3aa3ddcf89601 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Mar 17 00:26:56 2010 +0900</p>

<pre>   Bug fix: Bio::KEGG::GENES#pathway may fail, and other parse issues due to the format changes of KEGG GENES.

    * Bug fix: Bio::KEGG::GENES#pathway may return unexpected value
      after calling pathways, pathways_as_hash or pathways_as_string
      methods.
    * Bio::KEGG::GENES#eclinks, Bio::KEGG::Common::PathwaysAsHash,
      and Bio::KEGG::Common::OrthologsAsHash are modified due to
      the file format changes of KEGG::GENES.

lib/bio/db/kegg/common.rb |    9 +++++----
lib/bio/db/kegg/genes.rb  |   17 +++++++++++------
2 files changed, 16 insertions(+), 10 deletions(-)</pre>

<p>commit 364cd405a10d0742091281c5a16b77cb54a8087e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Mar 17 00:25:51 2010 +0900</p>

<pre>   New methods Bio::Location#== and Bio::Locations#==.

lib/bio/location.rb |   39 +++++++++++++++++++++++++++++++++++++++
1 files changed, 39 insertions(+), 0 deletions(-)</pre>

<p>commit 2c7ffd6808e572cf35b82d6e74790447d44d08cc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Mar 17 00:23:00 2010 +0900</p>

<pre>   Improved unit tests for Bio::KEGG::GENES with new test data.

test/data/KEGG/b0529.gene           |   47 +++++++
test/unit/bio/db/kegg/test_genes.rb |  254 ++++++++++++++++++++++++++++++++++-
2 files changed, 300 insertions(+), 1 deletions(-)
create mode 100644 test/data/KEGG/b0529.gene</pre>

<p>commit 764869fd42d1e3f96885b3499844bf4fadde80f1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Mar 17 00:10:51 2010 +0900</p>

<pre>   Bug fix: Bio::KEGG::GENOME parser issues for PLASMID, REFERENCE, and ORIGINAL_DB fields.

    * Bug fix: Fixed parse error for PLASMID fields due to the changes
      of the KEGG GENOME file format. For the bug fix, tag_get and
      tag_cut methods are redefined.
    * Bug fix: Fixed parse error for REFERENCE fields due to the changes
      of the file format.
    * New method Bio::KEGG::GENOME#original_databases is added to get
      ORIGINAL_DB record as an Array of String objects.

lib/bio/db/kegg/genome.rb |   69 +++++++++++++++++++++++++++++++++++++++-----
1 files changed, 61 insertions(+), 8 deletions(-)</pre>

<p>commit 75db7c6c7132f19e212be36d06643a0f48a7df44 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Mar 17 00:09:09 2010 +0900</p>

<pre>   New method Bio::Reference#==.

lib/bio/reference.rb |   24 ++++++++++++++++++++++++
1 files changed, 24 insertions(+), 0 deletions(-)</pre>

<p>commit 64a6bfb52ca1bb27bd38c86c060e2925f38924fb Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Mar 17 00:07:43 2010 +0900</p>

<pre>   Newly added unit tests for Bio::KEGG::GENOME with test data.

test/data/KEGG/T00005.genome         |  140 ++++++++++++
test/data/KEGG/T00070.genome         |   34 +++
test/unit/bio/db/kegg/test_genome.rb |  408 ++++++++++++++++++++++++++++++++++
3 files changed, 582 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/T00005.genome
create mode 100644 test/data/KEGG/T00070.genome
create mode 100644 test/unit/bio/db/kegg/test_genome.rb</pre>

<p>commit 21c92bb991c83dce27a4411382c456cdd6029a82 Author: Naohisa Goto
&lt;ngoto@gen-info.osaka-u.ac.jp&gt; Date:   Tue Mar 9 23:24:15 2010 +0900</p>

<pre>   Renamed Bio::KEGG::PATHWAY#keggmodules to pathway_modules_as_strings, etc.

    * Bio::KEGG::PATHWAY#keggmodules is renamed to
      pathway_modules_as_strings.
    * New method pathway_modules_as_hash and its alias method
      pathway_modules is added.
    * Bio::KEGG::PATHWAY#rel_pathways is renamed to
      rel_pathways_as_strings.
    * New method rel_pathways_as_hash is added, and rel_pathways
      is changed to be the alias of the rel_pathways_as_hash method.
    * Unit tests are also changed.

lib/bio/db/kegg/pathway.rb            |   42 +++++++++++++++++++++++++++++++-
test/unit/bio/db/kegg/test_pathway.rb |   29 ++++++++++++++++++++--
2 files changed, 66 insertions(+), 5 deletions(-)</pre>

<p>commit fa10f38716ec2eecd6fa8e8b027085377e9ee421 Author: Naohisa Goto
&lt;ngoto@gen-info.osaka-u.ac.jp&gt; Date:   Tue Mar 9 21:13:18 2010 +0900</p>

<pre>   Fixed text indentations.

lib/bio/db/kegg/pathway.rb            |    4 ++--
test/unit/bio/db/kegg/test_pathway.rb |   25 +++++++++++++------------
2 files changed, 15 insertions(+), 14 deletions(-)</pre>

<p>commit 0916b0cac5d17ce47ef5cc3382e3167293bcf4c2 Author: Kozo Nishida
&lt;kozo-ni@cg04.naist.jp&gt; Date:   Tue Feb 2 17:34:17 2010 +0900</p>

<pre>   Newly added Bio::KEGG::PATHWAY with test code and test data.

lib/bio/db/kegg/pathway.rb            |   73 +++++++++++++++++++++++++++++++++
test/data/KEGG/map00052.pathway       |   13 ++++++
test/unit/bio/db/kegg/test_pathway.rb |   57 +++++++++++++++++++++++++
3 files changed, 143 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/kegg/pathway.rb
create mode 100644 test/data/KEGG/map00052.pathway
create mode 100644 test/unit/bio/db/kegg/test_pathway.rb</pre>

<p>commit c3ceea339164754071f03ce13da4f65e08230f40 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Feb 19 00:43:38 2010 +0900</p>

<pre>   Tutorial.rd.html is regenerated.

doc/Tutorial.rd.html |   55 +++++++++++++++++++++++++++++++++++++++++++++++++-
1 files changed, 54 insertions(+), 1 deletions(-)</pre>

<p>commit 315da0213edfece696d22cc4648cb7a74f18ad34 Author: Ra
&lt;ilpuccio.febo@gmail.com&gt; Date:   Sun Feb 7 10:38:36 2010 +0100</p>

<pre>   Added BioSQL docs links

doc/Tutorial.rd |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 22374415873906f4bcd3e84950c14b5f0b6c7e61 Author: Ra
&lt;ilpuccio.febo@gmail.com&gt; Date:   Sun Feb 7 02:39:16 2010 +0100</p>

<pre>   Added link to BioSQL install doc.

doc/Tutorial.rd |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 4d18dd2f5a3f18348e5f4aa07b14c104d3a65f5b Author: Ra
&lt;ilpuccio.febo@gmail.com&gt; Date:   Fri Feb 5 21:30:36 2010 +0100</p>

<pre>   Added other examples about BioSQL

doc/Tutorial.rd |   36 ++++++++++++++++++++++++++++--------
1 files changed, 28 insertions(+), 8 deletions(-)</pre>

<p>commit b704f01cd0799ab1a7e3975119e9d6139ddfbd51 Author: Ra
&lt;ilpuccio.febo@gmail.com&gt; Date:   Wed Jan 27 21:08:25 2010 +0100</p>

<pre>   BioSQL tutorial continue...

doc/Tutorial.rd |   18 ++++++++++++++++--
1 files changed, 16 insertions(+), 2 deletions(-)</pre>

<p>commit 1993a1566b5ade937703d0291c4eaf2de673d170 Author: Ra
&lt;ilpuccio.febo@gmail.com&gt; Date:   Wed Jan 27 20:32:23 2010 +0100</p>

<pre>   BioSQL tutorial inital draft.

doc/Tutorial.rd |   25 ++++++++++++++++++++++++-
1 files changed, 24 insertions(+), 1 deletions(-)</pre>

<p>commit 09047b664a03492d7546d92b619faacee72d0cd5 Author: Jan Aerts
&lt;jan.aerts@gmail.com&gt; Date:   Sun Feb 7 17:58:59 2010 +0900</p>

<pre>   Added code example that will serve as basis for sequence/codon howto

doc/howtos/sequence_codon.txt |   38 ++++++++++++++++++++++++++++++++++++++
1 files changed, 38 insertions(+), 0 deletions(-)
create mode 100644 doc/howtos/sequence_codon.txt</pre>

<p>commit c1e2165ba801cccd52135b13ed36713517e1fa8a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Feb 5 12:50:38 2010 +0900</p>

<pre>   Suppressed &quot;warning: parenthesize argument(s) for future version&quot; in Ruby 1.8.5.

lib/bio/appl/paml/codeml/report.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit f7ce9ba6a2f4e680ee40017a21aa95d05baf34f4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Feb 4 20:26:00 2010 +0900</p>

<pre>   Added :startdoc: and removed an empty line for RDoc.

lib/bio/appl/paml/codeml/report.rb |    4 +++-
1 files changed, 3 insertions(+), 1 deletions(-)</pre>

<p>commit 6d4d2e1f37efb1e53091fbc9a0977568996788ff Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Feb 4 16:58:10 2010 +0900</p>

<pre>   New unit test for Bio::PAML::Codeml::Report.

    * New unit test for Bio::PAML::Codeml::Report and related classes.
      The test code is copied from the examples described in
      lib/bio/appl/paml/codeml/report.rb and modified for the unit test.

test/unit/bio/appl/paml/codeml/test_report.rb |  253 +++++++++++++++++++++++++
1 files changed, 253 insertions(+), 0 deletions(-)
create mode 100644 test/unit/bio/appl/paml/codeml/test_report.rb</pre>

<p>commit 8418549811293c3e20b91d4e95da2cb2a282a064 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Feb 4 16:47:37 2010 +0900</p>

<pre>   Changes due to the rename to report_single.rb.

.../bio/appl/paml/codeml/test_report_single.rb     |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 7bfb428da237709b243c8d0e4646bd41710d1519 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Feb 4 16:39:10 2010 +0900</p>

<pre>   Renamed codeml/test_report.rb to codeml/test_report_single.rb.

    * Renamed test/unit/bio/appl/paml/codeml/test_report.rb to
      test_report_single.rb.

test/unit/bio/appl/paml/codeml/test_report.rb      |   46 --------------------
.../bio/appl/paml/codeml/test_report_single.rb     |   46 ++++++++++++++++++++
2 files changed, 46 insertions(+), 46 deletions(-)
delete mode 100644 test/unit/bio/appl/paml/codeml/test_report.rb
create mode 100644 test/unit/bio/appl/paml/codeml/test_report_single.rb</pre>

<p>commit 762d38b1564da7d846e3dcd461cf465aa685a1ae Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Tue Jan 12 10:13:35 2010 +0100</p>

<pre>   Modified output of Bio::PAML::Codeml::PositiveSites#graph_to_s

    * Modified output of Bio::PAML::Codeml::PositiveSites#graph_to_s.
      (Part of commit ea350da85e5db2ba35cb8dd1e86e3d4323ee3fd1.
       Original commit message is:
       HtmlPaml: fixed some missing output
             use real greek omega in output)

lib/bio/appl/paml/codeml/report.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit f88645cd783b7027950133c0badb0a8da8e4fb95 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Tue Jan 12 09:24:46 2010 +0100</p>

<pre>   Codeml: no negative gaps

lib/bio/appl/paml/codeml/report.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 978b21cf90d0280e6e6c7d6e4fa65c49692bdd69 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 17:31:45 2010 +0100</p>

<pre>   Codeml: always raise an error when significance can not be calculated

lib/bio/appl/paml/codeml/report.rb |   15 ++++++++++-----
1 files changed, 10 insertions(+), 5 deletions(-)</pre>

<p>commit 12b5895f6f1819252d616bb0a38aa88a7828daff Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 17:22:34 2010 +0100</p>

<pre>   Codeml: oops

lib/bio/appl/paml/codeml/report.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit a8ff0a07fdbef72f72103f0bceb9c24a63162fc6 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 17:19:26 2010 +0100</p>

<pre>   Codeml: added significance testing for a few model combinations

lib/bio/appl/paml/codeml/report.rb |   57 +++++++++++++++++++++++++++++++++++-
1 files changed, 56 insertions(+), 1 deletions(-)</pre>

<p>commit 0e11af19450faca3568f89b23d5bd764688f75c0 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 16:24:51 2010 +0100</p>

<pre>   Codeml: raise error instead of a 'nil' error when buffer is incomplete

lib/bio/appl/paml/codeml/report.rb |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 1cb2aaaa701a1613812dd479201d27c1d7dcf016 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 14:37:52 2010 +0100</p>

<pre>   Bio::PAML::Codeml::PositiveSites#graph_to_s gets fill character

    * Bio::PAML::Codeml::PositiveSites#graph_to_s gets fill character
      as an argument.
      (Part of commit d67259c9f203dc92c68ad04b4112329a7093a259.
       Original commit message is:
       HtmlPaml: show colors for probabilities of positive selection)

lib/bio/appl/paml/codeml/report.rb |   15 +++++++++------
1 files changed, 9 insertions(+), 6 deletions(-)</pre>

<p>commit ae5b9cf9ee697cc237c77335b12b57709a0e7a46 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 13:37:52 2010 +0100</p>

<pre>   Codeml: return correct buffer

lib/bio/appl/paml/codeml/report.rb |   10 +++++-----
1 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit 44f2e28c3e0d382505b067ec3c7aa55cbb9f0a38 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 13:10:40 2010 +0100</p>

<pre>   Improvement of Bio::PAML::Codeml::PositiveSites#initialize, etc

    * Improved target analysis location detection in
      Bio::PAML::Codeml::PositiveSites#initialize.
    * Changed description inside Bio::PAML::Codeml::Report#nb_sites
      and sites methods.
    * This is part of commit e88ff474748b3295a8a4089356d3086638200d64.
      (Original commit message: HtmlPaml: improved output)

lib/bio/appl/paml/codeml/report.rb |   20 ++++++++++++--------
1 files changed, 12 insertions(+), 8 deletions(-)</pre>

<p>commit ee8973696d0434c591ceaffc580f1aa30fd036f9 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 12:54:40 2010 +0100</p>

<pre>   Codeml: fixed doctests

lib/bio/appl/paml/codeml/report.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit f7bbb0859e28cb51137d0a8f8d962821eb67db91 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 12:51:25 2010 +0100</p>

<pre>   New method Bio::PAML::Codeml::PositiveSites#to_s

    * New method Bio::PAML::Codeml::PositiveSites#to_s
      (part of the commit 82e933fd1961a2b31873bc37cbf3205adbf0a6de,
       original commit message: HtmlPaml: add facility for color output)

lib/bio/appl/paml/codeml/report.rb |    6 ++++++
1 files changed, 6 insertions(+), 0 deletions(-)</pre>

<p>commit d4f3dbaf78f623d870a1a76ab1353d786e0fb73b Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 12:34:59 2010 +0100</p>

<pre>   Codeml: HtmlPaml: minor tweaks

lib/bio/appl/paml/codeml/report.rb |    7 ++-----
1 files changed, 2 insertions(+), 5 deletions(-)</pre>

<p>commit 7d41b6acb41c5913622fde127a030f940a432cc5 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 12:19:15 2010 +0100</p>

<pre>   Codeml: add short description to positive sites line

lib/bio/appl/paml/codeml/report.rb |   13 +++++++++++++
1 files changed, 13 insertions(+), 0 deletions(-)</pre>

<p>commit a9d6765b3a5d23be7e8cf59954d67cd2354e5878 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 12:13:55 2010 +0100</p>

<pre>   Codeml: fixed bug in graph output

lib/bio/appl/paml/codeml/report.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit af124dcaa2adb446e273456f8dd0b84aff9b00db Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 11 12:11:00 2010 +0100</p>

<pre>   Codeml: added graph_seq, which shows the AA of the first sequence at positive sites

lib/bio/appl/paml/codeml/report.rb |   11 +++++++++--
1 files changed, 9 insertions(+), 2 deletions(-)</pre>

<p>commit 1924dcd951a9e655726bc1af72626526ed223258 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Jan 8 10:13:37 2010 +0100</p>

<pre>   Codeml: added :stopdoc: directive for rdoc

lib/bio/appl/paml/codeml/report.rb |   21 ++++++++++++---------
1 files changed, 12 insertions(+), 9 deletions(-)</pre>

<p>commit 7b68fd9d785723935f90144f67999fcf74bcc7c0 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Jan 8 10:01:49 2010 +0100</p>

<pre>   Codeml: fixed the doctests and added some info.
           all tests pass &amp;

lib/bio/appl/paml/codeml/report.rb |   43 ++++++++++++++++++++++++-----------
1 files changed, 29 insertions(+), 14 deletions(-)</pre>

<p>commit ce212c507e9e81120e7ad12be6df955e90d0ad33 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Jan 8 09:46:05 2010 +0100</p>

<pre>   Codeml: exclude TestFile class from RDoc generated documentation

lib/bio/appl/paml/codeml/report.rb |    9 ++++++---
1 files changed, 6 insertions(+), 3 deletions(-)</pre>

<p>commit 9d64a70b628cb3cda9cc1576b9966dae242e5230 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Jan 8 09:39:02 2010 +0100</p>

<pre>   Codeml: added many comments

lib/bio/appl/paml/codeml/report.rb |   87 +++++++++++++++++++++++++++++------
1 files changed, 72 insertions(+), 15 deletions(-)</pre>

<p>commit 2b92df64016865a2ab40c93650409cfd67a2a98e Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 4 17:57:32 2010 +0100</p>

<pre>   codeml: Added parser for full Bayesian sites
           all tests pass &amp;

lib/bio/appl/paml/codeml/report.rb |   52 ++++++++++++++++++++++++++++++-----
1 files changed, 44 insertions(+), 8 deletions(-)</pre>

<p>commit 198f0c014f7993fbe22825d0be67e9b1aa19d2de Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 4 17:37:15 2010 +0100</p>

<pre>   codeml: show graph

lib/bio/appl/paml/codeml/report.rb |   46 ++++++++++++++++++++++++++++++++++-
1 files changed, 44 insertions(+), 2 deletions(-)</pre>

<p>commit 0843bb4d79dc4d94a22d79a797a39dc2866222c5 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 4 17:09:27 2010 +0100</p>

<pre>   Codeml: added full support for positive selection sites
           doctests + unit tests pass &amp;

lib/bio/appl/paml/codeml/report.rb |  150 ++++++++++++++++++++++++++++--------
1 files changed, 118 insertions(+), 32 deletions(-)</pre>

<p>commit 1610c5d86cddd53f7f0300d0f7a137daaa61ef94 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 4 12:33:20 2010 +0100</p>

<pre>   codeml: added M3 classes

lib/bio/appl/paml/codeml/report.rb |   28 +++++++++++++++++++++++++---
1 files changed, 25 insertions(+), 3 deletions(-)</pre>

<p>commit efc939fbd300ceb371b342172382cdec9fcc74b7 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Mon Jan 4 12:02:47 2010 +0100</p>

<pre>   codeml: adding compatibility layer for single model (old type)
           unit tests pass &amp;

lib/bio/appl/paml/codeml/report.rb |   44 +++++++++++++++++++++++++++++++-----
1 files changed, 38 insertions(+), 6 deletions(-)</pre>

<p>commit b89990daa3ea26a9f9195ec16044fb2070bcdd1a Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sun Jan 3 12:34:53 2010 +0100</p>

<pre>   Implementation parsing one model - doctests for M0 pass

lib/bio/appl/paml/codeml/report.rb |   90 ++++++++++++++++++++++++++++++++----
1 files changed, 81 insertions(+), 9 deletions(-)</pre>

<p>commit dce447d3e81e738323e6fb6b2d28324e1fa62e7d Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sun Jan 3 11:27:54 2010 +0100</p>

<pre>   Codeml: use BioTestFile for locating test data in the doctest

lib/bio/appl/paml/codeml/report.rb |    9 ++++++++-
1 files changed, 8 insertions(+), 1 deletions(-)</pre>

<p>commit 27a7b558d60b7ec127df2c351542433c321704ac Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sat Jan 2 23:36:47 2010 +0100</p>

<pre>   Codeml: split new type report and old type report

lib/bio/appl/paml/codeml/report.rb |   19 +++++++++++++++----
1 files changed, 15 insertions(+), 4 deletions(-)</pre>

<p>commit 027808e4723ca77af3e15b461ddcc09faf692732 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sat Jan 2 23:25:22 2010 +0100</p>

<pre>   Added example files for PAML codeml dual model runs

test/data/paml/codeml/models/aa.aln         |   26 ++
test/data/paml/codeml/models/aa.dnd         |   13 +
test/data/paml/codeml/models/aa.ph          |   13 +
test/data/paml/codeml/models/alignment.phy  |   49 ++++
test/data/paml/codeml/models/results0-3.txt |  312 ++++++++++++++++++++++++
test/data/paml/codeml/models/results7-8.txt |  340 +++++++++++++++++++++++++++
6 files changed, 753 insertions(+), 0 deletions(-)
create mode 100644 test/data/paml/codeml/models/aa.aln
create mode 100644 test/data/paml/codeml/models/aa.dnd
create mode 100644 test/data/paml/codeml/models/aa.ph
create mode 100644 test/data/paml/codeml/models/alignment.phy
create mode 100644 test/data/paml/codeml/models/results0-3.txt
create mode 100644 test/data/paml/codeml/models/results7-8.txt</pre>

<p>commit 1d35e616ce411bf643ab6dcb7126a6e1aca1e186 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sat Jan 2 17:04:56 2010 +0100</p>

<pre>   Codeml::Report Added new description and reference

lib/bio/appl/paml/codeml/report.rb |  113 ++++++++++++++++++++++++++++++------
1 files changed, 96 insertions(+), 17 deletions(-)</pre>

<p>commit d21b26044e776fab44dbc95f181afd04b67abe28 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Feb 1 22:31:21 2010 +0900</p>

<pre>   Bug fix and Ruby 1.9 support: Bio::Command.call_command_fork etc.

    * Bug fix: In Bio::Command.call_command_fork, thread switching is
      disabled in the child process. Thanks to Andrew Grimm who
      reports the bug ([BioRuby] Thread-safety of alignment).
      Note that call_command_fork no longer works in Ruby 1.9 because
      it is changed to use Thread.critical which is removed in Ruby 1.9.
    * Ruby 1.9 support: In Ruby 1.9, Bio::Command.call_command_popen
      bypasses shell execution by passing command-line as an Array,
      which is a new feature added in Ruby 1.9. Now, call_command_popen
      is safe and robust enough with Ruby 1.9.
    * Ruby 1.9 support: In Ruby 1.9, Bio::Command.call_command and
      query_command use call_command_popen and query_command_popen,
      respectively.
    * RDoc for the above and related methods are modified.

lib/bio/command.rb                  |   80 +++++++++++++++++++++++++++++++----
test/functional/bio/test_command.rb |    4 ++
2 files changed, 76 insertions(+), 8 deletions(-)</pre>

<p>commit 981dc1c89049bf00e56a9e83ef352cb4c4b45d6a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Feb 2 22:47:36 2010 +0900</p>

<pre>   Bug fix: Bio::FastaNumericFormat#to_biosequence bug fix

    * Bug fix: New method Bio::FastaNumericFormat#to_biosequence is
      defined to avoid NomethodError occurred in the superclass'es
      method. For the purpose, a new module
      Bio::Sequence::Adapter::FastaNumericFormat is added.
      Thanks to Hiroyuki MISHIMA who reports the bug ([BioRuby] trouble
      on the FASTA.QUAL format (Bio::FastaNumericFormat)).
    * Newly added unit test for Bio::FastaNumericFormat#to_biosequence.

lib/bio/db/fasta/qual.rb                |   24 ++++++++++++++++++++++++
lib/bio/db/fasta/qual_to_biosequence.rb |   29 +++++++++++++++++++++++++++++
lib/bio/sequence/adapter.rb             |    1 +
test/unit/bio/db/test_qual.rb           |   11 +++++++++--
4 files changed, 63 insertions(+), 2 deletions(-)
create mode 100644 lib/bio/db/fasta/qual_to_biosequence.rb</pre>

<p>commit 29ed6870e453f54aac2ce9dcb7891186eb01c40d Author: Ben J Woodcroft
&lt;ben@reefedge.sols.uq.edu.au&gt; Date:   Wed Jan 13 14:38:13 2010 +1000</p>

<pre>   Bug fix: fixed uniprot GN parsing issue

lib/bio/db/embl/sptr.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit a3002a79ec012559f5847ba8ebe4faf6e7fa609e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Jan 8 22:14:15 2010 +0900</p>

<pre>   Tutorial.rd.html is regenerated.

doc/Tutorial.rd.html |   29 ++++++++++++++---------------
1 files changed, 14 insertions(+), 15 deletions(-)</pre>

<p>commit 9238c3cb0e8f1156d23a5dfb3ce4e299a91b9f23 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Jan 8 09:04:23 2010 +0100</p>

<pre>   Tutorial: removed bad links

doc/Tutorial.rd |   10 +---------
1 files changed, 1 insertions(+), 9 deletions(-)</pre>

<p>commit 60542fd9863c5fc1240a15cc76f8fa90644a15c8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jan 6 20:38:25 2010 +0900</p>

<pre>   Changed header and the depth of loading helper due to the rename.

test/unit/bio/appl/clustalw/test_report.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 68924e736df76fe3c77d9fe132b6df01fc0621fe Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jan 6 20:34:10 2010 +0900</p>

<pre>   Renamed test/unit/bio/db/test_clustalw.rb to test/unit/bio/appl/clustalw/test_report.rb.

test/unit/bio/appl/clustalw/test_report.rb |   61 ++++++++++++++++++++++++++++
test/unit/bio/db/test_clustalw.rb          |   61 ----------------------------
2 files changed, 61 insertions(+), 61 deletions(-)
create mode 100644 test/unit/bio/appl/clustalw/test_report.rb
delete mode 100644 test/unit/bio/db/test_clustalw.rb</pre>

<p>commit 8368eee50de51f6218ffc7b1bf1aad332702c4ba Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Tue Jan 5 12:54:43 2010 +0100</p>

<pre>   Clustal: unit tests according to Naohisa

lib/bio/appl/clustalw/report.rb   |    6 +++---
test/unit/bio/db/test_clustalw.rb |   10 +++++-----
2 files changed, 8 insertions(+), 8 deletions(-)</pre>

<p>commit ad525a01fa17052e9b7e9b7f30639c48596552ba Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Tue Jan 5 12:50:17 2010 +0100</p>

<pre>   Clustal: unit test uses File.read

test/unit/bio/db/test_clustalw.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 7ab517e05cc470b9ca57273092599adb8c00dc11 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Tue Jan 5 12:49:21 2010 +0100</p>

<pre>   Clustal: unit test, changed class name and copyright header

test/unit/bio/db/test_clustalw.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 0829ee91a97976eb6671a2feec7edfc524f44b2c Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Tue Jan 5 12:46:37 2010 +0100</p>

<pre>   Clustal: Changed [] to get_sequence, with method description

    * Clustal: Added copyright.
    * Changed [] to get_sequence, with method description.

lib/bio/appl/clustalw/report.rb |   12 ++++++++++--
1 files changed, 10 insertions(+), 2 deletions(-)</pre>

<p>commit 3b8968a6b7b98e0f03b0822849594262a8f4ac99 Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sun Dec 27 16:44:30 2009 +0100</p>

<pre>   ClustalW: Added [] method to reach sequence + definition

lib/bio/appl/clustalw/report.rb   |    9 +++++++++
test/unit/bio/db/test_clustalw.rb |    6 ++----
2 files changed, 11 insertions(+), 4 deletions(-)</pre>

<p>commit 3926fabbcc0636c6e4ed08233af3d647c620cd5b Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sun Dec 27 16:22:38 2009 +0100</p>

<pre>   ClustalW: Add ALN parser unit test

test/data/clustalw/example1.aln   |   58 ++++++++++++++++++++++++++++++++++
test/unit/bio/db/test_clustalw.rb |   63 +++++++++++++++++++++++++++++++++++++
2 files changed, 121 insertions(+), 0 deletions(-)
create mode 100644 test/data/clustalw/example1.aln
create mode 100644 test/unit/bio/db/test_clustalw.rb</pre>

<p>commit 2ef97f945b122dc279eb0ec0a34a2adb0c5f0cff Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sat Jan 2 13:24:33 2010 +0100</p>

<pre>   Tutorial: Fixed URLs

doc/Tutorial.rd |   10 +++++-----
1 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit 567ca8b010e15cbea9398ee74c78eae01fc6671d Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Fri Jan 1 12:08:50 2010 +0100</p>

<pre>   Tutorial: Added info on gem install

doc/Tutorial.rd |    9 +++++++--
1 files changed, 7 insertions(+), 2 deletions(-)</pre>

<p>commit 21070ab4928d9c7446d58f3003d43ee6235046aa Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Thu Dec 31 11:41:54 2009 +0100</p>

<pre>   Tutorial.rd: Added Naohisa's Ruby replacement for sed conversion

doc/Tutorial.rd |    6 +++++-
1 files changed, 5 insertions(+), 1 deletions(-)</pre>

<p>commit ebded2364f716fa03b0fdbec9887f807836eb789 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Jan 6 10:59:39 2010 +0900</p>

<pre>   Bio::BIORUBY_EXTRA_VERSION is changed to &quot;.5000&quot;.

bioruby.gemspec    |    2 +-
lib/bio/version.rb |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit a1bda9088662edec55af0106b4292c39e51c8b7b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 21:56:33 2009 +0900</p>

<pre>   BioRuby 1.4.0 is released.

ChangeLog       |   32 ++++++++++++++++++++++++++++++++
bioruby.gemspec |    3 ++-
2 files changed, 34 insertions(+), 1 deletions(-)</pre>

<p>commit 5c88896357e1eff0686ceb06cbec0a7837f85050 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 21:55:41 2009 +0900</p>

<pre>   Preparation for bioruby-1.4.0 release.

lib/bio/version.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 11f56d3d8efc2cf5d9408da865af044fa099b925 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 21:52:25 2009 +0900</p>

<pre>   Added about ChangeLog which is replaced by git-log.

RELEASE_NOTES.rdoc |    5 +++++
1 files changed, 5 insertions(+), 0 deletions(-)</pre>

<p>commit 17d5b1825b6c73d710d72903d8710caa9996353a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 20:11:49 2009 +0900</p>

<pre>   ChangeLog is autogenerated from git log.

    * ChangeLog is autogenerated from git log with the following command:
       % git log --stat --summary \
             3d1dfcc0e13ad582b9c70c7fdde3a89d0bacdc80..HEAD &gt; ChangeLog

ChangeLog | 2306 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 2306 insertions(+), 0 deletions(-)
create mode 100644 ChangeLog</pre>

<p>commit 02bf77af589ea62df81e9634df6fe949df2fd3ef Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 19:25:39 2009 +0900</p>

<pre>   test_output_size is disabled because it depends on html decorations

test/functional/bio/appl/test_pts1.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 5781fb402e85e73fd47948b4466c8129355b714b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 19:21:21 2009 +0900</p>

<pre>   The PTS1 Predictor's URL had been changed.

    * The PTS1 Predictor's URL had been changed.
    * Changed to use @uri instead of @host and @cgi_path.

lib/bio/appl/pts1.rb |    6 ++----
1 files changed, 2 insertions(+), 4 deletions(-)</pre>

<p>commit a4e691d913e1ae51eadb1a871efc2c8718ef5587 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 18:33:00 2009 +0900</p>

<pre>   Preparation of ChangeLog autogeneration: old ChangeLog is moved to doc/ChangeLog-before-1.3.1.

ChangeLog                  | 3961 --------------------------------------------
doc/ChangeLog-before-1.3.1 | 3961 ++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 3961 insertions(+), 3961 deletions(-)
delete mode 100644 ChangeLog
create mode 100644 doc/ChangeLog-before-1.3.1</pre>

<p>commit c011604766baa3cdf5ca2f4a776aa67c56460d29 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 17:53:51 2009 +0900</p>

<pre>   Tutorial.rd.html is regenerated.

doc/Tutorial.rd.html |   70 +++++++++++++------------------------------------
1 files changed, 19 insertions(+), 51 deletions(-)</pre>

<p>commit 6e2cdd13d61970aa4704475bfb5aefb70719c2e1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 17:42:25 2009 +0900</p>

<pre>   Added Bio::NCBI.default_email= in the example, and examples using deprecated Bio::PubMed methods are temporarily commented out.

doc/Tutorial.rd |    8 ++++++++
1 files changed, 8 insertions(+), 0 deletions(-)</pre>

<p>commit 8e6d5e9baf98be7e58f4dda8b5d043a42149874b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 28 17:15:09 2009 +0900</p>

<pre>   Reinserted &quot;==&gt;&quot; for Blast example, and removed duplicated Ruby Ensembl API example.

doc/Tutorial.rd |   25 ++-----------------------
1 files changed, 2 insertions(+), 23 deletions(-)</pre>

<p>commit 849edd7e8c5b26923cab47e7f5542948fab2b1fb Author: Pjotr Prins
&lt;pjotr.public01@thebird.nl&gt; Date:   Sun Dec 27 09:49:14 2009 +0100</p>

<pre>   Tutorial: Added info on how to run rubydoctest
             Removed bioruby&gt; prefix for one failing BLAST test

doc/Tutorial.rd |   69 ++++++++++++++++++++++++++++++++++++++----------------
1 files changed, 48 insertions(+), 21 deletions(-)</pre>

<p>commit a39fcf0ca1a5265789110f42cc616fc5d3c16414 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 25 12:30:18 2009 +0900</p>

<pre>   Modified for release notes and fixed typo.

RELEASE_NOTES.rdoc |   29 +++++++++++++++--------------
1 files changed, 15 insertions(+), 14 deletions(-)</pre>

<p>commit 3fa8b68f19fc2b6aaf8f54eb10517cc761b2193b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 25 12:10:34 2009 +0900</p>

<pre>   Changes following the rename to RELEASE_NOTES.rdoc.

README.rdoc         |    2 +-
bioruby.gemspec     |    6 +++---
bioruby.gemspec.erb |    2 +-
3 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit fd692a1165d368b9bdbe068ea6bf63fd91c9925c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 25 12:03:41 2009 +0900</p>

<pre>   Renamed doc/Changes-1.4.rdoc to RELEASE_NOTES.rdoc.

RELEASE_NOTES.rdoc   |  160 ++++++++++++++++++++++++++++++++++++++++++++++++++
doc/Changes-1.4.rdoc |  160 --------------------------------------------------
2 files changed, 160 insertions(+), 160 deletions(-)
create mode 100644 RELEASE_NOTES.rdoc
delete mode 100644 doc/Changes-1.4.rdoc</pre>

<p>commit 0e37f04dd8d34517693fdd4bc27f8bdada7c2f13 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 21:48:52 2009 +0900</p>

<pre>   Changed Bio::PhyloXML::Parser.new to open, and regenerated html.

doc/Tutorial.rd      |   10 ++--
doc/Tutorial.rd.html |  125 ++++++++++++++++++++++++++++++++++++++++++-------
2 files changed, 112 insertions(+), 23 deletions(-)</pre>

<p>commit aeacbbd425c2e88369c171bd92c60bf8e520a9e5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 19:26:49 2009 +0900</p>

<pre>   bioruby.gemspec is regenerated

bioruby.gemspec |    8 +++++++-
1 files changed, 7 insertions(+), 1 deletions(-)</pre>

<p>commit 1034205c199a638c359780922293f8b39c467356 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 19:24:56 2009 +0900</p>

<pre>   Version number changed to 1.4.0-rc1

lib/bio/version.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 04bf2da43f78fbb702b67323f3be1fe3bd2d0351 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 19:22:41 2009 +0900</p>

<pre>   Issues added and modified.

KNOWN_ISSUES.rdoc |   35 +++++++++++++++++++++++++++++++++--
1 files changed, 33 insertions(+), 2 deletions(-)</pre>

<p>commit f1a76157b009fb0ca94d9a0e0f8a85522c383b19 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 19:22:19 2009 +0900</p>

<pre>   Added news and incompatible changes.

doc/Changes-1.4.rdoc |  102 ++++++++++++++++++++++++++++++++++++++++++++++++--
1 files changed, 98 insertions(+), 4 deletions(-)</pre>

<p>commit 9c8ef18a20c49f17d5b89aa1db5819b2c8ee9b1d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 19:10:02 2009 +0900</p>

<pre>   Email address for NCBI Entrez is given with Bio::NCBI.default_email=.

bin/bioruby                           |    5 ++++-
sample/demo_ncbi_rest.rb              |    2 ++
sample/demo_pubmed.rb                 |    2 ++
sample/pmfetch.rb                     |    2 ++
sample/pmsearch.rb                    |    2 ++
test/functional/bio/io/test_pubmed.rb |    4 ++++
6 files changed, 16 insertions(+), 1 deletions(-)</pre>

<p>commit 7a7179665694da35ab0970909bfbda9ad1b057da Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 19:09:09 2009 +0900</p>

<pre>   Changed autoload hierarchy of Bio::NCBI.

lib/bio.rb             |   10 ++++++----
lib/bio/io/ncbisoap.rb |    3 ++-
2 files changed, 8 insertions(+), 5 deletions(-)</pre>

<p>commit f8dc0268d9edf699fd3f0cf18dd55a2b10ec3bcc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 24 18:58:18 2009 +0900</p>

<pre>   New singleton methods Bio::NCBI.default_email=, default_tool=, etc.

    * New singleton methods Bio::NCBI.default_email=, default_email,
      default_tools=, default_tools, etc., because email and tool
      parameters will be mandatory in Entrez eUtils.
    * Changed to raise error when email or tool is empty. Note that
      default email is nil and library users should always set their
      email address.
    * Default tool name is changed to include $0 and bioruby version ID.
    * Added multi-thread support for Bio::NCBI::REST#ncbi_access_wait.

lib/bio/io/ncbirest.rb |  161 ++++++++++++++++++++++++++++++++++++++---------
1 files changed, 130 insertions(+), 31 deletions(-)</pre>

<p>commit 2e311dc44290ef6bda48f0bcba09a3c22bf32d9a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 21 22:24:52 2009 +0900</p>

<pre>   Description about the incompatible change of Bio::KEGG::REACTION#rpairs.

doc/Changes-1.4.rdoc |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit d57ace3a89077caae3c681743da4b92d16b90af8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 21 22:17:46 2009 +0900</p>

<pre>   Bio::KEGG::R#ACTION#rpairs is changed to return a hash.

lib/bio/db/kegg/reaction.rb            |   65 ++++++++++++++++++++++++--------
test/unit/bio/db/kegg/test_reaction.rb |   27 ++++++++++++-
2 files changed, 74 insertions(+), 18 deletions(-)</pre>

<p>commit 60e4c77d184ee81c51668b446518cfbc9256be50 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Dec 21 22:15:44 2009 +0900</p>

<pre>   Document bug fix: return value mistake.

lib/bio/db/kegg/genes.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 6376dd55aa4995769746e556ca719d37f02975d6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Dec 20 17:32:52 2009 +0900</p>

<pre>   Added README.txt for FASTQ example data.

test/data/fastq/README.txt |  109 ++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 109 insertions(+), 0 deletions(-)
create mode 100644 test/data/fastq/README.txt</pre>

<p>commit 8dec18794c846726733d66c5a22170f5b2c4bb1a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Dec 15 13:51:13 2009 +0900</p>

<pre>   Newly added unit tests for Bio::KEGG::GLYCAN with test data.

test/data/KEGG/G00024.glycan         |   47 ++++++
test/data/KEGG/G01366.glycan         |   18 +++
test/unit/bio/db/kegg/test_glycan.rb |  260 ++++++++++++++++++++++++++++++++++
3 files changed, 325 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/G00024.glycan
create mode 100644 test/data/KEGG/G01366.glycan
create mode 100644 test/unit/bio/db/kegg/test_glycan.rb</pre>

<p>commit 90b97bfbcfb3f7e3d5c28b195bdb9b9c058df887 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Dec 15 11:42:39 2009 +0900</p>

<pre>   Newly added unit test for Bio::KEGG::DRUG with test data.

test/data/KEGG/D00063.drug         |  104 +++++++++++++++++++
test/unit/bio/db/kegg/test_drug.rb |  194 ++++++++++++++++++++++++++++++++++++
2 files changed, 298 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/D00063.drug
create mode 100644 test/unit/bio/db/kegg/test_drug.rb</pre>

<p>commit 443f778795b82a7f572cb8b85d2a8a8b3cea1334 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Dec 15 11:38:59 2009 +0900</p>

<pre>   New method Bio::KEGG::DRUG#products

    * New method Bio::KEGG::DRUG#products.
    * Improved RDoc.

lib/bio/db/kegg/drug.rb |   50 +++++++++++++++++++++++++++++++++++++---------
1 files changed, 40 insertions(+), 10 deletions(-)</pre>

<p>commit 48184d96b989f909ac0effb759cbc4b1ddc98dd1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 11 01:36:54 2009 +0900</p>

<pre>   Methods in Bio::KEGG::Common::* are changed to cache return values in instance variables.

lib/bio/db/kegg/common.rb |   62 ++++++++++++++++++++++++++------------------
1 files changed, 37 insertions(+), 25 deletions(-)</pre>

<p>commit f364ea609f1e01ca5270a5bd7404e0bbf752bc89 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 11 01:23:42 2009 +0900</p>

<pre>   Version is changed to 1.4.0-alpha1, and bioruby.gemspec is regenerated.

bioruby.gemspec     |  142 ++++++++++++++++++++++++++++++++++++++++++++++++++-
bioruby.gemspec.erb |    4 +-
lib/bio/version.rb  |    4 +-
3 files changed, 145 insertions(+), 5 deletions(-)</pre>

<p>commit 096b5fbf6b7ff906203aabf93eb9a0bd56ae9ba2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 11 01:22:59 2009 +0900</p>

<pre>   Added documents about Bio::KEGG incompatible changes.

doc/Changes-1.4.rdoc |   48 ++++++++++++++++++++++++++++++++++++++++++------
1 files changed, 42 insertions(+), 6 deletions(-)</pre>

<p>commit 72ed277fe30bb1033cbc16d462f137510afb84e6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 11 01:21:26 2009 +0900</p>

<pre>   Newly added unit tests for Bio::KEGG::ENZYME with test data.

test/data/KEGG/1.1.1.1.enzyme        |  935 ++++++++++++++++++++++++++++++++++
test/unit/bio/db/kegg/test_enzyme.rb |  241 +++++++++
2 files changed, 1176 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/1.1.1.1.enzyme
create mode 100644 test/unit/bio/db/kegg/test_enzyme.rb</pre>

<p>commit b99fcb39f7c5d2857cbb65283d85ea868ae8561d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Dec 11 01:09:03 2009 +0900</p>

<pre>   Changed Bio::KEGG::*#dblinks, pathways, orthologs, genes methods.

    * In Bio::KEGG::COMPOUND, DRUG, ENZYME, GLYCAN and ORTHOLOGY, the
      method dblinks is changed to return a Hash. The old methods  are
      renamed to dblinks_as_strings.
    * In Bio::KEGG::COMPOUND, DRUG, ENZYME, GENES, GLYCAN and REACTION,
      the method  pathways is changed to return a Hash. The old methods
      are renamed to pathways_as_strings except for GENES.
    * In Bio::KEGG::ENZYME, GENES, GLYCAN and REACTION, the method
      orthologs is changed to return a Hash. The old methods are renamed
      to orthologs_as_strings.
    * Bio::KEGG::ENZYME#genes and Bio::KEGG::ORTHOLOGY#genes is changed
      to return a Hash. The old methods are renamed to genes_as_strings.
    * Added Bio::KEGG::REACTION#rpairs_as_tokens, older behavior of rpairs.
    * Modules in lib/bio/db/kegg/common.rb are moved uner Bio::KEGG::Common
      namespace.
    * Refactoring.
    * Added documents.
    * Tests modified.

lib/bio/db/kegg/common.rb              |   40 +++++++++++++++++++++++++------
lib/bio/db/kegg/compound.rb            |   10 ++++---
lib/bio/db/kegg/drug.rb                |   27 +++++++++++++++------
lib/bio/db/kegg/enzyme.rb              |   31 ++++++++++++++++++++----
lib/bio/db/kegg/genes.rb               |   39 +++++++++++++++++++------------
lib/bio/db/kegg/glycan.rb              |   22 +++++++++++++++--
lib/bio/db/kegg/orthology.rb           |   25 +++++++------------
lib/bio/db/kegg/reaction.rb            |   16 +++++++++---
test/unit/bio/db/kegg/test_compound.rb |   27 ++++++++++++--------
test/unit/bio/db/kegg/test_reaction.rb |   13 +++++----
10 files changed, 170 insertions(+), 80 deletions(-)</pre>

<p>commit 2cc9d4e2f28f6b2bbcb8f714f9e2eb144c594fbf Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 10 16:02:54 2009 +0900</p>

<pre>   Bio::KEGG::GENES#structure no more adds PDB: prefix.

    * Bio::KEGG::GENES#structure no more adds PDB: prefix.
    * Added Bio::KEGG::GENES#structures as an alias of structure.

lib/bio/db/kegg/genes.rb            |    7 +++----
test/unit/bio/db/kegg/test_genes.rb |    7 ++++---
2 files changed, 7 insertions(+), 7 deletions(-)</pre>

<p>commit a8ceb23bdf19d6649aa4d879cba76a9e3f91d1d4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Dec 10 15:28:33 2009 +0900</p>

<pre>   Refactoring of Bio::KEGG::Orthology#dblinks and genes.

    * Refactoring of Bio::KEGG::Orthology#dblinks and genes: no need
      to treat @data because lines_fetch internally does so.

lib/bio/db/kegg/orthology.rb |   10 ++--------
1 files changed, 2 insertions(+), 8 deletions(-)</pre>

<p>commit 720e0bccdfdc6fac6222cac1a9f05d6e2419896c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 9 16:39:03 2009 +0900</p>

<pre>   Changed dummy lines for RDoc.

lib/bio/db/kegg/compound.rb  |    4 ++--
lib/bio/db/kegg/orthology.rb |    2 +-
lib/bio/db/kegg/reaction.rb  |    4 ++--
3 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit 20f8c03af92e5cfedcb49e8ed9fc6fda2b86e9c9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 9 15:17:39 2009 +0900</p>

<pre>   Refactoring of Bio::KEGG::REACTION#orthologs.

    * Refactoring of Bio::KEGG::REACTION#orthologs: no need to treat
      @data because lines_fetch internally does so.

lib/bio/db/kegg/reaction.rb |    5 +----
1 files changed, 1 insertions(+), 4 deletions(-)</pre>

<p>commit b924601bacd643f66b37dd991913e6862df704a9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Dec 6 15:51:03 2009 +0900</p>

<pre>   Bio::KEGG::GENES#pathways is changed to return raw lines as an Array of strings.

    * Bio::KEGG::GENES#pathways is changed to return raw lines as an
      Array of strings.
    * RDoc is added for Bio::KEGG::GENES.

lib/bio/db/kegg/genes.rb |   99 ++++++++++++++++++++++++++++++++++++++++++++-
1 files changed, 96 insertions(+), 3 deletions(-)</pre>

<p>commit 4c840dc6a539db1d854b23991269b3e6515f637e Author: Kozo Nishida
&lt;kozo-ni@cg05.naist.jp&gt; Date:   Wed Dec 2 17:02:00 2009 +0900</p>

<pre>   Added test methods.

test/unit/bio/db/kegg/test_compound.rb |   47 ++++++++++++++++++++++++++++++++
1 files changed, 47 insertions(+), 0 deletions(-)</pre>

<p>commit 105efa1ecd1bc99a54aac32710a97df15035119d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 23:31:07 2009 +0900</p>

<pre>   Refactoring: to use lib/bio/db/kegg/common.rb for dblinks_as_hash method.

lib/bio/db/kegg/orthology.rb |   16 +++++-----------
1 files changed, 5 insertions(+), 11 deletions(-)</pre>

<p>commit c394ead051c3a13ceb534f93816af7ad35be932a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 23:07:23 2009 +0900</p>

<pre>   Bio::KEGG::REACTION#orthologies is renamed to orthologs_as_hash with changing its return value to a hash.

    * Bio::KEGG::REACTION#orthologies is renamed to orthologs_as_hash
      with changing its return value to a hash.
    * The code of the orthologs_as_hash method is moved to
      lib/bio/db/kegg/common.rb.
    * Added new method Bio::KEGG::REACTION#orthologs, copied from
      lib/bio/db/kegg/glycan.rb.

lib/bio/db/kegg/common.rb              |   18 +++++++++++++++++-
lib/bio/db/kegg/reaction.rb            |   14 ++++++--------
sample/demo_kegg_reaction.rb           |    6 ++++--
test/unit/bio/db/kegg/test_reaction.rb |   12 ++++++++++--
4 files changed, 37 insertions(+), 13 deletions(-)</pre>

<p>commit 4e01fda27166faf066104ab9897904fd46f57123 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 22:48:06 2009 +0900</p>

<pre>   Added Bio::KEGG::REACTION#pathways_as_hash and reverted pathways method.

    * New method Bio::KEGG::REACTION#pathways_as_hash, using a module
      in lib/bio/db/kegg/common.rb.
    * Bio::KEGG::REACTION#pathways is reverted to return an array of
      string.

lib/bio/db/kegg/reaction.rb            |   18 +++++++++++-------
test/unit/bio/db/kegg/test_reaction.rb |    8 +++++++-
2 files changed, 18 insertions(+), 8 deletions(-)</pre>

<p>commit 0c2ce4b8462792d496ab3f58206fdbd47143e280 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 22:35:21 2009 +0900</p>

<pre>   New methods Bio::KEGG::COMPOUND#dblinks_as_hash and pathways_as_hash, using modules in lib/bio/db/kegg/common.rb.

lib/bio/db/kegg/compound.rb            |   14 +++++++++++
test/unit/bio/db/kegg/test_compound.rb |   38 ++++++++++++++++++++++++++++++++
2 files changed, 52 insertions(+), 0 deletions(-)</pre>

<p>commit 63df07e030120eb43de22555277529822b072270 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 22:25:20 2009 +0900</p>

<pre>   Methods commonly used from Bio::KEGG::* classes.

    * Modules containing methods commonly used from Bio::KEGG::* classes.
      The &quot;dblinks_as_hash&quot; method is copied from
      lib/bio/db/kegg/orthology.rb. The &quot;pathways_as_hash&quot; method is
      derived from the dblinks_as_hash and Bio::KEGG::REACTION#pathways
      methods.

lib/bio/db/kegg/common.rb |   60 +++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 60 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/kegg/common.rb</pre>

<p>commit 0e55c6701b09a52356ac55300181ee656773826f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 21:39:06 2009 +0900</p>

<pre>   Bio::KEGG::COMPOUND#dblinks is reverted to return an array of string.

lib/bio/db/kegg/compound.rb            |   11 ++---------
test/unit/bio/db/kegg/test_compound.rb |    8 +++++++-
2 files changed, 9 insertions(+), 10 deletions(-)</pre>

<p>commit a05adcddf6c7ed67c042f31ecd86848af1ba8a22 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Dec 2 21:13:39 2009 +0900</p>

<pre>   Bug fix: fixed a copy-and-paste mistake.

lib/bio/db/kegg/drug.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 86925f3c80730e3ea3377a23a70cadb3876258c4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Dec 1 21:31:40 2009 +0900</p>

<pre>   Bio::KEGG::ORTHOLOGY#dblinks_as_hash should preserve database names.

doc/Changes-1.4.rdoc                    |    4 ++++
lib/bio/db/kegg/orthology.rb            |    2 +-
test/unit/bio/db/kegg/test_orthology.rb |    2 +-
3 files changed, 6 insertions(+), 2 deletions(-)</pre>

<p>commit 60847cd2d0701fa38a499578649cb216c93993a2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Dec 1 20:41:51 2009 +0900</p>

<pre>   Test class names are changed to avoid potential class name conflict.

test/unit/bio/db/kegg/test_compound.rb  |    2 +-
test/unit/bio/db/kegg/test_genes.rb     |    4 ++--
test/unit/bio/db/kegg/test_orthology.rb |    2 +-
test/unit/bio/db/kegg/test_reaction.rb  |    2 +-
4 files changed, 5 insertions(+), 5 deletions(-)</pre>

<p>commit 2bda62af7a020c22379dd9ec3a42496d2a5b94cb Author: Kozo Nishida
&lt;kozo-ni@cg05.naist.jp&gt; Date:   Tue Dec 1 04:38:27 2009 +0900</p>

<pre>   Added unit tests for Bio::KEGG::ORTHOLOGY.

test/data/KEGG/K02338.orthology         |  902 +++++++++++++++++++++++++++++++
test/unit/bio/db/kegg/test_orthology.rb |   50 ++
2 files changed, 952 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/K02338.orthology
create mode 100644 test/unit/bio/db/kegg/test_orthology.rb</pre>

<p>commit acad9497caf5d737394568e911691fdad11ca091 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 30 21:39:32 2009 +0900</p>

<pre>   Changed to use BioRubyTestDataPath instead of __FILE__.

test/unit/bio/db/kegg/test_compound.rb |    3 +--
test/unit/bio/db/kegg/test_reaction.rb |    3 +--
2 files changed, 2 insertions(+), 4 deletions(-)</pre>

<p>commit 8e95f3fb60cd61b2bfad8e66caf03d3ff02a6dca Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 16:37:21 2009 +0900</p>

<pre>   Bio::Fastq::QualityScore is renamed to Bio::Sequence::QualityScore.

    * Bio::Fastq::QualityScore is renamed to Bio::Sequence::QualityScore.
    * Changes of filenames due to the previous file move.

lib/bio/db/fasta/format_qual.rb              |   18 ++++++++--------
lib/bio/db/fastq.rb                          |    7 ++---
lib/bio/sequence.rb                          |    3 +-
lib/bio/sequence/quality_score.rb            |   25 +++++++++++------------
test/unit/bio/sequence/test_quality_score.rb |   28 +++++++++++++-------------
5 files changed, 40 insertions(+), 41 deletions(-)</pre>

<p>commit 2b29654c1d7e927e445e7acdd525835a873c2a2a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 16:15:42 2009 +0900</p>

<pre>   lib/bio/db/fastq/quality_score.rb is moved to lib/bio/sequence/. The unit test is also moved.

    * lib/bio/db/fastq/quality_score.rb is moved to lib/bio/sequence/.
    * test/unit/bio/db/fastq/test_quality_score.rb is moved to
      test/unit/bio/sequence/.
    * The file contents will be modified with the following commit.

lib/bio/db/fastq/quality_score.rb            |  206 ----------------
lib/bio/sequence/quality_score.rb            |  206 ++++++++++++++++
test/unit/bio/db/fastq/test_quality_score.rb |  330 --------------------------
test/unit/bio/sequence/test_quality_score.rb |  330 ++++++++++++++++++++++++++
4 files changed, 536 insertions(+), 536 deletions(-)
delete mode 100644 lib/bio/db/fastq/quality_score.rb
create mode 100644 lib/bio/sequence/quality_score.rb
delete mode 100644 test/unit/bio/db/fastq/test_quality_score.rb
create mode 100644 test/unit/bio/sequence/test_quality_score.rb</pre>

<p>commit aa8d49bf31f90dd2796c18ee0aa6291979284ec2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 15:20:36 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_gff1.rb.

lib/bio/db/gff.rb   |   17 -----------------
sample/demo_gff1.rb |   49 +++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 49 insertions(+), 17 deletions(-)
create mode 100644 sample/demo_gff1.rb</pre>

<p>commit 76fffd2d2429346478fb3d8c88cdcd878a1047b1 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 15:06:41 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_tmhmm_report.rb.

lib/bio/appl/tmhmm/report.rb |   36 ----------------------
sample/demo_tmhmm_report.rb  |   68 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 68 insertions(+), 36 deletions(-)
create mode 100644 sample/demo_tmhmm_report.rb</pre>

<p>commit dfafb0a2bcec4c0b4cd3640374e151e2039056dc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 14:59:27 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_targetp_report.rb.

lib/bio/appl/targetp/report.rb |  105 +------------------------------
sample/demo_targetp_report.rb  |  135 ++++++++++++++++++++++++++++++++++++++++
2 files changed, 136 insertions(+), 104 deletions(-)
create mode 100644 sample/demo_targetp_report.rb</pre>

<p>commit 75f7c8527546f8ea3079f53b90a9b4d8260b4de0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 14:33:28 2009 +0900</p>

<pre>   Follow-up of the SOSUI server URL change.

lib/bio/appl/sosui/report.rb |    6 ++++--
1 files changed, 4 insertions(+), 2 deletions(-)</pre>

<p>commit 8022696295dc296462f73b40cc74ad5259bee387 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 14:32:11 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_sosui_report.rb.

lib/bio/appl/sosui/report.rb |   53 +------------------------
sample/demo_sosui_report.rb  |   89 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 90 insertions(+), 52 deletions(-)
create mode 100644 sample/demo_sosui_report.rb</pre>

<p>commit 4acfe7f565039b34a036682912a75f55da808b45 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 29 14:02:32 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_hmmer_report.rb.

lib/bio/appl/hmmer/report.rb |  100 ----------------------------
sample/demo_hmmer_report.rb  |  149 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 149 insertions(+), 100 deletions(-)
create mode 100644 sample/demo_hmmer_report.rb</pre>

<p>commit 4f7bd1b7628d90661d8b557ca854b14cc44fb99c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 15:49:21 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are removed because they are very short.

lib/bio/appl/fasta/format10.rb |   14 --------------
lib/bio/appl/hmmer.rb          |   16 +---------------
lib/bio/io/flatfile.rb         |    8 +-------
3 files changed, 2 insertions(+), 36 deletions(-)</pre>

<p>commit c2a72d195189755532e7e206af34d152ab6332d8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 15:20:28 2009 +0900</p>

<pre>   Bug fix: Failure of Bio::Fasta.remote due to the remote site changes.

lib/bio/appl/fasta.rb |    5 ++++-
1 files changed, 4 insertions(+), 1 deletions(-)</pre>

<p>commit 549112fb4dfb5f6b2fe3491fb161887a9f5262ac Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 15:13:10 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_fasta_remote.rb.

lib/bio/appl/fasta.rb       |   18 ---------------
sample/demo_fasta_remote.rb |   51 +++++++++++++++++++++++++++++++++++++++++++
2 files changed, 51 insertions(+), 18 deletions(-)
create mode 100644 sample/demo_fasta_remote.rb</pre>

<p>commit 0e4ca0db83692fdbbe93e90272a07bcbac89192c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 10:17:47 2009 +0900</p>

<pre>   Text indents for some comment lines are changed.

sample/demo_blast_report.rb  |    4 ++--
sample/demo_kegg_compound.rb |    4 ++--
sample/demo_prosite.rb       |    4 ++--
sample/demo_sirna.rb         |    4 ++--
4 files changed, 8 insertions(+), 8 deletions(-)</pre>

<p>commit c0cf91fe2a9247bc3705b20515f9d4fa14288d5a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 10:13:26 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_keggapi.rb.

    * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
      sample/demo_keggapi.rb.
    * Commented out demonstrations of deprecated methods:
      get_neighbors_by_gene, get_similarity_between_genes,
      get_ko_members, get_oc_members_by_gene, get_pc_members_by_gene.
    * Commented out demonstrations of methods internally using
      the deprecated methods: get_all_neighbors_by_gene,
      get_all_oc_members_by_gene, get_all_pc_members_by_gene.

lib/bio/io/keggapi.rb  |  442 ------------------------------------------
sample/demo_keggapi.rb |  502 ++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 502 insertions(+), 442 deletions(-)
create mode 100644 sample/demo_keggapi.rb</pre>

<p>commit 8b8206c1d8ee699185fdd19d3329311c85ee003c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 01:50:06 2009 +0900</p>

<pre>   Fixed the license line.

lib/bio/db/prosite.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit ebfeec8243abd4e2f65335fda1ead18efff66897 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 01:41:58 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_ncbi_rest.rb.

lib/bio/io/ncbirest.rb   |  101 ------------------------------------
sample/demo_ncbi_rest.rb |  128 ++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 128 insertions(+), 101 deletions(-)
create mode 100644 sample/demo_ncbi_rest.rb</pre>

<p>commit 5a0f8379a374650d12fc88fbbd5b28c38ae96395 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 01:33:07 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_prosite.rb.

lib/bio/db/prosite.rb  |   95 +-------------------------------------
sample/demo_prosite.rb |  120 ++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 121 insertions(+), 94 deletions(-)
create mode 100644 sample/demo_prosite.rb</pre>

<p>commit c560a5d0ba9d4919dbcca156ea620056dcb8f725 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 01:14:37 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_psort.rb.

lib/bio/appl/psort.rb |  111 ---------------------------------------
sample/demo_psort.rb  |  138 +++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 138 insertions(+), 111 deletions(-)
create mode 100644 sample/demo_psort.rb</pre>

<p>commit 1299a55d214784a536ae3cd8bfabdfd61fe1da86 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 01:04:29 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_psort_report.rb.

    * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
      sample/demo_psort_report.rb, without any checks.

lib/bio/appl/psort/report.rb |   46 +---------------------------
sample/demo_psort_report.rb  |   70 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 71 insertions(+), 45 deletions(-)
create mode 100644 sample/demo_psort_report.rb</pre>

<p>commit a2686fe3c5a93947c94d4602514a62a808c182d5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 00:53:54 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_genscan_report.rb.

lib/bio/appl/genscan/report.rb |  176 ----------------------------------
sample/demo_genscan_report.rb  |  202 ++++++++++++++++++++++++++++++++++++++++
2 files changed, 202 insertions(+), 176 deletions(-)
create mode 100644 sample/demo_genscan_report.rb</pre>

<p>commit 22f662ba69dd2d4a2273562dd7ea921f5cdd84bd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 00:28:01 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_ddbjxml.rb.

lib/bio/io/ddbjxml.rb  |  182 +-----------------------------------------
sample/demo_ddbjxml.rb |  212 ++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 213 insertions(+), 181 deletions(-)
create mode 100644 sample/demo_ddbjxml.rb</pre>

<p>commit ed3b34b6598f632c7b9b3f1a17b42406c19ca32d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 26 00:12:33 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_pubmed.rb.

    * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
      sample/demo_pubmed.rb.
    * Codes using Entrez CGI are disabled in the demo.

lib/bio/io/pubmed.rb  |   88 -------------------------------------
sample/demo_pubmed.rb |  116 +++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 116 insertions(+), 88 deletions(-)
create mode 100644 sample/demo_pubmed.rb</pre>

<p>commit 9e6d720f383e88e247eacab6f0e43f38140a62f2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 23:59:10 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are removed.

    * Demo codes in the &quot;if __FILE__ == $0&quot; are removed because their
      function have already been moved to sample/demo_blast_report.rb.

lib/bio/appl/blast/format0.rb |  193 --------------------------------------
lib/bio/appl/blast/report.rb  |  149 +-----------------------------
lib/bio/appl/blast/wublast.rb |  208 -----------------------------------------
3 files changed, 2 insertions(+), 548 deletions(-)</pre>

<p>commit bbba2812fa9131d01fc655eb174d84f06facd8b8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 23:49:36 2009 +0900</p>

<pre>   New demo code of BLAST parser based on codes in &quot;if __FILE__ ==$0&quot;

    * Newly added sample/demo_blast_report.rb, demonstration of
      BLAST parsers  Bio::Blast::Report, Bio::Blast::Default::Report,
      and Bio::Blast::WU::Report. It is based on the demonstration codes
      in the &quot;if __FILE__ == $0&quot; in lib/bio/appl/blast/report.rb,
      lib/bio/appl/blast/format0.rb, and lib/bio/appl/blast/wublast.rb.

sample/demo_blast_report.rb |  285 +++++++++++++++++++++++++++++++++++++++++++
1 files changed, 285 insertions(+), 0 deletions(-)
create mode 100644 sample/demo_blast_report.rb</pre>

<p>commit 5235ed15db8d3ba3e59d8dc3bbbcf1b5b9c58281 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 21:57:08 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_das.rb.

    * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
      sample/demo_das.rb.
    * Demo codes using UCSC DAS server is added.
    * Demo using the WormBase DAS server is temporarily disabled because
      it does not work well possibly because of the server trouble.

lib/bio/io/das.rb  |   44 ----------------------
sample/demo_das.rb |  105 ++++++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 105 insertions(+), 44 deletions(-)
create mode 100644 sample/demo_das.rb</pre>

<p>commit b7b0f7bef0505b9678673e54bb863d4ff7897dd5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 20:58:35 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_taxonomy.rb, although it does not work correctly now.

lib/bio/db/kegg/taxonomy.rb  |   53 +-----------------------
sample/demo_kegg_taxonomy.rb |   92 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 93 insertions(+), 52 deletions(-)
create mode 100644 sample/demo_kegg_taxonomy.rb</pre>

<p>commit 23da98ca19fce1f0b487e1f955ef4cd896839590 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 20:11:12 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_reaction.rb.

lib/bio/db/kegg/reaction.rb  |   16 +----------
sample/demo_kegg_reaction.rb |   64 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 65 insertions(+), 15 deletions(-)
create mode 100644 sample/demo_kegg_reaction.rb</pre>

<p>commit 9c0bfb857a6b41d8e6a42ff2cbf7b06ca1d38d78 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 19:12:00 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_orthology.rb.

lib/bio/db/kegg/orthology.rb  |   23 +--------------
sample/demo_kegg_orthology.rb |   62 +++++++++++++++++++++++++++++++++++++++++
2 files changed, 63 insertions(+), 22 deletions(-)
create mode 100644 sample/demo_kegg_orthology.rb</pre>

<p>commit 6f6f1eb3d87dea588ea333708c4d4486ac7136b6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 12:19:26 2009 +0900</p>

<pre>   Commented out demo for nonexistent method &quot;bindings&quot;.

sample/demo_kegg_glycan.rb |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 98a6d904058b5af4808f16bcb710d73bd97c9764 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 12:18:31 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_glycan.rb.

lib/bio/db/kegg/glycan.rb  |   21 -------------
sample/demo_kegg_glycan.rb |   72 ++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 72 insertions(+), 21 deletions(-)
create mode 100644 sample/demo_kegg_glycan.rb</pre>

<p>commit d26e835ca9def2287f1050f1b048892e3cafdaa0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 11:49:05 2009 +0900</p>

<pre>   Added references.

lib/bio/db/kegg/genome.rb |    2 ++
1 files changed, 2 insertions(+), 0 deletions(-)</pre>

<p>commit c3c460462481b5b8d6e9441216bcf6370b4890ef Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 11:45:31 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_genome.rb.

lib/bio/db/kegg/genome.rb  |   42 +------------------------
sample/demo_kegg_genome.rb |   74 ++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 75 insertions(+), 41 deletions(-)
create mode 100644 sample/demo_kegg_genome.rb</pre>

<p>commit 0d8e709b66bf18ead5944c27a50eb6cf2c47862f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 11:43:24 2009 +0900</p>

<pre>   Added document about downloading sample data.

sample/demo_kegg_drug.rb |   13 ++++++++++++-
1 files changed, 12 insertions(+), 1 deletions(-)</pre>

<p>commit 0608893198e9bc88521b6c013069d8c7a13bb0e5 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 00:10:48 2009 +0900</p>

<pre>   Added documents.

lib/bio/db/kegg/drug.rb |   15 +++++++++++++++
1 files changed, 15 insertions(+), 0 deletions(-)</pre>

<p>commit 0ecdc1ee0460f16dba1e4cd5ab575c92e1c6b1ac Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 25 00:06:02 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_drug.rb.

lib/bio/db/kegg/drug.rb  |   18 +--------------
sample/demo_kegg_drug.rb |   54 ++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 55 insertions(+), 17 deletions(-)
create mode 100644 sample/demo_kegg_drug.rb</pre>

<p>commit b0c349103f01a26f4741999bd696bf5b1c032e06 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 23:51:13 2009 +0900</p>

<pre>   Added documents.

lib/bio/db/kegg/compound.rb |   10 ++++++++++
1 files changed, 10 insertions(+), 0 deletions(-)</pre>

<p>commit e965b454c553ed9670bc83962a2a9d7c5de49929 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 23:45:15 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_kegg_compound.rb.

lib/bio/db/kegg/compound.rb  |   19 +-------------
sample/demo_kegg_compound.rb |   57 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 58 insertions(+), 18 deletions(-)
create mode 100644 sample/demo_kegg_compound.rb</pre>

<p>commit 7454db7c8b8ef7202736d311356d4ca350af336f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 23:06:21 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_litdb.rb.

lib/bio/db/litdb.rb  |   17 +----------------
sample/demo_litdb.rb |   42 ++++++++++++++++++++++++++++++++++++++++++
2 files changed, 43 insertions(+), 16 deletions(-)
create mode 100644 sample/demo_litdb.rb</pre>

<p>commit fde284248e013e44184ee2ba7da85e5b83155a69 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 22:57:01 2009 +0900</p>

<pre>   Ruby 1.9 support: String#each_line instead of String#each

lib/bio/db/go.rb |    6 +++---
1 files changed, 3 insertions(+), 3 deletions(-)</pre>

<p>commit 8b60099615790fe372b4fde27a391dedc767aab2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 22:53:12 2009 +0900</p>

<pre>   Sample code bug fix: fixed method names, and workaround for Zlib error.

    * Sample code bug fix: Following method name changes.
    * Workaround for Zlib::DataError.

sample/demo_go.rb |   13 +++++++++----
1 files changed, 9 insertions(+), 4 deletions(-)</pre>

<p>commit 737fec3db555811d127d2356e5ceef63b0413fb8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 19:47:14 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_go.rb.

lib/bio/db/go.rb  |   70 +---------------------------------------
sample/demo_go.rb |   93 +++++++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 94 insertions(+), 69 deletions(-)
create mode 100644 sample/demo_go.rb</pre>

<p>commit 8264b15690132d9e766f16d0829bb12cd122b900 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 19:20:52 2009 +0900</p>

<pre>   Document bug fix: Changed Bio::Bl2seq to Bio::Blast::Bl2seq in the RDoc.

    * Document bug fix: Changed Bio::Bl2seq to Bio::Blast::Bl2seq
      in the RDoc.
    * Modified copyright line.

lib/bio/appl/bl2seq/report.rb |   18 +++++++++---------
1 files changed, 9 insertions(+), 9 deletions(-)</pre>

<p>commit c572ff022fee43505355608f0a0e3ba2181e87e2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 19:17:04 2009 +0900</p>

<pre>   Bug fix: Failed to read Bio::Blast::Bl2seq::Report data by using Bio::FlatFile.

lib/bio/appl/bl2seq/report.rb |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 4f6b080623442dfcc5864e2aefde7e53ace068e8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 19:15:11 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_bl2seq_report.rb.

lib/bio/appl/bl2seq/report.rb |  194 +------------------------------------
sample/demo_bl2seq_report.rb  |  220 +++++++++++++++++++++++++++++++++++++++++
2 files changed, 221 insertions(+), 193 deletions(-)
create mode 100644 sample/demo_bl2seq_report.rb</pre>

<p>commit 2f03e8757383e0d1a26c0f6942c74a30f3b26d90 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 18:24:43 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_genbank.rb.

    * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
      sample/demo_genbank.rb, and modified as below.
       * To get sequences from the NCBI web service.
       * By default, arguments are sequence IDs (accession numbers).
       * New option &quot;--files&quot; (or &quot;-files&quot;, &quot;--file&quot;, or &quot;-file&quot;) to
         read sequences from file(s).

lib/bio/db/genbank/genbank.rb |   87 +--------------------------
sample/demo_genbank.rb        |  132 +++++++++++++++++++++++++++++++++++++++++
2 files changed, 133 insertions(+), 86 deletions(-)
create mode 100644 sample/demo_genbank.rb</pre>

<p>commit a2981c28fdb629a655c71c920f6588f8b80aff06 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 15:06:50 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_aaindex.rb.

lib/bio/db/aaindex.rb  |   39 +---------------------------
sample/demo_aaindex.rb |   67 ++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 68 insertions(+), 38 deletions(-)
create mode 100644 sample/demo_aaindex.rb</pre>

<p>commit b741d17ec5c5ac234bab35b8716fee072635de1a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 24 12:45:43 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_sirna.rb

    * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
      sample/demo_sirna.rb, and modified for reading normal sequence
      files instead of a raw sequence.

lib/bio/util/sirna.rb |   24 +------------------
sample/demo_sirna.rb  |   63 +++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 64 insertions(+), 23 deletions(-)
create mode 100644 sample/demo_sirna.rb</pre>

<p>commit 7cc778e78bc63ef73796ee15d6f0db8d6967aefe Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 23:00:42 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_pathway.rb.

lib/bio/pathway.rb     |  171 -----------------------------------------
sample/demo_pathway.rb |  196 ++++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 196 insertions(+), 171 deletions(-)
create mode 100644 sample/demo_pathway.rb</pre>

<p>commit 7e5510587abc0b50b6851f005a3236bf9dc79d08 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 22:49:13 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_locations.rb.

lib/bio/location.rb      |   73 ----------------------------------
sample/demo_locations.rb |   99 ++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 99 insertions(+), 73 deletions(-)
create mode 100644 sample/demo_locations.rb</pre>

<p>commit f1c02666f4b11d5cf208d6beb592d8ac962ce2da Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 22:35:50 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_codontable.rb.

lib/bio/data/codontable.rb |   96 +-----------------------------------
sample/demo_codontable.rb  |  119 ++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 120 insertions(+), 95 deletions(-)
create mode 100644 sample/demo_codontable.rb</pre>

<p>commit c11a7793f85faf3d66d630833c38358ffa34a698 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 16:35:16 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_nucleicacid.rb.

lib/bio/data/na.rb         |   27 +-----------------------
sample/demo_nucleicacid.rb |   49 ++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 50 insertions(+), 26 deletions(-)
create mode 100644 sample/demo_nucleicacid.rb</pre>

<p>commit de41b67c3f65baa0f122689b2e9f479d8a247934 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 16:25:05 2009 +0900</p>

<pre>   Demo codes in the &quot;if __FILE__ == $0&quot; are moved to sample/demo_aminoacid.rb.

lib/bio/data/aa.rb       |   78 +-----------------------------------
sample/demo_aminoacid.rb |  101 ++++++++++++++++++++++++++++++++++++++++++++++
2 files changed, 102 insertions(+), 77 deletions(-)
create mode 100644 sample/demo_aminoacid.rb</pre>

<p>commit e652dd44ecb6b6dad652e33a398f92bb8373e7dd Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 16:12:48 2009 +0900</p>

<pre>   Added an error message about encoding in Ruby 1.9.1

KNOWN_ISSUES.rdoc |    3 ++-
1 files changed, 2 insertions(+), 1 deletions(-)</pre>

<p>commit 003133b0d4e2234c27927c9d10b75185c354102e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 23 15:52:21 2009 +0900</p>

<pre>   changed recommended Ruby version

README.rdoc |    4 ++--
1 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit 408483d36b713678361cecf6c77ff7a2098f71fc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 22 17:07:37 2009 +0900</p>

<pre>   added information about doc/Changes-1.4.rdoc

README.rdoc |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit ef342933839e8c6cef9883045fcaf468aff5da23 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 22 16:50:55 2009 +0900</p>

<pre>   In PhyloXML support, added a link to GNOME Libxml2 and fixed RDoc syntax.

README.rdoc |    6 ++++--
1 files changed, 4 insertions(+), 2 deletions(-)</pre>

<p>commit 0237ef42d60c7a76cadf8ea78f4251bcfe89c95f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 19 09:43:15 2009 +0900</p>

<pre>   Ruby 1.9 support: String#each_line instead of String#each

lib/bio/appl/meme/mast/report.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit ec935ea9b19415bf3325bcc0763fbc22f3c71a3d Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 19 09:40:49 2009 +0900</p>

<pre>   The &quot;libpath magic&quot; is replaced by loading helper routine.

test/unit/bio/appl/meme/mast/test_report.rb |   11 ++++++-----
test/unit/bio/appl/meme/test_mast.rb        |   11 ++++++-----
test/unit/bio/appl/meme/test_motif.rb       |    8 +++++---
3 files changed, 17 insertions(+), 13 deletions(-)</pre>

<p>commit 3f65eeb503f3b2ef866cab4c73d2d700ca572835 Author: Adam Kraut
&lt;adamnkraut@gmail.com&gt; Date:   Tue Mar 17 19:41:31 2009 -0400</p>

<pre>   Added basic support for MEME/MAST applications

lib/bio/appl/meme/mast.rb                   |  156 +++++++++++++++++++++++++++
lib/bio/appl/meme/mast/report.rb            |   91 ++++++++++++++++
lib/bio/appl/meme/motif.rb                  |   48 ++++++++
test/data/meme/mast.out                     |   13 +++
test/data/meme/meme.out                     |    3 +
test/unit/bio/appl/meme/mast/test_report.rb |   45 ++++++++
test/unit/bio/appl/meme/test_mast.rb        |  102 +++++++++++++++++
test/unit/bio/appl/meme/test_motif.rb       |   36 ++++++
8 files changed, 494 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/appl/meme/mast.rb
create mode 100644 lib/bio/appl/meme/mast/report.rb
create mode 100644 lib/bio/appl/meme/motif.rb
create mode 100644 test/data/meme/db
create mode 100644 test/data/meme/mast
create mode 100644 test/data/meme/mast.out
create mode 100644 test/data/meme/meme.out
create mode 100644 test/unit/bio/appl/meme/mast/test_report.rb
create mode 100644 test/unit/bio/appl/meme/test_mast.rb
create mode 100644 test/unit/bio/appl/meme/test_motif.rb</pre>

<p>commit 3862f54fda0caec2a07e563a1f8a11913baca2e3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 18 20:29:56 2009 +0900</p>

<pre>   New version of PhyloXML schema, version 1.10.

    * Upgraded to New version of PhyloXML schema, version 1.10,
      developed by Christian M Zmasek.

lib/bio/db/phyloxml/phyloxml.xsd | 1155 +++++++++++++++++++-------------------
1 files changed, 582 insertions(+), 573 deletions(-)</pre>

<p>commit 45ffd9228d513b3dbf29e1011c6a6689a8bd1b08 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 18 00:26:44 2009 +0900</p>

<pre>   Newly added sample script to test big PhyloXML data

    * Newly added a sample script to test big PhyloXML data based on
      Diana Jaunzeikare's work.
      (http://github.com/latvianlinuxgirl/bioruby/blob/
      20627fc5a443d6c2e3dc73ed50e9c578ffcbc330/
      test/unit/bio/db/test_phyloxml_big.rb).

sample/test_phyloxml_big.rb |  205 +++++++++++++++++++++++++++++++++++++++++++
1 files changed, 205 insertions(+), 0 deletions(-)
create mode 100644 sample/test_phyloxml_big.rb</pre>

<p>commit 828a8971e057919b80508cf29fd9518828b74a2f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 17 23:54:37 2009 +0900</p>

<pre>   Speed up of Bio::Tree#children and parent: caching node's parent.

    * For speed up of Bio::Tree#children and parent, internal cache of
      the parent for each node is added. The cache is automatically
      cleared when the tree is modified. Note that the cache can only
      be accessed from inside Bio::Tree.
    * Bio::Tree#parent is changed to directly raise IndexError when
      both of the root specified in the argument and preset in the
      tree are nil (previously, the same error is raised in the path
      method which is internally called from the parent method).
    * Bio::Tree#path is changed not to call bfs_shortest_path if the
      node1 and node2 are adjacent.

lib/bio/tree.rb |   75 +++++++++++++++++++++++++++++++++++++++++++++++++++---
1 files changed, 70 insertions(+), 5 deletions(-)</pre>

<p>commit 75862212e6bb807a570338e39e19d527219b6f13 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 22:11:15 2009 +0900</p>

<pre>   Documented incompatible changes of Bio::KEGG::COMPOUND and Bio::KEGG:REACTION.

doc/Changes-1.4.rdoc |   10 ++++++++++
1 files changed, 10 insertions(+), 0 deletions(-)</pre>

<p>commit c74cfabd6414c8b50db0251739f967accd90773f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 21:20:42 2009 +0900</p>

<pre>   Ruby compatibility issue: Enumerable#each_slice(4).each does not work in Ruby 1.8.5.

lib/bio/db/kegg/reaction.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit e6a920e401a2b06c355174ccdc9b993a38f9d7ec Author: Mitsuteru Nakao
&lt;mitsuteru.nakao@gmail.com&gt; Date:   Wed Jul 22 22:50:22 2009 +0900</p>

<pre>   Added new method Bio::KEGG::GENES#structure with the unit tests.

lib/bio/db/kegg/genes.rb            |   12 ++++++++++++
test/unit/bio/db/kegg/test_genes.rb |   25 +++++++++++++++++++++++++
2 files changed, 37 insertions(+), 0 deletions(-)</pre>

<p>commit 9fee0c133d069348857014410983f682e468c1c7 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 21:04:02 2009 +0900</p>

<pre>   The &quot;libpath magic&quot; is replaced by loading helper routine.

test/unit/bio/db/kegg/test_compound.rb |    6 ++++--
test/unit/bio/db/kegg/test_reaction.rb |    6 ++++--
2 files changed, 8 insertions(+), 4 deletions(-)</pre>

<p>commit 1199330eab95b8434303e92c5f792818e96db814 Author: Kozo Nishida
&lt;kozo-ni@cg05.naist.jp&gt; Date:   Sat Nov 14 09:03:50 2009 +0900</p>

<pre>   Newly added unit tests for Bio::KEGG::COMPOUND and Bio::KEGG::REACTION

    * Newly added unit tests for Bio::KEGG::COMPOUND and
      Bio::KEGG::REACTION with test data. (Note that this is a
      combination of several commits made by Kozo Nishida,
      merged from git://github.com/kozo2/bioruby.git ).

test/data/KEGG/C00025.compound         |  102 ++++++++++++++++++++++++++++++++
test/data/KEGG/R00006.reaction         |   14 ++++
test/unit/bio/db/kegg/test_compound.rb |   49 +++++++++++++++
test/unit/bio/db/kegg/test_reaction.rb |   57 ++++++++++++++++++
4 files changed, 222 insertions(+), 0 deletions(-)
create mode 100644 test/data/KEGG/C00025.compound
create mode 100644 test/data/KEGG/R00006.reaction
create mode 100644 test/unit/bio/db/kegg/test_compound.rb
create mode 100644 test/unit/bio/db/kegg/test_reaction.rb</pre>

<p>commit 1b47640665d4332bafd9e9709628ee9722f1f3f4 Author: Kozo Nishida
&lt;kozo-ni@cg05.naist.jp&gt; Date:   Sat Nov 14 09:03:50 2009 +0900</p>

<pre>   Bio::KEGG::COMPOUND#dblinks changed to return hash list

    * Bio::KEGG::COMPOUND#dblinks is changed to return hash list (array
      containing hashes).

lib/bio/db/kegg/compound.rb |   11 +++++++++--
1 files changed, 9 insertions(+), 2 deletions(-)</pre>

<p>commit 2aa43a0aa765ee4502923c2102e352826a9a7abd Author: Kozo Nishida
&lt;kozo-ni@cg05.naist.jp&gt; Date:   Sat Nov 14 07:29:19 2009 +0900</p>

<pre>   Bio::KEGG:REACTION#rpair and pathways changed to return hash list, and added orthologies method.

    * New method: Bio::KEGG:REACTION#orthologies
    * Bio::KEGG:REACTION#rpair and pathways are changed to return hash
      list (array containing hashes).

lib/bio/db/kegg/reaction.rb |   33 ++++++++++++++++++++++++++++++---
1 files changed, 30 insertions(+), 3 deletions(-)</pre>

<p>commit a82f5d228370beeeb397be07e07394652fd7837e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 20:03:19 2009 +0900</p>

<pre>   Changed not to modify given argument

lib/bio/util/restriction_enzyme/single_strand.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit ff10d5540759a5e7eaaa71da020d95170b98e007 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 19:50:08 2009 +0900</p>

<pre>   Newly added a document for incompatible and/or important changes of the new release version.

    * Newly added a document for incompatible and/or important changes
      of the new release version.
    * Added description about Bio::RestrictionEnzyme validation is
      disabled (although very small change).

doc/Changes-1.4.rdoc |   16 ++++++++++++++++
1 files changed, 16 insertions(+), 0 deletions(-)
create mode 100644 doc/Changes-1.4.rdoc</pre>

<p>commit 629e537f90e0825fadeec2e0207f8caddfbed59a Author: trevor &lt;&gt;
Date:   Sat Sep 19 11:03:23 2009 -0500</p>

<pre>   speed-up serial calls to RestrictionEnzyme

lib/bio/db/rebase.rb                               |    2 +-
lib/bio/util/restriction_enzyme/single_strand.rb   |    3 +-
.../util/restriction_enzyme/test_single_strand.rb  |   24 ++++++++++---------
.../test_single_strand_complement.rb               |   24 ++++++++++---------
4 files changed, 29 insertions(+), 24 deletions(-)</pre>

<p>commit 9b55a92d5300294bef7b624d0f9aa3edd3e8d7fc Author: trevor &lt;&gt;
Date:   Sat Sep 19 10:46:21 2009 -0500</p>

<pre>   speed-up rebase library

lib/bio/db/rebase.rb |    9 ++++-----
1 files changed, 4 insertions(+), 5 deletions(-)</pre>

<p>commit 4aaa24b3fc3cf2d1f7cf8b6d974d2115958b5a1b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 15:08:31 2009 +0900</p>

<pre>   Ruby 1.9 support: Array#to_s is changed to join('')

lib/bio/db/sanger_chromatogram/scf.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 1cf924b81150545e807169144aeaca6a75f9731c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Mon Nov 16 12:59:10 2009 +0900</p>

<pre>   Ruby 1.9 support: Array#nitems (counts the number of non-nil elements) is removed in 1.9.

    * Ruby 1.9 support: Array#nitems (counts the number of non-nil
      elements) is removed in Ruby 1.9. In scf.rb, it seems that
      nil would never be included in the array, and simply replaced
      by Array#size.

lib/bio/db/sanger_chromatogram/scf.rb |    8 ++++----
1 files changed, 4 insertions(+), 4 deletions(-)</pre>

<p>commit a95994d9cfbf4fc89fa716358ac5b92d42a1307b Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 15 19:23:12 2009 +0900</p>

<pre>   Bug fix: error when quality_scores are larger than the sequence length, and added a require line.

    * Bug fix: error when sequence.quality_scores are larger than
      the sequence length.
    * Added a require line.

lib/bio/db/fasta/format_qual.rb |    5 ++++-
1 files changed, 4 insertions(+), 1 deletions(-)</pre>

<p>commit c5aafca19b58b1651080e81699b7020cd3fd3f47 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Nov 15 19:21:49 2009 +0900</p>

<pre>   Newly added unit tests for Bio::Sequence::Format::Formatter::Fasta_numeric and Qual.

test/unit/bio/db/fasta/test_format_qual.rb |  346 ++++++++++++++++++++++++++++
1 files changed, 346 insertions(+), 0 deletions(-)
create mode 100644 test/unit/bio/db/fasta/test_format_qual.rb</pre>

<p>commit 6e24170e29d2576ff69b18eaadc94e9769b8612a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Nov 14 02:41:13 2009 +0900</p>

<pre>   Newly added Bio::Sequence::Format::Formatter::Qual and Fasta_numeric, formatter for Qual format and FastaNumericFormat.

lib/bio/db/fasta/format_qual.rb |  201 +++++++++++++++++++++++++++++++++++++++
lib/bio/sequence/format.rb      |    7 ++
2 files changed, 208 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/fasta/format_qual.rb</pre>

<p>commit 6959fd359040b6ca9570111d515118dc2d472029 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Nov 14 02:19:17 2009 +0900</p>

<pre>   Split quality score methods in Bio::Fastq::FormatData into separete modules

    * Quality score calculation methods in Bio::Fastq::FormatData in
      lib/bio/db/fastq.rb is splitted into separate modules
      Bio::Fastq::QualityScore::Converter, Phred, and Solexa
      in lib/bio/db/fastq/quality_score.rb.
    * Unit tests for Bio::Fastq::QualityScore::* are newly added
      in test/unit/bio/db/fastq/test_quality_score.rb.
    * Possible bug fix: probability should be 0 &lt;= p &lt;= 1.

lib/bio/db/fastq.rb                          |  112 +--------
lib/bio/db/fastq/quality_score.rb            |  206 ++++++++++++++++
test/unit/bio/db/fastq/test_quality_score.rb |  330 ++++++++++++++++++++++++++
3 files changed, 544 insertions(+), 104 deletions(-)
create mode 100644 lib/bio/db/fastq/quality_score.rb
create mode 100644 test/unit/bio/db/fastq/test_quality_score.rb</pre>

<p>commit 98f7703c28f0c2c34e4fe1631de227e20b9666c3 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 23:48:19 2009 +0900</p>

<pre>   When no error_probabilities in the sequence and quality_score_type is nil, Fastq formatter implicitly assumes that the quality_score_type is :phred.

    * When no error_probabilities in the sequence and
      quality_score_type is nil, Fastq formatter implicitly assumes
      that the quality_score_type is :phred.
    * Bug fix: fixed typo in lib/bio/db/fastq/format_fastq.rb.

lib/bio/db/fastq/format_fastq.rb |    5 ++++-
lib/bio/sequence.rb              |    3 +++
2 files changed, 7 insertions(+), 1 deletions(-)</pre>

<p>commit f85a6aee9827bc573dcb735f4a1a1827926cc66c Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 23:29:19 2009 +0900</p>

<pre>   Bug fix: fixed typo for Bio::Sequence#quality_score_type.

lib/bio/sequence.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit d81a611d7c7c46a789b86e99cebe064ba559e3e0 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 20:42:12 2009 +0900</p>

<pre>   Splitting lib/bio/db/fasta.rb: FastaNumericFormat is moved to a new file, etc.

    * Splitting lib/bio/db/fasta.rb as follows:
       * Bio::FastaNumericFormat is moved to lib/bio/db/fasta/qual.rb.
       * Demo codes in the &quot;if __FILE__ == $0&quot; are moved to
         sample/demo_fastaformat.rb.
    * Unit tests for Bio::FastaNumericFormat are moved from
      test/unit/bio/db/test_fasta.rb to test/unit/bio/db/test_qual.rb.
    * lib/bio.rb is also modified for the autoload.
    * Bug fix: fixed incorrect autoload path for Bio::FastaDefline.

lib/bio.rb                     |    4 +-
lib/bio/db/fasta.rb            |  135 +---------------------------------------
lib/bio/db/fasta/qual.rb       |  102 ++++++++++++++++++++++++++++++
sample/demo_fastaformat.rb     |  105 +++++++++++++++++++++++++++++++
test/unit/bio/db/test_fasta.rb |   43 -------------
test/unit/bio/db/test_qual.rb  |   63 +++++++++++++++++++
6 files changed, 273 insertions(+), 179 deletions(-)
create mode 100644 lib/bio/db/fasta/qual.rb
create mode 100644 sample/demo_fastaformat.rb
create mode 100644 test/unit/bio/db/test_qual.rb</pre>

<p>commit c70bed5c3f828c94084fdeabe255fbb3930097d0 Author: Andrew Grimm
&lt;andrew.j.grimm@gmail.com&gt; Date:   Sun Aug 16 19:49:38 2009 +1000</p>

<pre>   Removed use of uninitialized variable in FastaNumericFormat.

lib/bio/db/fasta.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit ce6dcc344a3d7beec544d7164308dd97bafa8a19 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 13:04:11 2009 +0900</p>

<pre>   User data type should be stored as is, even if unknown data type.

lib/bio/db/sanger_chromatogram/abif.rb |    3 ++-
1 files changed, 2 insertions(+), 1 deletions(-)</pre>

<p>commit ca96e59f151c2e10b5cd8c0690b8297979e52036 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 12:50:59 2009 +0900</p>

<pre>   Removed Bio::Abif#method_missing and added alternative method

    * Removed Bio::Abif#method_missing, because method_missing can
      hide many errors related to method calls e.g. method name typo,
      and it is not suitable for only getting data.
    * New method Bio::Abif#data is added to get data (alternative of
      the method_missing).

lib/bio/db/sanger_chromatogram/abif.rb |   19 ++++++++++++-------
1 files changed, 12 insertions(+), 7 deletions(-)</pre>

<p>commit 8b0da27523998cb9a9df07f5e907cda6e3cef0dc Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 12:04:41 2009 +0900</p>

<pre>   removed a non-ascii character in comment

lib/bio/db/sanger_chromatogram/chromatogram.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 55c1a180fec97338bae8e3c5b5d5ceec64aed0f6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Nov 13 12:02:53 2009 +0900</p>

<pre>   Bug fix: Bio::SangerChromatogram#complement fails when the object is frozen.

lib/bio/db/sanger_chromatogram/chromatogram.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 4e7d8b0ba304d1ff01364fad68035f7ec9463fb9 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 12 22:23:47 2009 +0900</p>

<pre>   fixed a typo in a copyright line

test/unit/bio/util/test_sirna.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit f376124112d23ba9b0491dbd427d328edc81d872 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Nov 12 22:05:37 2009 +0900</p>

<pre>   The &quot;libpath magic&quot; in tests are replaced by the load of helper routine.

    * In all unit tests, the &quot;libpath magic&quot; are replaced by the load
      of helper routine.
    * Changed to use a constant BioRubyTestDataPath for generating test
      data file path.
    * Some &quot;require&quot; lines are modified.
    * &quot;File.open(...).read&quot; in some tests are replaced by &quot;File.read(...)&quot;.
    * Header comment lines of some tests with wrong filename and/or
      class/module name information are fixed.

test/functional/bio/appl/test_pts1.rb              |    6 ++++--
test/functional/bio/io/test_ensembl.rb             |    7 ++++---
test/functional/bio/io/test_pubmed.rb              |    8 +++++---
test/functional/bio/io/test_soapwsdl.rb            |    9 +++++----
test/functional/bio/io/test_togows.rb              |    9 +++++----
test/functional/bio/sequence/test_output_embl.rb   |   10 ++++++----
test/functional/bio/test_command.rb                |   10 +++++-----
test/runner.rb                                     |    8 +++++---
test/unit/bio/appl/bl2seq/test_report.rb           |    9 +++++----
test/unit/bio/appl/blast/test_ncbioptions.rb       |    6 ++++--
test/unit/bio/appl/blast/test_report.rb            |    9 +++++----
test/unit/bio/appl/blast/test_rpsblast.rb          |    9 +++++----
test/unit/bio/appl/gcg/test_msf.rb                 |   10 +++++-----
test/unit/bio/appl/genscan/test_report.rb          |   17 ++++++++---------
test/unit/bio/appl/hmmer/test_report.rb            |    9 +++++----
test/unit/bio/appl/iprscan/test_report.rb          |   11 ++++++-----
test/unit/bio/appl/mafft/test_report.rb            |   11 ++++++-----
test/unit/bio/appl/paml/codeml/test_rates.rb       |    9 +++++----
test/unit/bio/appl/paml/codeml/test_report.rb      |    9 +++++----
test/unit/bio/appl/paml/test_codeml.rb             |    9 +++++----
test/unit/bio/appl/sim4/test_report.rb             |    9 +++++----
test/unit/bio/appl/sosui/test_report.rb            |   11 ++++++-----
test/unit/bio/appl/targetp/test_report.rb          |    8 +++++---
test/unit/bio/appl/test_blast.rb                   |    9 +++++----
test/unit/bio/appl/test_fasta.rb                   |    6 ++++--
test/unit/bio/appl/test_pts1.rb                    |    6 ++++--
test/unit/bio/appl/tmhmm/test_report.rb            |   11 ++++++-----
test/unit/bio/data/test_aa.rb                      |    8 +++++---
test/unit/bio/data/test_codontable.rb              |    9 +++++----
test/unit/bio/data/test_na.rb                      |    8 +++++---
test/unit/bio/db/biosql/tc_biosql.rb               |    6 +++++-
test/unit/bio/db/embl/test_common.rb               |    6 ++++--
test/unit/bio/db/embl/test_embl.rb                 |   12 ++++++------
test/unit/bio/db/embl/test_embl_rel89.rb           |   12 ++++++------
test/unit/bio/db/embl/test_embl_to_bioseq.rb       |   15 +++++++--------
test/unit/bio/db/embl/test_sptr.rb                 |   14 ++++++--------
test/unit/bio/db/embl/test_uniprot.rb              |   11 ++++++-----
test/unit/bio/db/kegg/test_genes.rb                |    8 +++++---
test/unit/bio/db/pdb/test_pdb.rb                   |    6 ++++--
test/unit/bio/db/sanger_chromatogram/test_abif.rb  |    3 ++-
test/unit/bio/db/sanger_chromatogram/test_scf.rb   |    3 ++-
test/unit/bio/db/test_aaindex.rb                   |   12 ++++++------
test/unit/bio/db/test_fasta.rb                     |    8 +++++---
test/unit/bio/db/test_fastq.rb                     |   10 +++++-----
test/unit/bio/db/test_gff.rb                       |    6 ++++--
test/unit/bio/db/test_lasergene.rb                 |   12 +++++++-----
test/unit/bio/db/test_medline.rb                   |    6 ++++--
test/unit/bio/db/test_newick.rb                    |   12 ++++++------
test/unit/bio/db/test_nexus.rb                     |    6 ++++--
test/unit/bio/db/test_phyloxml.rb                  |   14 +++++++-------
test/unit/bio/db/test_phyloxml_writer.rb           |   15 +++++++--------
test/unit/bio/db/test_prosite.rb                   |   11 ++++++-----
test/unit/bio/db/test_rebase.rb                    |    8 +++++---
test/unit/bio/db/test_soft.rb                      |   13 +++++++------
test/unit/bio/io/flatfile/test_autodetection.rb    |   13 ++++++-------
test/unit/bio/io/flatfile/test_buffer.rb           |   11 ++++++-----
test/unit/bio/io/flatfile/test_splitter.rb         |    8 ++++----
test/unit/bio/io/test_ddbjxml.rb                   |    7 ++++---
test/unit/bio/io/test_ensembl.rb                   |    8 +++++---
test/unit/bio/io/test_fastacmd.rb                  |    7 ++++---
test/unit/bio/io/test_flatfile.rb                  |   11 ++++++-----
test/unit/bio/io/test_soapwsdl.rb                  |    7 ++++---
test/unit/bio/io/test_togows.rb                    |    6 ++++--
test/unit/bio/sequence/test_aa.rb                  |    8 +++++---
test/unit/bio/sequence/test_common.rb              |    6 ++++--
test/unit/bio/sequence/test_compat.rb              |    6 ++++--
test/unit/bio/sequence/test_dblink.rb              |    8 +++++---
test/unit/bio/sequence/test_na.rb                  |    6 ++++--
test/unit/bio/shell/plugin/test_seq.rb             |    8 +++++---
test/unit/bio/test_alignment.rb                    |    8 +++++---
test/unit/bio/test_command.rb                      |    7 ++++---
test/unit/bio/test_db.rb                           |    8 +++++---
test/unit/bio/test_feature.rb                      |    6 ++++--
test/unit/bio/test_location.rb                     |    6 ++++--
test/unit/bio/test_map.rb                          |    8 +++++---
test/unit/bio/test_pathway.rb                      |    6 ++++--
test/unit/bio/test_reference.rb                    |    6 ++++--
test/unit/bio/test_sequence.rb                     |    8 +++++---
test/unit/bio/test_shell.rb                        |    8 +++++---
test/unit/bio/test_tree.rb                         |   12 ++++++------
.../analysis/test_calculated_cuts.rb               |    6 ++++--
.../restriction_enzyme/analysis/test_cut_ranges.rb |    6 ++++--
.../analysis/test_sequence_range.rb                |    6 ++++--
.../double_stranded/test_aligned_strands.rb        |    6 ++++--
.../double_stranded/test_cut_location_pair.rb      |    6 ++++--
.../test_cut_location_pair_in_enzyme_notation.rb   |    6 ++++--
.../double_stranded/test_cut_locations.rb          |    6 ++++--
.../test_cut_locations_in_enzyme_notation.rb       |    6 ++++--
.../test_cut_locations_in_enzyme_notation.rb       |    6 ++++--
.../bio/util/restriction_enzyme/test_analysis.rb   |    6 ++++--
.../bio/util/restriction_enzyme/test_cut_symbol.rb |    6 ++++--
.../restriction_enzyme/test_double_stranded.rb     |    6 ++++--
.../util/restriction_enzyme/test_single_strand.rb  |    6 ++++--
.../test_single_strand_complement.rb               |    6 ++++--
.../restriction_enzyme/test_string_formatting.rb   |    6 ++++--
test/unit/bio/util/test_color_scheme.rb            |    8 +++++---
test/unit/bio/util/test_contingency_table.rb       |    8 +++++---
test/unit/bio/util/test_restriction_enzyme.rb      |    6 ++++--
test/unit/bio/util/test_sirna.rb                   |    8 +++++---
99 files changed, 479 insertions(+), 343 deletions(-)</pre>

<p>commit f4fa0a5edc6ff6fc35577d84bda86363014a57a4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 11 17:04:43 2009 +0900</p>

<pre>   test_chromatogram.rb is splitted into test_abif.rb and test_scf.rb

test/unit/bio/db/sanger_chromatogram/test_abif.rb  |   75 +++++++++++++++
.../db/sanger_chromatogram/test_chromatogram.rb    |  101 --------------------
test/unit/bio/db/sanger_chromatogram/test_scf.rb   |   97 +++++++++++++++++++
3 files changed, 172 insertions(+), 101 deletions(-)
create mode 100644 test/unit/bio/db/sanger_chromatogram/test_abif.rb
delete mode 100644 test/unit/bio/db/sanger_chromatogram/test_chromatogram.rb
create mode 100644 test/unit/bio/db/sanger_chromatogram/test_scf.rb</pre>

<p>commit d9cc613273cadc7f9fdfe2bafbd933efb1f403ca Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Nov 11 17:01:37 2009 +0900</p>

<pre>   Newly added unit test helper routine which aims to replace the libpath magic

test/bioruby_test_helper.rb |   61 +++++++++++++++++++++++++++++++++++++++++++
1 files changed, 61 insertions(+), 0 deletions(-)
create mode 100644 test/bioruby_test_helper.rb</pre>

<p>commit 10e76db2a8ec37bde541157d0735303b4ca8b3b8 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Nov 10 20:59:01 2009 +0900</p>

<pre>   Bio::SangerChromatogram#to_s is renamed to sequence_string.

lib/bio/db/sanger_chromatogram/chromatogram.rb |    2 +-
1 files changed, 1 insertions(+), 1 deletions(-)</pre>

<p>commit 0ec2c2f38c4b4a3e451841dc32540dfa10743bc2 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 30 22:49:19 2009 +0900</p>

<pre>   Renamed/moved files/directories following the rename of class names.

    * renamed: lib/bio/db/chromatogram.rb -&gt;
               lib/bio/db/sanger_chromatogram/chromatogram.rb
    * renamed: lib/bio/db/chromatogram/abi.rb -&gt;
               lib/bio/db/sanger_chromatogram/abif.rb
    * renamed: lib/bio/db/chromatogram/scf.rb -&gt;
               lib/bio/db/sanger_chromatogram/scf.rb
    * renamed: lib/bio/db/chromatogram/chromatogram_to_biosequence.rb -&gt;
               lib/bio/db/sanger_chromatogram/chromatogram_to_biosequence.rb
    * renamed: test/unit/bio/db/test_chromatogram.rb -&gt;
               test/unit/bio/db/sanger_chromatogram/test_chromatogram.rb
    * renamed: test/data/chromatogram/test_chromatogram_abi.ab1 -&gt;
               test/data/sanger_chromatogram/test_chromatogram_abif.ab1
    * renamed: test/data/chromatogram/*.scf -&gt;
               test/data/sanger_chromatogram/*.scf

lib/bio/db/chromatogram.rb                         |  133 -------------
lib/bio/db/chromatogram/abi.rb                     |  114 -----------
.../db/chromatogram/chromatogram_to_biosequence.rb |   32 ---
lib/bio/db/chromatogram/scf.rb                     |  210 --------------------
lib/bio/db/sanger_chromatogram/abif.rb             |  114 +++++++++++
lib/bio/db/sanger_chromatogram/chromatogram.rb     |  133 +++++++++++++
.../chromatogram_to_biosequence.rb                 |   32 +++
lib/bio/db/sanger_chromatogram/scf.rb              |  210 ++++++++++++++++++++
test/data/chromatogram/test_chromatogram_abi.ab1   |  Bin 228656 -&gt; 0 bytes
.../data/chromatogram/test_chromatogram_scf_v2.scf |  Bin 47503 -&gt; 0 bytes
.../data/chromatogram/test_chromatogram_scf_v3.scf |  Bin 47503 -&gt; 0 bytes
.../sanger_chromatogram/test_chromatogram_abif.ab1 |  Bin 0 -&gt; 228656 bytes
.../test_chromatogram_scf_v2.scf                   |  Bin 0 -&gt; 47503 bytes
.../test_chromatogram_scf_v3.scf                   |  Bin 0 -&gt; 47503 bytes
.../db/sanger_chromatogram/test_chromatogram.rb    |  101 ++++++++++
test/unit/bio/db/test_chromatogram.rb              |  101 ----------
16 files changed, 590 insertions(+), 590 deletions(-)
delete mode 100644 lib/bio/db/chromatogram.rb
delete mode 100644 lib/bio/db/chromatogram/abi.rb
delete mode 100644 lib/bio/db/chromatogram/chromatogram_to_biosequence.rb
delete mode 100644 lib/bio/db/chromatogram/scf.rb
create mode 100644 lib/bio/db/sanger_chromatogram/abif.rb
create mode 100644 lib/bio/db/sanger_chromatogram/chromatogram.rb
create mode 100644 lib/bio/db/sanger_chromatogram/chromatogram_to_biosequence.rb
create mode 100644 lib/bio/db/sanger_chromatogram/scf.rb
delete mode 100644 test/data/chromatogram/test_chromatogram_abi.ab1
delete mode 100644 test/data/chromatogram/test_chromatogram_scf_v2.scf
delete mode 100644 test/data/chromatogram/test_chromatogram_scf_v3.scf
create mode 100644 test/data/sanger_chromatogram/test_chromatogram_abif.ab1
create mode 100644 test/data/sanger_chromatogram/test_chromatogram_scf_v2.scf
create mode 100644 test/data/sanger_chromatogram/test_chromatogram_scf_v3.scf
create mode 100644 test/unit/bio/db/sanger_chromatogram/test_chromatogram.rb
delete mode 100644 test/unit/bio/db/test_chromatogram.rb</pre>

<p>commit 49bfe319e535c8414be32b47c07fe5204a24b398 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 30 22:02:08 2009 +0900</p>

<pre>   Renamed Chromatogram to SangerChromatogram and Abi to Abif, and preparation of filename changes.

    * Renamed Chromatogram to SangerChromatogram because the word
      &quot;chromatogram&quot; may be used by various experimental methods
      other than the Sanger chromatogram.
    * Renamed Abi to Abif because Applied Biosystems who determined
      the file format says that its name is ABIF.
    * Preparation of changing filenames. However, filenames are
      not really changed now because of recording history of file
      contents modification. The paths shown in the &quot;require&quot; lines
      and test data paths may not be existed now.

lib/bio.rb                                         |    6 ++--
lib/bio/db/chromatogram.rb                         |   24 +++++++-------
lib/bio/db/chromatogram/abi.rb                     |   15 +++++----
.../db/chromatogram/chromatogram_to_biosequence.rb |   10 +++---
lib/bio/db/chromatogram/scf.rb                     |   15 +++++----
lib/bio/sequence/adapter.rb                        |    3 +-
test/unit/bio/db/test_chromatogram.rb              |   32 ++++++++++---------
7 files changed, 57 insertions(+), 48 deletions(-)</pre>

<p>commit 6c020440663214014973ae8e5007ce2d31d8d45e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 24 00:44:47 2009 +0900</p>

<pre>   New class method Bio::PhyloXML::Parser.open(filename) and API change of new(), etc.

    * New class methods to create parser object from various data
      source are added: Bio::PhyloXML::Parser.open(filename),
      for_io(io), open_uri(uri).
    * API change of Bio::PhyloXML::Parser.new(). Now, new(filename)
      is deprecated and it can only take a XML-formatted string.
    * Tests are added and modified to reflect the above changes.
    * test/unit/bio/db/test_phyloxml_writer.rb: avoid using WeakRef
      for temporary directory maintenance.

lib/bio/db/phyloxml/phyloxml_parser.rb   |  224 +++++++++++++++++++++++++++---
lib/bio/db/phyloxml/phyloxml_writer.rb   |    4 +-
test/unit/bio/db/test_phyloxml.rb        |  178 ++++++++++++++++++++++--
test/unit/bio/db/test_phyloxml_writer.rb |   70 +++++-----
4 files changed, 408 insertions(+), 68 deletions(-)</pre>

<p>commit fca5e800fc051a38ac6d25652c684fdd4f9bff14 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 23 15:13:25 2009 +0900</p>

<pre>   Rearrangement of require and autoload so as to correctly load PhyloXML classes

lib/bio.rb                               |   11 +++++++----
lib/bio/db/phyloxml/phyloxml_elements.rb |   16 +++++++++++++++-
lib/bio/db/phyloxml/phyloxml_parser.rb   |   11 ++++++-----
lib/bio/db/phyloxml/phyloxml_writer.rb   |    5 ++++-
test/unit/bio/db/test_phyloxml.rb        |    5 -----
test/unit/bio/db/test_phyloxml_writer.rb |    3 ---
6 files changed, 32 insertions(+), 19 deletions(-)</pre>

<p>commit a291af62ef262ee04f3a0e1b6415d4e256c56a94 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Oct 23 00:08:44 2009 +0900</p>

<pre>   Fixed argument order of assert_equal(expected, actual), etc.

    * Test bug fix: Argument order of assert_equal must be
      assert_equal(expected, actual).
    * assert_instance_of() instead of assert_equal() in
      TestPhyloXML1#test_init.
    * Removed some commented-out tests which may not be needed.

test/unit/bio/db/test_phyloxml.rb        |  295 +++++++++++++++---------------
test/unit/bio/db/test_phyloxml_writer.rb |    8 +-
2 files changed, 147 insertions(+), 156 deletions(-)</pre>

<p>commit 152304dc9809102f56a2f1779c59111f84b9cd02 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 17 01:40:49 2009 +0900</p>

<pre>   Improvement of tests for Bio::Fastq and related classes.

test/unit/bio/db/test_fastq.rb |  372 ++++++++++++++++++++++++++--------------
1 files changed, 245 insertions(+), 127 deletions(-)</pre>

<p>commit 61556223a469a5f8b1bb4f343eca92c88c66cb9a Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 17 01:38:52 2009 +0900</p>

<pre>   FASTQ output support is added to Bio::Sequence.

lib/bio/db/fastq/format_fastq.rb |  172 ++++++++++++++++++++++++++++++++++++++
lib/bio/sequence/format.rb       |    9 ++
2 files changed, 181 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/fastq/format_fastq.rb</pre>

<p>commit ea4203ebb7ca268a5b6d6c50aeb63ed0eed5a803 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 17 01:32:50 2009 +0900</p>

<pre>   New attributes for genome sequencer data are added to Bio::Sequence.

    * New attributes for genome sequencer data are added to
      Bio::Sequence class: quality_scores, quality_scores_type,
      error_probabilities.

lib/bio/sequence.rb |   13 +++++++++++++
1 files changed, 13 insertions(+), 0 deletions(-)</pre>

<p>commit fce158b2194519081361e12c170882ec2e87fc5e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Oct 17 01:13:27 2009 +0900</p>

<pre>   New methods Bio::Fastq#to_biosequence, etc. and improvement of tolerance for overflows

    * Bio::Fastq#to_biosequence is newly added.
    * New methods: Bio::Fastq#seq, entry_id, quality_score_type.
    * Default behavior of Bio::Fastq::FormatData#scores2str is changed
      not to raise error but to truncate saturated values.
    * Improvement of tolerance for overflows, and preventing to calculate
      log of negative number.

lib/bio/db/fastq.rb                      |  105 ++++++++++++++++++++++++++++--
lib/bio/db/fastq/fastq_to_biosequence.rb |   40 +++++++++++
lib/bio/sequence/adapter.rb              |    1 +
3 files changed, 139 insertions(+), 7 deletions(-)
create mode 100644 lib/bio/db/fastq/fastq_to_biosequence.rb</pre>

<p>commit 0f189974d2027cecee575b27e969de7f62508309 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Oct 13 21:41:58 2009 +0900</p>

<pre>   Avoid using Numeric#fdiv because it can only be used in Ruby 1.8.7 or later

lib/bio/db/fastq.rb            |    6 +++---
test/unit/bio/db/test_fastq.rb |   10 +++++-----
2 files changed, 8 insertions(+), 8 deletions(-)</pre>

<p>commit 42999fc6230e52c4f241f411d299db941196f62e Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Tue Oct 13 21:30:25 2009 +0900</p>

<pre>   Bio::Fastq#qualities is renamed to quality_scores.

    * Bio::Fastq#qualities is renamed to Bio::Fastq#quality_scores, and
      the original method name is changed to be an alias of the new name.

lib/bio/db/fastq.rb            |   16 +++++++++-------
test/unit/bio/db/test_fastq.rb |   30 +++++++++++++++---------------
2 files changed, 24 insertions(+), 22 deletions(-)</pre>

<p>commit cc0ee2169f298046c5e55fcbadfeaac01f6bf704 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Oct 11 19:19:18 2009 +0900</p>

<pre>   Newly added unit tests for Bio::Fastq with test data

    * Newly added unit tests for Bio::Fastq with test data. The test data
      is created by P.J.A. Cock et al., and is also used in Biopython and
      BioPerl.

test/data/fastq/error_diff_ids.fastq               |   20 +
test/data/fastq/error_double_qual.fastq            |   22 +
test/data/fastq/error_double_seq.fastq             |   22 +
test/data/fastq/error_long_qual.fastq              |   20 +
test/data/fastq/error_no_qual.fastq                |   20 +
test/data/fastq/error_qual_del.fastq               |   20 +
test/data/fastq/error_qual_escape.fastq            |   20 +
test/data/fastq/error_qual_null.fastq              |  Bin 0 -&gt; 610 bytes
test/data/fastq/error_qual_space.fastq             |   21 +
test/data/fastq/error_qual_tab.fastq               |   21 +
test/data/fastq/error_qual_unit_sep.fastq          |   20 +
test/data/fastq/error_qual_vtab.fastq              |   20 +
test/data/fastq/error_short_qual.fastq             |   20 +
test/data/fastq/error_spaces.fastq                 |   20 +
test/data/fastq/error_tabs.fastq                   |   21 +
test/data/fastq/error_trunc_at_plus.fastq          |   19 +
test/data/fastq/error_trunc_at_qual.fastq          |   19 +
test/data/fastq/error_trunc_at_seq.fastq           |   18 +
test/data/fastq/error_trunc_in_plus.fastq          |   19 +
test/data/fastq/error_trunc_in_qual.fastq          |   20 +
test/data/fastq/error_trunc_in_seq.fastq           |   18 +
test/data/fastq/error_trunc_in_title.fastq         |   17 +
.../fastq/illumina_full_range_as_illumina.fastq    |    8 +
.../data/fastq/illumina_full_range_as_sanger.fastq |    8 +
.../data/fastq/illumina_full_range_as_solexa.fastq |    8 +
.../illumina_full_range_original_illumina.fastq    |    8 +
test/data/fastq/longreads_as_illumina.fastq        |   40 ++
test/data/fastq/longreads_as_sanger.fastq          |   40 ++
test/data/fastq/longreads_as_solexa.fastq          |   40 ++
test/data/fastq/longreads_original_sanger.fastq    |  120 ++++
test/data/fastq/misc_dna_as_illumina.fastq         |   16 +
test/data/fastq/misc_dna_as_sanger.fastq           |   16 +
test/data/fastq/misc_dna_as_solexa.fastq           |   16 +
test/data/fastq/misc_dna_original_sanger.fastq     |   16 +
test/data/fastq/misc_rna_as_illumina.fastq         |   16 +
test/data/fastq/misc_rna_as_sanger.fastq           |   16 +
test/data/fastq/misc_rna_as_solexa.fastq           |   16 +
test/data/fastq/misc_rna_original_sanger.fastq     |   16 +
.../data/fastq/sanger_full_range_as_illumina.fastq |    8 +
test/data/fastq/sanger_full_range_as_sanger.fastq  |    8 +
test/data/fastq/sanger_full_range_as_solexa.fastq  |    8 +
.../fastq/sanger_full_range_original_sanger.fastq  |    8 +
.../data/fastq/solexa_full_range_as_illumina.fastq |    8 +
test/data/fastq/solexa_full_range_as_sanger.fastq  |    8 +
test/data/fastq/solexa_full_range_as_solexa.fastq  |    8 +
.../fastq/solexa_full_range_original_solexa.fastq  |    8 +
test/data/fastq/wrapping_as_illumina.fastq         |   12 +
test/data/fastq/wrapping_as_sanger.fastq           |   12 +
test/data/fastq/wrapping_as_solexa.fastq           |   12 +
test/data/fastq/wrapping_original_sanger.fastq     |   24 +
test/unit/bio/db/test_fastq.rb                     |  711 ++++++++++++++++++++
51 files changed, 1652 insertions(+), 0 deletions(-)
create mode 100644 test/data/fastq/error_diff_ids.fastq
create mode 100644 test/data/fastq/error_double_qual.fastq
create mode 100644 test/data/fastq/error_double_seq.fastq
create mode 100644 test/data/fastq/error_long_qual.fastq
create mode 100644 test/data/fastq/error_no_qual.fastq
create mode 100644 test/data/fastq/error_qual_del.fastq
create mode 100644 test/data/fastq/error_qual_escape.fastq
create mode 100644 test/data/fastq/error_qual_null.fastq
create mode 100644 test/data/fastq/error_qual_space.fastq
create mode 100644 test/data/fastq/error_qual_tab.fastq
create mode 100644 test/data/fastq/error_qual_unit_sep.fastq
create mode 100644 test/data/fastq/error_qual_vtab.fastq
create mode 100644 test/data/fastq/error_short_qual.fastq
create mode 100644 test/data/fastq/error_spaces.fastq
create mode 100644 test/data/fastq/error_tabs.fastq
create mode 100644 test/data/fastq/error_trunc_at_plus.fastq
create mode 100644 test/data/fastq/error_trunc_at_qual.fastq
create mode 100644 test/data/fastq/error_trunc_at_seq.fastq
create mode 100644 test/data/fastq/error_trunc_in_plus.fastq
create mode 100644 test/data/fastq/error_trunc_in_qual.fastq
create mode 100644 test/data/fastq/error_trunc_in_seq.fastq
create mode 100644 test/data/fastq/error_trunc_in_title.fastq
create mode 100644 test/data/fastq/illumina_full_range_as_illumina.fastq
create mode 100644 test/data/fastq/illumina_full_range_as_sanger.fastq
create mode 100644 test/data/fastq/illumina_full_range_as_solexa.fastq
create mode 100644 test/data/fastq/illumina_full_range_original_illumina.fastq
create mode 100644 test/data/fastq/longreads_as_illumina.fastq
create mode 100644 test/data/fastq/longreads_as_sanger.fastq
create mode 100644 test/data/fastq/longreads_as_solexa.fastq
create mode 100644 test/data/fastq/longreads_original_sanger.fastq
create mode 100644 test/data/fastq/misc_dna_as_illumina.fastq
create mode 100644 test/data/fastq/misc_dna_as_sanger.fastq
create mode 100644 test/data/fastq/misc_dna_as_solexa.fastq
create mode 100644 test/data/fastq/misc_dna_original_sanger.fastq
create mode 100644 test/data/fastq/misc_rna_as_illumina.fastq
create mode 100644 test/data/fastq/misc_rna_as_sanger.fastq
create mode 100644 test/data/fastq/misc_rna_as_solexa.fastq
create mode 100644 test/data/fastq/misc_rna_original_sanger.fastq
create mode 100644 test/data/fastq/sanger_full_range_as_illumina.fastq
create mode 100644 test/data/fastq/sanger_full_range_as_sanger.fastq
create mode 100644 test/data/fastq/sanger_full_range_as_solexa.fastq
create mode 100644 test/data/fastq/sanger_full_range_original_sanger.fastq
create mode 100644 test/data/fastq/solexa_full_range_as_illumina.fastq
create mode 100644 test/data/fastq/solexa_full_range_as_sanger.fastq
create mode 100644 test/data/fastq/solexa_full_range_as_solexa.fastq
create mode 100644 test/data/fastq/solexa_full_range_original_solexa.fastq
create mode 100644 test/data/fastq/wrapping_as_illumina.fastq
create mode 100644 test/data/fastq/wrapping_as_sanger.fastq
create mode 100644 test/data/fastq/wrapping_as_solexa.fastq
create mode 100644 test/data/fastq/wrapping_original_sanger.fastq
create mode 100644 test/unit/bio/db/test_fastq.rb</pre>

<p>commit 951d8f7303a5c28783a2c8b25c9fb347730c1a8f Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Oct 11 19:10:15 2009 +0900</p>

<pre>   Bio::Fastq API changed.

    * Bio::Fastq API changed. Removed methods: phred_quality, solexa_quality.
      New methods: qualities, error_probabilities, format, format=,
      validate_format.
    * New exception classes Bio::Fastq::Error::* for errors.
    * Internal structure is also changed. Internal only classes
      Bio::Fastq::FormatData::* which store parameters for format variants.

lib/bio/db/fastq.rb |  519 +++++++++++++++++++++++++++++++++++++++++++++++++--
1 files changed, 501 insertions(+), 18 deletions(-)</pre>

<p>commit 9bb7f6ca762c615e50d98c35b60982a4caeea323 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Sep 25 23:36:13 2009 +0900</p>

<pre>   Bug fix: infinite loop in Bio::Fastq.new. Thanks to Hiroyuki Mishima for reporting the bug.

lib/bio/db/fastq.rb |   16 ++++++++++------
1 files changed, 10 insertions(+), 6 deletions(-)</pre>

<p>commit fca6aa5333a95db4dc87e8fc814bd028d5720de4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Mar 20 11:52:33 2009 +0900</p>

<pre>   Added file format autodetection for Bio::Fastq

lib/bio/io/flatfile/autodetection.rb |    6 ++++++
1 files changed, 6 insertions(+), 0 deletions(-)</pre>

<p>commit 1ba21545e7d49ae8b775fbed7a4e92b1daa54ac6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Mar 20 11:48:59 2009 +0900</p>

<pre>   Added autoload for Bio::Fastq

lib/bio.rb |    1 +
1 files changed, 1 insertions(+), 0 deletions(-)</pre>

<p>commit 380b99106d4c7955b9d07ee8668b53d384c974f4 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Mar 19 17:07:25 2009 +0900</p>

<pre>   Newly added FASTQ format parser (still a prototype)

lib/bio/db/fastq.rb |  162 +++++++++++++++++++++++++++++++++++++++++++++++++++
1 files changed, 162 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/fastq.rb</pre>

<p>commit 2c5df2a5f1b5ae1ea9e61c1dccc8bcd2f496f6ce Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Sep 20 19:08:55 2009 +0900</p>

<pre>   Removed &quot;require 'rubygems'&quot;.

lib/bio/db/phyloxml/phyloxml_parser.rb |    2 --
1 files changed, 0 insertions(+), 2 deletions(-)</pre>

<p>commit 67818d2550e5d53eeee0f3d710f66f7506fb8127 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sat Sep 19 17:06:21 2009 +0900</p>

<pre>   Use Bio::PubMed.esearch and efetch, etc.

    * Changed to use Bio::PubMed.esearch and efetch instead of
      deprecated methods.
    * Regular expression for extracting option is changed.

sample/pmfetch.rb  |   15 +++++++++++----
sample/pmsearch.rb |   17 +++++++++++++----
2 files changed, 24 insertions(+), 8 deletions(-)</pre>

<p>commit 0c95889bf69e3140b5f09ade1203d50136aee014 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Sep 18 17:58:17 2009 +0900</p>

<pre>   Changed to use temporary directory when writing a file, etc.

     * To avoid unexpected file corruption and possibly security risk,
       changed to use temporary directory when writing files. The
       temporary directory is normally removed when all tests end.
       To prevent removing the directory, set environment variable
       BIORUBY_TEST_DEBUG.
     * To avoid test class name conflict, TestPhyloXMLData is renamed
       to TestPhyloXMLWriterData.
     * Added a new test to check existence of libxml-ruby, and removed
       code to raise error when it is not found. The code of the new
       test is completely the same as of in test_phyloxml.rb, but it
       is added for the purpose when test_phyloxml_writer.rb is called
       independently.

test/unit/bio/db/test_phyloxml_writer.rb |  161 +++++++++++++++++++++---------
1 files changed, 112 insertions(+), 49 deletions(-)</pre>

<p>commit 520d0f5ed535f621aed60b71d8765a99e97306a6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Sun Sep 20 18:34:19 2009 +0900</p>

<pre>   Newly added internal-only class Bio::Command::Tmpdir to handle temporary directory

    * Newly added internal-only class Bio::Command::Tmpdir to handle
      temporary directory. It is BioRuby library internal use only.
    * Bio::Command.mktmpdir is changed to be completely compatible
      with Ruby 1.9.x's Dir.mktmpdir.

lib/bio/command.rb                  |  104 +++++++++++++++++++++++++++++++---
test/functional/bio/test_command.rb |   49 ++++++++++++++++
2 files changed, 143 insertions(+), 10 deletions(-)</pre>

<p>commit c813b60ae62f44d9688b21d47c84e4b7083547e6 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Fri Sep 18 17:55:14 2009 +0900</p>

<pre>   Added new test to check existence of libxml-ruby, instead of raising error.

test/unit/bio/db/test_phyloxml.rb |   30 ++++++++++++++++++++----------
1 files changed, 20 insertions(+), 10 deletions(-)</pre>

<p>commit 1b71dd9624640f3f775baab360eef0be92a86677 Author: Diana Jaunzeikare
&lt;latvianlinuxgirl@gmail.com&gt; Date:   Fri Sep 18 21:43:18 2009 -0400</p>

<pre>   Renamed output files generated by phyloxml_writer unit tests.

test/unit/bio/db/test_phyloxml_writer.rb |   13 ++++++++++---
1 files changed, 10 insertions(+), 3 deletions(-)</pre>

<p>commit f8e138cb9e28996f1024fa9cf7c68c8f08603941 Author: Diana Jaunzeikare
&lt;latvianlinuxgirl@gmail.com&gt; Date:   Fri Sep 18 21:33:35 2009 -0400</p>

<pre>   Added ncbi_taxonomy_mollusca_short.xml test file

.../data/phyloxml/ncbi_taxonomy_mollusca_short.xml |   65 ++++++++++++++++++++
1 files changed, 65 insertions(+), 0 deletions(-)
create mode 100644 test/data/phyloxml/ncbi_taxonomy_mollusca_short.xml</pre>

<p>commit be1be310b7581928581cde24303fe2e16c04e82f Author: Diana Jaunzeikare
&lt;latvianlinuxgirl@gmail.com&gt; Date:   Fri Sep 18 21:29:20 2009 -0400</p>

<pre>   Made the code compactible with libxml-ruby 1.1.3 (previous was 0.9.4) version.

lib/bio/db/phyloxml/phyloxml_elements.rb |   58 +++++++++++++++---------------
lib/bio/db/phyloxml/phyloxml_parser.rb   |   10 ++++-
lib/bio/db/phyloxml/phyloxml_writer.rb   |    8 +++--
3 files changed, 42 insertions(+), 34 deletions(-)</pre>

<p>commit a3441afd5650069a5ada64b202a0714e8723e911 Author: Diana Jaunzeikare
&lt;latvianlinuxgirl@gmail.com&gt; Date:   Tue May 26 00:55:47 2009 -0400</p>

<pre>   Newly added PhyloXML support written by Diana Jaunzeikare.

    * Newly added PhyloXML support written by Diana Jaunzeikare.
      It have been written during the Google Summer of Code 2009
      &quot;Implementing phyloXML support in BioRuby&quot;, mentored by
      Christian Zmasek et al. with NESCent. For details of development,
      see git://github.com/latvianlinuxgirl/bioruby.git and BioRuby
      mailing list archives.
    * This is a combination of 119 commits. The last commit date was
      Mon Aug 17 10:30:10 2009 -0400.

README.rdoc                              |    3 +
doc/Tutorial.rd                          |  120 ++-
lib/bio.rb                               |    6 +
lib/bio/db/phyloxml/phyloxml.xsd         |  573 ++++++++
lib/bio/db/phyloxml/phyloxml_elements.rb | 1160 +++++++++++++++++
lib/bio/db/phyloxml/phyloxml_parser.rb   |  767 +++++++++++
lib/bio/db/phyloxml/phyloxml_writer.rb   |  223 ++++
test/data/phyloxml/apaf.xml              |  666 ++++++++++
test/data/phyloxml/bcl_2.xml             | 2097 ++++++++++++++++++++++++++++++
test/data/phyloxml/made_up.xml           |  144 ++
test/data/phyloxml/phyloxml_examples.xml |  415 ++++++
test/unit/bio/db/test_phyloxml.rb        |  619 +++++++++
test/unit/bio/db/test_phyloxml_writer.rb |  258 ++++
13 files changed, 7050 insertions(+), 1 deletions(-)
create mode 100644 lib/bio/db/phyloxml/phyloxml.xsd
create mode 100644 lib/bio/db/phyloxml/phyloxml_elements.rb
create mode 100644 lib/bio/db/phyloxml/phyloxml_parser.rb
create mode 100644 lib/bio/db/phyloxml/phyloxml_writer.rb
create mode 100644 test/data/phyloxml/apaf.xml
create mode 100644 test/data/phyloxml/bcl_2.xml
create mode 100644 test/data/phyloxml/made_up.xml
create mode 100644 test/data/phyloxml/phyloxml_examples.xml
create mode 100644 test/unit/bio/db/test_phyloxml.rb
create mode 100644 test/unit/bio/db/test_phyloxml_writer.rb</pre>

<p>commit fd8281f03423ddf23f7d409863b4df647f1b1564 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Sep 9 21:08:15 2009 +0900</p>

<pre>   Newly added Chromatogram classes contributed by Anthony Underwood.

    * Newly added Chromatogram classes contributed by Anthony Underwood.
      See git://github.com/aunderwo/bioruby.git for details of development
      before this merge.

lib/bio.rb                                         |    3 +
lib/bio/db/chromatogram.rb                         |  133 +++++++++++++
lib/bio/db/chromatogram/abi.rb                     |  111 +++++++++++
.../db/chromatogram/chromatogram_to_biosequence.rb |   32 +++
lib/bio/db/chromatogram/scf.rb                     |  207 ++++++++++++++++++++
lib/bio/sequence/adapter.rb                        |    1 +
test/data/chromatogram/test_chromatogram_abi.ab1   |  Bin 0 -&gt; 228656 bytes
.../data/chromatogram/test_chromatogram_scf_v2.scf |  Bin 0 -&gt; 47503 bytes
.../data/chromatogram/test_chromatogram_scf_v3.scf |  Bin 0 -&gt; 47503 bytes
test/unit/bio/db/test_chromatogram.rb              |   99 ++++++++++
10 files changed, 586 insertions(+), 0 deletions(-)
create mode 100644 lib/bio/db/chromatogram.rb
create mode 100644 lib/bio/db/chromatogram/abi.rb
create mode 100644 lib/bio/db/chromatogram/chromatogram_to_biosequence.rb
create mode 100644 lib/bio/db/chromatogram/scf.rb
create mode 100644 test/data/chromatogram/test_chromatogram_abi.ab1
create mode 100644 test/data/chromatogram/test_chromatogram_scf_v2.scf
create mode 100644 test/data/chromatogram/test_chromatogram_scf_v3.scf
create mode 100644 test/unit/bio/db/test_chromatogram.rb</pre>

<p>commit 78f9463b764687401ff4a7480c1383c5594e5133 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Thu Sep 10 12:38:25 2009 +0900</p>

<pre>   Bio::BIORUBY_EXTRA_VERSION is changed to &quot;.5000&quot;.

bioruby.gemspec    |    2 +-
lib/bio/version.rb |    2 +-
2 files changed, 2 insertions(+), 2 deletions(-)</pre>

<p>commit e731c6e52bc9a672e4546eeca4f2d2d968bdba09 Author: Naohisa Goto
&lt;ng@bioruby.org&gt; Date:   Wed Sep 2 15:24:00 2009 +0900</p>

<pre>   BioRuby 1.3.1 is released.

   ChangeLog is modified, and bioruby.gemspec is regenerated.

ChangeLog       |   11 +++++++++++
bioruby.gemspec |    2 +-
2 files changed, 12 insertions(+), 1 deletions(-)</pre>

  </div>

  <div id="validator-badges">
    <p><small><a href="http://validator.w3.org/check/referer">[Validate]</a></small></p>
    <p><small>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish
      Rdoc Generator</a> 2</small>.</p>
  </div>
</body>
</html>

