<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1" />

<title>README_DEV - BioRuby API documentation</title>

  <meta name="keywords" content="ruby,documentation,README_DEV">
  <meta name="description" content="README_DEV: README. Copyright (C) 2005, 2006 Toshiaki Katayama &lt;k@bioruby.">

<script type="text/javascript">
  var rdoc_rel_prefix = "./";
  var index_rel_prefix = "./";
</script>

<script src="./js/navigation.js" defer></script>
<script src="./js/search.js" defer></script>
<script src="./js/search_index.js" defer></script>
<script src="./js/searcher.js" defer></script>
<script src="./js/darkfish.js" defer></script>

<link href="./css/fonts.css" rel="stylesheet">
<link href="./css/rdoc.css" rel="stylesheet">


<body id="top" role="document" class="file">
<div id="navigation-toggle" role="button" tabindex="0" aria-label="Toggle sidebar" aria-expanded="true" aria-controls="navigation">
  <span aria-hidden="true">&#9776;</span>
</div>


<nav id="navigation" role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="./index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="./table_of_contents.html#pages">Pages</a>
    <a href="./table_of_contents.html#classes">Classes</a>
    <a href="./table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search (/) for a class, method, ..." spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>



  <ul class="link-list" role="directory">
              <li>      <a href="#label-README.DEV">README.DEV</a>
          <li>
            <details open>
              <summary>      <a href="#label-HOW+TO+CONTRIBUTE+TO+THE+BIORUBY+PROJECT-3F">HOW TO CONTRIBUTE TO THE BIORUBY PROJECT?</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>
            <details open>
              <summary>      <a href="#label-Git">Git</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Preparation+before+sending+patches+or+pull+requests">Preparation before sending patches or pull requests</a>
          <li>
            <details open>
              <summary>      <a href="#label-Sending+your+contribution">Sending your contribution</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-With+patches">With patches</a>
          <li>      <a href="#label-With+pull+requests">With pull requests</a>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-Notes+for+the+treatment+of+contributions+in+the+blessed+repository">Notes for the treatment of contributions in the blessed repository</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Merging+policy">Merging policy</a>
          <li>      <a href="#label-Git+commit+management+policy">Git commit management policy</a>

              </ul>
            </details>
          </li>

              </ul>
            </details>
          </li>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-LICENSE">LICENSE</a>
          <li>
            <details open>
              <summary>      <a href="#label-CODING+STYLE">CODING STYLE</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Use+the+following+naming+conventions">Use the following naming conventions</a>
          <li>      <a href="#label-Indentation+must+not+include+tabs">Indentation must not include tabs</a>
          <li>      <a href="#label-Parenthesis+in+the+method+definition+line+should+be+written">Parenthesis in the method definition line should be written</a>
          <li>      <a href="#label-Comments">Comments</a>
          <li>      <a href="#label-Documentation+should+be+written+in+the+RDoc+format+in+the+source+code">Documentation should be written in the RDoc format in the source code</a>
          <li>
            <details open>
              <summary>      <a href="#label-Standard+documentation">Standard documentation</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-of+files">of files</a>
          <li>      <a href="#label-of+classes+and+methods+within+those+files">of classes and methods within those files</a>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-Exception+handling">Exception handling</a>
          <li>      <a href="#label-Testing+code+should+use+-27test-2Funit-27">Testing code should use ‘test/unit’</a>
          <li>      <a href="#label-Using+autoload">Using autoload</a>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-NAMESPACE">NAMESPACE</a>
          <li>      <a href="#label-MAINTENANCE">MAINTENANCE</a>
          <li>      <a href="#label-RUBY+VERSION+and+IMPLEMENTATION">RUBY VERSION and IMPLEMENTATION</a>
          <li>      <a href="#label-OS+and+ARCHITECTURE">OS and ARCHITECTURE</a>

  </ul>
</div>

  
<div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
    <li><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
    <li><a href="./README_rdoc.html">README</a>
    <li><a href="./README_DEV_rdoc.html">README_DEV</a>
    <li><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
    <li><details><summary>doc</summary>
    <ul class="link-list">
      <li><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_3_rdoc.html">RELEASE_NOTES-1.4.3</a>
      <li><a href="./doc/RELEASE_NOTES-1_5_0_rdoc.html">RELEASE_NOTES-1.5.0</a>
    </ul></details>
  </ul>
</div>


  <footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.10.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

</nav>

<main role="main" aria-label="Page README_DEV.rdoc">

<h1 id="label-README.DEV">README.DEV<span><a href="#label-README.DEV">&para;</a> <a href="#top">&uarr;</a></span></h1>
<dl class="rdoc-list note-list"><dt>Copyright</dt>
<dd>
<p>Copyright © 2005, 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>Copyright</dt>
<dd>
<p>Copyright © 2006, 2008 Jan Aerts &lt;jandot@bioruby.org&gt;</p>
</dd><dt>Copyright</dt>
<dd>
<p>Copyright © 2011, 2019 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd></dl>

<h1 id="label-HOW+TO+CONTRIBUTE+TO+THE+BIORUBY+PROJECT-3F">HOW TO CONTRIBUTE TO THE BIORUBY PROJECT?<span><a href="#label-HOW+TO+CONTRIBUTE+TO+THE+BIORUBY+PROJECT-3F">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>There are many possible ways to contribute to the BioRuby project, such as:</p>
<ul><li>
<p>Join the discussion on the BioRuby mailing list</p>
</li><li>
<p>Send a bug report or write a bug fix patch</p>
</li><li>
<p>Add and correct documentation</p>
</li><li>
<p>Develop code for new features, etc.</p>
</li></ul>

<p>All of these are welcome!  This document mainly focuses on the last option, how to contribute your code to the BioRuby distribution. This may also be helpful when you send large patches for existing codes.</p>

<p>We would like to include your contribution as long as the scope of your module meets the field of bioinformatics.</p>

<h2 id="label-Git">Git<span><a href="#label-Git">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Bioruby is now under git source control at <a href="http://github.com/bioruby/bioruby">github.com/bioruby/bioruby</a>. There are two basic ways to contribute: with patches or pull requests. Both are explained on the bioruby wiki at <a href="http://bioruby.open-bio.org/wiki">bioruby.open-bio.org/wiki</a>.</p>

<h3 id="label-Preparation+before+sending+patches+or+pull+requests">Preparation before sending patches or pull requests<span><a href="#label-Preparation+before+sending+patches+or+pull+requests">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Before sending patches or pull requests, rewriting history and reordering or selecting patches are recommended. See “Creating the perfect patch series” in the Git User’s Manual. <a href="http://www.kernel.org/pub/software/scm/git/docs/user-manual.html#patch-series">www.kernel.org/pub/software/scm/git/docs/user-manual.html#patch-series</a></p>

<h3 id="label-Sending+your+contribution">Sending your contribution<span><a href="#label-Sending+your+contribution">&para;</a> <a href="#top">&uarr;</a></span></h3>

<h4 id="label-With+patches">With patches<span><a href="#label-With+patches">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>You can send patches with git-format-patch. For a smaller change, unified diff (diff -u) without using git can also be accepted.</p>

<h4 id="label-With+pull+requests">With pull requests<span><a href="#label-With+pull+requests">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>We are happy if your commits can be pulled with fast-forward. For the purpose, using git-rebase before sending pull request is recommended. See “Keeping a patch series up to date using git rebase” in the Git User’s Manual. <a href="http://www.kernel.org/pub/software/scm/git/docs/user-manual.html#using-git-rebase">www.kernel.org/pub/software/scm/git/docs/user-manual.html#using-git-rebase</a></p>

<h3 id="label-Notes+for+the+treatment+of+contributions+in+the+blessed+repository">Notes for the treatment of contributions in the blessed repository<span><a href="#label-Notes+for+the+treatment+of+contributions+in+the+blessed+repository">&para;</a> <a href="#top">&uarr;</a></span></h3>

<h4 id="label-Merging+policy">Merging policy<span><a href="#label-Merging+policy">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>We do not always merge your commits as is.  We may edit, rewrite, reorder, select, and/or mix your commits before and/or after merging to the blessed repository.</p>

<h4 id="label-Git+commit+management+policy">Git commit management policy<span><a href="#label-Git+commit+management+policy">&para;</a> <a href="#top">&uarr;</a></span></h4>

<p>We want to keep the commit history linear as far as possible, because it is easy to find problems and regressions in commits.  See “Why bisecting merge commits can be harder than bisecting linear history” in the Git User’s Manual. <a href="http://www.kernel.org/pub/software/scm/git/docs/user-manual.html#bisect-merges">www.kernel.org/pub/software/scm/git/docs/user-manual.html#bisect-merges</a></p>

<p>Note that the above policy is only for the main ‘blessed’ repository, and it does not aim to restrict each user’s fork.</p>

<h1 id="label-LICENSE">LICENSE<span><a href="#label-LICENSE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>If you would like your module to be included in the BioRuby distribution, you need to give us right to change the license of your module to make it compatible with other modules in BioRuby.</p>

<p>BioRuby was previously distributed under the LGPL license, but now is distributed under the same terms as Ruby.</p>

<h1 id="label-CODING+STYLE">CODING STYLE<span><a href="#label-CODING+STYLE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>You will need to follow the typical coding styles of the BioRuby modules:</p>

<h2 id="label-Use+the+following+naming+conventions">Use the following naming conventions<span><a href="#label-Use+the+following+naming+conventions">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>CamelCase for module and class names</p>
</li><li>
<p>‘_’-separated_lowercase for method names</p>
</li><li>
<p>‘_’-separated_lowercase for variable names</p>
</li><li>
<p>all UPPERCASE for constants</p>
</li></ul>

<h2 id="label-Indentation+must+not+include+tabs">Indentation must not include tabs<span><a href="#label-Indentation+must+not+include+tabs">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Use 2 spaces for indentation.</p>
</li><li>
<p>Don’t replace spaces to tabs.</p>
</li></ul>

<h2 id="label-Parenthesis+in+the+method+definition+line+should+be+written">Parenthesis in the method definition line should be written<span><a href="#label-Parenthesis+in+the+method+definition+line+should+be+written">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Good: <code>def example(str, ary)</code></p>
</li><li>
<p>Discouraged: <code>def example str, ary</code></p>
</li></ul>

<h2 id="label-Comments">Comments<span><a href="#label-Comments">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Don’t use <code>=begin</code> and <code>=end</code> blocks for comments.  If you need to add comments, include it in the RDoc documentation.</p>

<h2 id="label-Documentation+should+be+written+in+the+RDoc+format+in+the+source+code">Documentation should be written in the RDoc format in the source code<span><a href="#label-Documentation+should+be+written+in+the+RDoc+format+in+the+source+code">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>The RDoc format is becoming the popular standard for Ruby documentation. We are now in transition from the previously used RD format to the RDoc format in API documentation.</p>

<p>Additional tutorial documentation and working examples are encouraged with your contribution.  You may use the header part of the file for this purpose as demonstrated in the previous section.</p>

<h2 id="label-Standard+documentation">Standard documentation<span><a href="#label-Standard+documentation">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-of+files">of files<span><a href="#label-of+files">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Each file should start with a header, which covers the following topics:</p>
<ul><li>
<p>copyright</p>
</li><li>
<p>license</p>
</li><li>
<p>description of the file (<em>not</em> the classes; see below)</p>
</li><li>
<p>any references, if appropriate</p>
</li></ul>

<p>The header should be formatted as follows:</p>

<pre>#
# = bio/db/hoge.rb - Hoge database parser classes
#
# Copyright::  Copyright (C) 2001, 2003-2005 Bio R. Hacker &lt;brh@example.org&gt;,
# Copyright::  Copyright (C) 2006 Chem R. Hacker &lt;crh@example.org&gt;
#
# License::    The Ruby License
#
# == Description
#
# This file contains classes that implement an interface to the Hoge database.
#
# == References
#
# * Hoge F. et al., The Hoge database, Nucleic. Acid. Res. 123:100--123 (2030)
# * http://hoge.db/
#

require &#39;foo&#39;

module Bio

  autoload :Bar, &#39;bio/bar&#39;

class Hoge
  :
end  # Hoge

end  # Bio</pre>

<h3 id="label-of+classes+and+methods+within+those+files">of classes and methods within those files<span><a href="#label-of+classes+and+methods+within+those+files">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Classes and methods should be documented in a standardized format, as in the following example (from lib/bio/sequence.rb):</p>

<pre class="ruby"><span class="ruby-comment"># == Description</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># Bio::Sequence objects represent annotated sequences in bioruby.</span>
<span class="ruby-comment"># A Bio::Sequence object is a wrapper around the actual sequence,</span>
<span class="ruby-comment"># represented as either a Bio::Sequence::NA or a Bio::Sequence::AA object.</span>
<span class="ruby-comment"># For most users, this encapsulation will be completely transparent.</span>
<span class="ruby-comment"># Bio::Sequence responds to all methods defined for Bio::Sequence::NA/AA</span>
<span class="ruby-comment"># objects using the same arguments and returning the same values (even though</span>
<span class="ruby-comment"># these methods are not documented specifically for Bio::Sequence).</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># == Usage</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment">#   require &#39;bio&#39;</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment">#   # Create a nucleic or amino acid sequence</span>
<span class="ruby-comment">#   dna = Bio::Sequence.auto(&#39;atgcatgcATGCATGCAAAA&#39;)</span>
<span class="ruby-comment">#   rna = Bio::Sequence.auto(&#39;augcaugcaugcaugcaaaa&#39;)</span>
<span class="ruby-comment">#   aa = Bio::Sequence.auto(&#39;ACDEFGHIKLMNPQRSTVWYU&#39;)</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment">#   # Print in FASTA format</span>
<span class="ruby-comment">#   puts dna.output(:fasta)</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment">#   # Print all codons</span>
<span class="ruby-comment">#   dna.window_search(3,3) do |codon|</span>
<span class="ruby-comment">#     puts codon</span>
<span class="ruby-comment">#   end</span>
<span class="ruby-comment">#</span>
<span class="ruby-keyword">class</span> <span class="ruby-constant">Sequence</span>

  <span class="ruby-comment"># Create a new Bio::Sequence object</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment">#   s = Bio::Sequence.new(&#39;atgc&#39;)</span>
  <span class="ruby-comment">#   puts s                                  # =&gt; &#39;atgc&#39;</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># Note that this method does not intialize the contained sequence</span>
  <span class="ruby-comment"># as any kind of bioruby object, only as a simple string</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment">#   puts s.seq.class                        # =&gt; String</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># See Bio::Sequence#na, Bio::Sequence#aa, and Bio::Sequence#auto</span>
  <span class="ruby-comment"># for methods to transform the basic String of a just created</span>
  <span class="ruby-comment"># Bio::Sequence object to a proper bioruby object</span>
  <span class="ruby-comment"># ---</span>
  <span class="ruby-comment"># *Arguments*:</span>
  <span class="ruby-comment"># * (required) _str_: String or Bio::Sequence::NA/AA object</span>
  <span class="ruby-comment"># *Returns*:: Bio::Sequence object</span>
  <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">initialize</span>(<span class="ruby-identifier">str</span>)
    <span class="ruby-ivar">@seq</span> = <span class="ruby-identifier">str</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-comment"># The sequence identifier.  For example, for a sequence</span>
  <span class="ruby-comment"># of Genbank origin, this is the accession number.</span>
  <span class="ruby-identifier">attr_accessor</span> <span class="ruby-value">:entry_id</span>

  <span class="ruby-comment"># An Array of Bio::Feature objects</span>
  <span class="ruby-identifier">attr_accessor</span> <span class="ruby-value">:features</span>
<span class="ruby-keyword">end</span> <span class="ruby-comment"># Sequence</span>
</pre>

<p>Preceding the class definition (<code>class Sequence</code>), there is at least a description and a usage example. Please use the <code>Description</code> and <code>Usage</code> headings. If appropriate, refer to other classes that interact with or are related to the class.</p>

<p>The code in the usage example should, if possible, be in a format that a user can copy-and-paste into a new script to run. It should illustrate the most important uses of the class. If possible and if it would not clutter up the example too much, try to provide any input data directly into the usage example, instead of refering to ARGV or ARGF for input.</p>

<pre class="ruby"><span class="ruby-identifier">dna</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;atgcatgcATGCATGCAAAA&#39;</span>)
</pre>

<p>Otherwise, describe the input shortly, for example:</p>

<pre class="ruby"><span class="ruby-comment"># input should be string consisting of nucleotides</span>
<span class="ruby-identifier">dna</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-constant">ARGF</span>.<span class="ruby-identifier">read</span>)
</pre>

<p>Methods should be preceded by a comment that describes what the method does, including any relevant usage examples. (In contrast to the documentation for the class itself, headings are not required.) In addition, any arguments should be listed, as well as the type of thing that is returned by the method. The format of this information is as follows:</p>

<pre class="ruby"><span class="ruby-comment"># ---</span>
<span class="ruby-comment"># *Arguments*:</span>
<span class="ruby-comment"># * (required) _str_: String or Bio::Sequence::NA</span>
<span class="ruby-comment"># * (optional) _nr_: a number that means something</span>
<span class="ruby-comment"># *Returns*:: true or false</span>
</pre>

<p>Attribute accessors can be preceded by a short description.</p>

<pre class="ruby"><span class="ruby-comment"># P-value (Float)</span>
<span class="ruby-identifier">attr_reader</span> <span class="ruby-value">:pvalue</span>
</pre>

<p>For writing rdoc documentation, putting two or more attributes in a line (such as <code>attr_reader :evalue, :pvalue</code>) is strongly discouraged.</p>

<p>Methods looks like attributes can also be preceded by a short description.</p>

<pre class="ruby"><span class="ruby-comment"># Scientific name (String)</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">scientific_name</span>
  <span class="ruby-comment">#...</span>
<span class="ruby-keyword">end</span>

<span class="ruby-comment"># Scientific name (String)</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">scientific_name=</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-comment">#...</span>
<span class="ruby-keyword">end</span>
</pre>

<h2 id="label-Exception+handling">Exception handling<span><a href="#label-Exception+handling">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Don’t use</p>

<pre class="ruby"><span class="ruby-identifier">$stderr</span>.<span class="ruby-identifier">puts</span> <span class="ruby-string">&quot;WARNING&quot;</span>
</pre>

<p>in your code. Instead, try to avoid printing error messages. For fatal errors, use <code>raise</code> with an appropriate message.</p>

<p>Kernel#warn can only be used to notice incompatible changes to programmers. Typically it may be used for deprecated or obsolete usage of a method. For example,</p>

<pre class="ruby"><span class="ruby-identifier">warn</span> <span class="ruby-string">&quot;The Foo#bar method is obsoleted. Use Foo#baz instead.&quot;</span>
</pre>

<h2 id="label-Testing+code+should+use+-27test-2Funit-27">Testing code should use ‘test/unit’<span><a href="#label-Testing+code+should+use+-27test-2Funit-27">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Unit tests should come with your modules by which you can assure what you meant to do with each method.  The test code is useful to make maintenance easy and ensure stability. The use of</p>

<pre>if __FILE__ == $0</pre>

<p>is deprecated.</p>

<h2 id="label-Using+autoload">Using autoload<span><a href="#label-Using+autoload">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>To quicken the initial load time we have replaced most of ‘require’ to ‘autoload’ since BioRuby version 0.7.  During this change, we have found some tips:</p>

<p>You should not separate the same namespace into several files.</p>
<ul><li>
<p>For example, if you have separated definitions of the Bio::Foo class into two files (e.g. ‘bio/foo.rb’ and ‘bio/bar.rb’), you need to resolve the dependencies (including the load order) yourself.</p>
</li><li>
<p>If you have a defined Bio::Foo in ‘bio/foo.rb’ and a defined Bio::Foo::Bar in ‘bio/foo/bar.rb’ add the following line in the ‘bio/foo.rb’ file:</p>

<pre class="ruby"><span class="ruby-identifier">autoload</span> <span class="ruby-value">:Bar</span>, <span class="ruby-string">&#39;bio/foo/bar&#39;</span>
</pre>
</li></ul>

<p>You should not put several top level namespaces in one file.</p>
<ul><li>
<p>For example, if you have Bio::A, Bio::B and Bio::C in the file ‘bio/foo.rb’, you need</p>

<pre class="ruby"><span class="ruby-identifier">autoload</span> <span class="ruby-value">:A</span>, <span class="ruby-string">&#39;bio/foo&#39;</span>
<span class="ruby-identifier">autoload</span> <span class="ruby-value">:B</span>, <span class="ruby-string">&#39;bio/foo&#39;</span>
<span class="ruby-identifier">autoload</span> <span class="ruby-value">:C</span>, <span class="ruby-string">&#39;bio/foo&#39;</span>
</pre>

<p>to load the module automatically (instead of require ‘bio/foo’). In this case, you should put them under the new namespace like Bio::Foo::A, Bio::Foo::B and Bio::Foo::C in the file ‘bio/foo’, then use</p>

<pre class="ruby"><span class="ruby-identifier">autoload</span> <span class="ruby-value">:Foo</span>, <span class="ruby-string">&#39;bio/foo&#39;</span>
</pre>

<p>so autoload can be written in 1 line.</p>
</li></ul>

<h1 id="label-NAMESPACE">NAMESPACE<span><a href="#label-NAMESPACE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Your module should be located under the top-level module <a href="Bio.html"><code>Bio</code></a> and put under the ‘bioruby/lib/bio’ directory.  The class/module names and the file names should be short and descriptive.</p>

<p>There are already several sub directories in ‘bioruby/lib’:</p>

<pre>bio/*.rb   -- general and widely used basic classes
bio/appl/  -- wrapper and parser for the external applications
bio/data/  -- basic biological data
bio/db/    -- flatfile database entry parsers
bio/io/    -- I/O interfaces for files, RDB, web services etc.
bio/util/  -- utilities and algorithms for bioinformatics</pre>

<p>If your module doesn’t match any of the above, please propose an appropriate directory name when you contribute. Please let the staff discuss on namespaces (class names), API (method names) before commiting a new module or making changes on existing modules.</p>

<h1 id="label-MAINTENANCE">MAINTENANCE<span><a href="#label-MAINTENANCE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Finally, please maintain the code you’ve contributed. Please let us know (on the bioruby list) before you commit, so that users can discuss on the change.</p>

<h1 id="label-RUBY+VERSION+and+IMPLEMENTATION">RUBY VERSION and IMPLEMENTATION<span><a href="#label-RUBY+VERSION+and+IMPLEMENTATION">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>We are mainly using Ruby MRI (Matz’ Ruby Implementation, or Matz’ Ruby Interpreter).  Please confirm that your code is running on current stable release versions of Ruby MRI.</p>

<p>See <a href="README_rdoc.html">README.rdoc</a> and <a href="RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a> for recommended Ruby versions.</p>

<p>It is welcome to support JRuby, Rubinius, etc, in addition to Ruby MRI.</p>

<p>Of course, it is strongly encouraged to write code that is not affected by differences between Ruby versions and/or implementations, as far as possible.</p>

<p>Although we no longer support Ruby 1.8, it might be useful if your code could also run on Ruby 1.8.7 in addition to supported Ruby versions.</p>

<h1 id="label-OS+and+ARCHITECTURE">OS and ARCHITECTURE<span><a href="#label-OS+and+ARCHITECTURE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>We hope BioRuby can be run on both UNIX (and UNIX-like OS) and Microsoft Windows.</p>

</main>

