<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>RELEASE_NOTES - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "./";
  var index_rel_prefix = "./";
</script>

<script src="./js/navigation.js" defer></script>
<script src="./js/search.js" defer></script>
<script src="./js/search_index.js" defer></script>
<script src="./js/searcher.js" defer></script>
<script src="./js/darkfish.js" defer></script>

<link href="./css/fonts.css" rel="stylesheet">
<link href="./css/rdoc.css" rel="stylesheet">


<body id="top" role="document" class="file">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="./index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="./table_of_contents.html#pages">Pages</a>
    <a href="./table_of_contents.html#classes">Classes</a>
    <a href="./table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>



  <ul class="link-list" role="directory">
              <li>
            <details open>
              <summary>      <a href="#label-BioRuby+2.0.5+RELEASE+NOTES">BioRuby 2.0.5 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Fixes">Fixes</a>
          <li>
            <details open>
              <summary>      <a href="#label-Changes+for+developers">Changes for developers</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Changes+of+CI+-28Continuous+Integration-29">Changes of CI (Continuous Integration)</a>
          <li>      <a href="#label-Changes+of+Gemfile+and+related+files">Changes of Gemfile and related files</a>
          <li>      <a href="#label-Others">Others</a>

              </ul>
            </details>
          </li>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-BioRuby+2.0.4+RELEASE+NOTES">BioRuby 2.0.4 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Fixes">Fixes</a>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-BioRuby+2.0.3+RELEASE+NOTES">BioRuby 2.0.3 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Bug+fixes">Bug fixes</a>
          <li>
            <details open>
              <summary>      <a href="#label-Incompatible+changes">Incompatible changes</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Bio-3A-3ASequence-3A-3A-2A+incompatible+changes+since+Ruby+3.0.0">Bio::Sequence::* incompatible changes since Ruby 3.0.0</a>

              </ul>
            </details>
          </li>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-BioRuby+2.0.2+RELEASE+NOTES">BioRuby 2.0.2 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Bug+fixes">Bug fixes</a>
          <li>      <a href="#label-Known+issues">Known issues</a>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-BioRuby+2.0.1+RELEASE+NOTES">BioRuby 2.0.1 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Bug+fixes">Bug fixes</a>
          <li>
            <details open>
              <summary>      <a href="#label-Improvement+of+sample+scripts">Improvement of sample scripts</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-CWL+-28Common+Workflow+Language-29+workflow+files+are+added">CWL (Common Workflow Language) workflow files are added</a>

              </ul>
            </details>
          </li>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-BioRuby+2.0.0+RELEASE+NOTES">BioRuby 2.0.0 RELEASE NOTES</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>
            <details open>
              <summary>      <a href="#label-Features+moved+to+separete+gems">Features moved to separete gems</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-BioRuby+Shell+is+moved+to+-22bio-shell-22">BioRuby Shell is moved to “bio-shell”</a>
          <li>      <a href="#label-Executable+files+are+moved+to+-22bio-executables-22">Executable files are moved to “bio-executables”</a>
          <li>      <a href="#label-Fast+BLAST+XML+result+parser+by+using+Expat+XML+Parser+is+moved+to+-22bio-blast-xmlparser-22">Fast BLAST XML result parser by using Expat XML Parser is moved to “bio-blast-xmlparser”</a>
          <li>      <a href="#label-Bio-3A-3APhyloXML+is+moved+to+-22bioruby-phyloxml-22">Bio::PhyloXML is moved to “bioruby-phyloxml”</a>
          <li>      <a href="#label-Bio-3A-3ASQL+is+moved+to+-22bio-biosql-22">Bio::SQL is moved to “bio-biosql”</a>

              </ul>
            </details>
          </li>
          <li>
            <details open>
              <summary>      <a href="#label-New+features+and+improvements">New features and improvements</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-HTTPS+is+used+to+access+NCBI+web+services">HTTPS is used to access NCBI web services</a>
          <li>      <a href="#label-KEGG-3A-3AGENES-23diseases+and+related+methods+are+added">KEGG::GENES#diseases and related methods are added</a>
          <li>      <a href="#label-Pre-calculated+ambiguity+codon+tables+in+Bio-3A-3ACodonTable">Pre-calculated ambiguity codon tables in Bio::CodonTable</a>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-Bug+fixes">Bug fixes</a>
          <li>
            <details open>
              <summary>      <a href="#label-Incompatible+changes">Incompatible changes</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Bio-3A-3ATaxonomy+is+removed+and+merged+to+Bio-3A-3APhyloXML-3A-3ATaxonomy">Bio::Taxonomy is removed and merged to Bio::PhyloXML::Taxonomy</a>
          <li>      <a href="#label-Some+features+are+moved+to+separete+gems">Some features are moved to separete gems</a>

              </ul>
            </details>
          </li>
          <li>      <a href="#label-Known+issues">Known issues</a>
          <li>
            <details open>
              <summary>      <a href="#label-Other+important+news">Other important news</a>
</summary>
              <ul class="link-list" role="directory">
                          <li>      <a href="#label-Ruby+1.8+is+no+longer+supported">Ruby 1.8 is no longer supported</a>
          <li>      <a href="#label-Installation+without+RubyGems+is+no+longer+supported">Installation without RubyGems is no longer supported</a>

              </ul>
            </details>
          </li>

              </ul>
            </details>
          </li>

  </ul>
</div>


  <div id="project-metadata">
    
<div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
    <li><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
    <li><a href="./README_rdoc.html">README</a>
    <li><a href="./README_DEV_rdoc.html">README_DEV</a>
    <li><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
    <li><details><summary>doc</summary>
    <ul class="link-list">
      <li><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
      <li><a href="./doc/RELEASE_NOTES-1_4_3_rdoc.html">RELEASE_NOTES-1.4.3</a>
      <li><a href="./doc/RELEASE_NOTES-1_5_0_rdoc.html">RELEASE_NOTES-1.5.0</a>
    </ul></details>
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-label="Page RELEASE_NOTES.rdoc">

<h1 id="label-BioRuby+2.0.5+RELEASE+NOTES">BioRuby 2.0.5 RELEASE NOTES<span><a href="#label-BioRuby+2.0.5+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Some fixes have been made in BioRuby 2.0.5 after the release of 2.0.4.</p>

<h2 id="label-Fixes">Fixes<span><a href="#label-Fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="Bio/UniProtKB.html#method-i-ft"><code>Bio::UniProtKB#ft</code></a>: Additional bug fixes for parsing FT lines since UniProt release 2019_11. (<a href="https://github.com/bioruby/bioruby/issues/147">github.com/bioruby/bioruby/issues/147</a> )</p>
</li><li>
<p><a href="Bio/UniProtKB.html#method-i-oh"><code>Bio::UniProtKB#oh</code></a>: Fixed parse error when parsing 1A_AMVLE (P03589).</p>
</li><li>
<p><a href="Bio/Blast/Remote/GenomeNet.html"><code>Bio::Blast::Remote::GenomeNet</code></a>: Fixed errors caused by the change of GenomeNet BLAST webpage’s internal URL (which is unofficial internal URL and may subject to be changed without notices).</p>
</li><li>
<p><a href="Bio/PTS1.html"><code>Bio::PTS1</code></a>: Fixed errors due to the PTS1 website URL change from http to https.</p>
</li></ul>

<h2 id="label-Changes+for+developers">Changes for developers<span><a href="#label-Changes+for+developers">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Changes+of+CI+-28Continuous+Integration-29">Changes of CI (Continuous Integration)<span><a href="#label-Changes+of+CI+-28Continuous+Integration-29">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>Started using GitHub Actions for CI.</p>
</li><li>
<p>Stopped using Travis-CI for CI.</p>
</li><li>
<p>Fixed AppVeyor CI errors.</p>
</li></ul>

<h3 id="label-Changes+of+Gemfile+and+related+files">Changes of Gemfile and related files<span><a href="#label-Changes+of+Gemfile+and+related+files">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>Gemfile: Added “matrix” and “rexml” gems that are excluded from the default gems in recent Ruby versions.</p>
</li><li>
<p>gemfiles/*: Removed because of the CI changes.</p>
</li></ul>

<h3 id="label-Others">Others<span><a href="#label-Others">&para;</a> <a href="#top">&uarr;</a></span></h3>
<ul><li>
<p>.gitignore is added.</p>
</li></ul>

<h1 id="label-BioRuby+2.0.4+RELEASE+NOTES">BioRuby 2.0.4 RELEASE NOTES<span><a href="#label-BioRuby+2.0.4+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Some fixes have been made in BioRuby 2.0.4 after the release of 2.0.3.</p>

<h2 id="label-Fixes">Fixes<span><a href="#label-Fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="Bio/UniProtKB.html#method-i-ft"><code>Bio::UniProtKB#ft</code></a>: Supports text format change of FT lines since UniProt release 2019_11. (<a href="https://github.com/bioruby/bioruby/issues/147">github.com/bioruby/bioruby/issues/147</a> )</p>
</li><li>
<p><a href="Bio/UniProtKB.html#method-i-protein_name"><code>Bio::UniProtKB#protein_name</code></a> and synonyms methods may raise error after calling <a href="Bio/UniProtKB.html#method-i-de"><code>Bio::UniProtKB#de</code></a> method.</p>
</li></ul>

<h1 id="label-BioRuby+2.0.3+RELEASE+NOTES">BioRuby 2.0.3 RELEASE NOTES<span><a href="#label-BioRuby+2.0.3+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Some bug fixes have been made in BioRuby 2.0.3 after the release of 2.0.2.</p>

<h2 id="label-Bug+fixes">Bug fixes<span><a href="#label-Bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Fix Ruby 3.0.0 Bio::Sequence::* issue. (<a href="https://github.com/bioruby/bioruby/issues/137">github.com/bioruby/bioruby/issues/137</a> )</p>
</li><li>
<p>Fix typo (<a href="https://github.com/bioruby/bioruby/pull/145">github.com/bioruby/bioruby/pull/145</a> )</p>
</li></ul>

<h2 id="label-Incompatible+changes">Incompatible changes<span><a href="#label-Incompatible+changes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ASequence-3A-3A-2A+incompatible+changes+since+Ruby+3.0.0">Bio::Sequence::* incompatible changes since Ruby 3.0.0<span><a href="#label-Bio-3A-3ASequence-3A-3A-2A+incompatible+changes+since+Ruby+3.0.0">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Since Ruby 3.0.0, the following methods in <a href="Bio/Sequence/NA.html"><code>Bio::Sequence::NA</code></a>, <a href="Bio/Sequence/AA.html"><code>Bio::Sequence::AA</code></a>, and Bio::Sequence::Generic return or yield String instance, instead of the Bio::Sequence::* instance.</p>
<ul><li>
<p>dump</p>
</li><li>
<p>scrub</p>
</li></ul>

<p>For details about Ruby 3.0.0 incompatible changes, see <a href="https://github.com/ruby/ruby/blob/v3_0_0/NEWS.md">News for Ruby 3.0.0</a>.</p>

<h1 id="label-BioRuby+2.0.2+RELEASE+NOTES">BioRuby 2.0.2 RELEASE NOTES<span><a href="#label-BioRuby+2.0.2+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Some bugs fixes have been made in BioRuby 2.0.2 after the release of 2.0.1.</p>

<h2 id="label-Bug+fixes">Bug fixes<span><a href="#label-Bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Fix NameError in <a href="Bio/Sequence/Format.html#method-i-output"><code>Bio::Sequence#output</code></a>(:embl) (<a href="https://github.com/bioruby/bioruby/issues/135">github.com/bioruby/bioruby/issues/135</a> )</p>
</li><li>
<p>Suppress warning: Gem::Specification#has_rdoc= is deprecated (<a href="https://github.com/bioruby/bioruby/issues/138">github.com/bioruby/bioruby/issues/138</a> )</p>
</li><li>
<p>Fix misspelling URL in <a href="README_rdoc.html">README.rdoc</a></p>
</li></ul>

<h2 id="label-Known+issues">Known issues<span><a href="#label-Known+issues">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>A known issue about Ruby 3.0 is added to <a href="KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a>. The issue will be fixed in the near future.</p>

<h1 id="label-BioRuby+2.0.1+RELEASE+NOTES">BioRuby 2.0.1 RELEASE NOTES<span><a href="#label-BioRuby+2.0.1+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>Some bug fixes and improvements have been made to the BioRuby 2.0.1 after the version 2.0.0 is released.</p>

<h2 id="label-Bug+fixes">Bug fixes<span><a href="#label-Bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p><a href="Bio/GFF/GFF2/Record.html#method-c-parse"><code>Bio::GFF::GFF2::Record.parse</code></a> did not return correct object.</p>
</li></ul>

<h2 id="label-Improvement+of+sample+scripts">Improvement of sample scripts<span><a href="#label-Improvement+of+sample+scripts">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>The following scripts in the sample/ directiry are newly added.</p>
<ul><li>
<p>color_scheme_aa.rb: Example of <a href="Bio/ColorScheme.html"><code>Bio::ColorScheme</code></a> for an amino acid sequence.</p>
</li><li>
<p>fastq2html.rb: Visualization of FASTQ sequences, colored by quality scores.</p>
</li><li>
<p>rev_comp.rb: Shows reverse-complement sequences of the given sequences.</p>
</li></ul>

<p>The floowing scripts are modified to fix bug and/or to improve features.</p>
<ul><li>
<p>na2aa.rb: Completely rewritten to fix bug. Shows translated sequences.</p>
</li><li>
<p>color_scheme_na.rb: Added support for various sequence formats.</p>
</li></ul>

<h3 id="label-CWL+-28Common+Workflow+Language-29+workflow+files+are+added">CWL (Common Workflow Language) workflow files are added<span><a href="#label-CWL+-28Common+Workflow+Language-29+workflow+files+are+added">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>CWL (Common Workflow Language) workflow files are added for some sample scripts. The usage of each sample script will be clarified with the CWL workflow files.</p>

<p>Two type of files are prepared for CWL workflow engine. *.cwl is a workflow definition file for each sample script. *.testdata.yaml describes sample input data for each CWL workflow.</p>

<p>In this version, cwl files for the 3 sample scripts are added.</p>
<ul><li>
<p>fastq2html.rb: fastq2html.cwl with fastq2html.testdata.yaml</p>
</li><li>
<p>na2aa.rb: na2aa.cwl with na2aa.testdata.yaml</p>
</li><li>
<p>rev_comp.rb: rev_comp.cwl with rev_comp.testdata.yaml</p>
</li></ul>

<h1 id="label-BioRuby+2.0.0+RELEASE+NOTES">BioRuby 2.0.0 RELEASE NOTES<span><a href="#label-BioRuby+2.0.0+RELEASE+NOTES">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p>A lot of changes have been made to the BioRuby 2.0.0 after the version 1.5.x is released. This document describes important and/or incompatible changes since the BioRuby 1.5.0 release.</p>

<p>For known problems, see <a href="KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a>.</p>

<h2 id="label-Features+moved+to+separete+gems">Features moved to separete gems<span><a href="#label-Features+moved+to+separete+gems">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Some features are moved to separate gems because of reducing complexity and/or to avoid external library dependency of BioRuby core.</p>

<h3 id="label-BioRuby+Shell+is+moved+to+-22bio-shell-22">BioRuby Shell is moved to “bio-shell”<span><a href="#label-BioRuby+Shell+is+moved+to+-22bio-shell-22">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>BioRuby Shell is split to “bio-shell” gem.</p>

<h3 id="label-Executable+files+are+moved+to+-22bio-executables-22">Executable files are moved to “bio-executables”<span><a href="#label-Executable+files+are+moved+to+-22bio-executables-22">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>To avoid unexpected loading of executable files by some Rails software, all executable commands are moved to “bio-executables” gem (except the “bioruby” command that is included in the above “bio-shell” gem).</p>

<h3 id="label-Fast+BLAST+XML+result+parser+by+using+Expat+XML+Parser+is+moved+to+-22bio-blast-xmlparser-22">Fast BLAST XML result parser by using Expat XML Parser is moved to “bio-blast-xmlparser”<span><a href="#label-Fast+BLAST+XML+result+parser+by+using+Expat+XML+Parser+is+moved+to+-22bio-blast-xmlparser-22">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Fast BLAST XML result parser by using Expat XML Parser is split to “bio-blast-xmlparser” gem, because of external C library dependency. Please install “bio-blast-xmlparser” gem if possible. If it is installed, BioRuby automatically use it.</p>

<h3 id="label-Bio-3A-3APhyloXML+is+moved+to+-22bioruby-phyloxml-22">Bio::PhyloXML is moved to “bioruby-phyloxml”<span><a href="#label-Bio-3A-3APhyloXML+is+moved+to+-22bioruby-phyloxml-22">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Bio::PhyloXML is split to “bioruby-phyloxml” gem.</p>

<p>NOTE: Please uninstall “bio-phyloxml” gem, that have been created as a preliminary trial of splitting a module in 2012 and have not been maintained after that.</p>

<h3 id="label-Bio-3A-3ASQL+is+moved+to+-22bio-biosql-22">Bio::SQL is moved to “bio-biosql”<span><a href="#label-Bio-3A-3ASQL+is+moved+to+-22bio-biosql-22">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Bio::SQL is split to “bio-biosql” gem.</p>

<h2 id="label-New+features+and+improvements">New features and improvements<span><a href="#label-New+features+and+improvements">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-HTTPS+is+used+to+access+NCBI+web+services">HTTPS is used to access NCBI web services<span><a href="#label-HTTPS+is+used+to+access+NCBI+web+services">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>As you may know, NCBI announced that all HTTP resources will be switched to HTTPS on September 30, 2016. To follow the transition, all URLs for accessing NCBI E-utilities in BioRuby are changed to use HTTPS.</p>

<p>In BioRuby, the following classes/modules are affected.</p>
<ul><li>
<p><a href="Bio/NCBI/REST.html"><code>Bio::NCBI::REST</code></a> and descending classes</p>
</li><li>
<p><a href="Bio/PubMed.html"><code>Bio::PubMed</code></a></p>
</li></ul>

<p>In some rare cases (especially when building Ruby and/or OpenSSL by yourself from source code), Ruby does not include SSL/TLS support, or Ruby fails to detect SSL root certificates. In such cases, you may need to reinstall or upgrade Ruby, OpenSSL (or alternatives), and/or SSL root certificates with appropriate configuration options. Alternatively, installing binary packages is generally a good idea.</p>

<h3 id="label-KEGG-3A-3AGENES-23diseases+and+related+methods+are+added">KEGG::GENES#diseases and related methods are added<span><a href="#label-KEGG-3A-3AGENES-23diseases+and+related+methods+are+added">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>The following methods are added to KEGG::GENES, contributed by @kojix2.</p>
<ul><li>
<p>networks_as_strings</p>
</li><li>
<p>diseases_as_strings</p>
</li><li>
<p>diseases_as_hash</p>
</li><li>
<p>diseases</p>
</li><li>
<p>drug_targets_as_strings</p>
</li></ul>

<h3 id="label-Pre-calculated+ambiguity+codon+tables+in+Bio-3A-3ACodonTable">Pre-calculated ambiguity codon tables in <a href="Bio/CodonTable.html"><code>Bio::CodonTable</code></a><span><a href="#label-Pre-calculated+ambiguity+codon+tables+in+Bio-3A-3ACodonTable">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Pre-calculated ambiguity codon tables are added, contributed by Tomoaki NISHIYAMA.</p>

<h2 id="label-Bug+fixes">Bug fixes<span><a href="#label-Bug+fixes">&para;</a> <a href="#top">&uarr;</a></span></h2>
<ul><li>
<p>Fixed a parser bug in <a href="Bio/Fasta/Report.html"><code>Bio::Fasta::Report</code></a>, FASTA output (-m 10) parser, contributed by William Van Etten and Mark Wilkinson via GitHub.</p>
</li><li>
<p>HTTPS is used to access GenomeNet BLAST web service, contributed by @ramadis via GitHub.</p>
</li><li>
<p><a href="Bio/AAindex.html"><code>Bio::AAindex</code></a> documentation fix, suggested by @kojix2 via GitHub.</p>
</li><li>
<p>Suppress warning messages in Ruby 2.4 and later.</p>
</li></ul>

<h2 id="label-Incompatible+changes">Incompatible changes<span><a href="#label-Incompatible+changes">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Bio-3A-3ATaxonomy+is+removed+and+merged+to+Bio-3A-3APhyloXML-3A-3ATaxonomy">Bio::Taxonomy is removed and merged to Bio::PhyloXML::Taxonomy<span><a href="#label-Bio-3A-3ATaxonomy+is+removed+and+merged+to+Bio-3A-3APhyloXML-3A-3ATaxonomy">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Bio::Taxonomy in lib/bio/db/phyloxml/phyloxml_elements.rb was written for PhyloXML in 2009. It was intended to become general taxonomy data class in BioRuby in these days. However, no efforts have been made to improve the Bio::Taxonomy class, and it still remains to be a PhyloXML specific class. Because Bio::PhyloXML is split as a different Gem (Biogem) package, we now decide to remove Bio::Taxonomy and merge it to Bio::PhyloXML::Taxonomy.</p>

<p>In the codes using Bio::Taxonomy directly, changing Bio::Taxonomy to Bio::PhyloXML::Taxonomy or adding the following monkey patch may be needed.</p>

<pre class="ruby"><span class="ruby-keyword">module</span> <span class="ruby-constant">Bio</span>
   <span class="ruby-keyword">unless</span> <span class="ruby-keyword">defined?</span> <span class="ruby-constant">Taxonomy</span>
      <span class="ruby-constant">Taxonomy</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">PhyloXML</span><span class="ruby-operator">::</span><span class="ruby-constant">Taxonomy</span>
   <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span>
</pre>

<p>In the future, Bio::Taxonomy might be added as general taxonomy data class. The new Bio::Taxonomy might be incompatible with the current Bio::Taxonmy.</p>

<h3 id="label-Some+features+are+moved+to+separete+gems">Some features are moved to separete gems<span><a href="#label-Some+features+are+moved+to+separete+gems">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Some features are split to separete gems and removed from this “bio” gem. See the above “Features moved to separete gems” topics for details.</p>

<h2 id="label-Known+issues">Known issues<span><a href="#label-Known+issues">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>The following issues are added or updated. See <a href="KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a> for other already known issues.</p>

<h2 id="label-Other+important+news">Other important news<span><a href="#label-Other+important+news">&para;</a> <a href="#top">&uarr;</a></span></h2>

<h3 id="label-Ruby+1.8+is+no+longer+supported">Ruby 1.8 is no longer supported<span><a href="#label-Ruby+1.8+is+no+longer+supported">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Ruby 1.8.x is no longer supported. Though unsupported, some components may still run on Ruby 1.8.7. Please use Ruby 1.8.7 at your own risk with this version of BioRuby.</p>

<h3 id="label-Installation+without+RubyGems+is+no+longer+supported">Installation without RubyGems is no longer supported<span><a href="#label-Installation+without+RubyGems+is+no+longer+supported">&para;</a> <a href="#top">&uarr;</a></span></h3>

<p>Installation by using setup.rb without RubyGems is no longer supported, and setup.rb is no longer included in BioRuby distribution.</p>

</main>



<footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.5.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

