<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>class Bio::RestrictionEnzyme::Analysis - BioRuby API documentation</title>

<link type="text/css" media="screen" href="../../rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "../../";
</script>

<script type="text/javascript" charset="utf-8" src="../../js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/darkfish.js"></script>


<body id="top" class="class">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="../../index.html">Home</a>
    <a href="../../table_of_contents.html#classes">Classes</a>
    <a href="../../table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio/util/restriction_enzyme/analysis.rb
    <li>lib/bio/util/restriction_enzyme/analysis_basic.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    <nav id="parent-class-section" class="section">
  <h3 class="section-header">Parent</h3>
  
  <p class="link"><a href="../../Object.html">Object</a>
  
</nav>

    
    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-cut">::cut</a>
    
    <li><a href="#method-c-cut_without_permutations">::cut_without_permutations</a>
    
    <li><a href="#method-i-create_enzyme_actions">#create_enzyme_actions</a>
    
    <li><a href="#method-i-cut">#cut</a>
    
    <li><a href="#method-i-cut_and_return_by_permutations">#cut_and_return_by_permutations</a>
    
    <li><a href="#method-i-cut_without_permutations">#cut_without_permutations</a>
    
    <li><a href="#method-i-find_match_locations">#find_match_locations</a>
    
    <li><a href="#method-i-fragments_for_display">#fragments_for_display</a>
    
    <li><a href="#method-i-permute">#permute</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="../../ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="../../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="../../LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="../../README_rdoc.html">README</a>
  
    <li class="file"><a href="../../README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="../../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="../../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="../../Bio.html">Bio</a>
  
    <li><a href="../../Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="../../Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="../../Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="../../Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="../../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="../../Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="../../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="../../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="../../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="../../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="../../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="../../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="../../Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="../../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="../../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="../../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="../../Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="../../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="../../Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="../../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="../../Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="../../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="../../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="../../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="../../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="../../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="../../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="../../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="../../Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="../../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="../../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="../../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="../../Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="../../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="../../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="../../Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="../../Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="../../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="../../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="../../Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="../../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="../../Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="../../Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="../../Bio/Command.html">Bio::Command</a>
  
    <li><a href="../../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="../../Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="../../Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="../../Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="../../Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="../../Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="../../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="../../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="../../Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="../../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="../../Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="../../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="../../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="../../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="../../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="../../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="../../Bio/DB.html">Bio::DB</a>
  
    <li><a href="../../Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="../../Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="../../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="../../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="../../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="../../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="../../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="../../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="../../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="../../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="../../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="../../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="../../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="../../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="../../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="../../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="../../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="../../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="../../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="../../Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="../../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="../../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="../../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="../../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="../../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="../../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="../../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="../../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="../../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="../../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="../../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="../../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="../../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="../../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="../../Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="../../Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="../../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="../../Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="../../Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="../../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="../../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="../../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="../../Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="../../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="../../Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="../../Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="../../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="../../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="../../Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="../../Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="../../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="../../Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="../../Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="../../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="../../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="../../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="../../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="../../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="../../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="../../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="../../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="../../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="../../Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="../../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="../../Bio/Features.html">Bio::Features</a>
  
    <li><a href="../../Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="../../Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="../../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="../../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="../../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="../../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="../../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="../../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="../../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="../../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="../../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="../../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="../../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="../../Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="../../Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="../../Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="../../Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="../../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="../../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="../../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="../../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="../../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="../../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="../../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="../../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="../../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="../../Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="../../Bio/GO.html">Bio::GO</a>
  
    <li><a href="../../Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="../../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="../../Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="../../Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="../../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="../../Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="../../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="../../Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="../../Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="../../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="../../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="../../Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="../../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="../../Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="../../Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="../../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="../../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="../../Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="../../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="../../Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="../../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="../../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="../../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="../../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="../../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="../../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="../../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="../../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="../../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="../../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="../../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="../../Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="../../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="../../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="../../Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="../../Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="../../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="../../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="../../Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="../../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="../../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="../../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="../../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="../../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="../../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="../../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="../../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="../../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="../../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="../../Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="../../Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="../../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="../../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="../../Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="../../Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="../../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="../../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="../../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="../../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="../../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="../../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="../../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="../../Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="../../Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="../../Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="../../Bio/Location.html">Bio::Location</a>
  
    <li><a href="../../Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="../../Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="../../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="../../Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="../../Bio/Map.html">Bio::Map</a>
  
    <li><a href="../../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="../../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="../../Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="../../Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="../../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="../../Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="../../Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="../../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="../../Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="../../Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="../../Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="../../Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="../../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="../../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="../../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="../../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="../../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="../../Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="../../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="../../Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="../../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="../../Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="../../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="../../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="../../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="../../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="../../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="../../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="../../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="../../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="../../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="../../Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="../../Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="../../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="../../Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="../../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="../../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="../../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="../../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="../../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="../../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="../../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="../../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="../../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="../../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="../../Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="../../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="../../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="../../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="../../Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="../../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="../../Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="../../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="../../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="../../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="../../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="../../Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="../../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="../../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="../../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="../../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="../../Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="../../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="../../Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="../../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="../../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="../../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="../../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="../../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="../../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="../../Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="../../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="../../Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="../../Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="../../Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="../../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="../../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="../../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="../../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="../../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="../../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="../../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="../../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="../../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="../../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="../../Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="../../Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="../../Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="../../Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="../../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="../../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="../../Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="../../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="../../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="../../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="../../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="../../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="../../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="../../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="../../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="../../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="../../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="../../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="../../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="../../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="../../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="../../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="../../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="../../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="../../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="../../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="../../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="../../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="../../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="../../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="../../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="../../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="../../Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="../../Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="../../Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="../../Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="../../Bio/References.html">Bio::References</a>
  
    <li><a href="../../Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="../../Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="../../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="../../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="../../Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="../../Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="../../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="../../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="../../Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="../../Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="../../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="../../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="../../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="../../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="../../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="../../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="../../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="../../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="../../Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="../../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="../../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="../../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="../../Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="../../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="../../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="../../Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="../../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="../../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="../../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="../../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="../../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="../../Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="../../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="../../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="../../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="../../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="../../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="../../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="../../Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="../../Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="../../Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="../../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="../../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="../../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="../../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="../../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="../../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="../../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="../../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="../../Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="../../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="../../Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="../../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="../../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="../../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="../../Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="../../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="../../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="../../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="../../Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="../../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="../../Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="../../Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="../../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="../../Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="../../Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="../../Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="../../Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="../../Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="../../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="../../Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="../../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="../../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="../../Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="../../Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="../../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="../../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="../../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="../../Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="../../Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="../../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="../../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="../../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="../../Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="../../Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="../../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="../../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="../../Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="../../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="../../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="../../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="../../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="../../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="../../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="../../Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="../../Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="../../Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="../../Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="../../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="../../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="../../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="../../Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="../../Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="../../Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="../../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="../../Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="../../Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="../../BiorubyController.html">BiorubyController</a>
  
    <li><a href="../../BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="../../BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="../../Hash.html">Hash</a>
  
    <li><a href="../../Object.html">Object</a>
  
    <li><a href="../../String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="class">class Bio::RestrictionEnzyme::Analysis</h1>

  <div id="description" class="description">
    
  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    

    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-cut" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cut</span><span
            class="method-args">( sequence, *args )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>See cut instance method</p>
          

          
          <div class="method-source-code" id="cut-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis.rb, line 23</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">cut</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>.<span class="ruby-identifier">cut</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
<span class="ruby-keyword">end</span></pre>
          </div><!-- cut-source -->
          
        </div>

        

        
      </div><!-- cut-method -->

    
      <div id="method-c-cut_without_permutations" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cut_without_permutations</span><span
            class="method-args">( sequence, *args )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>See <a
href="Analysis.html#method-c-cut_without_permutations">::cut_without_permutations</a>
instance method</p>
          

          
          <div class="method-source-code" id="cut_without_permutations-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis_basic.rb, line 21</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">cut_without_permutations</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>.<span class="ruby-identifier">cut_without_permutations</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
<span class="ruby-keyword">end</span></pre>
          </div><!-- cut_without_permutations-source -->
          
        </div>

        

        
      </div><!-- cut_without_permutations-method -->

    
    </section><!-- public-class-method-details -->
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Instance Methods</h3>

    
      <div id="method-i-cut" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cut</span><span
            class="method-args">( sequence, *args )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>See main documentation for <a
href="../RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></p>

<p><code>cut</code> takes into account permutations of cut variations based on
competitiveness of enzymes for an enzyme cutsite or enzyme bindsite on a
sequence.</p>

<p>Example:</p>

<p>FIXME add output</p>

<pre>Bio::RestrictionEnzyme::Analysis.cut('gaattc', 'EcoRI')</pre>

<p>_same as:_</p>

<pre>Bio::RestrictionEnzyme::Analysis.cut('gaattc', 'g^aattc')</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>sequence</code>: <code>String</code> kind of object that will be used
as a nucleic acid sequence.</p>
</li><li>
<p><code>args</code>: Series of enzyme names, enzymes sequences with cut
marks, or <a href="../RestrictionEnzyme.html">RestrictionEnzyme</a>
objects.</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Fragments.html">Bio::RestrictionEnzyme::Fragments</a> object
populated with Bio::RestrictionEnzyme::Fragment objects.   (Note: unrelated
to <a
href="Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>)
or a <code>Symbol</code> containing an error code</p>
</dd></dl>
          

          
          <div class="method-source-code" id="cut-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis.rb, line 48</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cut</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-identifier">view_ranges</span> = <span class="ruby-keyword">false</span>
  
  <span class="ruby-identifier">args</span>.<span class="ruby-identifier">select</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">i</span><span class="ruby-operator">|</span> <span class="ruby-identifier">i</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Hash</span> }.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">hsh</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">hsh</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">key</span>, <span class="ruby-identifier">value</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">key</span> <span class="ruby-operator">==</span> <span class="ruby-value">:view_ranges</span>
        <span class="ruby-keyword">unless</span> ( <span class="ruby-identifier">value</span>.<span class="ruby-identifier">kind_of?</span>(<span class="ruby-constant">TrueClass</span>) <span class="ruby-keyword">or</span> <span class="ruby-identifier">value</span>.<span class="ruby-identifier">kind_of?</span>(<span class="ruby-constant">FalseClass</span>) )
          <span class="ruby-identifier">raise</span> <span class="ruby-constant">ArgumentError</span>, <span class="ruby-node">&quot;view_ranges must be set to true or false, currently #{value.inspect}.&quot;</span>
        <span class="ruby-keyword">end</span>
        <span class="ruby-identifier">view_ranges</span> = <span class="ruby-identifier">value</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  
  <span class="ruby-identifier">res</span> = <span class="ruby-identifier">cut_and_return_by_permutations</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">res</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">res</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Symbol</span>
  <span class="ruby-comment"># Format the fragments for the user</span>
  <span class="ruby-identifier">fragments_for_display</span>( <span class="ruby-identifier">res</span>, <span class="ruby-identifier">view_ranges</span> )
<span class="ruby-keyword">end</span></pre>
          </div><!-- cut-source -->
          
        </div>

        

        
      </div><!-- cut-method -->

    
      <div id="method-i-cut_without_permutations" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cut_without_permutations</span><span
            class="method-args">( sequence, *args )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>See main documentation for <a
href="../RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></p>

<p><a
href="../RestrictionEnzyme.html#method-c-cut">Bio::RestrictionEnzyme.cut</a>
is preferred over this!</p>

<p>USE AT YOUR OWN RISK</p>

<p>This is a simpler version of method <code>cut</code>.  <code>cut</code>
takes into account permutations of cut variations based on competitiveness
of enzymes for an enzyme cutsite or enzyme bindsite on a sequence.  This
does not take into account those possibilities and is therefore faster, but
less likely to be accurate.</p>

<p>This code is mainly included as an academic example without having to wade
through the extra layer of complexity added by the permutations.</p>

<p>Example:</p>

<p>FIXME add output</p>

<pre>Bio::RestrictionEnzyme::Analysis.cut_without_permutations('gaattc', 'EcoRI')</pre>

<p>_same as:_</p>

<pre>Bio::RestrictionEnzyme::Analysis.cut_without_permutations('gaattc', 'g^aattc')</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>sequence</code>: <code>String</code> kind of object that will be used
as a nucleic acid sequence.</p>
</li><li>
<p><code>args</code>: Series of enzyme names, enzymes sequences with cut
marks, or <a href="../RestrictionEnzyme.html">RestrictionEnzyme</a>
objects.</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Fragments.html">Bio::RestrictionEnzyme::Fragments</a> object
populated with Bio::RestrictionEnzyme::Fragment objects. (Note: unrelated
to <a
href="Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>)</p>
</dd></dl>
          

          
          <div class="method-source-code" id="cut_without_permutations-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis_basic.rb, line 55</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cut_without_permutations</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">fragments_for_display</span>( {} ) <span class="ruby-keyword">if</span> <span class="ruby-operator">!</span><span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">kind_of?</span>(<span class="ruby-constant">String</span>) <span class="ruby-keyword">or</span> <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">sequence</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>( <span class="ruby-identifier">sequence</span> )

  <span class="ruby-comment"># create_enzyme_actions returns two seperate array elements, they're not</span>
  <span class="ruby-comment"># needed separated here so we put them into one array</span>
  <span class="ruby-identifier">enzyme_actions</span> = <span class="ruby-identifier">create_enzyme_actions</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> ).<span class="ruby-identifier">flatten</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">fragments_for_display</span>( {} ) <span class="ruby-keyword">if</span> <span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">empty?</span>
  
  <span class="ruby-comment"># Primary and complement strands are both measured from '0' to 'sequence.size-1' here</span>
  <span class="ruby-identifier">sequence_range</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Range</span><span class="ruby-operator">::</span><span class="ruby-constant">SequenceRange</span>.<span class="ruby-identifier">new</span>( <span class="ruby-value">0</span>, <span class="ruby-value">0</span>, <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">-</span><span class="ruby-value">1</span>, <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">-</span><span class="ruby-value">1</span> )
  
  <span class="ruby-comment"># Add the cuts to the sequence_range from each enzyme_action</span>
  <span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">enzyme_action</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">cut_range</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">add_cut_range</span>(<span class="ruby-identifier">cut_range</span>)
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-comment"># Fill in the source sequence for sequence_range so it knows what bases</span>
  <span class="ruby-comment"># to use</span>
  <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">primary</span> = <span class="ruby-identifier">sequence</span>
  <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">complement</span> = <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">forward_complement</span>
  
  <span class="ruby-comment"># Format the fragments for the user</span>
  <span class="ruby-identifier">fragments_for_display</span>( {<span class="ruby-value">0</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">sequence_range</span>} )
<span class="ruby-keyword">end</span></pre>
          </div><!-- cut_without_permutations-source -->
          
        </div>

        

        
      </div><!-- cut_without_permutations-method -->

    
    </section><!-- public-instance-method-details -->
  
     <section id="protected-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Protected Instance Methods</h3>

    
      <div id="method-i-create_enzyme_actions" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">create_enzyme_actions</span><span
            class="method-args">( sequence, *args )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Creates an array of EnzymeActions based on the DNA sequence and supplied
enzymes.</p>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>sequence</code>: The string of DNA to match the enzyme recognition
sites against</p>
</li><li><dl class="rdoc-list note-list"><dt><code>args</code>
<dd>
<p>The enzymes to use.</p>
</dd></dl>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><code>Array</code> with the first element being an array of EnzymeAction
objects that <code>sometimes_cut</code>, and are subject to competition. 
The second is an array of EnzymeAction objects that <code>always_cut</code>
and are not subject to competition.</p>
</dd></dl>
          

          
          <div class="method-source-code" id="create_enzyme_actions-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis_basic.rb, line 120</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">create_enzyme_actions</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-identifier">all_enzyme_actions</span> = []
  
  <span class="ruby-identifier">args</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">enzyme</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">enzyme</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">enzyme</span>) <span class="ruby-keyword">unless</span> <span class="ruby-identifier">enzyme</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">DoubleStranded</span>

    <span class="ruby-comment"># make sure pattern is the proper size</span>
    <span class="ruby-comment"># for more info see the internal documentation of </span>
    <span class="ruby-comment"># Bio::RestrictionEnzyme::DoubleStranded.create_action_at</span>
    <span class="ruby-identifier">pattern</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(
      <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">DoubleStranded</span><span class="ruby-operator">::</span><span class="ruby-constant">AlignedStrands</span>.<span class="ruby-identifier">align</span>(
        <span class="ruby-identifier">enzyme</span>.<span class="ruby-identifier">primary</span>, <span class="ruby-identifier">enzyme</span>.<span class="ruby-identifier">complement</span>
      ).<span class="ruby-identifier">primary</span>
    ).<span class="ruby-identifier">to_re</span>
    
    <span class="ruby-identifier">find_match_locations</span>( <span class="ruby-identifier">sequence</span>, <span class="ruby-identifier">pattern</span> ).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">offset</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">all_enzyme_actions</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">enzyme</span>.<span class="ruby-identifier">create_action_at</span>( <span class="ruby-identifier">offset</span> )
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  
  <span class="ruby-comment"># FIXME VerticalCutRange should really be called VerticalAndHorizontalCutRange</span>
  
  <span class="ruby-comment"># * all_enzyme_actions is now full of EnzymeActions at specific locations across </span>
  <span class="ruby-comment">#   the sequence.</span>
  <span class="ruby-comment"># * all_enzyme_actions will now be examined to see if any EnzymeActions may</span>
  <span class="ruby-comment">#   conflict with one another, and if they do they'll be made note of in</span>
  <span class="ruby-comment">#   indicies_of_sometimes_cut.  They will then be remove FIXME</span>
  <span class="ruby-comment"># * a conflict occurs if another enzyme's bind site is compromised do due</span>
  <span class="ruby-comment">#   to another enzyme's cut.  Enzyme's bind sites may overlap and not be</span>
  <span class="ruby-comment">#   competitive, however neither bind site may be part of the other</span>
  <span class="ruby-comment">#   enzyme's cut or else they do become competitive.</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># Take current EnzymeAction's entire bind site and compare it to all other</span>
  <span class="ruby-comment"># EzymeAction's cut ranges.  Only look for vertical cuts as boundaries</span>
  <span class="ruby-comment"># since trailing horizontal cuts would have no influence on the bind site.</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># If example Enzyme A makes this cut pattern (cut range 2..5):</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># 0 1 2|3 4 5 6 7</span>
  <span class="ruby-comment">#      +-----+</span>
  <span class="ruby-comment"># 0 1 2 3 4 5|6 7</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># Then the bind site (and EnzymeAction range) for Enzyme B would need it's</span>
  <span class="ruby-comment"># right side to be at index 2 or less, or it's left side to be 6 or greater.</span>
  
  <span class="ruby-identifier">competition_indexes</span> = <span class="ruby-constant">Set</span>.<span class="ruby-identifier">new</span>

  <span class="ruby-identifier">all_enzyme_actions</span>[<span class="ruby-value">0</span><span class="ruby-operator">..</span><span class="ruby-value">-2</span>].<span class="ruby-identifier">each_with_index</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">current_enzyme_action</span>, <span class="ruby-identifier">i</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">next</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">competition_indexes</span>.<span class="ruby-identifier">include?</span> <span class="ruby-identifier">i</span>
    <span class="ruby-keyword">next</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">current_enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">empty?</span>  <span class="ruby-comment"># no cuts, some enzymes are like this (ex. CjuI)</span>
    
    <span class="ruby-identifier">all_enzyme_actions</span>[<span class="ruby-identifier">i</span><span class="ruby-operator">+</span><span class="ruby-value">1</span><span class="ruby-operator">..</span><span class="ruby-value">-1</span>].<span class="ruby-identifier">each_with_index</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">comparison_enzyme_action</span>, <span class="ruby-identifier">j</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">j</span> <span class="ruby-operator">+=</span> (<span class="ruby-identifier">i</span> <span class="ruby-operator">+</span> <span class="ruby-value">1</span>)
      <span class="ruby-keyword">next</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">competition_indexes</span>.<span class="ruby-identifier">include?</span> <span class="ruby-identifier">j</span>
      <span class="ruby-keyword">next</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">comparison_enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">empty?</span>  <span class="ruby-comment"># no cuts</span>
      
      <span class="ruby-keyword">if</span> (<span class="ruby-identifier">current_enzyme_action</span>.<span class="ruby-identifier">right</span> <span class="ruby-operator">&lt;=</span> <span class="ruby-identifier">comparison_enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">min_vertical</span>) <span class="ruby-keyword">or</span>
         (<span class="ruby-identifier">current_enzyme_action</span>.<span class="ruby-identifier">left</span> <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">comparison_enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">max_vertical</span>)
        <span class="ruby-comment"># no conflict</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">competition_indexes</span> <span class="ruby-operator">+=</span> [<span class="ruby-identifier">i</span>, <span class="ruby-identifier">j</span>] <span class="ruby-comment"># merge both indexes into the flat set</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
      
  <span class="ruby-identifier">sometimes_cut</span> = <span class="ruby-identifier">all_enzyme_actions</span>.<span class="ruby-identifier">values_at</span>( *<span class="ruby-identifier">competition_indexes</span> )
  <span class="ruby-identifier">always_cut</span> = <span class="ruby-identifier">all_enzyme_actions</span>
  <span class="ruby-identifier">always_cut</span>.<span class="ruby-identifier">delete_if</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span> <span class="ruby-identifier">sometimes_cut</span>.<span class="ruby-identifier">include?</span> <span class="ruby-identifier">x</span> }

  [<span class="ruby-identifier">sometimes_cut</span>, <span class="ruby-identifier">always_cut</span>]
<span class="ruby-keyword">end</span></pre>
          </div><!-- create_enzyme_actions-source -->
          
        </div>

        

        
      </div><!-- create_enzyme_actions-method -->

    
      <div id="method-i-cut_and_return_by_permutations" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cut_and_return_by_permutations</span><span
            class="method-args">( sequence, *args )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>See cut instance method</p>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>sequence</code>: <code>String</code> kind of object that will be used
as a nucleic acid sequence.</p>
</li><li>
<p><code>args</code>: Series of enzyme names, enzymes sequences with cut
marks, or <a href="../RestrictionEnzyme.html">RestrictionEnzyme</a>
objects.</p>
</li></ul>

<p>May also supply a <code>Hash</code> with the key “:max_permutations” to
specificy how many permutations are allowed - a value of 0 indicates no
permutations are allowed.</p>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><code>Hash</code> Keys are a permutation ID, values are SequenceRange
objects that have cuts applied.</p>
</dd></dl>

<p><em>also</em> may return the <code>Symbol</code> ':sequence_empty',
':no_cuts_found', or ':too_many_permutations'</p>
          

          
          <div class="method-source-code" id="cut_and_return_by_permutations-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis.rb, line 81</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cut_and_return_by_permutations</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )
  <span class="ruby-identifier">my_hash</span> = {}
  <span class="ruby-identifier">maximum_permutations</span> = <span class="ruby-keyword">nil</span>

  <span class="ruby-identifier">hashes_in_args</span> = <span class="ruby-identifier">args</span>.<span class="ruby-identifier">select</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">i</span><span class="ruby-operator">|</span> <span class="ruby-identifier">i</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Hash</span> }
  <span class="ruby-identifier">args</span>.<span class="ruby-identifier">delete_if</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">i</span><span class="ruby-operator">|</span> <span class="ruby-identifier">i</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Hash</span> }
  <span class="ruby-identifier">hashes_in_args</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">hsh</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">hsh</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">key</span>, <span class="ruby-identifier">value</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">case</span> <span class="ruby-identifier">key</span>
      <span class="ruby-keyword">when</span> <span class="ruby-value">:max_permutations</span>, <span class="ruby-string">'max_permutations'</span>, <span class="ruby-value">:maximum_permutations</span>, <span class="ruby-string">'maximum_permutations'</span>
        <span class="ruby-identifier">maximum_permutations</span> = <span class="ruby-identifier">value</span>.<span class="ruby-identifier">to_i</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">value</span> <span class="ruby-operator">==</span> <span class="ruby-keyword">nil</span>
      <span class="ruby-keyword">when</span> <span class="ruby-value">:view_ranges</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">raise</span> <span class="ruby-constant">ArgumentError</span>, <span class="ruby-node">&quot;Received key #{key.inspect} in argument - I only know the key ':max_permutations' and ':view_ranges' currently.  Hash passed: #{hsh.inspect}&quot;</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  
  <span class="ruby-keyword">if</span> <span class="ruby-operator">!</span><span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">kind_of?</span>(<span class="ruby-constant">String</span>) <span class="ruby-keyword">or</span> <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">empty?</span>
    <span class="ruby-identifier">logger</span>.<span class="ruby-identifier">warn</span> <span class="ruby-string">&quot;The supplied sequence is empty.&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-identifier">logger</span>)
    <span class="ruby-keyword">return</span> <span class="ruby-value">:sequence_empty</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">sequence</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>( <span class="ruby-identifier">sequence</span> )
  
  <span class="ruby-identifier">enzyme_actions</span>, <span class="ruby-identifier">initial_cuts</span> = <span class="ruby-identifier">create_enzyme_actions</span>( <span class="ruby-identifier">sequence</span>, *<span class="ruby-identifier">args</span> )

  <span class="ruby-keyword">if</span> <span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">initial_cuts</span>.<span class="ruby-identifier">empty?</span>
    <span class="ruby-identifier">logger</span>.<span class="ruby-identifier">warn</span> <span class="ruby-string">&quot;This enzyme does not make any cuts on this sequence.&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-identifier">logger</span>)
    <span class="ruby-keyword">return</span> <span class="ruby-value">:no_cuts_found</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-comment"># * When enzyme_actions.size is equal to '1' that means there are no permutations.</span>
  <span class="ruby-comment"># * If enzyme_actions.size is equal to '2' there is one</span>
  <span class="ruby-comment">#   permutation (&quot;[0, 1]&quot;)</span>
  <span class="ruby-comment"># * If enzyme_actions.size is equal to '3' there are two</span>
  <span class="ruby-comment">#   permutations (&quot;[0, 1, 2]&quot;)</span>
  <span class="ruby-comment"># * and so on..</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">maximum_permutations</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">1</span>
    <span class="ruby-keyword">if</span> (<span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">-</span> <span class="ruby-value">1</span>) <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">maximum_permutations</span>.<span class="ruby-identifier">to_i</span>
      <span class="ruby-identifier">logger</span>.<span class="ruby-identifier">warn</span> <span class="ruby-node">&quot;More permutations than maximum, skipping.  Found: #{enzyme_actions.size-1}  Max: #{maximum_permutations.to_i}&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-identifier">logger</span>)
      <span class="ruby-keyword">return</span> <span class="ruby-value">:too_many_permutations</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">1</span>
    <span class="ruby-identifier">permutations</span> = <span class="ruby-identifier">permute</span>(<span class="ruby-identifier">enzyme_actions</span>.<span class="ruby-identifier">size</span>)
    
    <span class="ruby-identifier">permutations</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">permutation</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">previous_cut_ranges</span> = []
      <span class="ruby-comment"># Primary and complement strands are both measured from '0' to 'sequence.size-1' here</span>
      <span class="ruby-identifier">sequence_range</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Range</span><span class="ruby-operator">::</span><span class="ruby-constant">SequenceRange</span>.<span class="ruby-identifier">new</span>( <span class="ruby-value">0</span>, <span class="ruby-value">0</span>, <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">-</span><span class="ruby-value">1</span>, <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">-</span><span class="ruby-value">1</span> )

      <span class="ruby-comment"># Add the cuts to the sequence_range from each enzyme_action contained</span>
      <span class="ruby-comment"># in initial_cuts.  These are the cuts that have no competition so are</span>
      <span class="ruby-comment"># not subject to permutations.</span>
      <span class="ruby-identifier">initial_cuts</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">enzyme_action</span><span class="ruby-operator">|</span>
        <span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">cut_range</span><span class="ruby-operator">|</span>
          <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">add_cut_range</span>(<span class="ruby-identifier">cut_range</span>)
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>

      <span class="ruby-identifier">permutation</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">id</span><span class="ruby-operator">|</span>
        <span class="ruby-identifier">enzyme_action</span> = <span class="ruby-identifier">enzyme_actions</span>[<span class="ruby-identifier">id</span>]

        <span class="ruby-comment"># conflict is false if the current enzyme action may cut in it's range.</span>
        <span class="ruby-comment"># conflict is true if it cannot due to a previous enzyme action making</span>
        <span class="ruby-comment"># a cut where this enzyme action needs a whole recognition site.</span>
        <span class="ruby-identifier">conflict</span> = <span class="ruby-keyword">false</span>

        <span class="ruby-comment"># If current size of enzyme_action overlaps with previous cut_range, don't cut</span>
        <span class="ruby-comment"># note that the enzyme action may fall in the middle of a previous enzyme action</span>
        <span class="ruby-comment"># so all cut locations must be checked that would fall underneath.</span>
        <span class="ruby-identifier">previous_cut_ranges</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">cut_range</span><span class="ruby-operator">|</span>
          <span class="ruby-keyword">next</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">cut_range</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Range</span><span class="ruby-operator">::</span><span class="ruby-constant">VerticalCutRange</span>  <span class="ruby-comment"># we aren't concerned with horizontal cuts</span>
          <span class="ruby-identifier">previous_cut_left</span> = <span class="ruby-identifier">cut_range</span>.<span class="ruby-identifier">range</span>.<span class="ruby-identifier">first</span> 
          <span class="ruby-identifier">previous_cut_right</span> = <span class="ruby-identifier">cut_range</span>.<span class="ruby-identifier">range</span>.<span class="ruby-identifier">last</span>

          <span class="ruby-comment"># Keep in mind: </span>
          <span class="ruby-comment"># * The cut location is to the immediate right of the base located at the index.</span>
          <span class="ruby-comment">#   ex: at^gc -- the cut location is at index 1</span>
          <span class="ruby-comment"># * The enzyme action location is located at the base of the index.</span>
          <span class="ruby-comment">#   ex: atgc -- 0 =&gt; 'a', 1 =&gt; 't', 2 =&gt; 'g', 3 =&gt; 'c'</span>
          <span class="ruby-comment"># method create_enzyme_actions has similar commentary if interested</span>
          <span class="ruby-keyword">if</span> (<span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">right</span> <span class="ruby-operator">&lt;=</span> <span class="ruby-identifier">previous_cut_left</span>) <span class="ruby-keyword">or</span>
             (<span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">left</span> <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">previous_cut_right</span>) <span class="ruby-keyword">or</span>
             (<span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">left</span> <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">previous_cut_left</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">right</span> <span class="ruby-operator">&lt;=</span> <span class="ruby-identifier">previous_cut_right</span>) <span class="ruby-comment"># in between cuts</span>
            <span class="ruby-comment"># no conflict</span>
          <span class="ruby-keyword">else</span>
            <span class="ruby-identifier">conflict</span> = <span class="ruby-keyword">true</span>
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">end</span>

        <span class="ruby-keyword">next</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">conflict</span> <span class="ruby-operator">==</span> <span class="ruby-keyword">true</span>
        <span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">cut_range</span><span class="ruby-operator">|</span> <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">add_cut_range</span>(<span class="ruby-identifier">cut_range</span>) }
        <span class="ruby-identifier">previous_cut_ranges</span> <span class="ruby-operator">+=</span> <span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>        
      <span class="ruby-keyword">end</span> <span class="ruby-comment"># permutation.each</span>

      <span class="ruby-comment"># Fill in the source sequence for sequence_range so it knows what bases</span>
      <span class="ruby-comment"># to use</span>
      <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">primary</span> = <span class="ruby-identifier">sequence</span>
      <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">complement</span> = <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">forward_complement</span>
      <span class="ruby-identifier">my_hash</span>[<span class="ruby-identifier">permutation</span>] = <span class="ruby-identifier">sequence_range</span>
    <span class="ruby-keyword">end</span> <span class="ruby-comment"># permutations.each</span>
    
  <span class="ruby-keyword">else</span> <span class="ruby-comment"># if enzyme_actions.size == 1</span>
    <span class="ruby-comment"># no permutations, just do it</span>
    <span class="ruby-identifier">sequence_range</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Range</span><span class="ruby-operator">::</span><span class="ruby-constant">SequenceRange</span>.<span class="ruby-identifier">new</span>( <span class="ruby-value">0</span>, <span class="ruby-value">0</span>, <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">-</span><span class="ruby-value">1</span>, <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">size</span><span class="ruby-operator">-</span><span class="ruby-value">1</span> )
    <span class="ruby-identifier">initial_cuts</span>.<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">enzyme_action</span><span class="ruby-operator">|</span> <span class="ruby-identifier">enzyme_action</span>.<span class="ruby-identifier">cut_ranges</span>.<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">cut_range</span><span class="ruby-operator">|</span> <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">add_cut_range</span>(<span class="ruby-identifier">cut_range</span>) } }
    <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">primary</span> = <span class="ruby-identifier">sequence</span>
    <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">complement</span> = <span class="ruby-identifier">sequence</span>.<span class="ruby-identifier">forward_complement</span>
    <span class="ruby-identifier">my_hash</span>[<span class="ruby-value">0</span>] = <span class="ruby-identifier">sequence_range</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-identifier">my_hash</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- cut_and_return_by_permutations-source -->
          
        </div>

        

        
      </div><!-- cut_and_return_by_permutations-method -->

    
      <div id="method-i-find_match_locations" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">find_match_locations</span><span
            class="method-args">( string, re )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns an <code>Array</code> of the match indicies of a
<code>RegExp</code> to a string.</p>

<p>Example:</p>

<pre class="ruby"><span class="ruby-identifier">find_match_locations</span>(<span class="ruby-string">'abccdefeg'</span>, <span class="ruby-regexp">%r[ce]/</span>) <span class="ruby-comment"># =&gt; [2,3,5,7]</span>
</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>string</code>: The string to scan</p>
</li><li>
<p><code>re</code>: A RegExp to use</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><code>Array</code> with indicies of match locations</p>
</dd></dl>
          

          
          <div class="method-source-code" id="find_match_locations-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis_basic.rb, line 203</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">find_match_locations</span>( <span class="ruby-identifier">string</span>, <span class="ruby-identifier">re</span> )
  <span class="ruby-identifier">md</span> = <span class="ruby-identifier">string</span>.<span class="ruby-identifier">match</span>( <span class="ruby-identifier">re</span> )
  <span class="ruby-identifier">locations</span> = []
  <span class="ruby-identifier">counter</span> = <span class="ruby-value">0</span>
  <span class="ruby-keyword">while</span> <span class="ruby-identifier">md</span>
    <span class="ruby-comment"># save the match index relative to the original string</span>
    <span class="ruby-identifier">locations</span> <span class="ruby-operator">&lt;&lt;</span> (<span class="ruby-identifier">counter</span> <span class="ruby-operator">+=</span> <span class="ruby-identifier">md</span>.<span class="ruby-identifier">begin</span>(<span class="ruby-value">0</span>))
    <span class="ruby-comment"># find the next match</span>
    <span class="ruby-identifier">md</span> = <span class="ruby-identifier">string</span>[ (<span class="ruby-identifier">counter</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>)<span class="ruby-operator">..</span><span class="ruby-value">-1</span> ].<span class="ruby-identifier">match</span>( <span class="ruby-identifier">re</span> )
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">locations</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- find_match_locations-source -->
          
        </div>

        

        
      </div><!-- find_match_locations-method -->

    
      <div id="method-i-fragments_for_display" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">fragments_for_display</span><span
            class="method-args">( hsh, view_ranges=false )</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Take the fragments from SequenceRange objects generated from add_cut_range
and return unique results as a Bio::RestrictionEnzyme::Analysis::Fragment
object.</p>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>hsh</code>: <code>Hash</code>  Keys are a permutation ID, if any. 
Values are SequenceRange objects that have cuts applied.</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Bio::RestrictionEnzyme::Analysis::Fragments object populated with
Bio::RestrictionEnzyme::Analysis::Fragment objects.</p>
</dd></dl>
          

          
          <div class="method-source-code" id="fragments_for_display-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis_basic.rb, line 94</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">fragments_for_display</span>( <span class="ruby-identifier">hsh</span>, <span class="ruby-identifier">view_ranges</span>=<span class="ruby-keyword">false</span> )
  <span class="ruby-identifier">ary</span> = <span class="ruby-constant">Fragments</span>.<span class="ruby-identifier">new</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">ary</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">hsh</span>

  <span class="ruby-identifier">hsh</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">permutation_id</span>, <span class="ruby-identifier">sequence_range</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">sequence_range</span>.<span class="ruby-identifier">fragments</span>.<span class="ruby-identifier">for_display</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">fragment</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">view_ranges</span>
        <span class="ruby-identifier">ary</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Fragment</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">primary</span>, <span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">complement</span>, <span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">p_left</span>, <span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">p_right</span>, <span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">c_left</span>, <span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">c_right</span>)
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">ary</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">RestrictionEnzyme</span><span class="ruby-operator">::</span><span class="ruby-constant">Fragment</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">primary</span>, <span class="ruby-identifier">fragment</span>.<span class="ruby-identifier">complement</span>)
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  
  <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">uniq!</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">view_ranges</span>
  
  <span class="ruby-identifier">ary</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- fragments_for_display-source -->
          
        </div>

        

        
      </div><!-- fragments_for_display-method -->

    
      <div id="method-i-permute" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">permute</span><span
            class="method-args">(count, permutations = [[0]])</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns permutation orders for a given number of elements.</p>

<p>Examples:</p>

<pre class="ruby"><span class="ruby-identifier">permute</span>(<span class="ruby-value">0</span>) <span class="ruby-comment"># =&gt; [[0]]</span>
<span class="ruby-identifier">permute</span>(<span class="ruby-value">1</span>) <span class="ruby-comment"># =&gt; [[0]]</span>
<span class="ruby-identifier">permute</span>(<span class="ruby-value">2</span>) <span class="ruby-comment"># =&gt; [[1, 0], [0, 1]]</span>
<span class="ruby-identifier">permute</span>(<span class="ruby-value">3</span>) <span class="ruby-comment"># =&gt; [[2, 1, 0], [2, 0, 1], [1, 2, 0], [0, 2, 1], [1, 0, 2], [0, 1, 2]]</span>
<span class="ruby-identifier">permute</span>(<span class="ruby-value">4</span>) <span class="ruby-comment"># =&gt; [[3, 2, 1, 0],</span>
                 [<span class="ruby-value">3</span>, <span class="ruby-value">2</span>, <span class="ruby-value">0</span>, <span class="ruby-value">1</span>],
                 [<span class="ruby-value">3</span>, <span class="ruby-value">1</span>, <span class="ruby-value">2</span>, <span class="ruby-value">0</span>],
                 [<span class="ruby-value">3</span>, <span class="ruby-value">0</span>, <span class="ruby-value">2</span>, <span class="ruby-value">1</span>],
                 [<span class="ruby-value">3</span>, <span class="ruby-value">1</span>, <span class="ruby-value">0</span>, <span class="ruby-value">2</span>],
                 [<span class="ruby-value">3</span>, <span class="ruby-value">0</span>, <span class="ruby-value">1</span>, <span class="ruby-value">2</span>],
                 [<span class="ruby-value">2</span>, <span class="ruby-value">3</span>, <span class="ruby-value">1</span>, <span class="ruby-value">0</span>],
                 [<span class="ruby-value">2</span>, <span class="ruby-value">3</span>, <span class="ruby-value">0</span>, <span class="ruby-value">1</span>],
                 [<span class="ruby-value">1</span>, <span class="ruby-value">3</span>, <span class="ruby-value">2</span>, <span class="ruby-value">0</span>],
                 [<span class="ruby-value">0</span>, <span class="ruby-value">3</span>, <span class="ruby-value">2</span>, <span class="ruby-value">1</span>],
                 [<span class="ruby-value">1</span>, <span class="ruby-value">3</span>, <span class="ruby-value">0</span>, <span class="ruby-value">2</span>],
                 [<span class="ruby-value">0</span>, <span class="ruby-value">3</span>, <span class="ruby-value">1</span>, <span class="ruby-value">2</span>],
                 [<span class="ruby-value">2</span>, <span class="ruby-value">1</span>, <span class="ruby-value">3</span>, <span class="ruby-value">0</span>],
                 [<span class="ruby-value">2</span>, <span class="ruby-value">0</span>, <span class="ruby-value">3</span>, <span class="ruby-value">1</span>],
                 [<span class="ruby-value">1</span>, <span class="ruby-value">2</span>, <span class="ruby-value">3</span>, <span class="ruby-value">0</span>],
                 [<span class="ruby-value">0</span>, <span class="ruby-value">2</span>, <span class="ruby-value">3</span>, <span class="ruby-value">1</span>],
                 [<span class="ruby-value">1</span>, <span class="ruby-value">0</span>, <span class="ruby-value">3</span>, <span class="ruby-value">2</span>],
                 [<span class="ruby-value">0</span>, <span class="ruby-value">1</span>, <span class="ruby-value">3</span>, <span class="ruby-value">2</span>],
                 [<span class="ruby-value">2</span>, <span class="ruby-value">1</span>, <span class="ruby-value">0</span>, <span class="ruby-value">3</span>],
                 [<span class="ruby-value">2</span>, <span class="ruby-value">0</span>, <span class="ruby-value">1</span>, <span class="ruby-value">3</span>],
                 [<span class="ruby-value">1</span>, <span class="ruby-value">2</span>, <span class="ruby-value">0</span>, <span class="ruby-value">3</span>],
                 [<span class="ruby-value">0</span>, <span class="ruby-value">2</span>, <span class="ruby-value">1</span>, <span class="ruby-value">3</span>],
                 [<span class="ruby-value">1</span>, <span class="ruby-value">0</span>, <span class="ruby-value">2</span>, <span class="ruby-value">3</span>],
                 [<span class="ruby-value">0</span>, <span class="ruby-value">1</span>, <span class="ruby-value">2</span>, <span class="ruby-value">3</span>]]
</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong></p>
<ul><li>
<p><code>count</code>: <code>Number</code> of different elements to be
permuted</p>
</li><li>
<p><code>permutations</code>: ignore - for the recursive algorithm</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><code>Array</code> of <code>Array</code> objects with different possible
permutation orders.  See examples.</p>
</dd></dl>
          

          
          <div class="method-source-code" id="permute-source">
            <pre><span class="ruby-comment"># File lib/bio/util/restriction_enzyme/analysis.rb, line 235</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">permute</span>(<span class="ruby-identifier">count</span>, <span class="ruby-identifier">permutations</span> = [[<span class="ruby-value">0</span>]])
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">permutations</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">count</span> <span class="ruby-operator">&lt;=</span> <span class="ruby-value">1</span>
  <span class="ruby-identifier">new_arrays</span> = []
  <span class="ruby-identifier">new_array</span> = []

  (<span class="ruby-identifier">permutations</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">size</span> <span class="ruby-operator">+</span> <span class="ruby-value">1</span>).<span class="ruby-identifier">times</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">n</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">new_array</span>.<span class="ruby-identifier">clear</span>
    <span class="ruby-identifier">permutations</span>.<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">a</span><span class="ruby-operator">|</span> <span class="ruby-identifier">new_array</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">a</span>.<span class="ruby-identifier">dup</span> }
    <span class="ruby-identifier">new_array</span>.<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">e</span><span class="ruby-operator">|</span> <span class="ruby-identifier">e</span>.<span class="ruby-identifier">insert</span>(<span class="ruby-identifier">n</span>, <span class="ruby-identifier">permutations</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">size</span>) }
    <span class="ruby-identifier">new_arrays</span> <span class="ruby-operator">+=</span> <span class="ruby-identifier">new_array</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-identifier">permute</span>(<span class="ruby-identifier">count</span><span class="ruby-operator">-</span><span class="ruby-value">1</span>, <span class="ruby-identifier">new_arrays</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- permute-source -->
          
        </div>

        

        
      </div><!-- permute-method -->

    
    </section><!-- protected-instance-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

