<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>class Bio::PAML::Codeml::Report - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../../../";
  var index_rel_prefix = "../../../";
</script>

<script src="../../../js/navigation.js" defer></script>
<script src="../../../js/search.js" defer></script>
<script src="../../../js/search_index.js" defer></script>
<script src="../../../js/searcher.js" defer></script>
<script src="../../../js/darkfish.js" defer></script>

<link href="../../../css/fonts.css" rel="stylesheet">
<link href="../../../css/rdoc.css" rel="stylesheet">


<body id="top" role="document" class="class">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../../../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../../../table_of_contents.html#pages">Pages</a>
    <a href="../../../table_of_contents.html#classes">Classes</a>
    <a href="../../../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#class-Bio::PAML::Codeml::Report-label-Description">Description</a>
    <li><a href="#class-Bio::PAML::Codeml::Report-label-References">References</a>
    <li><a href="#class-Bio::PAML::Codeml::Report-label-Examples">Examples</a>
  </ul>
</div>


  <div id="class-metadata">
    
    
<div id="parent-class-section" class="nav-section">
  <h3>Parent</h3>

  <p class="link"><a href="../Common/Report.html">Bio::PAML::Common::Report</a>
</div>

    
    
    
<!-- Method Quickref -->
<div id="method-list-section" class="nav-section">
  <h3>Methods</h3>

  <ul class="link-list" role="directory">
    <li ><a href="#method-c-new">::new</a>
    <li ><a href="#method-i-descr">#descr</a>
    <li ><a href="#method-i-nb_sites">#nb_sites</a>
    <li ><a href="#method-i-num_codons">#num_codons</a>
    <li ><a href="#method-i-num_sequences">#num_sequences</a>
    <li ><a href="#method-i-significant">#significant</a>
    <li ><a href="#method-i-sites">#sites</a>
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-labelledby="class-Bio::PAML::Codeml::Report">
  <h1 id="class-Bio::PAML::Codeml::Report" class="class">
    class Bio::PAML::Codeml::Report
  </h1>

  <section class="description">
    
<h2 id="class-Bio::PAML::Codeml::Report-label-Description">Description<span><a href="#class-Bio::PAML::Codeml::Report-label-Description">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Run <a href="../../PAML.html"><code>PAML</code></a> codeml and get the results from the output file. The <a href="Report.html"><code>Codeml::Report</code></a> object is returned by <a href="../Codeml.html#method-i-query"><code>Bio::PAML::Codeml.query</code></a>. For example</p>

<pre class="ruby"><span class="ruby-identifier">codeml</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">PAML</span><span class="ruby-operator">::</span><span class="ruby-constant">Codeml</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;codeml&#39;</span>, <span class="ruby-value">:runmode</span> <span class="ruby-operator">=&gt;</span> <span class="ruby-value">0</span>, 
    <span class="ruby-value">:RateAncestor</span> <span class="ruby-operator">=&gt;</span> <span class="ruby-value">1</span>, <span class="ruby-value">:alpha</span> <span class="ruby-operator">=&gt;</span> <span class="ruby-value">0.5</span>, <span class="ruby-value">:fix_alpha</span> <span class="ruby-operator">=&gt;</span> <span class="ruby-value">0</span>)
<span class="ruby-identifier">result</span> = <span class="ruby-identifier">codeml</span>.<span class="ruby-identifier">query</span>(<span class="ruby-identifier">alignment</span>, <span class="ruby-identifier">tree</span>)
</pre>

<p>where alignment and tree are Bioruby objects. This class assumes we have a buffer containing the output of codeml. </p>

<h2 id="class-Bio::PAML::Codeml::Report-label-References"><a href="../../References.html"><code>References</code></a><span><a href="#class-Bio::PAML::Codeml::Report-label-References">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Phylogenetic Analysis by Maximum Likelihood (<a href="../../PAML.html"><code>PAML</code></a>) is a package of programs for phylogenetic analyses of DNA or protein sequences using maximum likelihood. It is maintained and distributed for academic use free of charge by Ziheng Yang. Suggestion citation </p>

<pre>Yang, Z. 1997
PAML: a program package for phylogenetic analysis by maximum likelihood 
CABIOS 13:555-556</pre>

<p><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">abacus.gene.ucl.ac.uk/software/paml.html</a></p>

<h2 id="class-Bio::PAML::Codeml::Report-label-Examples">Examples<span><a href="#class-Bio::PAML::Codeml::Report-label-Examples">&para;</a> <a href="#top">&uarr;</a></span></h2>

<p>Invoke Bioruby&#39;s <a href="../../PAML.html"><code>PAML</code></a> codeml parser, after having read the contents of the codeml result file into <em>buf</em> (for example using File.read)</p>

<pre>&gt;&gt; c = Bio::PAML::Codeml::Report.new(buf)</pre>

<p>Do we have two models?</p>

<pre>&gt;&gt; c.models.size
=&gt; 2
&gt;&gt; c.models[0].name
=&gt; &quot;M0&quot;
&gt;&gt; c.models[1].name
=&gt; &quot;M3&quot;</pre>

<p>Check the general information</p>

<pre>&gt;&gt; c.num_sequences
=&gt; 6
&gt;&gt; c.num_codons
=&gt; 134
&gt;&gt; c.descr
=&gt; &quot;M0-3&quot;</pre>

<p>Test whether the second model M3 is significant over M0</p>

<pre>&gt;&gt; c.significant
=&gt; true</pre>

<p>Now fetch the results of the first model M0, and check its values</p>

<pre>&gt;&gt; m0 = c.models[0]
&gt;&gt; m0.tree_length
=&gt; 1.90227
&gt;&gt; m0.lnL
=&gt; -1125.800375
&gt;&gt; m0.omega
=&gt; 0.58589
&gt;&gt; m0.dN_dS
=&gt; 0.58589
&gt;&gt; m0.kappa
=&gt; 2.14311
&gt;&gt; m0.alpha
=&gt; nil</pre>

<p>We also have a tree (as a string)</p>

<pre>&gt;&gt; m0.tree
=&gt; &quot;((((PITG_23265T0: 0.000004, PITG_23253T0: 0.400074): 0.000004, PITG_23257T0: 0.952614): 0.000004, PITG_23264T0: 0.445507): 0.000004, PITG_23267T0: 0.011814, PITG_23293T0: 0.092242);&quot;</pre>

<p>Check the M3 and its specific values</p>

<pre>&gt;&gt; m3 = c.models[1]
&gt;&gt; m3.lnL
=&gt; -1070.964046
&gt;&gt; m3.classes.size
=&gt; 3
&gt;&gt; m3.classes[0]
=&gt; {:w=&gt;0.00928, :p=&gt;0.56413}</pre>

<p>And the tree</p>

<pre>&gt;&gt; m3.tree
=&gt; &quot;((((PITG_23265T0: 0.000004, PITG_23253T0: 0.762597): 0.000004, PITG_23257T0: 2.721710): 0.000004, PITG_23264T0: 0.924326): 0.014562, PITG_23267T0: 0.000004, PITG_23293T0: 0.237433);&quot;</pre>

<p>Next take the overall posterior analysis</p>

<pre>&gt;&gt; c.nb_sites.size
=&gt; 44
&gt;&gt; c.nb_sites[0].to_a
=&gt; [17, &quot;I&quot;, 0.988, 3.293]</pre>

<p>or by field</p>

<pre>&gt;&gt; codon = c.nb_sites[0]
&gt;&gt; codon.position
=&gt; 17
&gt;&gt; codon.probability
=&gt; 0.988
&gt;&gt; codon.dN_dS
=&gt; 3.293</pre>

<p>with aliases</p>

<pre>&gt;&gt; codon.p
=&gt; 0.988
&gt;&gt; codon.w
=&gt; 3.293</pre>

<p>Now we generate special string &#39;graph&#39; for positive selection. The  following returns a string the length of the input alignment and  shows the locations of positive selection:</p>

<pre>&gt;&gt; c.nb_sites.graph[0..32]
=&gt; &quot;                **    *       * *&quot;</pre>

<p>And with dN/dS (high values are still an asterisk *)</p>

<pre>&gt;&gt; c.nb_sites.graph_omega[0..32]
=&gt; &quot;                3*    6       6 2&quot;</pre>

<p>We also provide the raw buffers to adhere to the principle of  unexpected use. Test the raw buffers for content:</p>

<pre>&gt;&gt; c.header.to_s =~ /seed/
=&gt; 1
&gt;&gt; m0.to_s =~ /one-ratio/
=&gt; 3
&gt;&gt; m3.to_s =~ /discrete/
=&gt; 3
&gt;&gt; c.footer.to_s =~ /Bayes/
=&gt; 16</pre>

<p>Finally we do a test on an M7+M8 run. Again, after loading the results file into <em>buf</em></p>

<p>Invoke Bioruby&#39;s <a href="../../PAML.html"><code>PAML</code></a> codeml parser</p>

<pre>&gt;&gt; c = Bio::PAML::Codeml::Report.new(buf78)</pre>

<p>Do we have two models?</p>

<pre>&gt;&gt; c.models.size
=&gt; 2
&gt;&gt; c.models[0].name
=&gt; &quot;M7&quot;
&gt;&gt; c.models[1].name
=&gt; &quot;M8&quot;</pre>

<p>Assert the results are significant</p>

<pre>&gt;&gt; c.significant
=&gt; true</pre>

<p>Compared to M0/M3 there are some differences. The important ones are the parameters and the full Bayesian result available for M7/M8. This is the naive Bayesian result:</p>

<pre>&gt;&gt; c.nb_sites.size
=&gt; 10</pre>

<p>And this is the full Bayesian result:</p>

<pre>&gt;&gt; c.sites.size
=&gt; 30
&gt;&gt; c.sites[0].to_a
=&gt; [17, &quot;I&quot;, 0.672, 2.847]
&gt;&gt; c.sites.graph[0..32]
=&gt; &quot;                **    *       * *&quot;</pre>

<p>Note the differences of omega with earlier M0-M3 naive Bayesian  analysis:</p>

<pre>&gt;&gt; c.sites.graph_omega[0..32]
=&gt; &quot;                24    3       3 2&quot;</pre>

<p>The locations are the same, but the omega differs.</p>

  </section>

  <section id="5Buntitled-5D" class="documentation-section">



    <section class="attribute-method-details" class="method-section">
      <header>
        <h3>Attributes</h3>
      </header>

      <div id="attribute-i-footer" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">footer</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        </div>
      </div>
      <div id="attribute-i-header" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">header</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        </div>
      </div>
      <div id="attribute-i-models" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">models</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        </div>
      </div>
    </section>


     <section id="public-class-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Public Class Methods</h3>
       </header>

      <div id="method-c-new" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(buf)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>Parse codeml output file passed with <code>buf</code>, where buf contains the content of a codeml result file</p>

          <div class="method-source-code" id="new-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">222</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">initialize</span> <span class="ruby-identifier">buf</span>
<span class="line-num">223</span>   <span class="ruby-comment"># split the main buffer into sections for each model, header and footer.</span>
<span class="line-num">224</span>   <span class="ruby-identifier">sections</span> = <span class="ruby-identifier">buf</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">&quot;\nModel &quot;</span>)
<span class="line-num">225</span>   <span class="ruby-identifier">model_num</span> = <span class="ruby-identifier">sections</span>.<span class="ruby-identifier">size</span><span class="ruby-value">-1</span>
<span class="line-num">226</span>   <span class="ruby-identifier">raise</span> <span class="ruby-constant">ReportError</span>,<span class="ruby-node">&quot;Incorrect codeml data models=#{model_num}&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">model_num</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">2</span>
<span class="line-num">227</span>   <span class="ruby-identifier">foot2</span> = <span class="ruby-identifier">sections</span>[<span class="ruby-identifier">model_num</span>].<span class="ruby-identifier">split</span>(<span class="ruby-string">&quot;\nNaive &quot;</span>)
<span class="line-num">228</span>   <span class="ruby-keyword">if</span> <span class="ruby-identifier">foot2</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">==</span> <span class="ruby-value">2</span>
<span class="line-num">229</span>     <span class="ruby-comment"># We have a dual model</span>
<span class="line-num">230</span>     <span class="ruby-identifier">sections</span>[<span class="ruby-identifier">model_num</span>] = <span class="ruby-identifier">foot2</span>[<span class="ruby-value">0</span>]
<span class="line-num">231</span>     <span class="ruby-ivar">@footer</span> = <span class="ruby-string">&#39;Naive &#39;</span><span class="ruby-operator">+</span><span class="ruby-identifier">foot2</span>[<span class="ruby-value">1</span>]
<span class="line-num">232</span>     <span class="ruby-ivar">@models</span> = []
<span class="line-num">233</span>     <span class="ruby-identifier">sections</span>[<span class="ruby-value">1</span><span class="ruby-operator">..</span><span class="ruby-value">-1</span>].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span> <span class="ruby-identifier">model_buf</span> <span class="ruby-operator">|</span>
<span class="line-num">234</span>       <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">push</span> <span class="ruby-constant">Model</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">model_buf</span>)
<span class="line-num">235</span>     <span class="ruby-keyword">end</span>
<span class="line-num">236</span>   <span class="ruby-keyword">else</span>
<span class="line-num">237</span>     <span class="ruby-comment"># A single model is run</span>
<span class="line-num">238</span>     <span class="ruby-identifier">sections</span> = <span class="ruby-identifier">buf</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">&quot;\nTREE #&quot;</span>)
<span class="line-num">239</span>     <span class="ruby-identifier">model_num</span> = <span class="ruby-identifier">sections</span>.<span class="ruby-identifier">size</span><span class="ruby-value">-1</span>
<span class="line-num">240</span>     <span class="ruby-identifier">raise</span> <span class="ruby-constant">ReportError</span>,<span class="ruby-string">&quot;Can not parse single model file&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">model_num</span> <span class="ruby-operator">!=</span> <span class="ruby-value">1</span>
<span class="line-num">241</span>     <span class="ruby-ivar">@models</span> = []
<span class="line-num">242</span>     <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">push</span> <span class="ruby-identifier">sections</span>[<span class="ruby-value">1</span>]
<span class="line-num">243</span>     <span class="ruby-ivar">@footer</span> = <span class="ruby-identifier">sections</span>[<span class="ruby-value">1</span>][<span class="ruby-regexp">/Time used/</span>,<span class="ruby-value">1</span>]
<span class="line-num">244</span>     <span class="ruby-ivar">@single</span> = <span class="ruby-constant">ReportSingle</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">buf</span>)
<span class="line-num">245</span>   <span class="ruby-keyword">end</span>
<span class="line-num">246</span>   <span class="ruby-ivar">@header</span> = <span class="ruby-identifier">sections</span>[<span class="ruby-value">0</span>]
<span class="line-num">247</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

    </section>

     <section id="public-instance-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Public Instance Methods</h3>
       </header>

      <div id="method-i-descr" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">descr</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>Give a short description of the models, for example &#39;M0-3&#39;</p>

          <div class="method-source-code" id="descr-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">250</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">descr</span>
<span class="line-num">251</span>   <span class="ruby-identifier">num</span> = <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">size</span>
<span class="line-num">252</span>   <span class="ruby-keyword">case</span> <span class="ruby-identifier">num</span>
<span class="line-num">253</span>     <span class="ruby-keyword">when</span> <span class="ruby-value">0</span> 
<span class="line-num">254</span>       <span class="ruby-string">&#39;No model&#39;</span>
<span class="line-num">255</span>     <span class="ruby-keyword">when</span> <span class="ruby-value">1</span> 
<span class="line-num">256</span>       <span class="ruby-ivar">@models</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">name</span>
<span class="line-num">257</span>     <span class="ruby-keyword">else</span> 
<span class="line-num">258</span>       <span class="ruby-ivar">@models</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">name</span> <span class="ruby-operator">+</span> <span class="ruby-string">&#39;-&#39;</span> <span class="ruby-operator">+</span> <span class="ruby-ivar">@models</span>[<span class="ruby-value">1</span>].<span class="ruby-identifier">modelnum</span>.<span class="ruby-identifier">to_s</span>
<span class="line-num">259</span>   <span class="ruby-keyword">end</span>
<span class="line-num">260</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

      <div id="method-i-nb_sites" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">nb_sites</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>Return a <a href="PositiveSites.html"><code>PositiveSites</code></a> (naive empirical bayesian) object</p>

          <div class="method-source-code" id="nb_sites-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">273</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">nb_sites</span>
<span class="line-num">274</span>   <span class="ruby-constant">PositiveSites</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&quot;Naive Empirical Bayes (NEB)&quot;</span>,<span class="ruby-ivar">@footer</span>,<span class="ruby-identifier">num_codons</span>)
<span class="line-num">275</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

      <div id="method-i-num_codons" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">num_codons</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>Return the number of condons in the codeml alignment</p>

          <div class="method-source-code" id="num_codons-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">263</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">num_codons</span>
<span class="line-num">264</span>   <span class="ruby-ivar">@header</span>.<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">/seed used = \d+\n\s+\d+\s+\d+/</span>).<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">split</span>[<span class="ruby-value">5</span>].<span class="ruby-identifier">to_i</span><span class="ruby-operator">/</span><span class="ruby-value">3</span>
<span class="line-num">265</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

      <div id="method-i-num_sequences" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">num_sequences</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>Return the number of sequences in the codeml alignment</p>

          <div class="method-source-code" id="num_sequences-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">268</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">num_sequences</span>
<span class="line-num">269</span>   <span class="ruby-ivar">@header</span>.<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">/seed used = \d+\n\s+\d+\s+\d+/</span>).<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">split</span>[<span class="ruby-value">4</span>].<span class="ruby-identifier">to_i</span>
<span class="line-num">270</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

      <div id="method-i-significant" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">significant</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>If the number of models is two we can calculate whether the result is statistically significant, or not, at the 1% significance level. For example, for M7-8 the LRT statistic, or twice the log likelihood difference between the two compared models, may be compared against chi-square, with critical value 9.21 at the 1% significance level.</p>

<p>Here we support a few likely combinations, M0-3, M1-2 and M7-8, used most often in literature. For other combinations, or a different  significance level, you&#39;ll have to calculate chi-square yourself.</p>

<p>Returns true or false. If no result is calculated this method raises an error</p>

          <div class="method-source-code" id="significant-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">294</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">significant</span>
<span class="line-num">295</span>   <span class="ruby-identifier">raise</span> <span class="ruby-constant">ReportError</span>,<span class="ruby-node">&quot;Wrong number of models #{@models.size}&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">!=</span> <span class="ruby-value">2</span>
<span class="line-num">296</span>   <span class="ruby-identifier">lnL1</span> = <span class="ruby-ivar">@models</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">lnL</span>
<span class="line-num">297</span>   <span class="ruby-identifier">model1</span> = <span class="ruby-ivar">@models</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">modelnum</span>
<span class="line-num">298</span>   <span class="ruby-identifier">lnL2</span> = <span class="ruby-ivar">@models</span>[<span class="ruby-value">1</span>].<span class="ruby-identifier">lnL</span>
<span class="line-num">299</span>   <span class="ruby-identifier">model2</span> = <span class="ruby-ivar">@models</span>[<span class="ruby-value">1</span>].<span class="ruby-identifier">modelnum</span>
<span class="line-num">300</span>   <span class="ruby-keyword">case</span> [<span class="ruby-identifier">model1</span>, <span class="ruby-identifier">model2</span>]
<span class="line-num">301</span>     <span class="ruby-keyword">when</span> [<span class="ruby-value">0</span>,<span class="ruby-value">3</span>]
<span class="line-num">302</span>       <span class="ruby-value">2</span><span class="ruby-operator">*</span>(<span class="ruby-identifier">lnL2</span><span class="ruby-operator">-</span><span class="ruby-identifier">lnL1</span>) <span class="ruby-operator">&gt;</span> <span class="ruby-value">13.2767</span>   <span class="ruby-comment"># chi2: p=0.01, df=4</span>
<span class="line-num">303</span>     <span class="ruby-keyword">when</span> [<span class="ruby-value">1</span>,<span class="ruby-value">2</span>]
<span class="line-num">304</span>       <span class="ruby-value">2</span><span class="ruby-operator">*</span>(<span class="ruby-identifier">lnL2</span><span class="ruby-operator">-</span><span class="ruby-identifier">lnL1</span>) <span class="ruby-operator">&gt;</span>  <span class="ruby-value">9.2103</span>   <span class="ruby-comment"># chi2: p=0.01, df=2</span>
<span class="line-num">305</span>     <span class="ruby-keyword">when</span> [<span class="ruby-value">7</span>,<span class="ruby-value">8</span>]
<span class="line-num">306</span>       <span class="ruby-value">2</span><span class="ruby-operator">*</span>(<span class="ruby-identifier">lnL2</span><span class="ruby-operator">-</span><span class="ruby-identifier">lnL1</span>) <span class="ruby-operator">&gt;</span>  <span class="ruby-value">9.2103</span>   <span class="ruby-comment"># chi2: p=0.01, df=2</span>
<span class="line-num">307</span>     <span class="ruby-keyword">else</span>
<span class="line-num">308</span>       <span class="ruby-identifier">raise</span> <span class="ruby-constant">ReportError</span>,<span class="ruby-node">&quot;Significance calculation for #{descr} not supported&quot;</span>
<span class="line-num">309</span>   <span class="ruby-keyword">end</span>
<span class="line-num">310</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

      <div id="method-i-sites" class="method-detail ">
        <div class="method-heading">
          <span class="method-name">sites</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>

        <div class="method-description">
          <p>Return a <a href="PositiveSites.html"><code>PositiveSites</code></a> Bayes Empirical Bayes (BEB) analysis</p>

          <div class="method-source-code" id="sites-source">
            <pre>    <span class="ruby-comment"># File lib/bio/appl/paml/codeml/report.rb</span>
<span class="line-num">278</span> <span class="ruby-keyword">def</span> <span class="ruby-identifier ruby-title">sites</span>
<span class="line-num">279</span>   <span class="ruby-constant">PositiveSites</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&quot;Bayes Empirical Bayes (BEB)&quot;</span>,<span class="ruby-ivar">@footer</span>,<span class="ruby-identifier">num_codons</span>)
<span class="line-num">280</span> <span class="ruby-keyword">end</span></pre>
          </div>
        </div>


      </div>

    </section>

  </section>
</main>


<footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.3.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

