<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />

  <title>Class: Bio::FlatFile</title>

  <link rel="stylesheet" href="../rdoc.css" type="text/css" media="screen" />

  <script src="../js/jquery.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/thickbox-compressed.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/quicksearch.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/darkfish.js" type="text/javascript" charset="utf-8"></script>

</head>
<body id="top" class="class">

  <div id="metadata">
    <div id="home-metadata">
      <div id="home-section" class="section">
        <h3 class="section-header">
          <a href="../index.html">Home</a>
          <a href="../index.html#classes">Classes</a>
          <a href="../index.html#methods">Methods</a>
        </h3>
      </div>
    </div>

    <div id="file-metadata">
      <div id="file-list-section" class="section">
        <h3 class="section-header">In Files</h3>
        <div class="section-body">
          <ul>
          
            <li><a href="../lib/bio/io/flatfile_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile.rb">lib/bio/io/flatfile.rb</a></li>
          
            <li><a href="../lib/bio/io/flatfile/autodetection_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/autodetection.rb">lib/bio/io/flatfile/autodetection.rb</a></li>
          
            <li><a href="../lib/bio/io/flatfile/buffer_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/buffer.rb">lib/bio/io/flatfile/buffer.rb</a></li>
          
            <li><a href="../lib/bio/io/flatfile/splitter_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/io/flatfile/splitter.rb">lib/bio/io/flatfile/splitter.rb</a></li>
          
          </ul>
        </div>
      </div>

      
    </div>

    <div id="class-metadata">
      
      <!-- Parent Class -->
      <div id="parent-class-section" class="section">
        <h3 class="section-header">Parent</h3>
        
        <p class="link"><a href="../Object.html">Object</a></p>
        
      </div>
      

      

      
      <!-- Namespace Contents -->
      <div id="namespace-list-section" class="section">
        <h3 class="section-header">Namespace</h3>
        <ul class="link-list">
          
          <li><span class="type">MODULE</span> <a href="FlatFile/Splitter.html">Bio::FlatFile::Splitter</a></li>
          
          <li><span class="type">CLASS</span> <a href="FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a></li>
          
          <li><span class="type">CLASS</span> <a href="FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a></li>
          
          <li><span class="type">CLASS</span> <a href="FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a></li>
          
        </ul>
      </div>
      

      
      <!-- Method Quickref -->
      <div id="method-list-section" class="section">
        <h3 class="section-header">Methods</h3>
        <ul class="link-list">
          
          <li><a href="#method-c-auto">::auto</a></li>
          
          <li><a href="#method-c-autodetect">::autodetect</a></li>
          
          <li><a href="#method-c-autodetect_file">::autodetect_file</a></li>
          
          <li><a href="#method-c-autodetect_io">::autodetect_io</a></li>
          
          <li><a href="#method-c-autodetect_stream">::autodetect_stream</a></li>
          
          <li><a href="#method-c-foreach">::foreach</a></li>
          
          <li><a href="#method-c-new">::new</a></li>
          
          <li><a href="#method-c-open">::open</a></li>
          
          <li><a href="#method-c-open_file">::open_file</a></li>
          
          <li><a href="#method-c-open_uri">::open_uri</a></li>
          
          <li><a href="#method-c-to_a">::to_a</a></li>
          
          <li><a href="#method-i-autodetect">#autodetect</a></li>
          
          <li><a href="#method-i-close">#close</a></li>
          
          <li><a href="#method-i-dbclass-3D">#dbclass=</a></li>
          
          <li><a href="#method-i-each">#each</a></li>
          
          <li><a href="#method-i-each_entry">#each_entry</a></li>
          
          <li><a href="#method-i-entry_ended_pos">#entry_ended_pos</a></li>
          
          <li><a href="#method-i-entry_pos_flag">#entry_pos_flag</a></li>
          
          <li><a href="#method-i-entry_pos_flag-3D">#entry_pos_flag=</a></li>
          
          <li><a href="#method-i-entry_raw">#entry_raw</a></li>
          
          <li><a href="#method-i-entry_start_pos">#entry_start_pos</a></li>
          
          <li><a href="#method-i-eof-3F">#eof?</a></li>
          
          <li><a href="#method-i-gets">#gets</a></li>
          
          <li><a href="#method-i-io">#io</a></li>
          
          <li><a href="#method-i-next_entry">#next_entry</a></li>
          
          <li><a href="#method-i-path">#path</a></li>
          
          <li><a href="#method-i-pos">#pos</a></li>
          
          <li><a href="#method-i-pos-3D">#pos=</a></li>
          
          <li><a href="#method-i-raw-3D">#raw=</a></li>
          
          <li><a href="#method-i-rewind">#rewind</a></li>
          
          <li><a href="#method-i-to_io">#to_io</a></li>
          
        </ul>
      </div>
      

      
      <!-- Included Modules -->
      <div id="includes-section" class="section">
        <h3 class="section-header">Included Modules</h3>
        <ul class="link-list">
        
        
          <li><span class="include">Enumerable</span></li>
        
        
        </ul>
      </div>
      
    </div>

    <div id="project-metadata">
      
      
      <div id="fileindex-section" class="section project-section">
        <h3 class="section-header">Files</h3>
        <ul>
        
          <li class="file"><a href="../ChangeLog.html">ChangeLog</a></li>
        
          <li class="file"><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a></li>
        
          <li class="file"><a href="../README_rdoc.html">README.rdoc</a></li>
        
          <li class="file"><a href="../README_DEV_rdoc.html">README_DEV.rdoc</a></li>
        
          <li class="file"><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a></li>
        
          <li class="file"><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3.rdoc</a></li>
        
          <li class="file"><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0.rdoc</a></li>
        
          <li class="file"><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1.rdoc</a></li>
        
        </ul>
      </div>
      

      <div id="classindex-section" class="section project-section">
        <h3 class="section-header">Class/Module Index
          <span class="search-toggle"><img src="../images/find.png"
            height="16" width="16" alt="[+]"
            title="show/hide quicksearch" /></span></h3>
        <form action="#" method="get" accept-charset="utf-8" class="initially-hidden">
        <fieldset>
          <legend>Quicksearch</legend>
          <input type="text" name="quicksearch" value=""
            class="quicksearch-field" />
        </fieldset>
        </form>

        <ul class="link-list">
        
          <li><a href="../Bio.html">Bio</a></li>
        
          <li><a href="../Bio/AAindex.html">Bio::AAindex</a></li>
        
          <li><a href="../Bio/AAindex1.html">Bio::AAindex1</a></li>
        
          <li><a href="../Bio/AAindex2.html">Bio::AAindex2</a></li>
        
          <li><a href="../Bio/Abif.html">Bio::Abif</a></li>
        
          <li><a href="../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a></li>
        
          <li><a href="../Bio/Alignment.html">Bio::Alignment</a></li>
        
          <li><a href="../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a></li>
        
          <li><a href="../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a></li>
        
          <li><a href="../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a></li>
        
          <li><a href="../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a></li>
        
          <li><a href="../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a></li>
        
          <li><a href="../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a></li>
        
          <li><a href="../Bio/Alignment/Output.html">Bio::Alignment::Output</a></li>
        
          <li><a href="../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a></li>
        
          <li><a href="../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a></li>
        
          <li><a href="../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a></li>
        
          <li><a href="../Bio/Alignment/Site.html">Bio::Alignment::Site</a></li>
        
          <li><a href="../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a></li>
        
          <li><a href="../Bio/AminoAcid.html">Bio::AminoAcid</a></li>
        
          <li><a href="../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a></li>
        
          <li><a href="../Bio/Blast.html">Bio::Blast</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Default.html">Bio::Blast::Default</a></li>
        
          <li><a href="../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a></li>
        
          <li><a href="../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a></li>
        
          <li><a href="../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Remote.html">Bio::Blast::Remote</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a></li>
        
          <li><a href="../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a></li>
        
          <li><a href="../Bio/Blast/Report.html">Bio::Blast::Report</a></li>
        
          <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a></li>
        
          <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a></li>
        
          <li><a href="../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a></li>
        
          <li><a href="../Bio/Blast/WU.html">Bio::Blast::WU</a></li>
        
          <li><a href="../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a></li>
        
          <li><a href="../Bio/Blat.html">Bio::Blat</a></li>
        
          <li><a href="../Bio/Blat/Report.html">Bio::Blat::Report</a></li>
        
          <li><a href="../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a></li>
        
          <li><a href="../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/ClustalW.html">Bio::ClustalW</a></li>
        
          <li><a href="../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a></li>
        
          <li><a href="../Bio/CodonTable.html">Bio::CodonTable</a></li>
        
          <li><a href="../Bio/ColorScheme.html">Bio::ColorScheme</a></li>
        
          <li><a href="../Bio/Command.html">Bio::Command</a></li>
        
          <li><a href="../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a></li>
        
          <li><a href="../Bio/ContingencyTable.html">Bio::ContingencyTable</a></li>
        
          <li><a href="../Bio/DAS.html">Bio::DAS</a></li>
        
          <li><a href="../Bio/DAS/DNA.html">Bio::DAS::DNA</a></li>
        
          <li><a href="../Bio/DAS/DSN.html">Bio::DAS::DSN</a></li>
        
          <li><a href="../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a></li>
        
          <li><a href="../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a></li>
        
          <li><a href="../Bio/DAS/GFF.html">Bio::DAS::GFF</a></li>
        
          <li><a href="../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a></li>
        
          <li><a href="../Bio/DAS/LINK.html">Bio::DAS::LINK</a></li>
        
          <li><a href="../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a></li>
        
          <li><a href="../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a></li>
        
          <li><a href="../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a></li>
        
          <li><a href="../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a></li>
        
          <li><a href="../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a></li>
        
          <li><a href="../Bio/DB.html">Bio::DB</a></li>
        
          <li><a href="../Bio/DBGET.html">Bio::DBGET</a></li>
        
          <li><a href="../Bio/DDBJ.html">Bio::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a></li>
        
          <li><a href="../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a></li>
        
          <li><a href="../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a></li>
        
          <li><a href="../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a></li>
        
          <li><a href="../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a></li>
        
          <li><a href="../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a></li>
        
          <li><a href="../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a></li>
        
          <li><a href="../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a></li>
        
          <li><a href="../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a></li>
        
          <li><a href="../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a></li>
        
          <li><a href="../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a></li>
        
          <li><a href="../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a></li>
        
          <li><a href="../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a></li>
        
          <li><a href="../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a></li>
        
          <li><a href="../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a></li>
        
          <li><a href="../Bio/EBI.html">Bio::EBI</a></li>
        
          <li><a href="../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a></li>
        
          <li><a href="../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a></li>
        
          <li><a href="../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a></li>
        
          <li><a href="../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a></li>
        
          <li><a href="../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a></li>
        
          <li><a href="../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a></li>
        
          <li><a href="../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a></li>
        
          <li><a href="../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a></li>
        
          <li><a href="../Bio/EMBL.html">Bio::EMBL</a></li>
        
          <li><a href="../Bio/EMBLDB.html">Bio::EMBLDB</a></li>
        
          <li><a href="../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a></li>
        
          <li><a href="../Bio/EMBOSS.html">Bio::EMBOSS</a></li>
        
          <li><a href="../Bio/Ensembl.html">Bio::Ensembl</a></li>
        
          <li><a href="../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a></li>
        
          <li><a href="../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a></li>
        
          <li><a href="../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a></li>
        
          <li><a href="../Bio/FANTOM.html">Bio::FANTOM</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a></li>
        
          <li><a href="../Bio/Fasta.html">Bio::Fasta</a></li>
        
          <li><a href="../Bio/Fasta/Report.html">Bio::Fasta::Report</a></li>
        
          <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a></li>
        
          <li><a href="../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a></li>
        
          <li><a href="../Bio/FastaDefline.html">Bio::FastaDefline</a></li>
        
          <li><a href="../Bio/FastaFormat.html">Bio::FastaFormat</a></li>
        
          <li><a href="../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a></li>
        
          <li><a href="../Bio/Fastq.html">Bio::Fastq</a></li>
        
          <li><a href="../Bio/Fastq/Error.html">Bio::Fastq::Error</a></li>
        
          <li><a href="../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a></li>
        
          <li><a href="../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a></li>
        
          <li><a href="../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a></li>
        
          <li><a href="../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a></li>
        
          <li><a href="../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a></li>
        
          <li><a href="../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a></li>
        
          <li><a href="../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a></li>
        
          <li><a href="../Bio/Feature.html">Bio::Feature</a></li>
        
          <li><a href="../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a></li>
        
          <li><a href="../Bio/Features.html">Bio::Features</a></li>
        
          <li><a href="../Bio/Fetch.html">Bio::Fetch</a></li>
        
          <li><a href="../Bio/FlatFile.html">Bio::FlatFile</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a></li>
        
          <li><a href="../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a></li>
        
          <li><a href="../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a></li>
        
          <li><a href="../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a></li>
        
          <li><a href="../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a></li>
        
          <li><a href="../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a></li>
        
          <li><a href="../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a></li>
        
          <li><a href="../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a></li>
        
          <li><a href="../Bio/GCG.html">Bio::GCG</a></li>
        
          <li><a href="../Bio/GCG/Msf.html">Bio::GCG::Msf</a></li>
        
          <li><a href="../Bio/GCG/Seq.html">Bio::GCG::Seq</a></li>
        
          <li><a href="../Bio/GFF.html">Bio::GFF</a></li>
        
          <li><a href="../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a></li>
        
          <li><a href="../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a></li>
        
          <li><a href="../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a></li>
        
          <li><a href="../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a></li>
        
          <li><a href="../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a></li>
        
          <li><a href="../Bio/GFF/Record.html">Bio::GFF::Record</a></li>
        
          <li><a href="../Bio/GO.html">Bio::GO</a></li>
        
          <li><a href="../Bio/GO/External2go.html">Bio::GO::External2go</a></li>
        
          <li><a href="../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a></li>
        
          <li><a href="../Bio/GO/Ontology.html">Bio::GO::Ontology</a></li>
        
          <li><a href="../Bio/GenBank.html">Bio::GenBank</a></li>
        
          <li><a href="../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a></li>
        
          <li><a href="../Bio/GenPept.html">Bio::GenPept</a></li>
        
          <li><a href="../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a></li>
        
          <li><a href="../Bio/Genscan.html">Bio::Genscan</a></li>
        
          <li><a href="../Bio/Genscan/Report.html">Bio::Genscan::Report</a></li>
        
          <li><a href="../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a></li>
        
          <li><a href="../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a></li>
        
          <li><a href="../Bio/HGC.html">Bio::HGC</a></li>
        
          <li><a href="../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a></li>
        
          <li><a href="../Bio/HMMER.html">Bio::HMMER</a></li>
        
          <li><a href="../Bio/HMMER/Report.html">Bio::HMMER::Report</a></li>
        
          <li><a href="../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a></li>
        
          <li><a href="../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a></li>
        
          <li><a href="../Bio/Hinv.html">Bio::Hinv</a></li>
        
          <li><a href="../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a></li>
        
          <li><a href="../Bio/Hinv/Common.html">Bio::Hinv::Common</a></li>
        
          <li><a href="../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a></li>
        
          <li><a href="../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a></li>
        
          <li><a href="../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a></li>
        
          <li><a href="../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a></li>
        
          <li><a href="../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a></li>
        
          <li><a href="../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a></li>
        
          <li><a href="../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a></li>
        
          <li><a href="../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a></li>
        
          <li><a href="../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a></li>
        
          <li><a href="../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a></li>
        
          <li><a href="../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a></li>
        
          <li><a href="../Bio/Iprscan.html">Bio::Iprscan</a></li>
        
          <li><a href="../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a></li>
        
          <li><a href="../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a></li>
        
          <li><a href="../Bio/KEGG.html">Bio::KEGG</a></li>
        
          <li><a href="../Bio/KEGG/API.html">Bio::KEGG::API</a></li>
        
          <li><a href="../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a></li>
        
          <li><a href="../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a></li>
        
          <li><a href="../Bio/KEGG/Common.html">Bio::KEGG::Common</a></li>
        
          <li><a href="../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a></li>
        
          <li><a href="../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a></li>
        
          <li><a href="../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a></li>
        
          <li><a href="../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a></li>
        
          <li><a href="../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a></li>
        
          <li><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a></li>
        
          <li><a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a></li>
        
          <li><a href="../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a></li>
        
          <li><a href="../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a></li>
        
          <li><a href="../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a></li>
        
          <li><a href="../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a></li>
        
          <li><a href="../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a></li>
        
          <li><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a></li>
        
          <li><a href="../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a></li>
        
          <li><a href="../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a></li>
        
          <li><a href="../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a></li>
        
          <li><a href="../Bio/KEGGDB.html">Bio::KEGGDB</a></li>
        
          <li><a href="../Bio/LITDB.html">Bio::LITDB</a></li>
        
          <li><a href="../Bio/Lasergene.html">Bio::Lasergene</a></li>
        
          <li><a href="../Bio/Location.html">Bio::Location</a></li>
        
          <li><a href="../Bio/Locations.html">Bio::Locations</a></li>
        
          <li><a href="../Bio/MAFFT.html">Bio::MAFFT</a></li>
        
          <li><a href="../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a></li>
        
          <li><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a></li>
        
          <li><a href="../Bio/Map.html">Bio::Map</a></li>
        
          <li><a href="../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a></li>
        
          <li><a href="../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a></li>
        
          <li><a href="../Bio/Map/Mapping.html">Bio::Map::Mapping</a></li>
        
          <li><a href="../Bio/Map/Marker.html">Bio::Map::Marker</a></li>
        
          <li><a href="../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a></li>
        
          <li><a href="../Bio/Meme.html">Bio::Meme</a></li>
        
          <li><a href="../Bio/Meme/Mast.html">Bio::Meme::Mast</a></li>
        
          <li><a href="../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a></li>
        
          <li><a href="../Bio/Meme/Motif.html">Bio::Meme::Motif</a></li>
        
          <li><a href="../Bio/Muscle.html">Bio::Muscle</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/NBRF.html">Bio::NBRF</a></li>
        
          <li><a href="../Bio/NCBI.html">Bio::NCBI</a></li>
        
          <li><a href="../Bio/NCBI/REST.html">Bio::NCBI::REST</a></li>
        
          <li><a href="../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a></li>
        
          <li><a href="../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a></li>
        
          <li><a href="../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a></li>
        
          <li><a href="../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a></li>
        
          <li><a href="../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a></li>
        
          <li><a href="../Bio/NCBIDB.html">Bio::NCBIDB</a></li>
        
          <li><a href="../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a></li>
        
          <li><a href="../Bio/Newick.html">Bio::Newick</a></li>
        
          <li><a href="../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a></li>
        
          <li><a href="../Bio/Nexus.html">Bio::Nexus</a></li>
        
          <li><a href="../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a></li>
        
          <li><a href="../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a></li>
        
          <li><a href="../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a></li>
        
          <li><a href="../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a></li>
        
          <li><a href="../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a></li>
        
          <li><a href="../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a></li>
        
          <li><a href="../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a></li>
        
          <li><a href="../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a></li>
        
          <li><a href="../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a></li>
        
          <li><a href="../Bio/Nexus/Util.html">Bio::Nexus::Util</a></li>
        
          <li><a href="../Bio/NucleicAcid.html">Bio::NucleicAcid</a></li>
        
          <li><a href="../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a></li>
        
          <li><a href="../Bio/PAML.html">Bio::PAML</a></li>
        
          <li><a href="../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a></li>
        
          <li><a href="../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a></li>
        
          <li><a href="../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a></li>
        
          <li><a href="../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a></li>
        
          <li><a href="../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a></li>
        
          <li><a href="../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a></li>
        
          <li><a href="../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a></li>
        
          <li><a href="../Bio/PAML/Common.html">Bio::PAML::Common</a></li>
        
          <li><a href="../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a></li>
        
          <li><a href="../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a></li>
        
          <li><a href="../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a></li>
        
          <li><a href="../Bio/PDB.html">Bio::PDB</a></li>
        
          <li><a href="../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a></li>
        
          <li><a href="../Bio/PDB/Chain.html">Bio::PDB::Chain</a></li>
        
          <li><a href="../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a></li>
        
          <li><a href="../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a></li>
        
          <li><a href="../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a></li>
        
          <li><a href="../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a></li>
        
          <li><a href="../Bio/PDB/DataType.html">Bio::PDB::DataType</a></li>
        
          <li><a href="../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a></li>
        
          <li><a href="../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a></li>
        
          <li><a href="../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a></li>
        
          <li><a href="../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a></li>
        
          <li><a href="../Bio/PDB/Model.html">Bio::PDB::Model</a></li>
        
          <li><a href="../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a></li>
        
          <li><a href="../Bio/PDB/Record.html">Bio::PDB::Record</a></li>
        
          <li><a href="../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a></li>
        
          <li><a href="../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a></li>
        
          <li><a href="../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a></li>
        
          <li><a href="../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a></li>
        
          <li><a href="../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a></li>
        
          <li><a href="../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a></li>
        
          <li><a href="../Bio/PDB/Residue.html">Bio::PDB::Residue</a></li>
        
          <li><a href="../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a></li>
        
          <li><a href="../Bio/PDB/Utils.html">Bio::PDB::Utils</a></li>
        
          <li><a href="../Bio/PROSITE.html">Bio::PROSITE</a></li>
        
          <li><a href="../Bio/PSORT.html">Bio::PSORT</a></li>
        
          <li><a href="../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a></li>
        
          <li><a href="../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a></li>
        
          <li><a href="../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a></li>
        
          <li><a href="../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a></li>
        
          <li><a href="../Bio/PTS1.html">Bio::PTS1</a></li>
        
          <li><a href="../Bio/PTS1/Report.html">Bio::PTS1::Report</a></li>
        
          <li><a href="../Bio/Pathway.html">Bio::Pathway</a></li>
        
          <li><a href="../Bio/Phylip.html">Bio::Phylip</a></li>
        
          <li><a href="../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a></li>
        
          <li><a href="../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></li>
        
          <li><a href="../Bio/PhyloXML.html">Bio::PhyloXML</a></li>
        
          <li><a href="../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a></li>
        
          <li><a href="../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a></li>
        
          <li><a href="../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a></li>
        
          <li><a href="../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a></li>
        
          <li><a href="../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a></li>
        
          <li><a href="../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a></li>
        
          <li><a href="../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a></li>
        
          <li><a href="../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a></li>
        
          <li><a href="../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a></li>
        
          <li><a href="../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a></li>
        
          <li><a href="../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a></li>
        
          <li><a href="../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a></li>
        
          <li><a href="../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a></li>
        
          <li><a href="../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a></li>
        
          <li><a href="../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a></li>
        
          <li><a href="../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a></li>
        
          <li><a href="../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a></li>
        
          <li><a href="../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a></li>
        
          <li><a href="../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a></li>
        
          <li><a href="../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a></li>
        
          <li><a href="../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a></li>
        
          <li><a href="../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a></li>
        
          <li><a href="../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a></li>
        
          <li><a href="../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a></li>
        
          <li><a href="../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a></li>
        
          <li><a href="../Bio/Probcons.html">Bio::Probcons</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/PubMed.html">Bio::PubMed</a></li>
        
          <li><a href="../Bio/REBASE.html">Bio::REBASE</a></li>
        
          <li><a href="../Bio/RefSeq.html">Bio::RefSeq</a></li>
        
          <li><a href="../Bio/Reference.html">Bio::Reference</a></li>
        
          <li><a href="../Bio/References.html">Bio::References</a></li>
        
          <li><a href="../Bio/Registry.html">Bio::Registry</a></li>
        
          <li><a href="../Bio/Registry/DB.html">Bio::Registry::DB</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a></li>
        
          <li><a href="../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a></li>
        
          <li><a href="../Bio/SOFT.html">Bio::SOFT</a></li>
        
          <li><a href="../Bio/SOSUI.html">Bio::SOSUI</a></li>
        
          <li><a href="../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a></li>
        
          <li><a href="../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a></li>
        
          <li><a href="../Bio/SPTR.html">Bio::SPTR</a></li>
        
          <li><a href="../Bio/SQL.html">Bio::SQL</a></li>
        
          <li><a href="../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a></li>
        
          <li><a href="../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a></li>
        
          <li><a href="../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a></li>
        
          <li><a href="../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a></li>
        
          <li><a href="../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a></li>
        
          <li><a href="../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a></li>
        
          <li><a href="../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a></li>
        
          <li><a href="../Bio/SQL/Comment.html">Bio::SQL::Comment</a></li>
        
          <li><a href="../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a></li>
        
          <li><a href="../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a></li>
        
          <li><a href="../Bio/SQL/Location.html">Bio::SQL::Location</a></li>
        
          <li><a href="../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a></li>
        
          <li><a href="../Bio/SQL/Reference.html">Bio::SQL::Reference</a></li>
        
          <li><a href="../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a></li>
        
          <li><a href="../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a></li>
        
          <li><a href="../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a></li>
        
          <li><a href="../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a></li>
        
          <li><a href="../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a></li>
        
          <li><a href="../Bio/SQL/Term.html">Bio::SQL::Term</a></li>
        
          <li><a href="../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a></li>
        
          <li><a href="../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a></li>
        
          <li><a href="../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a></li>
        
          <li><a href="../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a></li>
        
          <li><a href="../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a></li>
        
          <li><a href="../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a></li>
        
          <li><a href="../Bio/Scf.html">Bio::Scf</a></li>
        
          <li><a href="../Bio/Sequence.html">Bio::Sequence</a></li>
        
          <li><a href="../Bio/Sequence/AA.html">Bio::Sequence::AA</a></li>
        
          <li><a href="../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a></li>
        
          <li><a href="../Bio/Sequence/Common.html">Bio::Sequence::Common</a></li>
        
          <li><a href="../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a></li>
        
          <li><a href="../Bio/Sequence/Format.html">Bio::Sequence::Format</a></li>
        
          <li><a href="../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a></li>
        
          <li><a href="../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a></li>
        
          <li><a href="../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a></li>
        
          <li><a href="../Bio/Sequence/NA.html">Bio::Sequence::NA</a></li>
        
          <li><a href="../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a></li>
        
          <li><a href="../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a></li>
        
          <li><a href="../Bio/Shell.html">Bio::Shell</a></li>
        
          <li><a href="../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a></li>
        
          <li><a href="../Bio/Shell/Core.html">Bio::Shell::Core</a></li>
        
          <li><a href="../Bio/Shell/Demo.html">Bio::Shell::Demo</a></li>
        
          <li><a href="../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a></li>
        
          <li><a href="../Bio/Shell/Irb.html">Bio::Shell::Irb</a></li>
        
          <li><a href="../Bio/Shell/Private.html">Bio::Shell::Private</a></li>
        
          <li><a href="../Bio/Shell/Script.html">Bio::Shell::Script</a></li>
        
          <li><a href="../Bio/Shell/Setup.html">Bio::Shell::Setup</a></li>
        
          <li><a href="../Bio/Shell/Web.html">Bio::Shell::Web</a></li>
        
          <li><a href="../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a></li>
        
          <li><a href="../Bio/SiRNA.html">Bio::SiRNA</a></li>
        
          <li><a href="../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a></li>
        
          <li><a href="../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a></li>
        
          <li><a href="../Bio/Sim4.html">Bio::Sim4</a></li>
        
          <li><a href="../Bio/Sim4/Report.html">Bio::Sim4::Report</a></li>
        
          <li><a href="../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a></li>
        
          <li><a href="../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a></li>
        
          <li><a href="../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/Spidey.html">Bio::Spidey</a></li>
        
          <li><a href="../Bio/Spidey/Report.html">Bio::Spidey::Report</a></li>
        
          <li><a href="../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a></li>
        
          <li><a href="../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a></li>
        
          <li><a href="../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/SwissProt.html">Bio::SwissProt</a></li>
        
          <li><a href="../Bio/TMHMM.html">Bio::TMHMM</a></li>
        
          <li><a href="../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a></li>
        
          <li><a href="../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a></li>
        
          <li><a href="../Bio/TRANSFAC.html">Bio::TRANSFAC</a></li>
        
          <li><a href="../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a></li>
        
          <li><a href="../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a></li>
        
          <li><a href="../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a></li>
        
          <li><a href="../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a></li>
        
          <li><a href="../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a></li>
        
          <li><a href="../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a></li>
        
          <li><a href="../Bio/TargetP.html">Bio::TargetP</a></li>
        
          <li><a href="../Bio/TargetP/Report.html">Bio::TargetP::Report</a></li>
        
          <li><a href="../Bio/Taxonomy.html">Bio::Taxonomy</a></li>
        
          <li><a href="../Bio/Tcoffee.html">Bio::Tcoffee</a></li>
        
          <li><a href="../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/TogoWS.html">Bio::TogoWS</a></li>
        
          <li><a href="../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a></li>
        
          <li><a href="../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a></li>
        
          <li><a href="../Bio/TrEMBL.html">Bio::TrEMBL</a></li>
        
          <li><a href="../Bio/Tree.html">Bio::Tree</a></li>
        
          <li><a href="../Bio/Tree/Edge.html">Bio::Tree::Edge</a></li>
        
          <li><a href="../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a></li>
        
          <li><a href="../Bio/Tree/Node.html">Bio::Tree::Node</a></li>
        
          <li><a href="../Bio/UniProt.html">Bio::UniProt</a></li>
        
          <li><a href="../BiorubyController.html">BiorubyController</a></li>
        
          <li><a href="../BiorubyGenerator.html">BiorubyGenerator</a></li>
        
          <li><a href="../BiorubyHelper.html">BiorubyHelper</a></li>
        
          <li><a href="../Hash.html">Hash</a></li>
        
          <li><a href="../Object.html">Object</a></li>
        
          <li><a href="../String.html">String</a></li>
        
        </ul>
        <div id="no-class-search-results" style="display: none;">No matching classes.</div>
      </div>

      
    </div>
  </div>

  <div id="documentation">
    <h1 class="class">Bio::FlatFile</h1>

    <div id="description" class="description">
      
<p><a href="FlatFile.html">Bio::FlatFile</a> is a helper and wrapper class to
read a biological data file. It acts like a IO object. It can automatically
detect data format, and users do not need to tell the class what the data
is.</p>

    </div><!-- description -->

    
    
    
    <div id="5Buntitled-5D" class="documentation-section">
      

      

      

      
      <!-- Attributes -->
      <div id="attribute-method-details" class="method-section section">
        <h3 class="section-header">Attributes</h3>

        
        <div id="dbclass-attribute-method" class="method-detail">
          <a name="dbclass"></a>
          
          <div class="method-heading attribute-method-heading">
            <span class="method-name">dbclass</span><span
              class="attribute-access-type">[R]</span>
          </div>

          <div class="method-description">
          
          <p>Returns database class which is automatically detected or given in
FlatFile#initialize.</p>
          
          </div>
        </div>
        
        <div id="entry-attribute-method" class="method-detail">
          <a name="entry"></a>
          
          <div class="method-heading attribute-method-heading">
            <span class="method-name">entry</span><span
              class="attribute-access-type">[R]</span>
          </div>

          <div class="method-description">
          
          
          
          </div>
        </div>
        
        <div id="raw-attribute-method" class="method-detail">
          <a name="raw"></a>
          
          <div class="method-heading attribute-method-heading">
            <span class="method-name">raw</span><span
              class="attribute-access-type">[R]</span>
          </div>

          <div class="method-description">
          
          <p>If true, raw mode.</p>
          
          </div>
        </div>
        
        <div id="skip_leader_mode-attribute-method" class="method-detail">
          <a name="skip_leader_mode"></a>
          
          <a name="skip_leader_mode="></a>
          
          <div class="method-heading attribute-method-heading">
            <span class="method-name">skip_leader_mode</span><span
              class="attribute-access-type">[RW]</span>
          </div>

          <div class="method-description">
          
          <p>The mode how to skip leader of the data.</p>
<table class="rdoc-list"><tr><td class="rdoc-term"><p>:firsttime </p></td>
<td>
<p>(DEFAULT) only head of file (= first time to read)</p>
</td></tr><tr><td class="rdoc-term"><p>:everytime </p></td>
<td>
<p>everytime to read entry</p>
</td></tr><tr><td class="rdoc-term"><p>nil </p></td>
<td>
<p>never skip</p>
</td></tr></table>
          
          </div>
        </div>
        
      </div><!-- attribute-method-details -->
      

      <!-- Methods -->
      
      <div id="public-class-method-details" class="method-section section">
        <h3 class="section-header">Public Class Methods</h3>

      
        <div id="auto-method" class="method-detail ">
          <a name="method-c-auto"></a>

          
          <div class="method-heading">
            <span class="method-name">auto</span><span
              class="method-args">(*arg, &block)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Same as <a href="FlatFile.html#method-c-open">Bio::FlatFile.open</a>(nil,
filename_or_stream, mode, perm, options).</p>
<ul><li>
<p>Example 1</p>

<pre>Bio::FlatFile.auto(ARGF)</pre>
</li><li>
<p>Example 2</p>

<pre>Bio::FlatFile.auto(&quot;embl/est_hum17.dat&quot;)</pre>
</li><li>
<p>Example 3</p>

<pre>Bio::FlatFile.auto(IO.popen(&quot;gzip -dc nc1101.flat.gz&quot;))</pre>
</li></ul>
            

            
            <div class="method-source-code" id="auto-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 122</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">auto</span>(*<span class="ruby-identifier">arg</span>, &amp;<span class="ruby-identifier">block</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">open</span>(<span class="ruby-keyword">nil</span>, *<span class="ruby-identifier">arg</span>, &amp;<span class="ruby-identifier">block</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- auto-source -->
            
          </div>

          

          
        </div><!-- auto-method -->

      
        <div id="autodetect-method" class="method-detail ">
          <a name="method-c-autodetect"></a>

          
          <div class="method-heading">
            <span class="method-name">autodetect</span><span
              class="method-args">(text)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Detects database class (== file format) of given string. If fails to
determine, returns false or nil.</p>
            

            
            <div class="method-source-code" id="autodetect-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 460</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">autodetect</span>(<span class="ruby-identifier">text</span>)
  <span class="ruby-constant">AutoDetect</span>.<span class="ruby-identifier">default</span>.<span class="ruby-identifier">autodetect</span>(<span class="ruby-identifier">text</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- autodetect-source -->
            
          </div>

          

          
        </div><!-- autodetect-method -->

      
        <div id="autodetect_file-method" class="method-detail ">
          <a name="method-c-autodetect_file"></a>

          
          <div class="method-heading">
            <span class="method-name">autodetect_file</span><span
              class="method-args">(filename)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Detects database class (== file format) of given file. If fails to
determine, returns nil.</p>
            

            
            <div class="method-source-code" id="autodetect_file-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 440</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">autodetect_file</span>(<span class="ruby-identifier">filename</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">open_file</span>(<span class="ruby-identifier">filename</span>).<span class="ruby-identifier">dbclass</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- autodetect_file-source -->
            
          </div>

          

          
        </div><!-- autodetect_file-method -->

      
        <div id="autodetect_io-method" class="method-detail ">
          <a name="method-c-autodetect_io"></a>

          
          <div class="method-heading">
            <span class="method-name">autodetect_io</span><span
              class="method-args">(io)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Detects database class (== file format) of given input stream. If fails to
determine, returns nil. Caution: the method reads some data from the input
stream, and the data will be lost.</p>
            

            
            <div class="method-source-code" id="autodetect_io-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 448</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">autodetect_io</span>(<span class="ruby-identifier">io</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">nil</span>, <span class="ruby-identifier">io</span>).<span class="ruby-identifier">dbclass</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- autodetect_io-source -->
            
          </div>

          

          
        </div><!-- autodetect_io-method -->

      
        <div id="autodetect_stream-method" class="method-detail ">
          <a name="method-c-autodetect_stream"></a>

          
          <div class="method-heading">
            <span class="method-name">autodetect_stream</span><span
              class="method-args">(io)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>This is OBSOLETED. Please use <a
href="FlatFile.html#method-c-autodetect_io">autodetect_io</a>(io) instead.</p>
            

            
            <div class="method-source-code" id="autodetect_stream-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 453</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">autodetect_stream</span>(<span class="ruby-identifier">io</span>)
  <span class="ruby-identifier">$stderr</span>.<span class="ruby-identifier">print</span> <span class="ruby-string">&quot;Bio::FlatFile.autodetect_stream will be deprecated.&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$VERBOSE</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">autodetect_io</span>(<span class="ruby-identifier">io</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- autodetect_stream-source -->
            
          </div>

          

          
        </div><!-- autodetect_stream-method -->

      
        <div id="foreach-method" class="method-detail ">
          <a name="method-c-foreach"></a>

          
          <div class="method-heading">
            <span class="method-name">foreach</span><span
              class="method-args">(*arg)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Executes the block for every entry in the stream. Same as <a
href="FlatFile.html#method-c-open">FlatFile.open(*arg)</a> { |ff| ff.each {
|entry| … }}.</p>
<ul><li>
<p>Example</p>

<pre>Bio::FlatFile.foreach('test.fst') { |e| puts e.definition }</pre>
</li></ul>
            

            
            <div class="method-source-code" id="foreach-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 194</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">foreach</span>(*<span class="ruby-identifier">arg</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">open</span>(*<span class="ruby-identifier">arg</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">flatfileobj</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">flatfileobj</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">entry</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">yield</span> <span class="ruby-identifier">entry</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- foreach-source -->
            
          </div>

          

          
        </div><!-- foreach-method -->

      
        <div id="new-method" class="method-detail ">
          <a name="method-c-new"></a>

          
          <div class="method-heading">
            <span class="method-name">new</span><span
              class="method-args">(dbclass, stream)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Same as <a href="FlatFile.html#method-c-open">FlatFile.open</a>, except
that ‘stream’ should be a opened stream object (IO, File, …, who have the
‘gets’ method).</p>
<ul><li>
<p>Example 1</p>

<pre>Bio::FlatFile.new(Bio::GenBank, ARGF)</pre>
</li><li>
<p>Example 2</p>

<pre>Bio::FlatFile.new(Bio::GenBank, IO.popen(&quot;gzip -dc nc1101.flat.gz&quot;))</pre>
</li></ul>

<p>Compatibility Note: Now, you cannot specify “:raw =&gt; true” or “:raw
=&gt; false”. Below styles are DEPRECATED.</p>
<ul><li>
<p>Example 3 (deprecated)</p>

<pre># Bio::FlatFile.new(nil, $stdin, :raw=&gt;true) # =&gt; ERROR
# Please rewrite as below.
ff = Bio::FlatFile.new(nil, $stdin)
ff.raw = true</pre>
</li><li>
<p>Example 3 in old style (deprecated)</p>

<pre># Bio::FlatFile.new(nil, $stdin, true) # =&gt; ERROR
# Please rewrite as below.
ff = Bio::FlatFile.new(nil, $stdin)
ff.raw = true</pre>
</li></ul>
            

            
            <div class="method-source-code" id="new-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 225</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">dbclass</span>, <span class="ruby-identifier">stream</span>)
  <span class="ruby-comment"># 2nd arg: IO object</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">stream</span>.<span class="ruby-identifier">kind_of?</span>(<span class="ruby-constant">BufferedInputStream</span>)
    <span class="ruby-ivar">@stream</span> = <span class="ruby-identifier">stream</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@stream</span> = <span class="ruby-constant">BufferedInputStream</span>.<span class="ruby-identifier">for_io</span>(<span class="ruby-identifier">stream</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-comment"># 1st arg: database class (or file format autodetection)</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">dbclass</span> <span class="ruby-keyword">then</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">dbclass</span> = <span class="ruby-identifier">dbclass</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">autodetect</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-ivar">@skip_leader_mode</span> = <span class="ruby-value">:firsttime</span>
  <span class="ruby-ivar">@firsttime_flag</span> = <span class="ruby-keyword">true</span>
  <span class="ruby-comment"># default raw mode is false</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">raw</span> = <span class="ruby-keyword">false</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- new-source -->
            
          </div>

          

          
        </div><!-- new-method -->

      
        <div id="open-method" class="method-detail ">
          <a name="method-c-open"></a>

          
          <div class="method-heading">
            <span class="method-name">open</span><span
              class="method-args">(*arg, &block)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <pre>Bio::FlatFile.open(file, *arg)
Bio::FlatFile.open(dbclass, file, *arg)</pre>

<p>Creates a new <a href="FlatFile.html">Bio::FlatFile</a> object to read a
file or a stream which contains <em>dbclass</em> data.</p>

<p><em>dbclass</em> should be a class (or module) or nil. e.g. <a
href="GenBank.html">Bio::GenBank</a>, <a
href="FastaFormat.html">Bio::FastaFormat</a>.</p>

<p>If <em>file</em> is a filename (which doesn’t have gets method), the method
opens a local file named <em>file</em> with <tt>File.open(filename,
*arg)</tt>.</p>

<p>When <em>dbclass</em> is omitted or nil is given to <em>dbclass</em>, the
method tries to determine database class (file format) automatically. When
it fails to determine, dbclass is set to nil and <a
href="FlatFile.html#method-i-next_entry">FlatFile#next_entry</a> would
fail. You can still set dbclass using <a
href="FlatFile.html#method-i-dbclass-3D">FlatFile#dbclass=</a> method.</p>
<ul><li>
<p>Example 1</p>

<pre>Bio::FlatFile.open(Bio::GenBank, &quot;genbank/gbest40.seq&quot;)</pre>
</li><li>
<p>Example 2</p>

<pre>Bio::FlatFile.open(nil, &quot;embl/est_hum17.dat&quot;)</pre>
</li><li>
<p>Example 3</p>

<pre>Bio::FlatFile.open(&quot;genbank/gbest40.seq&quot;)</pre>
</li><li>
<p>Example 4</p>

<pre>Bio::FlatFile.open(Bio::GenBank, $stdin)</pre>
</li></ul>

<p>If it is called with a block, the block will be executed with a new <a
href="FlatFile.html">Bio::FlatFile</a> object. If filename is given, the
file is automatically closed when leaving the block.</p>
<ul><li>
<p>Example 5</p>

<pre>Bio::FlatFile.open(nil, 'test4.fst') do |ff|
    ff.each { |e| print e.definition, &quot;\n&quot; }
end</pre>
</li><li>
<p>Example 6</p>

<pre>Bio::FlatFile.open('test4.fst') do |ff|
    ff.each { |e| print e.definition, &quot;\n&quot; }
end</pre>
</li></ul>

<p>Compatibility Note: <em>*arg</em> is completely passed to the
<tt>File.open</tt> and you cannot specify “:raw =&gt; true” or “:raw =&gt;
false”.</p>
            

            
            <div class="method-source-code" id="open-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 80</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">open</span>(*<span class="ruby-identifier">arg</span>, &amp;<span class="ruby-identifier">block</span>)
  <span class="ruby-comment"># FlatFile.open(dbclass, file, mode, perm)</span>
  <span class="ruby-comment"># FlatFile.open(file, mode, perm)</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">arg</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&lt;=</span> <span class="ruby-value">0</span>
    <span class="ruby-identifier">raise</span> <span class="ruby-constant">ArgumentError</span>, <span class="ruby-string">'wrong number of arguments (0 for 1)'</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">x</span> = <span class="ruby-identifier">arg</span>.<span class="ruby-identifier">shift</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">x</span>.<span class="ruby-identifier">is_a?</span>(<span class="ruby-constant">Module</span>) <span class="ruby-keyword">then</span>
    <span class="ruby-comment"># FlatFile.open(dbclass, filename_or_io, ...)</span>
    <span class="ruby-identifier">dbclass</span> = <span class="ruby-identifier">x</span>
  <span class="ruby-keyword">elsif</span> <span class="ruby-identifier">x</span>.<span class="ruby-identifier">nil?</span> <span class="ruby-keyword">then</span>
    <span class="ruby-comment"># FlatFile.open(nil, filename_or_io, ...)</span>
    <span class="ruby-identifier">dbclass</span> = <span class="ruby-keyword">nil</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-comment"># FlatFile.open(filename, ...)</span>
    <span class="ruby-identifier">dbclass</span> = <span class="ruby-keyword">nil</span>
    <span class="ruby-identifier">arg</span>.<span class="ruby-identifier">unshift</span>(<span class="ruby-identifier">x</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">arg</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&lt;=</span> <span class="ruby-value">0</span>
    <span class="ruby-identifier">raise</span> <span class="ruby-constant">ArgumentError</span>, <span class="ruby-string">'wrong number of arguments (1 for 2)'</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">file</span> = <span class="ruby-identifier">arg</span>.<span class="ruby-identifier">shift</span>
  <span class="ruby-comment"># check if file is filename or IO object</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-identifier">file</span>.<span class="ruby-identifier">respond_to?</span>(<span class="ruby-value">:gets</span>)
    <span class="ruby-comment"># 'file' is a filename</span>
    <span class="ruby-identifier">_open_file</span>(<span class="ruby-identifier">dbclass</span>, <span class="ruby-identifier">file</span>, *<span class="ruby-identifier">arg</span>, &amp;<span class="ruby-identifier">block</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-comment"># 'file' is a IO object</span>
    <span class="ruby-identifier">ff</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">dbclass</span>, <span class="ruby-identifier">file</span>)
    <span class="ruby-identifier">block_given?</span> <span class="ruby-operator">?</span> (<span class="ruby-keyword">yield</span> <span class="ruby-identifier">ff</span>) <span class="ruby-operator">:</span> <span class="ruby-identifier">ff</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- open-source -->
            
          </div>

          

          
        </div><!-- open-method -->

      
        <div id="open_file-method" class="method-detail ">
          <a name="method-c-open_file"></a>

          
          <div class="method-heading">
            <span class="method-name">open_file</span><span
              class="method-args">(filename, *arg)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Same as <a href="FlatFile.html#method-c-auto">FlatFile.auto</a>(filename,
*arg), except that it only accept filename and doesn’t accept IO object.
File format is automatically determined.</p>

<p>It can accept a block. If a block is given, it returns the block’s return
value. Otherwise, it returns a new <a href="FlatFile.html">FlatFile</a>
object.</p>
            

            
            <div class="method-source-code" id="open_file-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 144</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">open_file</span>(<span class="ruby-identifier">filename</span>, *<span class="ruby-identifier">arg</span>)
  <span class="ruby-identifier">_open_file</span>(<span class="ruby-keyword">nil</span>, <span class="ruby-identifier">filename</span>, *<span class="ruby-identifier">arg</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- open_file-source -->
            
          </div>

          

          
        </div><!-- open_file-method -->

      
        <div id="open_uri-method" class="method-detail ">
          <a name="method-c-open_uri"></a>

          
          <div class="method-heading">
            <span class="method-name">open_uri</span><span
              class="method-args">(uri, *arg)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Opens URI specified as <em>uri</em>. <em>uri</em> must be a <a
href="../String.html">String</a> or URI object. *arg is passed to
OpenURI.open_uri or URI#open.</p>

<p>Like FlatFile#open, it can accept a block.</p>

<p>Note that you MUST explicitly require ‘open-uri’. Because open-uri.rb
modifies existing class, it isn’t required by default.</p>
            

            
            <div class="method-source-code" id="open_uri-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 177</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">open_uri</span>(<span class="ruby-identifier">uri</span>, *<span class="ruby-identifier">arg</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">block_given?</span> <span class="ruby-keyword">then</span>
    <span class="ruby-constant">BufferedInputStream</span>.<span class="ruby-identifier">open_uri</span>(<span class="ruby-identifier">uri</span>, *<span class="ruby-identifier">arg</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">stream</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">yield</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">nil</span>, <span class="ruby-identifier">stream</span>)
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">stream</span> = <span class="ruby-constant">BufferedInputStream</span>.<span class="ruby-identifier">open_uri</span>(<span class="ruby-identifier">uri</span>, *<span class="ruby-identifier">arg</span>)
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">nil</span>, <span class="ruby-identifier">stream</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- open_uri-source -->
            
          </div>

          

          
        </div><!-- open_uri-method -->

      
        <div id="to_a-method" class="method-detail ">
          <a name="method-c-to_a"></a>

          
          <div class="method-heading">
            <span class="method-name">to_a</span><span
              class="method-args">(*arg)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Same as <a
href="FlatFile.html#method-c-auto">FlatFile.auto</a>(filename_or_stream,
*arg).<a href="FlatFile.html#method-c-to_a">to_a</a></p>

<p>(This method might be OBSOLETED in the future.)</p>
            

            
            <div class="method-source-code" id="to_a-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 129</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">to_a</span>(*<span class="ruby-identifier">arg</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">auto</span>(*<span class="ruby-identifier">arg</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">ff</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">raise</span> <span class="ruby-string">'cannot determine file format'</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">ff</span>.<span class="ruby-identifier">dbclass</span>
    <span class="ruby-identifier">ff</span>.<span class="ruby-identifier">to_a</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- to_a-source -->
            
          </div>

          

          
        </div><!-- to_a-method -->

      
      </div><!-- public-class-method-details -->
    
      <div id="public-instance-method-details" class="method-section section">
        <h3 class="section-header">Public Instance Methods</h3>

      
        <div id="autodetect-method" class="method-detail ">
          <a name="method-i-autodetect"></a>

          
          <div class="method-heading">
            <span class="method-name">autodetect</span><span
              class="method-args">(lines = 31, ad = AutoDetect.default)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Performs determination of database class (file format). Pre-reads
<tt>lines</tt> lines for format determination (default 31 lines). If fails,
returns nil or false. Otherwise, returns database class.</p>

<p>The method can be called anytime if you want (but not recommended). This
might be useful if input file is a mixture of muitiple format data.</p>
            

            
            <div class="method-source-code" id="autodetect-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 429</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">autodetect</span>(<span class="ruby-identifier">lines</span> = <span class="ruby-value">31</span>, <span class="ruby-identifier">ad</span> = <span class="ruby-constant">AutoDetect</span>.<span class="ruby-identifier">default</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">r</span> = <span class="ruby-identifier">ad</span>.<span class="ruby-identifier">autodetect_flatfile</span>(<span class="ruby-keyword">self</span>, <span class="ruby-identifier">lines</span>)
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">dbclass</span> = <span class="ruby-identifier">r</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">dbclass</span> = <span class="ruby-keyword">nil</span> <span class="ruby-keyword">unless</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">dbclass</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">r</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- autodetect-source -->
            
          </div>

          

          
        </div><!-- autodetect-method -->

      
        <div id="close-method" class="method-detail ">
          <a name="method-i-close"></a>

          
          <div class="method-heading">
            <span class="method-name">close</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Closes input stream. (similar to IO#close)</p>
            

            
            <div class="method-source-code" id="close-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 351</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">close</span>
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">close</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- close-source -->
            
          </div>

          

          
        </div><!-- close-method -->

      
        <div id="dbclass-3D-method" class="method-detail ">
          <a name="method-i-dbclass-3D"></a>

          
          <div class="method-heading">
            <span class="method-name">dbclass=</span><span
              class="method-args">(klass)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Sets database class. Plese use only if autodetect fails.</p>
            

            
            <div class="method-source-code" id="dbclass-3D-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 400</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dbclass=</span>(<span class="ruby-identifier">klass</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">klass</span> <span class="ruby-keyword">then</span>
    <span class="ruby-ivar">@dbclass</span> = <span class="ruby-identifier">klass</span>
    <span class="ruby-keyword">begin</span>
      <span class="ruby-ivar">@splitter</span> = <span class="ruby-ivar">@dbclass</span>.<span class="ruby-identifier">flatfile_splitter</span>(<span class="ruby-ivar">@dbclass</span>, <span class="ruby-ivar">@stream</span>)
    <span class="ruby-keyword">rescue</span> <span class="ruby-constant">NameError</span>, <span class="ruby-constant">NoMethodError</span>
      <span class="ruby-keyword">begin</span>
        <span class="ruby-identifier">splitter_class</span> = <span class="ruby-ivar">@dbclass</span><span class="ruby-operator">::</span><span class="ruby-constant">FLATFILE_SPLITTER</span>
      <span class="ruby-keyword">rescue</span> <span class="ruby-constant">NameError</span>
        <span class="ruby-identifier">splitter_class</span> = <span class="ruby-constant">Splitter</span><span class="ruby-operator">::</span><span class="ruby-constant">Default</span>
      <span class="ruby-keyword">end</span>
      <span class="ruby-ivar">@splitter</span> = <span class="ruby-identifier">splitter_class</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">klass</span>, <span class="ruby-ivar">@stream</span>)
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@dbclass</span> = <span class="ruby-keyword">nil</span>
    <span class="ruby-ivar">@splitter</span> = <span class="ruby-keyword">nil</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- dbclass-3D-source -->
            
          </div>

          

          
        </div><!-- dbclass-3D-method -->

      
        <div id="each-method" class="method-detail method-alias">
          <a name="method-i-each"></a>

          
          <div class="method-heading">
            <span class="method-name">each</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="FlatFile.html#method-i-each_entry">each_entry</a>
          </div>
          
        </div><!-- each-method -->

      
        <div id="each_entry-method" class="method-detail ">
          <a name="method-i-each_entry"></a>

          
          <div class="method-heading">
            <span class="method-name">each_entry</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Iterates over each entry in the flatfile.</p>
<ul><li>
<p>Example</p>

<pre>include Bio
ff = FlatFile.open(GenBank, &quot;genbank/gbhtg14.seq&quot;)
ff.each_entry do |x|
  puts x.definition
end</pre>
</li></ul>
            

            
            <div class="method-source-code" id="each_entry-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 334</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">each_entry</span>
  <span class="ruby-keyword">while</span> <span class="ruby-identifier">e</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">next_entry</span>
    <span class="ruby-keyword">yield</span> <span class="ruby-identifier">e</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- each_entry-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="FlatFile.html#method-i-each">each</a>
          </div>
          

          
        </div><!-- each_entry-method -->

      
        <div id="entry_ended_pos-method" class="method-detail ">
          <a name="method-i-entry_ended_pos"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_ended_pos</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>(end position of the last entry) + 1</p>
            

            
            <div class="method-source-code" id="entry_ended_pos-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 322</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_ended_pos</span>
  <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">entry_ended_pos</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- entry_ended_pos-source -->
            
          </div>

          

          
        </div><!-- entry_ended_pos-method -->

      
        <div id="entry_pos_flag-method" class="method-detail ">
          <a name="method-i-entry_pos_flag"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_pos_flag</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>a flag to write down entry start and end positions</p>
            

            
            <div class="method-source-code" id="entry_pos_flag-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 307</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_pos_flag</span>
  <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">entry_pos_flag</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- entry_pos_flag-source -->
            
          </div>

          

          
        </div><!-- entry_pos_flag-method -->

      
        <div id="entry_pos_flag-3D-method" class="method-detail ">
          <a name="method-i-entry_pos_flag-3D"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_pos_flag=</span><span
              class="method-args">(x)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Sets flag to write down entry start and end positions</p>
            

            
            <div class="method-source-code" id="entry_pos_flag-3D-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 312</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_pos_flag=</span>(<span class="ruby-identifier">x</span>)
  <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">entry_pos_flag</span> = <span class="ruby-identifier">x</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- entry_pos_flag-3D-source -->
            
          </div>

          

          
        </div><!-- entry_pos_flag-3D-method -->

      
        <div id="entry_raw-method" class="method-detail ">
          <a name="method-i-entry_raw"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_raw</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns the last raw entry as a string.</p>
            

            
            <div class="method-source-code" id="entry_raw-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 302</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_raw</span>
  <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">entry</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- entry_raw-source -->
            
          </div>

          

          
        </div><!-- entry_raw-method -->

      
        <div id="entry_start_pos-method" class="method-detail ">
          <a name="method-i-entry_start_pos"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_start_pos</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>start position of the last entry</p>
            

            
            <div class="method-source-code" id="entry_start_pos-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 317</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_start_pos</span>
  <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">entry_start_pos</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- entry_start_pos-source -->
            
          </div>

          

          
        </div><!-- entry_start_pos-method -->

      
        <div id="eof-3F-method" class="method-detail ">
          <a name="method-i-eof-3F"></a>

          
          <div class="method-heading">
            <span class="method-name">eof?</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns true if input stream is end-of-file. Otherwise, returns false.
(Similar to IO#eof?, but may not be equal to io.eof?, because <a
href="FlatFile.html">FlatFile</a> has its own internal buffer.)</p>
            

            
            <div class="method-source-code" id="eof-3F-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 380</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">eof?</span>
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">eof?</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- eof-3F-source -->
            
          </div>

          

          
        </div><!-- eof-3F-method -->

      
        <div id="gets-method" class="method-detail ">
          <a name="method-i-gets"></a>

          
          <div class="method-heading">
            <span class="method-name">gets</span><span
              class="method-args">(*arg)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Similar to IO#gets. Internal use only. Users should not call it directly.</p>
            

            
            <div class="method-source-code" id="gets-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 395</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gets</span>(*<span class="ruby-identifier">arg</span>)
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">gets</span>(*<span class="ruby-identifier">arg</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- gets-source -->
            
          </div>

          

          
        </div><!-- gets-method -->

      
        <div id="io-method" class="method-detail ">
          <a name="method-i-io"></a>

          
          <div class="method-heading">
            <span class="method-name">io</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>(DEPRECATED) IO object in the flatfile object.</p>

<p>Compatibility Note: <a
href="FlatFile.html#method-i-io">Bio::FlatFile#io</a> is deprecated. Please
use <a href="FlatFile.html#method-i-to_io">Bio::FlatFile#to_io</a> instead.</p>
            

            
            <div class="method-source-code" id="io-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 255</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">io</span>
  <span class="ruby-identifier">warn</span> <span class="ruby-node">&quot;Bio::FlatFile#io is deprecated.&quot;</span>
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">to_io</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- io-source -->
            
          </div>

          

          
        </div><!-- io-method -->

      
        <div id="next_entry-method" class="method-detail ">
          <a name="method-i-next_entry"></a>

          
          <div class="method-heading">
            <span class="method-name">next_entry</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Get next entry.</p>
            

            
            <div class="method-source-code" id="next_entry-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 277</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">next_entry</span>
  <span class="ruby-identifier">raise</span> <span class="ruby-constant">UnknownDataFormatError</span>, 
  <span class="ruby-string">'file format auto-detection failed?'</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@dbclass</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@skip_leader_mode</span> <span class="ruby-keyword">and</span>
      ((<span class="ruby-ivar">@firsttime_flag</span> <span class="ruby-keyword">and</span> <span class="ruby-ivar">@skip_leader_mode</span> <span class="ruby-operator">==</span> <span class="ruby-value">:firsttime</span>) <span class="ruby-keyword">or</span>
         <span class="ruby-ivar">@skip_leader_mode</span> <span class="ruby-operator">==</span> <span class="ruby-value">:everytime</span>)
    <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">skip_leader</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">raw</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">r</span> = <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">get_entry</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">r</span> = <span class="ruby-ivar">@splitter</span>.<span class="ruby-identifier">get_parsed_entry</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@firsttime_flag</span> = <span class="ruby-keyword">false</span>
  <span class="ruby-keyword">return</span> <span class="ruby-keyword">nil</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">r</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">raw</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">r</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@entry</span> = <span class="ruby-identifier">r</span>
    <span class="ruby-ivar">@entry</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- next_entry-source -->
            
          </div>

          

          
        </div><!-- next_entry-method -->

      
        <div id="path-method" class="method-detail ">
          <a name="method-i-path"></a>

          
          <div class="method-heading">
            <span class="method-name">path</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Pathname, filename or URI (or nil).</p>
            

            
            <div class="method-source-code" id="path-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 268</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">path</span>
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">path</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- path-source -->
            
          </div>

          

          
        </div><!-- path-method -->

      
        <div id="pos-method" class="method-detail ">
          <a name="method-i-pos"></a>

          
          <div class="method-heading">
            <span class="method-name">pos</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns current position of input stream. If the input stream is not a
normal file, the result is not guaranteed. It is similar to IO#pos. Note
that it will not be equal to io.pos, because <a
href="FlatFile.html">FlatFile</a> has its own internal buffer.</p>
            

            
            <div class="method-source-code" id="pos-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 361</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pos</span>
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">pos</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- pos-source -->
            
          </div>

          

          
        </div><!-- pos-method -->

      
        <div id="pos-3D-method" class="method-detail ">
          <a name="method-i-pos-3D"></a>

          
          <div class="method-heading">
            <span class="method-name">pos=</span><span
              class="method-args">(p)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>(Not recommended to use it.) Sets position of input stream. If the input
stream is not a normal file, the result is not guaranteed. It is similar to
IO#pos=. Note that it will not be equal to io.pos=, because <a
href="FlatFile.html">FlatFile</a> has its own internal buffer.</p>
            

            
            <div class="method-source-code" id="pos-3D-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 372</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pos=</span>(<span class="ruby-identifier">p</span>)
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">pos</span>=(<span class="ruby-identifier">p</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- pos-3D-source -->
            
          </div>

          

          
        </div><!-- pos-3D-method -->

      
        <div id="raw-3D-method" class="method-detail ">
          <a name="method-i-raw-3D"></a>

          
          <div class="method-heading">
            <span class="method-name">raw=</span><span
              class="method-args">(bool)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>If true is given, the <a
href="FlatFile.html#method-i-next_entry">next_entry</a> method returns a
entry as a text, whereas if false, returns as a parsed object.</p>
            

            
            <div class="method-source-code" id="raw-3D-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 386</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">raw=</span>(<span class="ruby-identifier">bool</span>)
  <span class="ruby-ivar">@raw</span> = (<span class="ruby-identifier">bool</span> <span class="ruby-operator">?</span> <span class="ruby-keyword">true</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">false</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- raw-3D-source -->
            
          </div>

          

          
        </div><!-- raw-3D-method -->

      
        <div id="rewind-method" class="method-detail ">
          <a name="method-i-rewind"></a>

          
          <div class="method-heading">
            <span class="method-name">rewind</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Resets file pointer to the start of the flatfile. (similar to IO#rewind)</p>
            

            
            <div class="method-source-code" id="rewind-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 343</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">rewind</span>
  <span class="ruby-identifier">r</span> = (<span class="ruby-ivar">@splitter</span> <span class="ruby-operator">||</span> <span class="ruby-ivar">@stream</span>).<span class="ruby-identifier">rewind</span>
  <span class="ruby-ivar">@firsttime_flag</span> = <span class="ruby-keyword">true</span>
  <span class="ruby-identifier">r</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- rewind-source -->
            
          </div>

          

          
        </div><!-- rewind-method -->

      
        <div id="to_io-method" class="method-detail ">
          <a name="method-i-to_io"></a>

          
          <div class="method-heading">
            <span class="method-name">to_io</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>IO object in the flatfile object.</p>

<p>Compatibility Note: <a
href="FlatFile.html#method-i-io">Bio::FlatFile#io</a> is deprecated.</p>
            

            
            <div class="method-source-code" id="to_io-source">
<pre>
<span class="ruby-comment"># File lib/bio/io/flatfile.rb, line 263</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_io</span>
  <span class="ruby-ivar">@stream</span>.<span class="ruby-identifier">to_io</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- to_io-source -->
            
          </div>

          

          
        </div><!-- to_io-method -->

      
      </div><!-- public-instance-method-details -->
    
    </div><!-- 5Buntitled-5D -->
  

  </div><!-- documentation -->

  <div id="validator-badges">
    <p><small><a href="http://validator.w3.org/check/referer">[Validate]</a></small></p>
    <p><small>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish
      Rdoc Generator</a> 2</small>.</p>
  </div>

</body>
</html>

