<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>class Bio::Sequence - BioRuby API documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
  var index_rel_prefix = "../";
</script>

<script src="../js/jquery.js"></script>
<script src="../js/darkfish.js"></script>

<link href="../css/fonts.css" rel="stylesheet">
<link href="../css/rdoc.css" rel="stylesheet">



<body id="top" role="document" class="class">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../table_of_contents.html#pages">Pages</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  
<div class="nav-section">
  <h3>Table of Contents</h3>

  <ul class="link-list" role="directory">
    <li><a href="#class-Bio::Sequence-label-DESCRIPTION">DESCRIPTION</a>
    <li><a href="#class-Bio::Sequence-label-USAGE">USAGE</a>
  </ul>
</div>


  <div id="class-metadata">
    
    <div id="parent-class-section" class="nav-section">
  <h3>Parent</h3>

  
  <p class="link"><a href="../Object.html">Object</a>
  
</div>

    <div id="includes-section" class="nav-section">
  <h3>Included Modules</h3>

  <ul class="link-list">
  
  
    <li><a class="include" href="Sequence/Format.html">Bio::Sequence::Format</a>
  
  
  
    <li><a class="include" href="Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
  
  </ul>
</div>

    <div id="extends-section" class="nav-section">
  <h3>Extended With Modules</h3>

  <ul class="link-list">
    
  
    <li><a class="extend" href="Sequence/Common.html">Bio::Sequence::Common</a>
  
  
  </ul>
</div>

    <!-- Method Quickref -->
<div id="method-list-section" class="nav-section">
  <h3>Methods</h3>

  <ul class="link-list" role="directory">
    
    <li ><a href="#method-c-adapter">::adapter</a>
    
    <li ><a href="#method-c-auto">::auto</a>
    
    <li ><a href="#method-c-guess">::guess</a>
    
    <li ><a href="#method-c-input">::input</a>
    
    <li ><a href="#method-c-new">::new</a>
    
    <li ><a href="#method-c-read">::read</a>
    
    <li ><a href="#method-i-aa">#aa</a>
    
    <li ><a href="#method-i-accessions">#accessions</a>
    
    <li ><a href="#method-i-auto">#auto</a>
    
    <li ><a href="#method-i-guess">#guess</a>
    
    <li ><a href="#method-i-na">#na</a>
    
    <li ><a href="#method-i-to_s">#to_s</a>
    
    <li ><a href="#method-i-to_str">#to_str</a>
    
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-labelledby="class-Bio::Sequence">
  <h1 id="class-Bio::Sequence" class="class">
    class Bio::Sequence
  </h1>

  <section class="description">
    
<h1 id="class-Bio::Sequence-label-DESCRIPTION">DESCRIPTION<span><a href="#class-Bio::Sequence-label-DESCRIPTION">&para;</a> <a href="#top">&uarr;</a></span></h1>

<p><a href="Sequence.html">Bio::Sequence</a> objects represent annotated
sequences in bioruby. A <a href="Sequence.html">Bio::Sequence</a> object is
a wrapper around the actual sequence,  represented as either a <a
href="Sequence/NA.html">Bio::Sequence::NA</a> or a <a
href="Sequence/AA.html">Bio::Sequence::AA</a> object. For most users, this
encapsulation will be completely transparent. <a
href="Sequence.html">Bio::Sequence</a> responds to all methods defined for
Bio::Sequence::NA/AA objects using the same arguments and returning the
same values (even though  these methods are not documented specifically for
<a href="Sequence.html">Bio::Sequence</a>).</p>

<h1 id="class-Bio::Sequence-label-USAGE">USAGE<span><a href="#class-Bio::Sequence-label-USAGE">&para;</a> <a href="#top">&uarr;</a></span></h1>

<pre class="ruby"><span class="ruby-comment"># Create a nucleic or amino acid sequence</span>
<span class="ruby-identifier">dna</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;atgcatgcATGCATGCAAAA&#39;</span>)
<span class="ruby-identifier">rna</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;augcaugcaugcaugcaaaa&#39;</span>)
<span class="ruby-identifier">aa</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;ACDEFGHIKLMNPQRSTVWYU&#39;</span>)

<span class="ruby-comment"># Print it out</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">to_s</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">aa</span>.<span class="ruby-identifier">to_s</span>

<span class="ruby-comment"># Get a subsequence, bioinformatics style (first nucleotide is &#39;1&#39;)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">subseq</span>(<span class="ruby-value">2</span>,<span class="ruby-value">6</span>)

<span class="ruby-comment"># Get a subsequence, informatics style (first nucleotide is &#39;0&#39;)</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>[<span class="ruby-value">2</span>,<span class="ruby-value">6</span>]

<span class="ruby-comment"># Print in FASTA format</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">output</span>(:<span class="ruby-identifier">fasta</span>)

<span class="ruby-comment"># Print all codons</span>
<span class="ruby-identifier">dna</span>.<span class="ruby-identifier">window_search</span>(<span class="ruby-value">3</span>,<span class="ruby-value">3</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">codon</span><span class="ruby-operator">|</span>
  <span class="ruby-identifier">puts</span> <span class="ruby-identifier">codon</span>
<span class="ruby-keyword">end</span>

<span class="ruby-comment"># Splice or otherwise mangle your sequence</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">splicing</span>(<span class="ruby-string">&quot;complement(join(1..5,16..20))&quot;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">rna</span>.<span class="ruby-identifier">splicing</span>(<span class="ruby-string">&quot;complement(join(1..5,16..20))&quot;</span>)

<span class="ruby-comment"># Convert a sequence containing ambiguity codes into a </span>
<span class="ruby-comment"># regular expression you can use for subsequent searching</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">aa</span>.<span class="ruby-identifier">to_re</span>

<span class="ruby-comment"># These should speak for themselves</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">complement</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">composition</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">molecular_weight</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">translate</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">dna</span>.<span class="ruby-identifier">gc_percent</span>
</pre>

  </section>

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    

    
    <section class="attribute-method-details" class="method-section">
      <header>
        <h3>Attributes</h3>
      </header>

      
      <div id="attribute-i-classification" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">classification</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Organism classification, taxonomic classification of the source organism.
(Array of <a href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-comments" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">comments</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Comments (String or an Array of <a href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-data_class" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">data_class</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Data Class defined by <a href="EMBL.html">EMBL</a> (String) See <a
href="http://www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html#3_1">www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html#3_1</a></p>
        
        </div>
      </div>
      
      <div id="attribute-i-date_created" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">date_created</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Created date of the sequence entry (Date, DateTime, Time, or <a
href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-date_modified" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">date_modified</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Last modified date of the sequence entry (Date, DateTime, Time, or <a
href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-dblinks" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">dblinks</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Links to other database entries. (An Array of <a
href="Sequence/DBLink.html">Bio::Sequence::DBLink</a> objects)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-definition" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">definition</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>A <a href="../String.html">String</a> with a description of the sequence
(String)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-division" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">division</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Taxonomic Division defined by EMBL/GenBank/DDBJ (String) See <a
href="http://www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html#3_2">www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html#3_2</a></p>
        
        </div>
      </div>
      
      <div id="attribute-i-entry_id" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">entry_id</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>The sequence identifier (String).  For example, for a sequence of Genbank
origin, this is the locus name. For a sequence of <a
href="EMBL.html">EMBL</a> origin, this is the primary accession number.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-entry_version" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">entry_version</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Version of the entry (String or Integer). Unlike
<code>sequence_version</code>, <code>entry_version</code> is a database
maintainer&#39;s internal version number. The version number will be
changed when the database maintainer modifies the entry. The same enrty in
<a href="EMBL.html">EMBL</a>, <a href="GenBank.html">GenBank</a>, and <a
href="DDBJ.html">DDBJ</a> may have different entry_version.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-error_probabilities" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">error_probabilities</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Error probabilities of the bases/residues in the sequence. (Array
containing Float, or nil)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-features" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">features</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p><a href="Features.html">Features</a> (An Array of <a
href="Feature.html">Bio::Feature</a> objects)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-id_namespace" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">id_namespace</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Namespace of the sequence IDs described in <a
href="Sequence.html#attribute-i-entry_id">#entry_id</a>, <a
href="Sequence.html#attribute-i-primary_accession">#primary_accession</a>,
and <a
href="Sequence.html#attribute-i-secondary_accessions">#secondary_accessions</a>
methods (String). For example, &#39;EMBL&#39;, &#39;GenBank&#39;,
&#39;DDBJ&#39;, &#39;RefSeq&#39;.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-keywords" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">keywords</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Keywords (An Array of <a href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-molecule_type" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">molecule_type</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>molecular type (String). “DNA” or “RNA” for nucleotide sequence.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-moltype" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">moltype</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Bio::Sequence::NA/AA</p>
        
        </div>
      </div>
      
      <div id="attribute-i-organelle" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">organelle</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>(not well supported) Organelle information (String).</p>
        
        </div>
      </div>
      
      <div id="attribute-i-other_seqids" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">other_seqids</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p><a href="Sequence.html">Sequence</a> identifiers which are not described in
<a href="Sequence.html#attribute-i-entry_id">#entry_id</a>, <a
href="Sequence.html#attribute-i-primary_accession">#primary_accession</a>,and
<a
href="Sequence.html#attribute-i-secondary_accessions">#secondary_accessions</a>
methods (Array of <a href="Sequence/DBLink.html">Bio::Sequence::DBLink</a>
objects). For example, <a href="NCBI.html">NCBI</a> GI number can be
stored. Note that only identifiers of the entry itself should be stored.
For database cross references, <code>dblinks</code> should be used.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-primary_accession" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">primary_accession</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Primary accession number (String)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-quality_score_type" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">quality_score_type</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>The meaning (calculation method) of the quality scores stored in the
<code>quality_scores</code> attribute. Maybe one of :phred, :solexa, or
nil.</p>

<p>Note that if it is nil, and <code>error_probabilities</code> is empty, some
methods implicitly assumes that it is :phred (PHRED score).</p>
        
        </div>
      </div>
      
      <div id="attribute-i-quality_scores" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">quality_scores</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Quality scores of the bases/residues in the sequence. (Array containing
Integer, or nil)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-references" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">references</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p><a href="References.html">References</a> (An Array of <a
href="Reference.html">Bio::Reference</a> objects)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-release_created" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">release_created</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Release information when created (String)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-release_modified" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">release_modified</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Release information when last-modified (String)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-secondary_accessions" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">secondary_accessions</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Secondary accession numbers (Array of <a href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-seq" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">seq</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>The sequence object, usually Bio::Sequence::NA/AA,  but could be a simple
<a href="../String.html">String</a></p>
        
        </div>
      </div>
      
      <div id="attribute-i-sequence_version" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">sequence_version</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Version number of the sequence (String or Integer). Unlike
<code>entry_version</code>, <code>sequence_version</code> will be changed
when the submitter of the sequence updates the entry. Normally, the same
entry taken from different databases (EMBL, <a
href="GenBank.html">GenBank</a>, and <a href="DDBJ.html">DDBJ</a>) may have
the same sequence_version.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-species" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">species</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Organism species (String). For example, “Escherichia coli”.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-strandedness" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">strandedness</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Strandedness (String). “single” (single-stranded), “double”
(double-stranded), “mixed” (mixed-stranded), or nil.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-taxonomy" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">taxonomy</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Organism classification, taxonomic classification of the source organism.
(Array of <a href="../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-topology" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">topology</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Topology (String). “circular”, “linear”, or nil.</p>
        
        </div>
      </div>
      
    </section>
    

    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Public Class Methods</h3>
       </header>

    
      <div id="method-c-adapter" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">adapter</span><span
            class="method-args">(source_data, adapter_module)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Normally, users should not call this method directly. Use
Bio::*#to_biosequence (e.g. <a
href="GenBank.html#method-i-to_biosequence">Bio::GenBank#to_biosequence</a>).</p>

<p>Creates a new <a href="Sequence.html">Bio::Sequence</a> object from
database data with an adapter module.</p>
          
          

          
          <div class="method-source-code" id="adapter-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 463</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">adapter</span>(<span class="ruby-identifier">source_data</span>, <span class="ruby-identifier">adapter_module</span>)
  <span class="ruby-identifier">biosequence</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">instance_eval</span> {
    <span class="ruby-identifier">remove_instance_variable</span>(<span class="ruby-value">:@seq</span>)
    <span class="ruby-ivar">@source_data</span> = <span class="ruby-identifier">source_data</span>
  }
  <span class="ruby-identifier">biosequence</span>.<span class="ruby-identifier">extend</span>(<span class="ruby-identifier">adapter_module</span>)
  <span class="ruby-identifier">biosequence</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-c-auto" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">auto</span><span
            class="method-args">(str)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Given a sequence <a href="../String.html">String</a>, guess its type, Amino
Acid or Nucleic Acid, and return a new <a
href="Sequence.html">Bio::Sequence</a> object wrapping a sequence of the
guessed type (either <a href="Sequence/AA.html">Bio::Sequence::AA</a> or <a
href="Sequence/NA.html">Bio::Sequence::NA</a>)</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>str</em>: <a href="../String.html">String</a>
<strong>or</strong> Bio::Sequence::NA/AA object</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Sequence.html">Bio::Sequence</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="auto-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 283</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">auto</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-identifier">seq</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">auto</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">seq</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-c-guess" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">guess</span><span
            class="method-args">(str, *args)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Guess the class of a given sequence.  Returns the class (Bio::Sequence::AA
or <a href="Sequence/NA.html">Bio::Sequence::NA</a>) guessed.  In general,
used by developers only, but if you know what you are doing, feel free.</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> .<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;atgc&#39;</span>)        <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>

<p>There are three optional parameters: `threshold`, `length`, and `index`.</p>

<p>The `threshold` value (defaults to 0.9) is the frequency of  nucleic acid
bases [AGCTUagctu] required in the sequence for this method to produce a <a
href="Sequence/NA.html">Bio::Sequence::NA</a> “guess”.  In the default
case, if less than 90% of the bases (after excluding [Nn]) are in the set
[AGCTUagctu], then the guess is <a
href="Sequence/AA.html">Bio::Sequence::AA</a>.</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;atgcatgcqq&#39;</span>)      <span class="ruby-comment">#=&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;atgcatgcqq&#39;</span>, <span class="ruby-value">0.8</span>) <span class="ruby-comment">#=&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;atgcatgcqq&#39;</span>, <span class="ruby-value">0.7</span>) <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>

<p>The `length` value is how much of the total sequence to use in the guess
(default 10000).  If your sequence is very long, you may  want to use a
smaller amount to reduce the computational burden.</p>

<pre class="ruby"><span class="ruby-comment"># limit the guess to the first 1000 positions</span>
<span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;A VERY LONG SEQUENCE&#39;</span>, <span class="ruby-value">0.9</span>, <span class="ruby-value">1000</span>)
</pre>

<p>The `index` value is where to start the guess.  Perhaps you know there are
a lot of gaps at the start…</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;-----atgcc&#39;</span>)             <span class="ruby-comment">#=&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">puts</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-string">&#39;-----atgcc&#39;</span>,<span class="ruby-value">0.9</span>,<span class="ruby-value">10000</span>,<span class="ruby-value">5</span>) <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>str</em>: <a href="../String.html">String</a>
<strong>or</strong> Bio::Sequence::NA/AA object</p>
</li><li>
<p>(optional) <em>threshold</em>: Float in range 0,1 (default 0.9)</p>
</li><li>
<p>(optional) <em>length</em>: Fixnum (default 10000)</p>
</li><li>
<p>(optional) <em>index</em>: Fixnum (default 1)</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Bio::Sequence::NA/AA</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="guess-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 381</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-identifier">str</span>, <span class="ruby-operator">*</span><span class="ruby-identifier">args</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">str</span>).<span class="ruby-identifier">guess</span>(<span class="ruby-operator">*</span><span class="ruby-identifier">args</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-c-input" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">input</span><span
            class="method-args">(str, format = nil)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Create a new <a href="Sequence.html">Bio::Sequence</a> object from a
formatted string (GenBank, <a href="EMBL.html">EMBL</a>, fasta format,
etc.)</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">input</span>(<span class="ruby-identifier">str</span>)
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>str</em>: string</p>
</li><li>
<p>(optional) <em>format</em>: format specification (class or nil)</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Sequence.html">Bio::Sequence</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="input-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 436</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">input</span>(<span class="ruby-identifier">str</span>, <span class="ruby-identifier">format</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">format</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">klass</span> = <span class="ruby-identifier">format</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">klass</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">FlatFile</span><span class="ruby-operator">::</span><span class="ruby-constant">AutoDetect</span>.<span class="ruby-identifier">default</span>.<span class="ruby-identifier">autodetect</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">obj</span> = <span class="ruby-identifier">klass</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-identifier">obj</span>.<span class="ruby-identifier">to_biosequence</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(str)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Create a new <a href="Sequence.html">Bio::Sequence</a> object</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
</pre>

<p>Note that this method does not intialize the contained sequence as any kind
of bioruby object, only as a simple string</p>

<pre class="ruby"><span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; String</span>
</pre>

<p>See <a href="Sequence.html#method-i-na">#na</a>, <a
href="Sequence.html#method-i-aa">#aa</a>, and <a
href="Sequence.html#method-i-auto">#auto</a>  for methods to transform the
basic <a href="../String.html">String</a> of a just created  <a
href="Sequence.html">Bio::Sequence</a> object to a proper bioruby object</p>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>str</em>: <a href="../String.html">String</a> or
Bio::Sequence::NA/AA object</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Sequence.html">Bio::Sequence</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 99</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-ivar">@seq</span> = <span class="ruby-identifier">str</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-c-read" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">read</span><span
            class="method-args">(str, format = nil)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>alias of <a href="Sequence.html#method-c-input">::input</a></p>
          
          

          
          <div class="method-source-code" id="read-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 447</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">read</span>(<span class="ruby-identifier">str</span>, <span class="ruby-identifier">format</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">input</span>(<span class="ruby-identifier">str</span>, <span class="ruby-identifier">format</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
    </section>
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section">
       <header>
         <h3>Public Instance Methods</h3>
       </header>

    
      <div id="method-i-aa" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">aa</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Transform the sequence wrapped in the current <a
href="Sequence.html">Bio::Sequence</a> object into a <a
href="Sequence/NA.html">Bio::Sequence::NA</a> object.  This method will
change the current object.  This method does not validate your choice, so
be careful!</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; String</span>
<span class="ruby-identifier">s</span>.<span class="ruby-identifier">aa</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; Bio::Sequence::AA !!!</span>
</pre>

<p>However, if you know your sequence type, this method may be  constructively
used after initialization,</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;RRLE&#39;</span>)
<span class="ruby-identifier">s</span>.<span class="ruby-identifier">aa</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Sequence/AA.html">Bio::Sequence::AA</a></p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="aa-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 422</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">aa</span>
  <span class="ruby-ivar">@seq</span> = <span class="ruby-constant">AA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">seq</span>)
  <span class="ruby-ivar">@moltype</span> = <span class="ruby-constant">AA</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-accessions" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">accessions</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>accession numbers of the sequence</p>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Array of <a href="../String.html">String</a></p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="accessions-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 454</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">accessions</span>
  [ <span class="ruby-identifier">primary_accession</span>, <span class="ruby-identifier">secondary_accessions</span> ].<span class="ruby-identifier">flatten</span>.<span class="ruby-identifier">compact</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-auto" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">auto</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Guess the type of sequence, Amino Acid or Nucleic Acid, and create a  new
sequence object (Bio::Sequence::AA or <a
href="Sequence/NA.html">Bio::Sequence::NA</a>) on the basis of this guess. 
This method will change the current <a
href="Sequence.html">Bio::Sequence</a> object.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; String</span>
<span class="ruby-identifier">s</span>.<span class="ruby-identifier">auto</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Bio::Sequence::NA/AA object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="auto-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 264</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">auto</span>
  <span class="ruby-ivar">@moltype</span> = <span class="ruby-identifier">guess</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@moltype</span> <span class="ruby-operator">==</span> <span class="ruby-constant">NA</span>
    <span class="ruby-ivar">@seq</span> = <span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">seq</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@seq</span> = <span class="ruby-constant">AA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">seq</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-guess" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">guess</span><span
            class="method-args">(threshold = 0.9, length = 10000, index = 0)</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Guess the class of the current sequence.  Returns the class
(Bio::Sequence::AA or <a href="Sequence/NA.html">Bio::Sequence::NA</a>)
guessed.  In general, used by developers only, but if you know what you are
doing, feel free.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>                            <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>

<p>There are three parameters: `threshold`, `length`, and `index`.</p>

<p>The `threshold` value (defaults to 0.9) is the frequency of  nucleic acid
bases [AGCTUagctu] required in the sequence for this method to produce a <a
href="Sequence/NA.html">Bio::Sequence::NA</a> “guess”.  In the default
case, if less than 90% of the bases (after excluding [Nn]) are in the set
[AGCTUagctu], then the guess is <a
href="Sequence/AA.html">Bio::Sequence::AA</a>.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgcatgcqq&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>                            <span class="ruby-comment">#=&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-value">0.8</span>)                       <span class="ruby-comment">#=&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-value">0.7</span>)                       <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>

<p>The `length` value is how much of the total sequence to use in the guess
(default 10000).  If your sequence is very long, you may  want to use a
smaller amount to reduce the computational burden.</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-constant">A</span> <span class="ruby-constant">VERY</span> <span class="ruby-constant">LONG</span> <span class="ruby-constant">SEQUENCE</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-value">0.9</span>, <span class="ruby-value">1000</span>)  <span class="ruby-comment"># limit the guess to the first 1000 positions</span>
</pre>

<p>The `index` value is where to start the guess.  Perhaps you know there are
a lot of gaps at the start…</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;-----atgcc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>                            <span class="ruby-comment">#=&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">guess</span>(<span class="ruby-value">0.9</span>,<span class="ruby-value">10000</span>,<span class="ruby-value">5</span>)               <span class="ruby-comment">#=&gt; Bio::Sequence::NA</span>
</pre>
<hr>

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>threshold</em>: Float in range 0,1 (default 0.9)</p>
</li><li>
<p>(optional) <em>length</em>: Fixnum (default 10000)</p>
</li><li>
<p>(optional) <em>index</em>: Fixnum (default 1)</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>Bio::Sequence::NA/AA</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="guess-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 328</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">guess</span>(<span class="ruby-identifier">threshold</span> = <span class="ruby-value">0.9</span>, <span class="ruby-identifier">length</span> = <span class="ruby-value">10000</span>, <span class="ruby-identifier">index</span> = <span class="ruby-value">0</span>)
  <span class="ruby-identifier">str</span> = <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">to_s</span>[<span class="ruby-identifier">index</span>,<span class="ruby-identifier">length</span>].<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">extend</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">Common</span>
  <span class="ruby-identifier">cmp</span> = <span class="ruby-identifier">str</span>.<span class="ruby-identifier">composition</span>

  <span class="ruby-identifier">bases</span> = <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;A&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;T&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;G&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;C&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;U&#39;</span>] <span class="ruby-operator">+</span>
          <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;a&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;t&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;g&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;c&#39;</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;u&#39;</span>]

  <span class="ruby-identifier">total</span> = <span class="ruby-identifier">str</span>.<span class="ruby-identifier">length</span> <span class="ruby-operator">-</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;N&#39;</span>] <span class="ruby-operator">-</span> <span class="ruby-identifier">cmp</span>[<span class="ruby-string">&#39;n&#39;</span>]

  <span class="ruby-keyword">if</span> <span class="ruby-identifier">bases</span>.<span class="ruby-identifier">to_f</span> <span class="ruby-operator">/</span> <span class="ruby-identifier">total</span> <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">threshold</span>
    <span class="ruby-keyword">return</span> <span class="ruby-constant">NA</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">return</span> <span class="ruby-constant">AA</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-na" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">na</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Transform the sequence wrapped in the current <a
href="Sequence.html">Bio::Sequence</a> object into a <a
href="Sequence/NA.html">Bio::Sequence::NA</a> object.  This method will
change the current object.  This method does not validate your choice, so
be careful!</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;RRLE&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; String</span>
<span class="ruby-identifier">s</span>.<span class="ruby-identifier">na</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">seq</span>.<span class="ruby-identifier">class</span>                        <span class="ruby-comment">#=&gt; Bio::Sequence::NA !!!</span>
</pre>

<p>However, if you know your sequence type, this method may be  constructively
used after initialization,</p>

<pre class="ruby"><span class="ruby-identifier">s</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">s</span>.<span class="ruby-identifier">na</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Sequence/NA.html">Bio::Sequence::NA</a></p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="na-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence.rb, line 401</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">na</span>
  <span class="ruby-ivar">@seq</span> = <span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">seq</span>)
  <span class="ruby-ivar">@moltype</span> = <span class="ruby-constant">NA</span>
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        

        
      </div>

    
      <div id="method-i-to_s" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_s</span><span
            class="method-args">()</span>
          
          <span class="method-click-advice">click to toggle source</span>
          
        </div>
        

        <div class="method-description">
          
          <p>Return sequence as  <a
href="http://corelib.rubyonrails.org/classes/String.html">String</a>. The
original sequence is unchanged.</p>

<pre class="ruby"><span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&#39;atgc&#39;</span>)
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">to_s</span>                             <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">class</span>                       <span class="ruby-comment">#=&gt; String</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>                                  <span class="ruby-comment">#=&gt; &#39;atgc&#39;</span>
<span class="ruby-identifier">puts</span> <span class="ruby-identifier">s</span>.<span class="ruby-identifier">class</span>                            <span class="ruby-comment">#=&gt; Bio::Sequence</span>
</pre>
<hr>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a> object</p>
</dd></dl>
          
          

          
          <div class="method-source-code" id="to_s-source">
            <pre><span class="ruby-comment"># File lib/bio/sequence/compat.rb, line 27</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_s</span>
  <span class="ruby-constant">String</span>.<span class="ruby-identifier">new</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">seq</span>)
<span class="ruby-keyword">end</span></pre>
          </div>
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Sequence.html#method-i-to_str">to_str</a>
        </div>
        

        
      </div>

    
      <div id="method-i-to_str" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">to_str</span><span
            class="method-args">()</span>
          
        </div>
        

        <div class="method-description">
          
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Sequence.html#method-i-to_s">to_s</a>
        </div>
        
      </div>

    
    </section>
  
  </section>
</main>


<footer id="validator-badges" role="contentinfo">
  <p><a href="http://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://rdoc.github.io/rdoc">RDoc</a> 5.0.0.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

